// @btcv/ui/dist/_shared/index-C0gNQvxa.js
import * as n from "react";
import y from "react";
import { a as G, c as h, u as q, d as z } from "./index-CXeb1OMI.js";
import { c as Z, u as O } from "./index-DLcqcWxM.js";
import { jsx as R } from "react/jsx-runtime";
import { u as J } from "./index-BC7vfx-u.js";
import { P as K } from "./index-V-Ajw7Ac.js";
// @__NO_SIDE_EFFECTS__
function L(e) {
  const t = /* @__PURE__ */ Q(e), r = n.forwardRef((o, s) => {
    const { children: l, ...d } = o, u = n.Children.toArray(l), g = u.find(ee);
    if (g) {
      const p = g.props.children, a = u.map((S) => S === g ? n.Children.count(p) > 1 ? n.Children.only(null) : n.isValidElement(p) ? p.props.children : null : S);
      return /* @__PURE__ */ R(t, { ...d, ref: s, children: n.isValidElement(p) ? n.cloneElement(p, void 0, a) : null });
    }
    return /* @__PURE__ */ R(t, { ...d, ref: s, children: l });
  });
  return r.displayName = `${e}.Slot`, r;
}
// @__NO_SIDE_EFFECTS__
function Q(e) {
  const t = n.forwardRef((r, o) => {
    const { children: s, ...l } = r;
    if (n.isValidElement(s)) {
      const d = oe(s), u = te(l, s.props);
      return s.type !== n.Fragment && (u.ref = o ? Z(o, d) : d), n.cloneElement(s, u);
    }
    return n.Children.count(s) > 1 ? n.Children.only(null) : null;
  });
  return t.displayName = `${e}.SlotClone`, t;
}
var X = /* @__PURE__ */ Symbol("radix.slottable");
function ee(e) {
  return n.isValidElement(e) && typeof e.type == "function" && "__radixId" in e.type && e.type.__radixId === X;
}
function te(e, t) {
  const r = { ...t };
  for (const o in t) {
    const s = e[o], l = t[o];
    /^on[A-Z]/.test(o) ? s && l ? r[o] = (...u) => {
      const g = l(...u);
      return s(...u), g;
    } : s && (r[o] = s) : o === "style" ? r[o] = { ...s, ...l } : o === "className" && (r[o] = [s, l].filter(Boolean).join(" "));
  }
  return { ...e, ...r };
}
function oe(e) {
  let t = Object.getOwnPropertyDescriptor(e.props, "ref")?.get, r = t && "isReactWarning" in t && t.isReactWarning;
  return r ? e.ref : (t = Object.getOwnPropertyDescriptor(e, "ref")?.get, r = t && "isReactWarning" in t && t.isReactWarning, r ? e.props.ref : e.props.ref || e.ref);
}
function re(e) {
  const t = e + "CollectionProvider", [r, o] = G(t), [s, l] = r(
    t,
    { collectionRef: { current: null }, itemMap: /* @__PURE__ */ new Map() }
  ), d = (C) => {
    const { scope: c, children: m } = C, v = y.useRef(null), i = y.useRef(/* @__PURE__ */ new Map()).current;
    return /* @__PURE__ */ R(s, { scope: c, itemMap: i, collectionRef: v, children: m });
  };
  d.displayName = t;
  const u = e + "CollectionSlot", g = /* @__PURE__ */ L(u), p = y.forwardRef(
    (C, c) => {
      const { scope: m, children: v } = C, i = l(u, m), f = O(c, i.collectionRef);
      return /* @__PURE__ */ R(g, { ref: f, children: v });
    }
  );
  p.displayName = u;
  const a = e + "CollectionItemSlot", S = "data-radix-collection-item", F = /* @__PURE__ */ L(a), b = y.forwardRef(
    (C, c) => {
      const { scope: m, children: v, ...i } = C, f = y.useRef(null), w = O(c, f), _ = l(a, m);
      return y.useEffect(() => (_.itemMap.set(f, { ref: f, ...i }), () => {
        _.itemMap.delete(f);
      })), /* @__PURE__ */ R(F, { [S]: "", ref: w, children: v });
    }
  );
  b.displayName = a;
  function T(C) {
    const c = l(e + "CollectionConsumer", C);
    return y.useCallback(() => {
      const v = c.collectionRef.current;
      if (!v) return [];
      const i = Array.from(v.querySelectorAll(`[${S}]`));
      return Array.from(c.itemMap.values()).sort(
        (_, x) => i.indexOf(_.ref.current) - i.indexOf(x.ref.current)
      );
    }, [c.collectionRef, c.itemMap]);
  }
  return [
    { Provider: d, Slot: p, ItemSlot: b },
    T,
    o
  ];
}
var ne = n.createContext(void 0);
function ce(e) {
  const t = n.useContext(ne);
  return e || t || "ltr";
}
var P = "rovingFocusGroup.onEntryFocus", se = { bubbles: !1, cancelable: !0 }, A = "RovingFocusGroup", [D, k, ie] = re(A), [le, ye] = G(
  A,
  [ie]
), [ae, ue] = le(A), V = n.forwardRef(
  (e, t) => /* @__PURE__ */ R(D.Provider, { scope: e.__scopeRovingFocusGroup, children: /* @__PURE__ */ R(D.Slot, { scope: e.__scopeRovingFocusGroup, children: /* @__PURE__ */ R(fe, { ...e, ref: t }) }) })
);
V.displayName = A;
var fe = n.forwardRef((e, t) => {
  const {
    __scopeRovingFocusGroup: r,
    orientation: o,
    loop: s = !1,
    dir: l,
    currentTabStopId: d,
    defaultCurrentTabStopId: u,
    onCurrentTabStopIdChange: g,
    onEntryFocus: p,
    preventScrollOnEntryFocus: a = !1,
    ...S
  } = e, F = n.useRef(null), b = O(t, F), T = ce(l), [C, c] = q({
    prop: d,
    defaultProp: u ?? null,
    onChange: g,
    caller: A
  }), [m, v] = n.useState(!1), i = z(p), f = k(r), w = n.useRef(!1), [_, x] = n.useState(0);
  return n.useEffect(() => {
    const I = F.current;
    if (I)
      return I.addEventListener(P, i), () => I.removeEventListener(P, i);
  }, [i]), /* @__PURE__ */ R(
    ae,
    {
      scope: r,
      orientation: o,
      dir: T,
      loop: s,
      currentTabStopId: C,
      onItemFocus: n.useCallback(
        (I) => c(I),
        [c]
      ),
      onItemShiftTab: n.useCallback(() => v(!0), []),
      onFocusableItemAdd: n.useCallback(
        () => x((I) => I + 1),
        []
      ),
      onFocusableItemRemove: n.useCallback(
        () => x((I) => I - 1),
        []
      ),
      children: /* @__PURE__ */ R(
        K.div,
        {
          tabIndex: m || _ === 0 ? -1 : 0,
          "data-orientation": o,
          ...S,
          ref: b,
          style: { outline: "none", ...e.style },
          onMouseDown: h(e.onMouseDown, () => {
            w.current = !0;
          }),
          onFocus: h(e.onFocus, (I) => {
            const j = !w.current;
            if (I.target === I.currentTarget && j && !m) {
              const M = new CustomEvent(P, se);
              if (I.currentTarget.dispatchEvent(M), !M.defaultPrevented) {
                const N = f().filter((E) => E.focusable), H = N.find((E) => E.active), Y = N.find((E) => E.id === C), $ = [H, Y, ...N].filter(
                  Boolean
                ).map((E) => E.ref.current);
                W($, a);
              }
            }
            w.current = !1;
          }),
          onBlur: h(e.onBlur, () => v(!1))
        }
      )
    }
  );
}), U = "RovingFocusGroupItem", B = n.forwardRef(
  (e, t) => {
    const {
      __scopeRovingFocusGroup: r,
      focusable: o = !0,
      active: s = !1,
      tabStopId: l,
      children: d,
      ...u
    } = e, g = J(), p = l || g, a = ue(U, r), S = a.currentTabStopId === p, F = k(r), { onFocusableItemAdd: b, onFocusableItemRemove: T, currentTabStopId: C } = a;
    return n.useEffect(() => {
      if (o)
        return b(), () => T();
    }, [o, b, T]), /* @__PURE__ */ R(
      D.ItemSlot,
      {
        scope: r,
        id: p,
        focusable: o,
        active: s,
        children: /* @__PURE__ */ R(
          K.span,
          {
            tabIndex: S ? 0 : -1,
            "data-orientation": a.orientation,
            ...u,
            ref: t,
            onMouseDown: h(e.onMouseDown, (c) => {
              o ? a.onItemFocus(p) : c.preventDefault();
            }),
            onFocus: h(e.onFocus, () => a.onItemFocus(p)),
            onKeyDown: h(e.onKeyDown, (c) => {
              if (c.key === "Tab" && c.shiftKey) {
                a.onItemShiftTab();
                return;
              }
              if (c.target !== c.currentTarget) return;
              const m = me(c, a.orientation, a.dir);
              if (m !== void 0) {
                if (c.metaKey || c.ctrlKey || c.altKey || c.shiftKey) return;
                c.preventDefault();
                let i = F().filter((f) => f.focusable).map((f) => f.ref.current);
                if (m === "last") i.reverse();
                else if (m === "prev" || m === "next") {
                  m === "prev" && i.reverse();
                  const f = i.indexOf(c.currentTarget);
                  i = a.loop ? Ie(i, f + 1) : i.slice(f + 1);
                }
                setTimeout(() => W(i));
              }
            }),
            children: typeof d == "function" ? d({ isCurrentTabStop: S, hasTabStop: C != null }) : d
          }
        )
      }
    );
  }
);
B.displayName = U;
var de = {
  ArrowLeft: "prev",
  ArrowUp: "prev",
  ArrowRight: "next",
  ArrowDown: "next",
  PageUp: "first",
  Home: "first",
  PageDown: "last",
  End: "last"
};
function pe(e, t) {
  return t !== "rtl" ? e : e === "ArrowLeft" ? "ArrowRight" : e === "ArrowRight" ? "ArrowLeft" : e;
}
function me(e, t, r) {
  const o = pe(e.key, r);
  if (!(t === "vertical" && ["ArrowLeft", "ArrowRight"].includes(o)) && !(t === "horizontal" && ["ArrowUp", "ArrowDown"].includes(o)))
    return de[o];
}
function W(e, t = !1) {
  const r = document.activeElement;
  for (const o of e)
    if (o === r || (o.focus({ preventScroll: t }), document.activeElement !== r)) return;
}
function Ie(e, t) {
  return e.map((r, o) => e[(t + o) % e.length]);
}
var Fe = V, Te = B;
export {
  Te as I,
  Fe as R,
  re as a,
  ye as c,
  ce as u
};
