// @btcv/ui/dist/_shared/index-C3aZemLI.js
import * as a from "react";
import { P as O, c as h, a as L, u as W } from "./index-CXeb1OMI.js";
import { c as H, u as _ } from "./index-DLcqcWxM.js";
import { P as j, D as $ } from "./index-B03TCNO5.js";
import { h as G, R as B, u as K, F as Z } from "./Combination-D_l4PLF_.js";
import { u as U } from "./index-BC7vfx-u.js";
import { c as E, C as q, R as z, A as y, a as J } from "./index-uu9PT5Nu.js";
import { P as x } from "./index-V-Ajw7Ac.js";
import { jsx as l } from "react/jsx-runtime";
// @__NO_SIDE_EFFECTS__
function Q(e) {
  const r = /* @__PURE__ */ X(e), t = a.forwardRef((n, o) => {
    const { children: s, ...c } = n, i = a.Children.toArray(s), p = i.find(ee);
    if (p) {
      const u = p.props.children, P = i.map((d) => d === p ? a.Children.count(u) > 1 ? a.Children.only(null) : a.isValidElement(u) ? u.props.children : null : d);
      return /* @__PURE__ */ l(r, { ...c, ref: o, children: a.isValidElement(u) ? a.cloneElement(u, void 0, P) : null });
    }
    return /* @__PURE__ */ l(r, { ...c, ref: o, children: s });
  });
  return t.displayName = `${e}.Slot`, t;
}
// @__NO_SIDE_EFFECTS__
function X(e) {
  const r = a.forwardRef((t, n) => {
    const { children: o, ...s } = t;
    if (a.isValidElement(o)) {
      const c = re(o), i = oe(s, o.props);
      return o.type !== a.Fragment && (i.ref = n ? H(n, c) : c), a.cloneElement(o, i);
    }
    return a.Children.count(o) > 1 ? a.Children.only(null) : null;
  });
  return r.displayName = `${e}.SlotClone`, r;
}
var Y = /* @__PURE__ */ Symbol("radix.slottable");
function ee(e) {
  return a.isValidElement(e) && typeof e.type == "function" && "__radixId" in e.type && e.type.__radixId === Y;
}
function oe(e, r) {
  const t = { ...r };
  for (const n in r) {
    const o = e[n], s = r[n];
    /^on[A-Z]/.test(n) ? o && s ? t[n] = (...i) => {
      const p = s(...i);
      return o(...i), p;
    } : o && (t[n] = o) : n === "style" ? t[n] = { ...o, ...s } : n === "className" && (t[n] = [o, s].filter(Boolean).join(" "));
  }
  return { ...e, ...t };
}
function re(e) {
  let r = Object.getOwnPropertyDescriptor(e.props, "ref")?.get, t = r && "isReactWarning" in r && r.isReactWarning;
  return t ? e.ref : (r = Object.getOwnPropertyDescriptor(e, "ref")?.get, t = r && "isReactWarning" in r && r.isReactWarning, t ? e.props.ref : e.props.ref || e.ref);
}
var m = "Popover", [S] = L(m, [
  E
]), C = E(), [te, f] = S(m), w = (e) => {
  const {
    __scopePopover: r,
    children: t,
    open: n,
    defaultOpen: o,
    onOpenChange: s,
    modal: c = !1
  } = e, i = C(r), p = a.useRef(null), [u, P] = a.useState(!1), [d, v] = W({
    prop: n,
    defaultProp: o ?? !1,
    onChange: s,
    caller: m
  });
  return /* @__PURE__ */ l(z, { ...i, children: /* @__PURE__ */ l(
    te,
    {
      scope: r,
      contentId: U(),
      triggerRef: p,
      open: d,
      onOpenChange: v,
      onOpenToggle: a.useCallback(() => v((R) => !R), [v]),
      hasCustomAnchor: u,
      onCustomAnchorAdd: a.useCallback(() => P(!0), []),
      onCustomAnchorRemove: a.useCallback(() => P(!1), []),
      modal: c,
      children: t
    }
  ) });
};
w.displayName = m;
var b = "PopoverAnchor", ne = a.forwardRef(
  (e, r) => {
    const { __scopePopover: t, ...n } = e, o = f(b, t), s = C(t), { onCustomAnchorAdd: c, onCustomAnchorRemove: i } = o;
    return a.useEffect(() => (c(), () => i()), [c, i]), /* @__PURE__ */ l(y, { ...s, ...n, ref: r });
  }
);
ne.displayName = b;
var N = "PopoverTrigger", F = a.forwardRef(
  (e, r) => {
    const { __scopePopover: t, ...n } = e, o = f(N, t), s = C(t), c = _(r, o.triggerRef), i = /* @__PURE__ */ l(
      x.button,
      {
        type: "button",
        "aria-haspopup": "dialog",
        "aria-expanded": o.open,
        "aria-controls": o.contentId,
        "data-state": V(o.open),
        ...n,
        ref: c,
        onClick: h(e.onClick, o.onOpenToggle)
      }
    );
    return o.hasCustomAnchor ? i : /* @__PURE__ */ l(y, { asChild: !0, ...s, children: i });
  }
);
F.displayName = N;
var A = "PopoverPortal", [se, ae] = S(A, {
  forceMount: void 0
}), D = (e) => {
  const { __scopePopover: r, forceMount: t, children: n, container: o } = e, s = f(A, r);
  return /* @__PURE__ */ l(se, { scope: r, forceMount: t, children: /* @__PURE__ */ l(O, { present: t || s.open, children: /* @__PURE__ */ l(j, { asChild: !0, container: o, children: n }) }) });
};
D.displayName = A;
var g = "PopoverContent", I = a.forwardRef(
  (e, r) => {
    const t = ae(g, e.__scopePopover), { forceMount: n = t.forceMount, ...o } = e, s = f(g, e.__scopePopover);
    return /* @__PURE__ */ l(O, { present: n || s.open, children: s.modal ? /* @__PURE__ */ l(ie, { ...o, ref: r }) : /* @__PURE__ */ l(le, { ...o, ref: r }) });
  }
);
I.displayName = g;
var ce = /* @__PURE__ */ Q("PopoverContent.RemoveScroll"), ie = a.forwardRef(
  (e, r) => {
    const t = f(g, e.__scopePopover), n = a.useRef(null), o = _(r, n), s = a.useRef(!1);
    return a.useEffect(() => {
      const c = n.current;
      if (c) return G(c);
    }, []), /* @__PURE__ */ l(B, { as: ce, allowPinchZoom: !0, children: /* @__PURE__ */ l(
      T,
      {
        ...e,
        ref: o,
        trapFocus: t.open,
        disableOutsidePointerEvents: !0,
        onCloseAutoFocus: h(e.onCloseAutoFocus, (c) => {
          c.preventDefault(), s.current || t.triggerRef.current?.focus();
        }),
        onPointerDownOutside: h(
          e.onPointerDownOutside,
          (c) => {
            const i = c.detail.originalEvent, p = i.button === 0 && i.ctrlKey === !0, u = i.button === 2 || p;
            s.current = u;
          },
          { checkForDefaultPrevented: !1 }
        ),
        onFocusOutside: h(
          e.onFocusOutside,
          (c) => c.preventDefault(),
          { checkForDefaultPrevented: !1 }
        )
      }
    ) });
  }
), le = a.forwardRef(
  (e, r) => {
    const t = f(g, e.__scopePopover), n = a.useRef(!1), o = a.useRef(!1);
    return /* @__PURE__ */ l(
      T,
      {
        ...e,
        ref: r,
        trapFocus: !1,
        disableOutsidePointerEvents: !1,
        onCloseAutoFocus: (s) => {
          e.onCloseAutoFocus?.(s), s.defaultPrevented || (n.current || t.triggerRef.current?.focus(), s.preventDefault()), n.current = !1, o.current = !1;
        },
        onInteractOutside: (s) => {
          e.onInteractOutside?.(s), s.defaultPrevented || (n.current = !0, s.detail.originalEvent.type === "pointerdown" && (o.current = !0));
          const c = s.target;
          t.triggerRef.current?.contains(c) && s.preventDefault(), s.detail.originalEvent.type === "focusin" && o.current && s.preventDefault();
        }
      }
    );
  }
), T = a.forwardRef(
  (e, r) => {
    const {
      __scopePopover: t,
      trapFocus: n,
      onOpenAutoFocus: o,
      onCloseAutoFocus: s,
      disableOutsidePointerEvents: c,
      onEscapeKeyDown: i,
      onPointerDownOutside: p,
      onFocusOutside: u,
      onInteractOutside: P,
      ...d
    } = e, v = f(g, t), R = C(t);
    return K(), /* @__PURE__ */ l(
      Z,
      {
        asChild: !0,
        loop: !0,
        trapped: n,
        onMountAutoFocus: o,
        onUnmountAutoFocus: s,
        children: /* @__PURE__ */ l(
          $,
          {
            asChild: !0,
            disableOutsidePointerEvents: c,
            onInteractOutside: P,
            onEscapeKeyDown: i,
            onPointerDownOutside: p,
            onFocusOutside: u,
            onDismiss: () => v.onOpenChange(!1),
            children: /* @__PURE__ */ l(
              q,
              {
                "data-state": V(v.open),
                role: "dialog",
                id: v.contentId,
                ...R,
                ...d,
                ref: r,
                style: {
                  ...d.style,
                  "--radix-popover-content-transform-origin": "var(--radix-popper-transform-origin)",
                  "--radix-popover-content-available-width": "var(--radix-popper-available-width)",
                  "--radix-popover-content-available-height": "var(--radix-popper-available-height)",
                  "--radix-popover-trigger-width": "var(--radix-popper-anchor-width)",
                  "--radix-popover-trigger-height": "var(--radix-popper-anchor-height)"
                }
              }
            )
          }
        )
      }
    );
  }
), M = "PopoverClose", k = a.forwardRef(
  (e, r) => {
    const { __scopePopover: t, ...n } = e, o = f(M, t);
    return /* @__PURE__ */ l(
      x.button,
      {
        type: "button",
        ...n,
        ref: r,
        onClick: h(e.onClick, () => o.onOpenChange(!1))
      }
    );
  }
);
k.displayName = M;
var pe = "PopoverArrow", ue = a.forwardRef(
  (e, r) => {
    const { __scopePopover: t, ...n } = e, o = C(t);
    return /* @__PURE__ */ l(J, { ...o, ...n, ref: r });
  }
);
ue.displayName = pe;
function V(e) {
  return e ? "open" : "closed";
}
var Re = w, Ae = F, Oe = D, _e = I, Ee = k;
export {
  _e as C,
  Oe as P,
  Re as R,
  Ae as T,
  Ee as a
};
