// @btcv/ui/dist/_shared/index-BrLJJgkl.js
import * as o from "react";
import { c as b } from "./index-DLcqcWxM.js";
import { jsx as u, Fragment as g } from "react/jsx-runtime";
var E = /* @__PURE__ */ Symbol.for("react.lazy"), p = o[" use ".trim().toString()];
function _(t) {
  return typeof t == "object" && t !== null && "then" in t;
}
function d(t) {
  return t != null && typeof t == "object" && "$$typeof" in t && t.$$typeof === E && "_payload" in t && _(t._payload);
}
// @__NO_SIDE_EFFECTS__
function C(t) {
  const r = /* @__PURE__ */ R(t), n = o.forwardRef((l, e) => {
    let { children: i, ...s } = l;
    d(i) && typeof p == "function" && (i = p(i._payload));
    const a = o.Children.toArray(i), c = a.find(P);
    if (c) {
      const f = c.props.children, m = a.map((y) => y === c ? o.Children.count(f) > 1 ? o.Children.only(null) : o.isValidElement(f) ? f.props.children : null : y);
      return /* @__PURE__ */ u(r, { ...s, ref: e, children: o.isValidElement(f) ? o.cloneElement(f, void 0, m) : null });
    }
    return /* @__PURE__ */ u(r, { ...s, ref: e, children: i });
  });
  return n.displayName = `${t}.Slot`, n;
}
var I = /* @__PURE__ */ C("Slot");
// @__NO_SIDE_EFFECTS__
function R(t) {
  const r = o.forwardRef((n, l) => {
    let { children: e, ...i } = n;
    if (d(e) && typeof p == "function" && (e = p(e._payload)), o.isValidElement(e)) {
      const s = j(e), a = $(i, e.props);
      return e.type !== o.Fragment && (a.ref = l ? b(l, s) : s), o.cloneElement(e, a);
    }
    return o.Children.count(e) > 1 ? o.Children.only(null) : null;
  });
  return r.displayName = `${t}.SlotClone`, r;
}
var S = /* @__PURE__ */ Symbol("radix.slottable");
// @__NO_SIDE_EFFECTS__
function h(t) {
  const r = ({ children: n }) => /* @__PURE__ */ u(g, { children: n });
  return r.displayName = `${t}.Slottable`, r.__radixId = S, r;
}
var V = /* @__PURE__ */ h("Slottable");
function P(t) {
  return o.isValidElement(t) && typeof t.type == "function" && "__radixId" in t.type && t.type.__radixId === S;
}
function $(t, r) {
  const n = { ...r };
  for (const l in r) {
    const e = t[l], i = r[l];
    /^on[A-Z]/.test(l) ? e && i ? n[l] = (...a) => {
      const c = i(...a);
      return e(...a), c;
    } : e && (n[l] = e) : l === "style" ? n[l] = { ...e, ...i } : l === "className" && (n[l] = [e, i].filter(Boolean).join(" "));
  }
  return { ...t, ...n };
}
function j(t) {
  let r = Object.getOwnPropertyDescriptor(t.props, "ref")?.get, n = r && "isReactWarning" in r && r.isReactWarning;
  return n ? t.ref : (r = Object.getOwnPropertyDescriptor(t, "ref")?.get, n = r && "isReactWarning" in r && r.isReactWarning, n ? t.props.ref : t.props.ref || t.ref);
}
export {
  I as S,
  V as a,
  C as c
};
