// @btcv/ui/dist/_shared/index-CukUn3R0.js
import * as r from "react";
import { P as j, c as v, a as We, d as le } from "./index-CXeb1OMI.js";
import { c as de, a as Ye, I as Xe, R as He, u as $e } from "./index-C0gNQvxa.js";
import { c as fe, u as N } from "./index-DLcqcWxM.js";
import { P as ze, D as Ze } from "./index-B03TCNO5.js";
import { h as qe, u as Je, R as Qe, F as et } from "./Combination-D_l4PLF_.js";
import { c as pe, C as tt, A as nt, R as ot, a as rt } from "./index-uu9PT5Nu.js";
import { P as k, d as ct } from "./index-V-Ajw7Ac.js";
import { jsx as u } from "react/jsx-runtime";
// @__NO_SIDE_EFFECTS__
function at(e) {
  const n = /* @__PURE__ */ st(e), t = r.forwardRef((o, c) => {
    const { children: a, ...l } = o, f = r.Children.toArray(a), m = f.find(it);
    if (m) {
      const p = m.props.children, d = f.map((s) => s === m ? r.Children.count(p) > 1 ? r.Children.only(null) : r.isValidElement(p) ? p.props.children : null : s);
      return /* @__PURE__ */ u(n, { ...l, ref: c, children: r.isValidElement(p) ? r.cloneElement(p, void 0, d) : null });
    }
    return /* @__PURE__ */ u(n, { ...l, ref: c, children: a });
  });
  return t.displayName = `${e}.Slot`, t;
}
// @__NO_SIDE_EFFECTS__
function st(e) {
  const n = r.forwardRef((t, o) => {
    const { children: c, ...a } = t;
    if (r.isValidElement(c)) {
      const l = dt(c), f = lt(a, c.props);
      return c.type !== r.Fragment && (f.ref = o ? fe(o, l) : l), r.cloneElement(c, f);
    }
    return r.Children.count(c) > 1 ? r.Children.only(null) : null;
  });
  return n.displayName = `${e}.SlotClone`, n;
}
var ut = /* @__PURE__ */ Symbol("radix.slottable");
function it(e) {
  return r.isValidElement(e) && typeof e.type == "function" && "__radixId" in e.type && e.type.__radixId === ut;
}
function lt(e, n) {
  const t = { ...n };
  for (const o in n) {
    const c = e[o], a = n[o];
    /^on[A-Z]/.test(o) ? c && a ? t[o] = (...f) => {
      const m = a(...f);
      return c(...f), m;
    } : c && (t[o] = c) : o === "style" ? t[o] = { ...c, ...a } : o === "className" && (t[o] = [c, a].filter(Boolean).join(" "));
  }
  return { ...e, ...t };
}
function dt(e) {
  let n = Object.getOwnPropertyDescriptor(e.props, "ref")?.get, t = n && "isReactWarning" in n && n.isReactWarning;
  return t ? e.ref : (n = Object.getOwnPropertyDescriptor(e, "ref")?.get, t = n && "isReactWarning" in n && n.isReactWarning, t ? e.props.ref : e.props.ref || e.ref);
}
var Z = ["Enter", " "], ft = ["ArrowDown", "PageUp", "Home"], me = ["ArrowUp", "PageDown", "End"], pt = [...ft, ...me], mt = {
  ltr: [...Z, "ArrowRight"],
  rtl: [...Z, "ArrowLeft"]
}, ht = {
  ltr: ["ArrowLeft"],
  rtl: ["ArrowRight"]
}, F = "Menu", [O, Ct, vt] = Ye(F), [E, $t] = We(F, [
  vt,
  pe,
  de
]), W = pe(), he = de(), [Mt, P] = E(F), [gt, L] = E(F), Ce = (e) => {
  const { __scopeMenu: n, open: t = !1, children: o, dir: c, onOpenChange: a, modal: l = !0 } = e, f = W(n), [m, p] = r.useState(null), d = r.useRef(!1), s = le(a), h = $e(c);
  return r.useEffect(() => {
    const g = () => {
      d.current = !0, document.addEventListener("pointerdown", C, { capture: !0, once: !0 }), document.addEventListener("pointermove", C, { capture: !0, once: !0 });
    }, C = () => d.current = !1;
    return document.addEventListener("keydown", g, { capture: !0 }), () => {
      document.removeEventListener("keydown", g, { capture: !0 }), document.removeEventListener("pointerdown", C, { capture: !0 }), document.removeEventListener("pointermove", C, { capture: !0 });
    };
  }, []), /* @__PURE__ */ u(ot, { ...f, children: /* @__PURE__ */ u(
    Mt,
    {
      scope: n,
      open: t,
      onOpenChange: s,
      content: m,
      onContentChange: p,
      children: /* @__PURE__ */ u(
        gt,
        {
          scope: n,
          onClose: r.useCallback(() => s(!1), [s]),
          isUsingKeyboardRef: d,
          dir: h,
          modal: l,
          children: o
        }
      )
    }
  ) });
};
Ce.displayName = F;
var _t = "MenuAnchor", q = r.forwardRef(
  (e, n) => {
    const { __scopeMenu: t, ...o } = e, c = W(t);
    return /* @__PURE__ */ u(nt, { ...c, ...o, ref: n });
  }
);
q.displayName = _t;
var J = "MenuPortal", [Rt, ve] = E(J, {
  forceMount: void 0
}), Me = (e) => {
  const { __scopeMenu: n, forceMount: t, children: o, container: c } = e, a = P(J, n);
  return /* @__PURE__ */ u(Rt, { scope: n, forceMount: t, children: /* @__PURE__ */ u(j, { present: t || a.open, children: /* @__PURE__ */ u(ze, { asChild: !0, container: c, children: o }) }) });
};
Me.displayName = J;
var _ = "MenuContent", [Et, Q] = E(_), ge = r.forwardRef(
  (e, n) => {
    const t = ve(_, e.__scopeMenu), { forceMount: o = t.forceMount, ...c } = e, a = P(_, e.__scopeMenu), l = L(_, e.__scopeMenu);
    return /* @__PURE__ */ u(O.Provider, { scope: e.__scopeMenu, children: /* @__PURE__ */ u(j, { present: o || a.open, children: /* @__PURE__ */ u(O.Slot, { scope: e.__scopeMenu, children: l.modal ? /* @__PURE__ */ u(Pt, { ...c, ref: n }) : /* @__PURE__ */ u(St, { ...c, ref: n }) }) }) });
  }
), Pt = r.forwardRef(
  (e, n) => {
    const t = P(_, e.__scopeMenu), o = r.useRef(null), c = N(n, o);
    return r.useEffect(() => {
      const a = o.current;
      if (a) return qe(a);
    }, []), /* @__PURE__ */ u(
      ee,
      {
        ...e,
        ref: c,
        trapFocus: t.open,
        disableOutsidePointerEvents: t.open,
        disableOutsideScroll: !0,
        onFocusOutside: v(
          e.onFocusOutside,
          (a) => a.preventDefault(),
          { checkForDefaultPrevented: !1 }
        ),
        onDismiss: () => t.onOpenChange(!1)
      }
    );
  }
), St = r.forwardRef((e, n) => {
  const t = P(_, e.__scopeMenu);
  return /* @__PURE__ */ u(
    ee,
    {
      ...e,
      ref: n,
      trapFocus: !1,
      disableOutsidePointerEvents: !1,
      disableOutsideScroll: !1,
      onDismiss: () => t.onOpenChange(!1)
    }
  );
}), yt = /* @__PURE__ */ at("MenuContent.ScrollLock"), ee = r.forwardRef(
  (e, n) => {
    const {
      __scopeMenu: t,
      loop: o = !1,
      trapFocus: c,
      onOpenAutoFocus: a,
      onCloseAutoFocus: l,
      disableOutsidePointerEvents: f,
      onEntryFocus: m,
      onEscapeKeyDown: p,
      onPointerDownOutside: d,
      onFocusOutside: s,
      onInteractOutside: h,
      onDismiss: g,
      disableOutsideScroll: C,
      ...S
    } = e, b = P(_, t), y = L(_, t), Fe = W(t), Le = he(t), re = Ct(t), [Ke, ce] = r.useState(null), K = r.useRef(null), Ge = N(n, K, b.onContentChange), G = r.useRef(0), U = r.useRef(""), Ue = r.useRef(0), X = r.useRef(null), ae = r.useRef("right"), H = r.useRef(0), Ve = C ? Qe : r.Fragment, Be = C ? { as: yt, allowPinchZoom: !0 } : void 0, je = (i) => {
      const w = U.current + i, R = re().filter((M) => !M.disabled), x = document.activeElement, $ = R.find((M) => M.ref.current === x)?.textValue, z = R.map((M) => M.textValue), se = Lt(z, w, $), T = R.find((M) => M.textValue === se)?.ref.current;
      (function M(ue) {
        U.current = ue, window.clearTimeout(G.current), ue !== "" && (G.current = window.setTimeout(() => M(""), 1e3));
      })(w), T && setTimeout(() => T.focus());
    };
    r.useEffect(() => () => window.clearTimeout(G.current), []), Je();
    const I = r.useCallback((i) => ae.current === X.current?.side && Gt(i, X.current?.area), []);
    return /* @__PURE__ */ u(
      Et,
      {
        scope: t,
        searchRef: U,
        onItemEnter: r.useCallback(
          (i) => {
            I(i) && i.preventDefault();
          },
          [I]
        ),
        onItemLeave: r.useCallback(
          (i) => {
            I(i) || (K.current?.focus(), ce(null));
          },
          [I]
        ),
        onTriggerLeave: r.useCallback(
          (i) => {
            I(i) && i.preventDefault();
          },
          [I]
        ),
        pointerGraceTimerRef: Ue,
        onPointerGraceIntentChange: r.useCallback((i) => {
          X.current = i;
        }, []),
        children: /* @__PURE__ */ u(Ve, { ...Be, children: /* @__PURE__ */ u(
          et,
          {
            asChild: !0,
            trapped: c,
            onMountAutoFocus: v(a, (i) => {
              i.preventDefault(), K.current?.focus({ preventScroll: !0 });
            }),
            onUnmountAutoFocus: l,
            children: /* @__PURE__ */ u(
              Ze,
              {
                asChild: !0,
                disableOutsidePointerEvents: f,
                onEscapeKeyDown: p,
                onPointerDownOutside: d,
                onFocusOutside: s,
                onInteractOutside: h,
                onDismiss: g,
                children: /* @__PURE__ */ u(
                  He,
                  {
                    asChild: !0,
                    ...Le,
                    dir: y.dir,
                    orientation: "vertical",
                    loop: o,
                    currentTabStopId: Ke,
                    onCurrentTabStopIdChange: ce,
                    onEntryFocus: v(m, (i) => {
                      y.isUsingKeyboardRef.current || i.preventDefault();
                    }),
                    preventScrollOnEntryFocus: !0,
                    children: /* @__PURE__ */ u(
                      tt,
                      {
                        role: "menu",
                        "aria-orientation": "vertical",
                        "data-state": ke(b.open),
                        "data-radix-menu-content": "",
                        dir: y.dir,
                        ...Fe,
                        ...S,
                        ref: Ge,
                        style: { outline: "none", ...S.style },
                        onKeyDown: v(S.onKeyDown, (i) => {
                          const R = i.target.closest("[data-radix-menu-content]") === i.currentTarget, x = i.ctrlKey || i.altKey || i.metaKey, $ = i.key.length === 1;
                          R && (i.key === "Tab" && i.preventDefault(), !x && $ && je(i.key));
                          const z = K.current;
                          if (i.target !== z || !pt.includes(i.key)) return;
                          i.preventDefault();
                          const T = re().filter((M) => !M.disabled).map((M) => M.ref.current);
                          me.includes(i.key) && T.reverse(), kt(T);
                        }),
                        onBlur: v(e.onBlur, (i) => {
                          i.currentTarget.contains(i.target) || (window.clearTimeout(G.current), U.current = "");
                        }),
                        onPointerMove: v(
                          e.onPointerMove,
                          D((i) => {
                            const w = i.target, R = H.current !== i.clientX;
                            if (i.currentTarget.contains(w) && R) {
                              const x = i.clientX > H.current ? "right" : "left";
                              ae.current = x, H.current = i.clientX;
                            }
                          })
                        )
                      }
                    )
                  }
                )
              }
            )
          }
        ) })
      }
    );
  }
);
ge.displayName = _;
var It = "MenuGroup", te = r.forwardRef(
  (e, n) => {
    const { __scopeMenu: t, ...o } = e;
    return /* @__PURE__ */ u(k.div, { role: "group", ...o, ref: n });
  }
);
te.displayName = It;
var wt = "MenuLabel", _e = r.forwardRef(
  (e, n) => {
    const { __scopeMenu: t, ...o } = e;
    return /* @__PURE__ */ u(k.div, { ...o, ref: n });
  }
);
_e.displayName = wt;
var V = "MenuItem", ie = "menu.itemSelect", Y = r.forwardRef(
  (e, n) => {
    const { disabled: t = !1, onSelect: o, ...c } = e, a = r.useRef(null), l = L(V, e.__scopeMenu), f = Q(V, e.__scopeMenu), m = N(n, a), p = r.useRef(!1), d = () => {
      const s = a.current;
      if (!t && s) {
        const h = new CustomEvent(ie, { bubbles: !0, cancelable: !0 });
        s.addEventListener(ie, (g) => o?.(g), { once: !0 }), ct(s, h), h.defaultPrevented ? p.current = !1 : l.onClose();
      }
    };
    return /* @__PURE__ */ u(
      Re,
      {
        ...c,
        ref: m,
        disabled: t,
        onClick: v(e.onClick, d),
        onPointerDown: (s) => {
          e.onPointerDown?.(s), p.current = !0;
        },
        onPointerUp: v(e.onPointerUp, (s) => {
          p.current || s.currentTarget?.click();
        }),
        onKeyDown: v(e.onKeyDown, (s) => {
          const h = f.searchRef.current !== "";
          t || h && s.key === " " || Z.includes(s.key) && (s.currentTarget.click(), s.preventDefault());
        })
      }
    );
  }
);
Y.displayName = V;
var Re = r.forwardRef(
  (e, n) => {
    const { __scopeMenu: t, disabled: o = !1, textValue: c, ...a } = e, l = Q(V, t), f = he(t), m = r.useRef(null), p = N(n, m), [d, s] = r.useState(!1), [h, g] = r.useState("");
    return r.useEffect(() => {
      const C = m.current;
      C && g((C.textContent ?? "").trim());
    }, [a.children]), /* @__PURE__ */ u(
      O.ItemSlot,
      {
        scope: t,
        disabled: o,
        textValue: c ?? h,
        children: /* @__PURE__ */ u(Xe, { asChild: !0, ...f, focusable: !o, children: /* @__PURE__ */ u(
          k.div,
          {
            role: "menuitem",
            "data-highlighted": d ? "" : void 0,
            "aria-disabled": o || void 0,
            "data-disabled": o ? "" : void 0,
            ...a,
            ref: p,
            onPointerMove: v(
              e.onPointerMove,
              D((C) => {
                o ? l.onItemLeave(C) : (l.onItemEnter(C), C.defaultPrevented || C.currentTarget.focus({ preventScroll: !0 }));
              })
            ),
            onPointerLeave: v(
              e.onPointerLeave,
              D((C) => l.onItemLeave(C))
            ),
            onFocus: v(e.onFocus, () => s(!0)),
            onBlur: v(e.onBlur, () => s(!1))
          }
        ) })
      }
    );
  }
), bt = "MenuCheckboxItem", Ee = r.forwardRef(
  (e, n) => {
    const { checked: t = !1, onCheckedChange: o, ...c } = e;
    return /* @__PURE__ */ u(we, { scope: e.__scopeMenu, checked: t, children: /* @__PURE__ */ u(
      Y,
      {
        role: "menuitemcheckbox",
        "aria-checked": B(t) ? "mixed" : t,
        ...c,
        ref: n,
        "data-state": oe(t),
        onSelect: v(
          c.onSelect,
          () => o?.(B(t) ? !0 : !t),
          { checkForDefaultPrevented: !1 }
        )
      }
    ) });
  }
);
Ee.displayName = bt;
var Pe = "MenuRadioGroup", [xt, Tt] = E(
  Pe,
  { value: void 0, onValueChange: () => {
  } }
), Se = r.forwardRef(
  (e, n) => {
    const { value: t, onValueChange: o, ...c } = e, a = le(o);
    return /* @__PURE__ */ u(xt, { scope: e.__scopeMenu, value: t, onValueChange: a, children: /* @__PURE__ */ u(te, { ...c, ref: n }) });
  }
);
Se.displayName = Pe;
var ye = "MenuRadioItem", Ie = r.forwardRef(
  (e, n) => {
    const { value: t, ...o } = e, c = Tt(ye, e.__scopeMenu), a = t === c.value;
    return /* @__PURE__ */ u(we, { scope: e.__scopeMenu, checked: a, children: /* @__PURE__ */ u(
      Y,
      {
        role: "menuitemradio",
        "aria-checked": a,
        ...o,
        ref: n,
        "data-state": oe(a),
        onSelect: v(
          o.onSelect,
          () => c.onValueChange?.(t),
          { checkForDefaultPrevented: !1 }
        )
      }
    ) });
  }
);
Ie.displayName = ye;
var ne = "MenuItemIndicator", [we, At] = E(
  ne,
  { checked: !1 }
), be = r.forwardRef(
  (e, n) => {
    const { __scopeMenu: t, forceMount: o, ...c } = e, a = At(ne, t);
    return /* @__PURE__ */ u(
      j,
      {
        present: o || B(a.checked) || a.checked === !0,
        children: /* @__PURE__ */ u(
          k.span,
          {
            ...c,
            ref: n,
            "data-state": oe(a.checked)
          }
        )
      }
    );
  }
);
be.displayName = ne;
var Ot = "MenuSeparator", xe = r.forwardRef(
  (e, n) => {
    const { __scopeMenu: t, ...o } = e;
    return /* @__PURE__ */ u(
      k.div,
      {
        role: "separator",
        "aria-orientation": "horizontal",
        ...o,
        ref: n
      }
    );
  }
);
xe.displayName = Ot;
var Dt = "MenuArrow", Te = r.forwardRef(
  (e, n) => {
    const { __scopeMenu: t, ...o } = e, c = W(t);
    return /* @__PURE__ */ u(rt, { ...c, ...o, ref: n });
  }
);
Te.displayName = Dt;
var Nt = "MenuSub", [zt, Ae] = E(Nt), A = "MenuSubTrigger", Oe = r.forwardRef(
  (e, n) => {
    const t = P(A, e.__scopeMenu), o = L(A, e.__scopeMenu), c = Ae(A, e.__scopeMenu), a = Q(A, e.__scopeMenu), l = r.useRef(null), { pointerGraceTimerRef: f, onPointerGraceIntentChange: m } = a, p = { __scopeMenu: e.__scopeMenu }, d = r.useCallback(() => {
      l.current && window.clearTimeout(l.current), l.current = null;
    }, []);
    return r.useEffect(() => d, [d]), r.useEffect(() => {
      const s = f.current;
      return () => {
        window.clearTimeout(s), m(null);
      };
    }, [f, m]), /* @__PURE__ */ u(q, { asChild: !0, ...p, children: /* @__PURE__ */ u(
      Re,
      {
        id: c.triggerId,
        "aria-haspopup": "menu",
        "aria-expanded": t.open,
        "aria-controls": c.contentId,
        "data-state": ke(t.open),
        ...e,
        ref: fe(n, c.onTriggerChange),
        onClick: (s) => {
          e.onClick?.(s), !(e.disabled || s.defaultPrevented) && (s.currentTarget.focus(), t.open || t.onOpenChange(!0));
        },
        onPointerMove: v(
          e.onPointerMove,
          D((s) => {
            a.onItemEnter(s), !s.defaultPrevented && !e.disabled && !t.open && !l.current && (a.onPointerGraceIntentChange(null), l.current = window.setTimeout(() => {
              t.onOpenChange(!0), d();
            }, 100));
          })
        ),
        onPointerLeave: v(
          e.onPointerLeave,
          D((s) => {
            d();
            const h = t.content?.getBoundingClientRect();
            if (h) {
              const g = t.content?.dataset.side, C = g === "right", S = C ? -5 : 5, b = h[C ? "left" : "right"], y = h[C ? "right" : "left"];
              a.onPointerGraceIntentChange({
                area: [
                  // Apply a bleed on clientX to ensure that our exit point is
                  // consistently within polygon bounds
                  { x: s.clientX + S, y: s.clientY },
                  { x: b, y: h.top },
                  { x: y, y: h.top },
                  { x: y, y: h.bottom },
                  { x: b, y: h.bottom }
                ],
                side: g
              }), window.clearTimeout(f.current), f.current = window.setTimeout(
                () => a.onPointerGraceIntentChange(null),
                300
              );
            } else {
              if (a.onTriggerLeave(s), s.defaultPrevented) return;
              a.onPointerGraceIntentChange(null);
            }
          })
        ),
        onKeyDown: v(e.onKeyDown, (s) => {
          const h = a.searchRef.current !== "";
          e.disabled || h && s.key === " " || mt[o.dir].includes(s.key) && (t.onOpenChange(!0), t.content?.focus(), s.preventDefault());
        })
      }
    ) });
  }
);
Oe.displayName = A;
var De = "MenuSubContent", Ne = r.forwardRef(
  (e, n) => {
    const t = ve(_, e.__scopeMenu), { forceMount: o = t.forceMount, ...c } = e, a = P(_, e.__scopeMenu), l = L(_, e.__scopeMenu), f = Ae(De, e.__scopeMenu), m = r.useRef(null), p = N(n, m);
    return /* @__PURE__ */ u(O.Provider, { scope: e.__scopeMenu, children: /* @__PURE__ */ u(j, { present: o || a.open, children: /* @__PURE__ */ u(O.Slot, { scope: e.__scopeMenu, children: /* @__PURE__ */ u(
      ee,
      {
        id: f.contentId,
        "aria-labelledby": f.triggerId,
        ...c,
        ref: p,
        align: "start",
        side: l.dir === "rtl" ? "left" : "right",
        disableOutsidePointerEvents: !1,
        disableOutsideScroll: !1,
        trapFocus: !1,
        onOpenAutoFocus: (d) => {
          l.isUsingKeyboardRef.current && m.current?.focus(), d.preventDefault();
        },
        onCloseAutoFocus: (d) => d.preventDefault(),
        onFocusOutside: v(e.onFocusOutside, (d) => {
          d.target !== f.trigger && a.onOpenChange(!1);
        }),
        onEscapeKeyDown: v(e.onEscapeKeyDown, (d) => {
          l.onClose(), d.preventDefault();
        }),
        onKeyDown: v(e.onKeyDown, (d) => {
          const s = d.currentTarget.contains(d.target), h = ht[l.dir].includes(d.key);
          s && h && (a.onOpenChange(!1), f.trigger?.focus(), d.preventDefault());
        })
      }
    ) }) }) });
  }
);
Ne.displayName = De;
function ke(e) {
  return e ? "open" : "closed";
}
function B(e) {
  return e === "indeterminate";
}
function oe(e) {
  return B(e) ? "indeterminate" : e ? "checked" : "unchecked";
}
function kt(e) {
  const n = document.activeElement;
  for (const t of e)
    if (t === n || (t.focus(), document.activeElement !== n)) return;
}
function Ft(e, n) {
  return e.map((t, o) => e[(n + o) % e.length]);
}
function Lt(e, n, t) {
  const c = n.length > 1 && Array.from(n).every((p) => p === n[0]) ? n[0] : n, a = t ? e.indexOf(t) : -1;
  let l = Ft(e, Math.max(a, 0));
  c.length === 1 && (l = l.filter((p) => p !== t));
  const m = l.find(
    (p) => p.toLowerCase().startsWith(c.toLowerCase())
  );
  return m !== t ? m : void 0;
}
function Kt(e, n) {
  const { x: t, y: o } = e;
  let c = !1;
  for (let a = 0, l = n.length - 1; a < n.length; l = a++) {
    const f = n[a], m = n[l], p = f.x, d = f.y, s = m.x, h = m.y;
    d > o != h > o && t < (s - p) * (o - d) / (h - d) + p && (c = !c);
  }
  return c;
}
function Gt(e, n) {
  if (!n) return !1;
  const t = { x: e.clientX, y: e.clientY };
  return Kt(t, n);
}
function D(e) {
  return (n) => n.pointerType === "mouse" ? e(n) : void 0;
}
var Zt = Ce, qt = q, Jt = Me, Qt = ge, en = te, tn = _e, nn = Y, on = Ee, rn = Se, cn = Ie, an = be, sn = xe, un = Te, ln = Oe, dn = Ne;
export {
  qt as A,
  Qt as C,
  en as G,
  nn as I,
  tn as L,
  Jt as P,
  Zt as R,
  sn as S,
  on as a,
  rn as b,
  $t as c,
  cn as d,
  an as e,
  un as f,
  ln as g,
  dn as h
};
