// @btcv/ui/dist/_shared/index-uu9PT5Nu.js
import * as O from "react";
import { useLayoutEffect as Re } from "react";
import * as Oe from "react-dom";
import { P as Ot } from "./index-V-Ajw7Ac.js";
import { jsx as Y } from "react/jsx-runtime";
import { u as jt } from "./index-DLcqcWxM.js";
import { b as vt, a as Ce, d as Pe } from "./index-CXeb1OMI.js";
function Se(t) {
  const [e, n] = O.useState(void 0);
  return vt(() => {
    if (t) {
      n({ width: t.offsetWidth, height: t.offsetHeight });
      const o = new ResizeObserver((r) => {
        if (!Array.isArray(r) || !r.length)
          return;
        const i = r[0];
        let s, c;
        if ("borderBoxSize" in i) {
          const l = i.borderBoxSize, f = Array.isArray(l) ? l[0] : l;
          s = f.inlineSize, c = f.blockSize;
        } else
          s = t.offsetWidth, c = t.offsetHeight;
        n({ width: s, height: c });
      });
      return o.observe(t, { box: "border-box" }), () => o.unobserve(t);
    } else
      n(void 0);
  }, [t]), e;
}
const Ee = ["top", "right", "bottom", "left"], K = Math.min, F = Math.max, at = Math.round, ct = Math.floor, I = (t) => ({
  x: t,
  y: t
}), De = {
  left: "right",
  right: "left",
  bottom: "top",
  top: "bottom"
};
function bt(t, e, n) {
  return F(t, K(e, n));
}
function j(t, e) {
  return typeof t == "function" ? t(e) : t;
}
function q(t) {
  return t.split("-")[0];
}
function et(t) {
  return t.split("-")[1];
}
function Ct(t) {
  return t === "x" ? "y" : "x";
}
function Pt(t) {
  return t === "y" ? "height" : "width";
}
function V(t) {
  const e = t[0];
  return e === "t" || e === "b" ? "y" : "x";
}
function St(t) {
  return Ct(V(t));
}
function Le(t, e, n) {
  n === void 0 && (n = !1);
  const o = et(t), r = St(t), i = Pt(r);
  let s = r === "x" ? o === (n ? "end" : "start") ? "right" : "left" : o === "start" ? "bottom" : "top";
  return e.reference[i] > e.floating[i] && (s = ft(s)), [s, ft(s)];
}
function Me(t) {
  const e = ft(t);
  return [At(t), e, At(e)];
}
function At(t) {
  return t.includes("start") ? t.replace("start", "end") : t.replace("end", "start");
}
const Wt = ["left", "right"], kt = ["right", "left"], Te = ["top", "bottom"], $e = ["bottom", "top"];
function Ne(t, e, n) {
  switch (t) {
    case "top":
    case "bottom":
      return n ? e ? kt : Wt : e ? Wt : kt;
    case "left":
    case "right":
      return e ? Te : $e;
    default:
      return [];
  }
}
function We(t, e, n, o) {
  const r = et(t);
  let i = Ne(q(t), n === "start", o);
  return r && (i = i.map((s) => s + "-" + r), e && (i = i.concat(i.map(At)))), i;
}
function ft(t) {
  const e = q(t);
  return De[e] + t.slice(e.length);
}
function ke(t) {
  return {
    top: 0,
    right: 0,
    bottom: 0,
    left: 0,
    ...t
  };
}
function qt(t) {
  return typeof t != "number" ? ke(t) : {
    top: t,
    right: t,
    bottom: t,
    left: t
  };
}
function ut(t) {
  const {
    x: e,
    y: n,
    width: o,
    height: r
  } = t;
  return {
    width: o,
    height: r,
    top: n,
    left: e,
    right: e + o,
    bottom: n + r,
    x: e,
    y: n
  };
}
function Ft(t, e, n) {
  let {
    reference: o,
    floating: r
  } = t;
  const i = V(e), s = St(e), c = Pt(s), l = q(e), f = i === "y", a = o.x + o.width / 2 - r.width / 2, d = o.y + o.height / 2 - r.height / 2, m = o[c] / 2 - r[c] / 2;
  let u;
  switch (l) {
    case "top":
      u = {
        x: a,
        y: o.y - r.height
      };
      break;
    case "bottom":
      u = {
        x: a,
        y: o.y + o.height
      };
      break;
    case "right":
      u = {
        x: o.x + o.width,
        y: d
      };
      break;
    case "left":
      u = {
        x: o.x - r.width,
        y: d
      };
      break;
    default:
      u = {
        x: o.x,
        y: o.y
      };
  }
  switch (et(e)) {
    case "start":
      u[s] -= m * (n && f ? -1 : 1);
      break;
    case "end":
      u[s] += m * (n && f ? -1 : 1);
      break;
  }
  return u;
}
async function Fe(t, e) {
  var n;
  e === void 0 && (e = {});
  const {
    x: o,
    y: r,
    platform: i,
    rects: s,
    elements: c,
    strategy: l
  } = t, {
    boundary: f = "clippingAncestors",
    rootBoundary: a = "viewport",
    elementContext: d = "floating",
    altBoundary: m = !1,
    padding: u = 0
  } = j(e, t), h = qt(u), x = c[m ? d === "floating" ? "reference" : "floating" : d], g = ut(await i.getClippingRect({
    element: (n = await (i.isElement == null ? void 0 : i.isElement(x))) == null || n ? x : x.contextElement || await (i.getDocumentElement == null ? void 0 : i.getDocumentElement(c.floating)),
    boundary: f,
    rootBoundary: a,
    strategy: l
  })), v = d === "floating" ? {
    x: o,
    y: r,
    width: s.floating.width,
    height: s.floating.height
  } : s.reference, y = await (i.getOffsetParent == null ? void 0 : i.getOffsetParent(c.floating)), w = await (i.isElement == null ? void 0 : i.isElement(y)) ? await (i.getScale == null ? void 0 : i.getScale(y)) || {
    x: 1,
    y: 1
  } : {
    x: 1,
    y: 1
  }, A = ut(i.convertOffsetParentRelativeRectToViewportRelativeRect ? await i.convertOffsetParentRelativeRectToViewportRelativeRect({
    elements: c,
    rect: v,
    offsetParent: y,
    strategy: l
  }) : v);
  return {
    top: (g.top - A.top + h.top) / w.y,
    bottom: (A.bottom - g.bottom + h.bottom) / w.y,
    left: (g.left - A.left + h.left) / w.x,
    right: (A.right - g.right + h.right) / w.x
  };
}
const _e = 50, He = async (t, e, n) => {
  const {
    placement: o = "bottom",
    strategy: r = "absolute",
    middleware: i = [],
    platform: s
  } = n, c = s.detectOverflow ? s : {
    ...s,
    detectOverflow: Fe
  }, l = await (s.isRTL == null ? void 0 : s.isRTL(e));
  let f = await s.getElementRects({
    reference: t,
    floating: e,
    strategy: r
  }), {
    x: a,
    y: d
  } = Ft(f, o, l), m = o, u = 0;
  const h = {};
  for (let p = 0; p < i.length; p++) {
    const x = i[p];
    if (!x)
      continue;
    const {
      name: g,
      fn: v
    } = x, {
      x: y,
      y: w,
      data: A,
      reset: R
    } = await v({
      x: a,
      y: d,
      initialPlacement: o,
      placement: m,
      strategy: r,
      middlewareData: h,
      rects: f,
      platform: c,
      elements: {
        reference: t,
        floating: e
      }
    });
    a = y ?? a, d = w ?? d, h[g] = {
      ...h[g],
      ...A
    }, R && u < _e && (u++, typeof R == "object" && (R.placement && (m = R.placement), R.rects && (f = R.rects === !0 ? await s.getElementRects({
      reference: t,
      floating: e,
      strategy: r
    }) : R.rects), {
      x: a,
      y: d
    } = Ft(f, m, l)), p = -1);
  }
  return {
    x: a,
    y: d,
    placement: m,
    strategy: r,
    middlewareData: h
  };
}, ze = (t) => ({
  name: "arrow",
  options: t,
  async fn(e) {
    const {
      x: n,
      y: o,
      placement: r,
      rects: i,
      platform: s,
      elements: c,
      middlewareData: l
    } = e, {
      element: f,
      padding: a = 0
    } = j(t, e) || {};
    if (f == null)
      return {};
    const d = qt(a), m = {
      x: n,
      y: o
    }, u = St(r), h = Pt(u), p = await s.getDimensions(f), x = u === "y", g = x ? "top" : "left", v = x ? "bottom" : "right", y = x ? "clientHeight" : "clientWidth", w = i.reference[h] + i.reference[u] - m[u] - i.floating[h], A = m[u] - i.reference[u], R = await (s.getOffsetParent == null ? void 0 : s.getOffsetParent(f));
    let C = R ? R[y] : 0;
    (!C || !await (s.isElement == null ? void 0 : s.isElement(R))) && (C = c.floating[y] || i.floating[h]);
    const M = w / 2 - A / 2, W = C / 2 - p[h] / 2 - 1, L = K(d[g], W), $ = K(d[v], W), N = L, S = C - p[h] - $, P = C / 2 - p[h] / 2 + M, k = bt(N, P, S), E = !l.arrow && et(r) != null && P !== k && i.reference[h] / 2 - (P < N ? L : $) - p[h] / 2 < 0, D = E ? P < N ? P - N : P - S : 0;
    return {
      [u]: m[u] + D,
      data: {
        [u]: k,
        centerOffset: P - k - D,
        ...E && {
          alignmentOffset: D
        }
      },
      reset: E
    };
  }
}), Be = function(t) {
  return t === void 0 && (t = {}), {
    name: "flip",
    options: t,
    async fn(e) {
      var n, o;
      const {
        placement: r,
        middlewareData: i,
        rects: s,
        initialPlacement: c,
        platform: l,
        elements: f
      } = e, {
        mainAxis: a = !0,
        crossAxis: d = !0,
        fallbackPlacements: m,
        fallbackStrategy: u = "bestFit",
        fallbackAxisSideDirection: h = "none",
        flipAlignment: p = !0,
        ...x
      } = j(t, e);
      if ((n = i.arrow) != null && n.alignmentOffset)
        return {};
      const g = q(r), v = V(c), y = q(c) === c, w = await (l.isRTL == null ? void 0 : l.isRTL(f.floating)), A = m || (y || !p ? [ft(c)] : Me(c)), R = h !== "none";
      !m && R && A.push(...We(c, p, h, w));
      const C = [c, ...A], M = await l.detectOverflow(e, x), W = [];
      let L = ((o = i.flip) == null ? void 0 : o.overflows) || [];
      if (a && W.push(M[g]), d) {
        const P = Le(r, s, w);
        W.push(M[P[0]], M[P[1]]);
      }
      if (L = [...L, {
        placement: r,
        overflows: W
      }], !W.every((P) => P <= 0)) {
        var $, N;
        const P = ((($ = i.flip) == null ? void 0 : $.index) || 0) + 1, k = C[P];
        if (k && (!(d === "alignment" ? v !== V(k) : !1) || // We leave the current main axis only if every placement on that axis
        // overflows the main axis.
        L.every((b) => V(b.placement) === v ? b.overflows[0] > 0 : !0)))
          return {
            data: {
              index: P,
              overflows: L
            },
            reset: {
              placement: k
            }
          };
        let E = (N = L.filter((D) => D.overflows[0] <= 0).sort((D, b) => D.overflows[1] - b.overflows[1])[0]) == null ? void 0 : N.placement;
        if (!E)
          switch (u) {
            case "bestFit": {
              var S;
              const D = (S = L.filter((b) => {
                if (R) {
                  const T = V(b.placement);
                  return T === v || // Create a bias to the `y` side axis due to horizontal
                  // reading directions favoring greater width.
                  T === "y";
                }
                return !0;
              }).map((b) => [b.placement, b.overflows.filter((T) => T > 0).reduce((T, B) => T + B, 0)]).sort((b, T) => b[1] - T[1])[0]) == null ? void 0 : S[0];
              D && (E = D);
              break;
            }
            case "initialPlacement":
              E = c;
              break;
          }
        if (r !== E)
          return {
            reset: {
              placement: E
            }
          };
      }
      return {};
    }
  };
};
function _t(t, e) {
  return {
    top: t.top - e.height,
    right: t.right - e.width,
    bottom: t.bottom - e.height,
    left: t.left - e.width
  };
}
function Ht(t) {
  return Ee.some((e) => t[e] >= 0);
}
const Ve = function(t) {
  return t === void 0 && (t = {}), {
    name: "hide",
    options: t,
    async fn(e) {
      const {
        rects: n,
        platform: o
      } = e, {
        strategy: r = "referenceHidden",
        ...i
      } = j(t, e);
      switch (r) {
        case "referenceHidden": {
          const s = await o.detectOverflow(e, {
            ...i,
            elementContext: "reference"
          }), c = _t(s, n.reference);
          return {
            data: {
              referenceHiddenOffsets: c,
              referenceHidden: Ht(c)
            }
          };
        }
        case "escaped": {
          const s = await o.detectOverflow(e, {
            ...i,
            altBoundary: !0
          }), c = _t(s, n.floating);
          return {
            data: {
              escapedOffsets: c,
              escaped: Ht(c)
            }
          };
        }
        default:
          return {};
      }
    }
  };
}, Ut = /* @__PURE__ */ new Set(["left", "top"]);
async function Ie(t, e) {
  const {
    placement: n,
    platform: o,
    elements: r
  } = t, i = await (o.isRTL == null ? void 0 : o.isRTL(r.floating)), s = q(n), c = et(n), l = V(n) === "y", f = Ut.has(s) ? -1 : 1, a = i && l ? -1 : 1, d = j(e, t);
  let {
    mainAxis: m,
    crossAxis: u,
    alignmentAxis: h
  } = typeof d == "number" ? {
    mainAxis: d,
    crossAxis: 0,
    alignmentAxis: null
  } : {
    mainAxis: d.mainAxis || 0,
    crossAxis: d.crossAxis || 0,
    alignmentAxis: d.alignmentAxis
  };
  return c && typeof h == "number" && (u = c === "end" ? h * -1 : h), l ? {
    x: u * a,
    y: m * f
  } : {
    x: m * f,
    y: u * a
  };
}
const Xe = function(t) {
  return t === void 0 && (t = 0), {
    name: "offset",
    options: t,
    async fn(e) {
      var n, o;
      const {
        x: r,
        y: i,
        placement: s,
        middlewareData: c
      } = e, l = await Ie(e, t);
      return s === ((n = c.offset) == null ? void 0 : n.placement) && (o = c.arrow) != null && o.alignmentOffset ? {} : {
        x: r + l.x,
        y: i + l.y,
        data: {
          ...l,
          placement: s
        }
      };
    }
  };
}, Ye = function(t) {
  return t === void 0 && (t = {}), {
    name: "shift",
    options: t,
    async fn(e) {
      const {
        x: n,
        y: o,
        placement: r,
        platform: i
      } = e, {
        mainAxis: s = !0,
        crossAxis: c = !1,
        limiter: l = {
          fn: (g) => {
            let {
              x: v,
              y
            } = g;
            return {
              x: v,
              y
            };
          }
        },
        ...f
      } = j(t, e), a = {
        x: n,
        y: o
      }, d = await i.detectOverflow(e, f), m = V(q(r)), u = Ct(m);
      let h = a[u], p = a[m];
      if (s) {
        const g = u === "y" ? "top" : "left", v = u === "y" ? "bottom" : "right", y = h + d[g], w = h - d[v];
        h = bt(y, h, w);
      }
      if (c) {
        const g = m === "y" ? "top" : "left", v = m === "y" ? "bottom" : "right", y = p + d[g], w = p - d[v];
        p = bt(y, p, w);
      }
      const x = l.fn({
        ...e,
        [u]: h,
        [m]: p
      });
      return {
        ...x,
        data: {
          x: x.x - n,
          y: x.y - o,
          enabled: {
            [u]: s,
            [m]: c
          }
        }
      };
    }
  };
}, je = function(t) {
  return t === void 0 && (t = {}), {
    options: t,
    fn(e) {
      const {
        x: n,
        y: o,
        placement: r,
        rects: i,
        middlewareData: s
      } = e, {
        offset: c = 0,
        mainAxis: l = !0,
        crossAxis: f = !0
      } = j(t, e), a = {
        x: n,
        y: o
      }, d = V(r), m = Ct(d);
      let u = a[m], h = a[d];
      const p = j(c, e), x = typeof p == "number" ? {
        mainAxis: p,
        crossAxis: 0
      } : {
        mainAxis: 0,
        crossAxis: 0,
        ...p
      };
      if (l) {
        const y = m === "y" ? "height" : "width", w = i.reference[m] - i.floating[y] + x.mainAxis, A = i.reference[m] + i.reference[y] - x.mainAxis;
        u < w ? u = w : u > A && (u = A);
      }
      if (f) {
        var g, v;
        const y = m === "y" ? "width" : "height", w = Ut.has(q(r)), A = i.reference[d] - i.floating[y] + (w && ((g = s.offset) == null ? void 0 : g[d]) || 0) + (w ? 0 : x.crossAxis), R = i.reference[d] + i.reference[y] + (w ? 0 : ((v = s.offset) == null ? void 0 : v[d]) || 0) - (w ? x.crossAxis : 0);
        h < A ? h = A : h > R && (h = R);
      }
      return {
        [m]: u,
        [d]: h
      };
    }
  };
}, qe = function(t) {
  return t === void 0 && (t = {}), {
    name: "size",
    options: t,
    async fn(e) {
      var n, o;
      const {
        placement: r,
        rects: i,
        platform: s,
        elements: c
      } = e, {
        apply: l = () => {
        },
        ...f
      } = j(t, e), a = await s.detectOverflow(e, f), d = q(r), m = et(r), u = V(r) === "y", {
        width: h,
        height: p
      } = i.floating;
      let x, g;
      d === "top" || d === "bottom" ? (x = d, g = m === (await (s.isRTL == null ? void 0 : s.isRTL(c.floating)) ? "start" : "end") ? "left" : "right") : (g = d, x = m === "end" ? "top" : "bottom");
      const v = p - a.top - a.bottom, y = h - a.left - a.right, w = K(p - a[x], v), A = K(h - a[g], y), R = !e.middlewareData.shift;
      let C = w, M = A;
      if ((n = e.middlewareData.shift) != null && n.enabled.x && (M = y), (o = e.middlewareData.shift) != null && o.enabled.y && (C = v), R && !m) {
        const L = F(a.left, 0), $ = F(a.right, 0), N = F(a.top, 0), S = F(a.bottom, 0);
        u ? M = h - 2 * (L !== 0 || $ !== 0 ? L + $ : F(a.left, a.right)) : C = p - 2 * (N !== 0 || S !== 0 ? N + S : F(a.top, a.bottom));
      }
      await l({
        ...e,
        availableWidth: M,
        availableHeight: C
      });
      const W = await s.getDimensions(c.floating);
      return h !== W.width || p !== W.height ? {
        reset: {
          rects: !0
        }
      } : {};
    }
  };
};
function mt() {
  return typeof window < "u";
}
function nt(t) {
  return Kt(t) ? (t.nodeName || "").toLowerCase() : "#document";
}
function _(t) {
  var e;
  return (t == null || (e = t.ownerDocument) == null ? void 0 : e.defaultView) || window;
}
function X(t) {
  var e;
  return (e = (Kt(t) ? t.ownerDocument : t.document) || window.document) == null ? void 0 : e.documentElement;
}
function Kt(t) {
  return mt() ? t instanceof Node || t instanceof _(t).Node : !1;
}
function H(t) {
  return mt() ? t instanceof Element || t instanceof _(t).Element : !1;
}
function U(t) {
  return mt() ? t instanceof HTMLElement || t instanceof _(t).HTMLElement : !1;
}
function zt(t) {
  return !mt() || typeof ShadowRoot > "u" ? !1 : t instanceof ShadowRoot || t instanceof _(t).ShadowRoot;
}
function it(t) {
  const {
    overflow: e,
    overflowX: n,
    overflowY: o,
    display: r
  } = z(t);
  return /auto|scroll|overlay|hidden|clip/.test(e + o + n) && r !== "inline" && r !== "contents";
}
function Ue(t) {
  return /^(table|td|th)$/.test(nt(t));
}
function ht(t) {
  try {
    if (t.matches(":popover-open"))
      return !0;
  } catch {
  }
  try {
    return t.matches(":modal");
  } catch {
    return !1;
  }
}
const Ke = /transform|translate|scale|rotate|perspective|filter/, Ze = /paint|layout|strict|content/, G = (t) => !!t && t !== "none";
let wt;
function Et(t) {
  const e = H(t) ? z(t) : t;
  return G(e.transform) || G(e.translate) || G(e.scale) || G(e.rotate) || G(e.perspective) || !Dt() && (G(e.backdropFilter) || G(e.filter)) || Ke.test(e.willChange || "") || Ze.test(e.contain || "");
}
function Ge(t) {
  let e = Z(t);
  for (; U(e) && !tt(e); ) {
    if (Et(e))
      return e;
    if (ht(e))
      return null;
    e = Z(e);
  }
  return null;
}
function Dt() {
  return wt == null && (wt = typeof CSS < "u" && CSS.supports && CSS.supports("-webkit-backdrop-filter", "none")), wt;
}
function tt(t) {
  return /^(html|body|#document)$/.test(nt(t));
}
function z(t) {
  return _(t).getComputedStyle(t);
}
function pt(t) {
  return H(t) ? {
    scrollLeft: t.scrollLeft,
    scrollTop: t.scrollTop
  } : {
    scrollLeft: t.scrollX,
    scrollTop: t.scrollY
  };
}
function Z(t) {
  if (nt(t) === "html")
    return t;
  const e = (
    // Step into the shadow DOM of the parent of a slotted node.
    t.assignedSlot || // DOM Element detected.
    t.parentNode || // ShadowRoot detected.
    zt(t) && t.host || // Fallback.
    X(t)
  );
  return zt(e) ? e.host : e;
}
function Zt(t) {
  const e = Z(t);
  return tt(e) ? t.ownerDocument ? t.ownerDocument.body : t.body : U(e) && it(e) ? e : Zt(e);
}
function rt(t, e, n) {
  var o;
  e === void 0 && (e = []), n === void 0 && (n = !0);
  const r = Zt(t), i = r === ((o = t.ownerDocument) == null ? void 0 : o.body), s = _(r);
  if (i) {
    const c = Rt(s);
    return e.concat(s, s.visualViewport || [], it(r) ? r : [], c && n ? rt(c) : []);
  } else
    return e.concat(r, rt(r, [], n));
}
function Rt(t) {
  return t.parent && Object.getPrototypeOf(t.parent) ? t.frameElement : null;
}
function Gt(t) {
  const e = z(t);
  let n = parseFloat(e.width) || 0, o = parseFloat(e.height) || 0;
  const r = U(t), i = r ? t.offsetWidth : n, s = r ? t.offsetHeight : o, c = at(n) !== i || at(o) !== s;
  return c && (n = i, o = s), {
    width: n,
    height: o,
    $: c
  };
}
function Lt(t) {
  return H(t) ? t : t.contextElement;
}
function Q(t) {
  const e = Lt(t);
  if (!U(e))
    return I(1);
  const n = e.getBoundingClientRect(), {
    width: o,
    height: r,
    $: i
  } = Gt(e);
  let s = (i ? at(n.width) : n.width) / o, c = (i ? at(n.height) : n.height) / r;
  return (!s || !Number.isFinite(s)) && (s = 1), (!c || !Number.isFinite(c)) && (c = 1), {
    x: s,
    y: c
  };
}
const Je = /* @__PURE__ */ I(0);
function Jt(t) {
  const e = _(t);
  return !Dt() || !e.visualViewport ? Je : {
    x: e.visualViewport.offsetLeft,
    y: e.visualViewport.offsetTop
  };
}
function Qe(t, e, n) {
  return e === void 0 && (e = !1), !n || e && n !== _(t) ? !1 : e;
}
function J(t, e, n, o) {
  e === void 0 && (e = !1), n === void 0 && (n = !1);
  const r = t.getBoundingClientRect(), i = Lt(t);
  let s = I(1);
  e && (o ? H(o) && (s = Q(o)) : s = Q(t));
  const c = Qe(i, n, o) ? Jt(i) : I(0);
  let l = (r.left + c.x) / s.x, f = (r.top + c.y) / s.y, a = r.width / s.x, d = r.height / s.y;
  if (i) {
    const m = _(i), u = o && H(o) ? _(o) : o;
    let h = m, p = Rt(h);
    for (; p && o && u !== h; ) {
      const x = Q(p), g = p.getBoundingClientRect(), v = z(p), y = g.left + (p.clientLeft + parseFloat(v.paddingLeft)) * x.x, w = g.top + (p.clientTop + parseFloat(v.paddingTop)) * x.y;
      l *= x.x, f *= x.y, a *= x.x, d *= x.y, l += y, f += w, h = _(p), p = Rt(h);
    }
  }
  return ut({
    width: a,
    height: d,
    x: l,
    y: f
  });
}
function gt(t, e) {
  const n = pt(t).scrollLeft;
  return e ? e.left + n : J(X(t)).left + n;
}
function Qt(t, e) {
  const n = t.getBoundingClientRect(), o = n.left + e.scrollLeft - gt(t, n), r = n.top + e.scrollTop;
  return {
    x: o,
    y: r
  };
}
function tn(t) {
  let {
    elements: e,
    rect: n,
    offsetParent: o,
    strategy: r
  } = t;
  const i = r === "fixed", s = X(o), c = e ? ht(e.floating) : !1;
  if (o === s || c && i)
    return n;
  let l = {
    scrollLeft: 0,
    scrollTop: 0
  }, f = I(1);
  const a = I(0), d = U(o);
  if ((d || !d && !i) && ((nt(o) !== "body" || it(s)) && (l = pt(o)), d)) {
    const u = J(o);
    f = Q(o), a.x = u.x + o.clientLeft, a.y = u.y + o.clientTop;
  }
  const m = s && !d && !i ? Qt(s, l) : I(0);
  return {
    width: n.width * f.x,
    height: n.height * f.y,
    x: n.x * f.x - l.scrollLeft * f.x + a.x + m.x,
    y: n.y * f.y - l.scrollTop * f.y + a.y + m.y
  };
}
function en(t) {
  return Array.from(t.getClientRects());
}
function nn(t) {
  const e = X(t), n = pt(t), o = t.ownerDocument.body, r = F(e.scrollWidth, e.clientWidth, o.scrollWidth, o.clientWidth), i = F(e.scrollHeight, e.clientHeight, o.scrollHeight, o.clientHeight);
  let s = -n.scrollLeft + gt(t);
  const c = -n.scrollTop;
  return z(o).direction === "rtl" && (s += F(e.clientWidth, o.clientWidth) - r), {
    width: r,
    height: i,
    x: s,
    y: c
  };
}
const Bt = 25;
function on(t, e) {
  const n = _(t), o = X(t), r = n.visualViewport;
  let i = o.clientWidth, s = o.clientHeight, c = 0, l = 0;
  if (r) {
    i = r.width, s = r.height;
    const a = Dt();
    (!a || a && e === "fixed") && (c = r.offsetLeft, l = r.offsetTop);
  }
  const f = gt(o);
  if (f <= 0) {
    const a = o.ownerDocument, d = a.body, m = getComputedStyle(d), u = a.compatMode === "CSS1Compat" && parseFloat(m.marginLeft) + parseFloat(m.marginRight) || 0, h = Math.abs(o.clientWidth - d.clientWidth - u);
    h <= Bt && (i -= h);
  } else f <= Bt && (i += f);
  return {
    width: i,
    height: s,
    x: c,
    y: l
  };
}
function rn(t, e) {
  const n = J(t, !0, e === "fixed"), o = n.top + t.clientTop, r = n.left + t.clientLeft, i = U(t) ? Q(t) : I(1), s = t.clientWidth * i.x, c = t.clientHeight * i.y, l = r * i.x, f = o * i.y;
  return {
    width: s,
    height: c,
    x: l,
    y: f
  };
}
function Vt(t, e, n) {
  let o;
  if (e === "viewport")
    o = on(t, n);
  else if (e === "document")
    o = nn(X(t));
  else if (H(e))
    o = rn(e, n);
  else {
    const r = Jt(t);
    o = {
      x: e.x - r.x,
      y: e.y - r.y,
      width: e.width,
      height: e.height
    };
  }
  return ut(o);
}
function te(t, e) {
  const n = Z(t);
  return n === e || !H(n) || tt(n) ? !1 : z(n).position === "fixed" || te(n, e);
}
function sn(t, e) {
  const n = e.get(t);
  if (n)
    return n;
  let o = rt(t, [], !1).filter((c) => H(c) && nt(c) !== "body"), r = null;
  const i = z(t).position === "fixed";
  let s = i ? Z(t) : t;
  for (; H(s) && !tt(s); ) {
    const c = z(s), l = Et(s);
    !l && c.position === "fixed" && (r = null), (i ? !l && !r : !l && c.position === "static" && !!r && (r.position === "absolute" || r.position === "fixed") || it(s) && !l && te(t, s)) ? o = o.filter((a) => a !== s) : r = c, s = Z(s);
  }
  return e.set(t, o), o;
}
function cn(t) {
  let {
    element: e,
    boundary: n,
    rootBoundary: o,
    strategy: r
  } = t;
  const s = [...n === "clippingAncestors" ? ht(e) ? [] : sn(e, this._c) : [].concat(n), o], c = Vt(e, s[0], r);
  let l = c.top, f = c.right, a = c.bottom, d = c.left;
  for (let m = 1; m < s.length; m++) {
    const u = Vt(e, s[m], r);
    l = F(u.top, l), f = K(u.right, f), a = K(u.bottom, a), d = F(u.left, d);
  }
  return {
    width: f - d,
    height: a - l,
    x: d,
    y: l
  };
}
function ln(t) {
  const {
    width: e,
    height: n
  } = Gt(t);
  return {
    width: e,
    height: n
  };
}
function an(t, e, n) {
  const o = U(e), r = X(e), i = n === "fixed", s = J(t, !0, i, e);
  let c = {
    scrollLeft: 0,
    scrollTop: 0
  };
  const l = I(0);
  function f() {
    l.x = gt(r);
  }
  if (o || !o && !i)
    if ((nt(e) !== "body" || it(r)) && (c = pt(e)), o) {
      const u = J(e, !0, i, e);
      l.x = u.x + e.clientLeft, l.y = u.y + e.clientTop;
    } else r && f();
  i && !o && r && f();
  const a = r && !o && !i ? Qt(r, c) : I(0), d = s.left + c.scrollLeft - l.x - a.x, m = s.top + c.scrollTop - l.y - a.y;
  return {
    x: d,
    y: m,
    width: s.width,
    height: s.height
  };
}
function xt(t) {
  return z(t).position === "static";
}
function It(t, e) {
  if (!U(t) || z(t).position === "fixed")
    return null;
  if (e)
    return e(t);
  let n = t.offsetParent;
  return X(t) === n && (n = n.ownerDocument.body), n;
}
function ee(t, e) {
  const n = _(t);
  if (ht(t))
    return n;
  if (!U(t)) {
    let r = Z(t);
    for (; r && !tt(r); ) {
      if (H(r) && !xt(r))
        return r;
      r = Z(r);
    }
    return n;
  }
  let o = It(t, e);
  for (; o && Ue(o) && xt(o); )
    o = It(o, e);
  return o && tt(o) && xt(o) && !Et(o) ? n : o || Ge(t) || n;
}
const fn = async function(t) {
  const e = this.getOffsetParent || ee, n = this.getDimensions, o = await n(t.floating);
  return {
    reference: an(t.reference, await e(t.floating), t.strategy),
    floating: {
      x: 0,
      y: 0,
      width: o.width,
      height: o.height
    }
  };
};
function un(t) {
  return z(t).direction === "rtl";
}
const dn = {
  convertOffsetParentRelativeRectToViewportRelativeRect: tn,
  getDocumentElement: X,
  getClippingRect: cn,
  getOffsetParent: ee,
  getElementRects: fn,
  getClientRects: en,
  getDimensions: ln,
  getScale: Q,
  isElement: H,
  isRTL: un
};
function ne(t, e) {
  return t.x === e.x && t.y === e.y && t.width === e.width && t.height === e.height;
}
function mn(t, e) {
  let n = null, o;
  const r = X(t);
  function i() {
    var c;
    clearTimeout(o), (c = n) == null || c.disconnect(), n = null;
  }
  function s(c, l) {
    c === void 0 && (c = !1), l === void 0 && (l = 1), i();
    const f = t.getBoundingClientRect(), {
      left: a,
      top: d,
      width: m,
      height: u
    } = f;
    if (c || e(), !m || !u)
      return;
    const h = ct(d), p = ct(r.clientWidth - (a + m)), x = ct(r.clientHeight - (d + u)), g = ct(a), y = {
      rootMargin: -h + "px " + -p + "px " + -x + "px " + -g + "px",
      threshold: F(0, K(1, l)) || 1
    };
    let w = !0;
    function A(R) {
      const C = R[0].intersectionRatio;
      if (C !== l) {
        if (!w)
          return s();
        C ? s(!1, C) : o = setTimeout(() => {
          s(!1, 1e-7);
        }, 1e3);
      }
      C === 1 && !ne(f, t.getBoundingClientRect()) && s(), w = !1;
    }
    try {
      n = new IntersectionObserver(A, {
        ...y,
        // Handle <iframe>s
        root: r.ownerDocument
      });
    } catch {
      n = new IntersectionObserver(A, y);
    }
    n.observe(t);
  }
  return s(!0), i;
}
function hn(t, e, n, o) {
  o === void 0 && (o = {});
  const {
    ancestorScroll: r = !0,
    ancestorResize: i = !0,
    elementResize: s = typeof ResizeObserver == "function",
    layoutShift: c = typeof IntersectionObserver == "function",
    animationFrame: l = !1
  } = o, f = Lt(t), a = r || i ? [...f ? rt(f) : [], ...e ? rt(e) : []] : [];
  a.forEach((g) => {
    r && g.addEventListener("scroll", n, {
      passive: !0
    }), i && g.addEventListener("resize", n);
  });
  const d = f && c ? mn(f, n) : null;
  let m = -1, u = null;
  s && (u = new ResizeObserver((g) => {
    let [v] = g;
    v && v.target === f && u && e && (u.unobserve(e), cancelAnimationFrame(m), m = requestAnimationFrame(() => {
      var y;
      (y = u) == null || y.observe(e);
    })), n();
  }), f && !l && u.observe(f), e && u.observe(e));
  let h, p = l ? J(t) : null;
  l && x();
  function x() {
    const g = J(t);
    p && !ne(p, g) && n(), p = g, h = requestAnimationFrame(x);
  }
  return n(), () => {
    var g;
    a.forEach((v) => {
      r && v.removeEventListener("scroll", n), i && v.removeEventListener("resize", n);
    }), d?.(), (g = u) == null || g.disconnect(), u = null, l && cancelAnimationFrame(h);
  };
}
const pn = Xe, gn = Ye, wn = Be, xn = qe, yn = Ve, Xt = ze, vn = je, bn = (t, e, n) => {
  const o = /* @__PURE__ */ new Map(), r = {
    platform: dn,
    ...n
  }, i = {
    ...r.platform,
    _c: o
  };
  return He(t, e, {
    ...r,
    platform: i
  });
};
var An = typeof document < "u", Rn = function() {
}, lt = An ? Re : Rn;
function dt(t, e) {
  if (t === e)
    return !0;
  if (typeof t != typeof e)
    return !1;
  if (typeof t == "function" && t.toString() === e.toString())
    return !0;
  let n, o, r;
  if (t && e && typeof t == "object") {
    if (Array.isArray(t)) {
      if (n = t.length, n !== e.length) return !1;
      for (o = n; o-- !== 0; )
        if (!dt(t[o], e[o]))
          return !1;
      return !0;
    }
    if (r = Object.keys(t), n = r.length, n !== Object.keys(e).length)
      return !1;
    for (o = n; o-- !== 0; )
      if (!{}.hasOwnProperty.call(e, r[o]))
        return !1;
    for (o = n; o-- !== 0; ) {
      const i = r[o];
      if (!(i === "_owner" && t.$$typeof) && !dt(t[i], e[i]))
        return !1;
    }
    return !0;
  }
  return t !== t && e !== e;
}
function oe(t) {
  return typeof window > "u" ? 1 : (t.ownerDocument.defaultView || window).devicePixelRatio || 1;
}
function Yt(t, e) {
  const n = oe(t);
  return Math.round(e * n) / n;
}
function yt(t) {
  const e = O.useRef(t);
  return lt(() => {
    e.current = t;
  }), e;
}
function On(t) {
  t === void 0 && (t = {});
  const {
    placement: e = "bottom",
    strategy: n = "absolute",
    middleware: o = [],
    platform: r,
    elements: {
      reference: i,
      floating: s
    } = {},
    transform: c = !0,
    whileElementsMounted: l,
    open: f
  } = t, [a, d] = O.useState({
    x: 0,
    y: 0,
    strategy: n,
    placement: e,
    middlewareData: {},
    isPositioned: !1
  }), [m, u] = O.useState(o);
  dt(m, o) || u(o);
  const [h, p] = O.useState(null), [x, g] = O.useState(null), v = O.useCallback((b) => {
    b !== R.current && (R.current = b, p(b));
  }, []), y = O.useCallback((b) => {
    b !== C.current && (C.current = b, g(b));
  }, []), w = i || h, A = s || x, R = O.useRef(null), C = O.useRef(null), M = O.useRef(a), W = l != null, L = yt(l), $ = yt(r), N = yt(f), S = O.useCallback(() => {
    if (!R.current || !C.current)
      return;
    const b = {
      placement: e,
      strategy: n,
      middleware: m
    };
    $.current && (b.platform = $.current), bn(R.current, C.current, b).then((T) => {
      const B = {
        ...T,
        // The floating element's position may be recomputed while it's closed
        // but still mounted (such as when transitioning out). To ensure
        // `isPositioned` will be `false` initially on the next open, avoid
        // setting it to `true` when `open === false` (must be specified).
        isPositioned: N.current !== !1
      };
      P.current && !dt(M.current, B) && (M.current = B, Oe.flushSync(() => {
        d(B);
      }));
    });
  }, [m, e, n, $, N]);
  lt(() => {
    f === !1 && M.current.isPositioned && (M.current.isPositioned = !1, d((b) => ({
      ...b,
      isPositioned: !1
    })));
  }, [f]);
  const P = O.useRef(!1);
  lt(() => (P.current = !0, () => {
    P.current = !1;
  }), []), lt(() => {
    if (w && (R.current = w), A && (C.current = A), w && A) {
      if (L.current)
        return L.current(w, A, S);
      S();
    }
  }, [w, A, S, L, W]);
  const k = O.useMemo(() => ({
    reference: R,
    floating: C,
    setReference: v,
    setFloating: y
  }), [v, y]), E = O.useMemo(() => ({
    reference: w,
    floating: A
  }), [w, A]), D = O.useMemo(() => {
    const b = {
      position: n,
      left: 0,
      top: 0
    };
    if (!E.floating)
      return b;
    const T = Yt(E.floating, a.x), B = Yt(E.floating, a.y);
    return c ? {
      ...b,
      transform: "translate(" + T + "px, " + B + "px)",
      ...oe(E.floating) >= 1.5 && {
        willChange: "transform"
      }
    } : {
      position: n,
      left: T,
      top: B
    };
  }, [n, c, E.floating, a.x, a.y]);
  return O.useMemo(() => ({
    ...a,
    update: S,
    refs: k,
    elements: E,
    floatingStyles: D
  }), [a, S, k, E, D]);
}
const Cn = (t) => {
  function e(n) {
    return {}.hasOwnProperty.call(n, "current");
  }
  return {
    name: "arrow",
    options: t,
    fn(n) {
      const {
        element: o,
        padding: r
      } = typeof t == "function" ? t(n) : t;
      return o && e(o) ? o.current != null ? Xt({
        element: o.current,
        padding: r
      }).fn(n) : {} : o ? Xt({
        element: o,
        padding: r
      }).fn(n) : {};
    }
  };
}, Pn = (t, e) => {
  const n = pn(t);
  return {
    name: n.name,
    fn: n.fn,
    options: [t, e]
  };
}, Sn = (t, e) => {
  const n = gn(t);
  return {
    name: n.name,
    fn: n.fn,
    options: [t, e]
  };
}, En = (t, e) => ({
  fn: vn(t).fn,
  options: [t, e]
}), Dn = (t, e) => {
  const n = wn(t);
  return {
    name: n.name,
    fn: n.fn,
    options: [t, e]
  };
}, Ln = (t, e) => {
  const n = xn(t);
  return {
    name: n.name,
    fn: n.fn,
    options: [t, e]
  };
}, Mn = (t, e) => {
  const n = yn(t);
  return {
    name: n.name,
    fn: n.fn,
    options: [t, e]
  };
}, Tn = (t, e) => {
  const n = Cn(t);
  return {
    name: n.name,
    fn: n.fn,
    options: [t, e]
  };
};
var $n = "Arrow", re = O.forwardRef((t, e) => {
  const { children: n, width: o = 10, height: r = 5, ...i } = t;
  return /* @__PURE__ */ Y(
    Ot.svg,
    {
      ...i,
      ref: e,
      width: o,
      height: r,
      viewBox: "0 0 30 10",
      preserveAspectRatio: "none",
      children: t.asChild ? n : /* @__PURE__ */ Y("polygon", { points: "0,0 30,0 15,10" })
    }
  );
});
re.displayName = $n;
var Nn = re, Mt = "Popper", [ie, jn] = Ce(Mt), [Wn, se] = ie(Mt), ce = (t) => {
  const { __scopePopper: e, children: n } = t, [o, r] = O.useState(null);
  return /* @__PURE__ */ Y(Wn, { scope: e, anchor: o, onAnchorChange: r, children: n });
};
ce.displayName = Mt;
var le = "PopperAnchor", ae = O.forwardRef(
  (t, e) => {
    const { __scopePopper: n, virtualRef: o, ...r } = t, i = se(le, n), s = O.useRef(null), c = jt(e, s), l = O.useRef(null);
    return O.useEffect(() => {
      const f = l.current;
      l.current = o?.current || s.current, f !== l.current && i.onAnchorChange(l.current);
    }), o ? null : /* @__PURE__ */ Y(Ot.div, { ...r, ref: c });
  }
);
ae.displayName = le;
var Tt = "PopperContent", [kn, Fn] = ie(Tt), fe = O.forwardRef(
  (t, e) => {
    const {
      __scopePopper: n,
      side: o = "bottom",
      sideOffset: r = 0,
      align: i = "center",
      alignOffset: s = 0,
      arrowPadding: c = 0,
      avoidCollisions: l = !0,
      collisionBoundary: f = [],
      collisionPadding: a = 0,
      sticky: d = "partial",
      hideWhenDetached: m = !1,
      updatePositionStrategy: u = "optimized",
      onPlaced: h,
      ...p
    } = t, x = se(Tt, n), [g, v] = O.useState(null), y = jt(e, (ot) => v(ot)), [w, A] = O.useState(null), R = Se(w), C = R?.width ?? 0, M = R?.height ?? 0, W = o + (i !== "center" ? "-" + i : ""), L = typeof a == "number" ? a : { top: 0, right: 0, bottom: 0, left: 0, ...a }, $ = Array.isArray(f) ? f : [f], N = $.length > 0, S = {
      padding: L,
      boundary: $.filter(Hn),
      // with `strategy: 'fixed'`, this is the only way to get it to respect boundaries
      altBoundary: N
    }, { refs: P, floatingStyles: k, placement: E, isPositioned: D, middlewareData: b } = On({
      // default to `fixed` strategy so users don't have to pick and we also avoid focus scroll issues
      strategy: "fixed",
      placement: W,
      whileElementsMounted: (...ot) => hn(...ot, {
        animationFrame: u === "always"
      }),
      elements: {
        reference: x.anchor
      },
      middleware: [
        Pn({ mainAxis: r + M, alignmentAxis: s }),
        l && Sn({
          mainAxis: !0,
          crossAxis: !1,
          limiter: d === "partial" ? En() : void 0,
          ...S
        }),
        l && Dn({ ...S }),
        Ln({
          ...S,
          apply: ({ elements: ot, rects: Nt, availableWidth: ye, availableHeight: ve }) => {
            const { width: be, height: Ae } = Nt.reference, st = ot.floating.style;
            st.setProperty("--radix-popper-available-width", `${ye}px`), st.setProperty("--radix-popper-available-height", `${ve}px`), st.setProperty("--radix-popper-anchor-width", `${be}px`), st.setProperty("--radix-popper-anchor-height", `${Ae}px`);
          }
        }),
        w && Tn({ element: w, padding: c }),
        zn({ arrowWidth: C, arrowHeight: M }),
        m && Mn({ strategy: "referenceHidden", ...S })
      ]
    }), [T, B] = me(E), $t = Pe(h);
    vt(() => {
      D && $t?.();
    }, [D, $t]);
    const he = b.arrow?.x, pe = b.arrow?.y, ge = b.arrow?.centerOffset !== 0, [we, xe] = O.useState();
    return vt(() => {
      g && xe(window.getComputedStyle(g).zIndex);
    }, [g]), /* @__PURE__ */ Y(
      "div",
      {
        ref: P.setFloating,
        "data-radix-popper-content-wrapper": "",
        style: {
          ...k,
          transform: D ? k.transform : "translate(0, -200%)",
          // keep off the page when measuring
          minWidth: "max-content",
          zIndex: we,
          "--radix-popper-transform-origin": [
            b.transformOrigin?.x,
            b.transformOrigin?.y
          ].join(" "),
          // hide the content if using the hide middleware and should be hidden
          // set visibility to hidden and disable pointer events so the UI behaves
          // as if the PopperContent isn't there at all
          ...b.hide?.referenceHidden && {
            visibility: "hidden",
            pointerEvents: "none"
          }
        },
        dir: t.dir,
        children: /* @__PURE__ */ Y(
          kn,
          {
            scope: n,
            placedSide: T,
            onArrowChange: A,
            arrowX: he,
            arrowY: pe,
            shouldHideArrow: ge,
            children: /* @__PURE__ */ Y(
              Ot.div,
              {
                "data-side": T,
                "data-align": B,
                ...p,
                ref: y,
                style: {
                  ...p.style,
                  // if the PopperContent hasn't been placed yet (not all measurements done)
                  // we prevent animations so that users's animation don't kick in too early referring wrong sides
                  animation: D ? void 0 : "none"
                }
              }
            )
          }
        )
      }
    );
  }
);
fe.displayName = Tt;
var ue = "PopperArrow", _n = {
  top: "bottom",
  right: "left",
  bottom: "top",
  left: "right"
}, de = O.forwardRef(function(e, n) {
  const { __scopePopper: o, ...r } = e, i = Fn(ue, o), s = _n[i.placedSide];
  return (
    // we have to use an extra wrapper because `ResizeObserver` (used by `useSize`)
    // doesn't report size as we'd expect on SVG elements.
    // it reports their bounding box which is effectively the largest path inside the SVG.
    /* @__PURE__ */ Y(
      "span",
      {
        ref: i.onArrowChange,
        style: {
          position: "absolute",
          left: i.arrowX,
          top: i.arrowY,
          [s]: 0,
          transformOrigin: {
            top: "",
            right: "0 0",
            bottom: "center 0",
            left: "100% 0"
          }[i.placedSide],
          transform: {
            top: "translateY(100%)",
            right: "translateY(50%) rotate(90deg) translateX(-50%)",
            bottom: "rotate(180deg)",
            left: "translateY(50%) rotate(-90deg) translateX(50%)"
          }[i.placedSide],
          visibility: i.shouldHideArrow ? "hidden" : void 0
        },
        children: /* @__PURE__ */ Y(
          Nn,
          {
            ...r,
            ref: n,
            style: {
              ...r.style,
              // ensures the element can be measured correctly (mostly for if SVG)
              display: "block"
            }
          }
        )
      }
    )
  );
});
de.displayName = ue;
function Hn(t) {
  return t !== null;
}
var zn = (t) => ({
  name: "transformOrigin",
  options: t,
  fn(e) {
    const { placement: n, rects: o, middlewareData: r } = e, s = r.arrow?.centerOffset !== 0, c = s ? 0 : t.arrowWidth, l = s ? 0 : t.arrowHeight, [f, a] = me(n), d = { start: "0%", center: "50%", end: "100%" }[a], m = (r.arrow?.x ?? 0) + c / 2, u = (r.arrow?.y ?? 0) + l / 2;
    let h = "", p = "";
    return f === "bottom" ? (h = s ? d : `${m}px`, p = `${-l}px`) : f === "top" ? (h = s ? d : `${m}px`, p = `${o.floating.height + l}px`) : f === "right" ? (h = `${-l}px`, p = s ? d : `${u}px`) : f === "left" && (h = `${o.floating.width + l}px`, p = s ? d : `${u}px`), { data: { x: h, y: p } };
  }
});
function me(t) {
  const [e, n = "center"] = t.split("-");
  return [e, n];
}
var qn = ce, Un = ae, Kn = fe, Zn = de;
export {
  Un as A,
  Kn as C,
  qn as R,
  Zn as a,
  jn as c,
  Se as u
};
