// @btcv/ui/dist/_shared/index-CXeb1OMI.js
import * as o from "react";
import { jsx as S } from "react/jsx-runtime";
import { u as E } from "./index-DLcqcWxM.js";
function I(e, n, { checkForDefaultPrevented: t = !0 } = {}) {
  return function(r) {
    if (e?.(r), t === !1 || !r.defaultPrevented)
      return n?.(r);
  };
}
function U(e, n) {
  const t = o.createContext(n), s = (c) => {
    const { children: i, ...u } = c, a = o.useMemo(() => u, Object.values(u));
    return /* @__PURE__ */ S(t.Provider, { value: a, children: i });
  };
  s.displayName = e + "Provider";
  function r(c) {
    const i = o.useContext(t);
    if (i) return i;
    if (n !== void 0) return n;
    throw new Error(`\`${c}\` must be used within \`${e}\``);
  }
  return [s, r];
}
function _(e, n = []) {
  let t = [];
  function s(c, i) {
    const u = o.createContext(i), a = t.length;
    t = [...t, i];
    const f = (l) => {
      const { scope: p, children: m, ...v } = l, C = p?.[e]?.[a] || u, g = o.useMemo(() => v, Object.values(v));
      return /* @__PURE__ */ S(C.Provider, { value: g, children: m });
    };
    f.displayName = c + "Provider";
    function d(l, p) {
      const m = p?.[e]?.[a] || u, v = o.useContext(m);
      if (v) return v;
      if (i !== void 0) return i;
      throw new Error(`\`${l}\` must be used within \`${c}\``);
    }
    return [f, d];
  }
  const r = () => {
    const c = t.map((i) => o.createContext(i));
    return function(u) {
      const a = u?.[e] || c;
      return o.useMemo(
        () => ({ [`__scope${e}`]: { ...u, [e]: a } }),
        [u, a]
      );
    };
  };
  return r.scopeName = e, [s, M(r, ...n)];
}
function M(...e) {
  const n = e[0];
  if (e.length === 1) return n;
  const t = () => {
    const s = e.map((r) => ({
      useScope: r(),
      scopeName: r.scopeName
    }));
    return function(c) {
      const i = s.reduce((u, { useScope: a, scopeName: f }) => {
        const l = a(c)[`__scope${f}`];
        return { ...u, ...l };
      }, {});
      return o.useMemo(() => ({ [`__scope${n.scopeName}`]: i }), [i]);
    };
  };
  return t.scopeName = n.scopeName, t;
}
var h = globalThis?.document ? o.useLayoutEffect : () => {
}, P = o[" useInsertionEffect ".trim().toString()] || h;
function $({
  prop: e,
  defaultProp: n,
  onChange: t = () => {
  },
  caller: s
}) {
  const [r, c, i] = R({
    defaultProp: n,
    onChange: t
  }), u = e !== void 0, a = u ? e : r;
  {
    const d = o.useRef(e !== void 0);
    o.useEffect(() => {
      const l = d.current;
      l !== u && console.warn(
        `${s} is changing from ${l ? "controlled" : "uncontrolled"} to ${u ? "controlled" : "uncontrolled"}. Components should not switch from controlled to uncontrolled (or vice versa). Decide between using a controlled or uncontrolled value for the lifetime of the component.`
      ), d.current = u;
    }, [u, s]);
  }
  const f = o.useCallback(
    (d) => {
      if (u) {
        const l = A(d) ? d(e) : d;
        l !== e && i.current?.(l);
      } else
        c(d);
    },
    [u, e, c, i]
  );
  return [a, f];
}
function R({
  defaultProp: e,
  onChange: n
}) {
  const [t, s] = o.useState(e), r = o.useRef(t), c = o.useRef(n);
  return P(() => {
    c.current = n;
  }, [n]), o.useEffect(() => {
    r.current !== t && (c.current?.(t), r.current = t);
  }, [t, r]), [t, s, c];
}
function A(e) {
  return typeof e == "function";
}
function L(e) {
  const n = o.useRef(e);
  return o.useEffect(() => {
    n.current = e;
  }), o.useMemo(() => (...t) => n.current?.(...t), []);
}
function w(e, n) {
  return o.useReducer((t, s) => n[t][s] ?? t, e);
}
var x = (e) => {
  const { present: n, children: t } = e, s = y(n), r = typeof t == "function" ? t({ present: s.isPresent }) : o.Children.only(t), c = E(s.ref, O(r));
  return typeof t == "function" || s.isPresent ? o.cloneElement(r, { ref: c }) : null;
};
x.displayName = "Presence";
function y(e) {
  const [n, t] = o.useState(), s = o.useRef(null), r = o.useRef(e), c = o.useRef("none"), i = e ? "mounted" : "unmounted", [u, a] = w(i, {
    mounted: {
      UNMOUNT: "unmounted",
      ANIMATION_OUT: "unmountSuspended"
    },
    unmountSuspended: {
      MOUNT: "mounted",
      ANIMATION_END: "unmounted"
    },
    unmounted: {
      MOUNT: "mounted"
    }
  });
  return o.useEffect(() => {
    const f = N(s.current);
    c.current = u === "mounted" ? f : "none";
  }, [u]), h(() => {
    const f = s.current, d = r.current;
    if (d !== e) {
      const p = c.current, m = N(f);
      e ? a("MOUNT") : m === "none" || f?.display === "none" ? a("UNMOUNT") : a(d && p !== m ? "ANIMATION_OUT" : "UNMOUNT"), r.current = e;
    }
  }, [e, a]), h(() => {
    if (n) {
      let f;
      const d = n.ownerDocument.defaultView ?? window, l = (m) => {
        const C = N(s.current).includes(CSS.escape(m.animationName));
        if (m.target === n && C && (a("ANIMATION_END"), !r.current)) {
          const g = n.style.animationFillMode;
          n.style.animationFillMode = "forwards", f = d.setTimeout(() => {
            n.style.animationFillMode === "forwards" && (n.style.animationFillMode = g);
          });
        }
      }, p = (m) => {
        m.target === n && (c.current = N(s.current));
      };
      return n.addEventListener("animationstart", p), n.addEventListener("animationcancel", l), n.addEventListener("animationend", l), () => {
        d.clearTimeout(f), n.removeEventListener("animationstart", p), n.removeEventListener("animationcancel", l), n.removeEventListener("animationend", l);
      };
    } else
      a("ANIMATION_END");
  }, [n, a]), {
    isPresent: ["mounted", "unmountSuspended"].includes(u),
    ref: o.useCallback((f) => {
      s.current = f ? getComputedStyle(f) : null, t(f);
    }, [])
  };
}
function N(e) {
  return e?.animationName || "none";
}
function O(e) {
  let n = Object.getOwnPropertyDescriptor(e.props, "ref")?.get, t = n && "isReactWarning" in n && n.isReactWarning;
  return t ? e.ref : (n = Object.getOwnPropertyDescriptor(e, "ref")?.get, t = n && "isReactWarning" in n && n.isReactWarning, t ? e.props.ref : e.props.ref || e.ref);
}
export {
  x as P,
  _ as a,
  h as b,
  I as c,
  L as d,
  U as e,
  $ as u
};
