// @btcv/ui/dist/_shared/index-DlSuDb9N.js
import * as i from "react";
import { P as h, c as m, a as H, e as Z, u as q } from "./index-CXeb1OMI.js";
import { c as K, u as E } from "./index-DLcqcWxM.js";
import { u as y } from "./index-BC7vfx-u.js";
import { P as U, D as Y } from "./index-B03TCNO5.js";
import { R as z, h as J, u as Q, F as X } from "./Combination-D_l4PLF_.js";
import { P as D } from "./index-V-Ajw7Ac.js";
import { jsx as s, jsxs as N, Fragment as I } from "react/jsx-runtime";
// @__NO_SIDE_EFFECTS__
function ee(e) {
  const t = /* @__PURE__ */ te(e), n = i.forwardRef((r, o) => {
    const { children: a, ...c } = r, l = i.Children.toArray(a), u = l.find(ne);
    if (u) {
      const f = u.props.children, g = l.map((v) => v === u ? i.Children.count(f) > 1 ? i.Children.only(null) : i.isValidElement(f) ? f.props.children : null : v);
      return /* @__PURE__ */ s(t, { ...c, ref: o, children: i.isValidElement(f) ? i.cloneElement(f, void 0, g) : null });
    }
    return /* @__PURE__ */ s(t, { ...c, ref: o, children: a });
  });
  return n.displayName = `${e}.Slot`, n;
}
// @__NO_SIDE_EFFECTS__
function te(e) {
  const t = i.forwardRef((n, r) => {
    const { children: o, ...a } = n;
    if (i.isValidElement(o)) {
      const c = ae(o), l = re(a, o.props);
      return o.type !== i.Fragment && (l.ref = r ? K(r, c) : c), i.cloneElement(o, l);
    }
    return i.Children.count(o) > 1 ? i.Children.only(null) : null;
  });
  return t.displayName = `${e}.SlotClone`, t;
}
var oe = /* @__PURE__ */ Symbol("radix.slottable");
function ne(e) {
  return i.isValidElement(e) && typeof e.type == "function" && "__radixId" in e.type && e.type.__radixId === oe;
}
function re(e, t) {
  const n = { ...t };
  for (const r in t) {
    const o = e[r], a = t[r];
    /^on[A-Z]/.test(r) ? o && a ? n[r] = (...l) => {
      const u = a(...l);
      return o(...l), u;
    } : o && (n[r] = o) : r === "style" ? n[r] = { ...o, ...a } : r === "className" && (n[r] = [o, a].filter(Boolean).join(" "));
  }
  return { ...e, ...n };
}
function ae(e) {
  let t = Object.getOwnPropertyDescriptor(e.props, "ref")?.get, n = t && "isReactWarning" in t && t.isReactWarning;
  return n ? e.ref : (t = Object.getOwnPropertyDescriptor(e, "ref")?.get, n = t && "isReactWarning" in t && t.isReactWarning, n ? e.props.ref : e.props.ref || e.ref);
}
var R = "Dialog", [b] = H(R), [ie, d] = b(R), x = (e) => {
  const {
    __scopeDialog: t,
    children: n,
    open: r,
    defaultOpen: o,
    onOpenChange: a,
    modal: c = !0
  } = e, l = i.useRef(null), u = i.useRef(null), [f, g] = q({
    prop: r,
    defaultProp: o ?? !1,
    onChange: a,
    caller: R
  });
  return /* @__PURE__ */ s(
    ie,
    {
      scope: t,
      triggerRef: l,
      contentRef: u,
      contentId: y(),
      titleId: y(),
      descriptionId: y(),
      open: f,
      onOpenChange: g,
      onOpenToggle: i.useCallback(() => g((v) => !v), [g]),
      modal: c,
      children: n
    }
  );
};
x.displayName = R;
var A = "DialogTrigger", T = i.forwardRef(
  (e, t) => {
    const { __scopeDialog: n, ...r } = e, o = d(A, n), a = E(t, o.triggerRef);
    return /* @__PURE__ */ s(
      D.button,
      {
        type: "button",
        "aria-haspopup": "dialog",
        "aria-expanded": o.open,
        "aria-controls": o.contentId,
        "data-state": O(o.open),
        ...r,
        ref: a,
        onClick: m(e.onClick, o.onOpenToggle)
      }
    );
  }
);
T.displayName = A;
var _ = "DialogPortal", [se, S] = b(_, {
  forceMount: void 0
}), M = (e) => {
  const { __scopeDialog: t, forceMount: n, children: r, container: o } = e, a = d(_, t);
  return /* @__PURE__ */ s(se, { scope: t, forceMount: n, children: i.Children.map(r, (c) => /* @__PURE__ */ s(h, { present: n || a.open, children: /* @__PURE__ */ s(U, { asChild: !0, container: o, children: c }) })) });
};
M.displayName = _;
var C = "DialogOverlay", w = i.forwardRef(
  (e, t) => {
    const n = S(C, e.__scopeDialog), { forceMount: r = n.forceMount, ...o } = e, a = d(C, e.__scopeDialog);
    return a.modal ? /* @__PURE__ */ s(h, { present: r || a.open, children: /* @__PURE__ */ s(ce, { ...o, ref: t }) }) : null;
  }
);
w.displayName = C;
var le = /* @__PURE__ */ ee("DialogOverlay.RemoveScroll"), ce = i.forwardRef(
  (e, t) => {
    const { __scopeDialog: n, ...r } = e, o = d(C, n);
    return (
      // Make sure `Content` is scrollable even when it doesn't live inside `RemoveScroll`
      // ie. when `Overlay` and `Content` are siblings
      /* @__PURE__ */ s(z, { as: le, allowPinchZoom: !0, shards: [o.contentRef], children: /* @__PURE__ */ s(
        D.div,
        {
          "data-state": O(o.open),
          ...r,
          ref: t,
          style: { pointerEvents: "auto", ...r.style }
        }
      ) })
    );
  }
), p = "DialogContent", F = i.forwardRef(
  (e, t) => {
    const n = S(p, e.__scopeDialog), { forceMount: r = n.forceMount, ...o } = e, a = d(p, e.__scopeDialog);
    return /* @__PURE__ */ s(h, { present: r || a.open, children: a.modal ? /* @__PURE__ */ s(ue, { ...o, ref: t }) : /* @__PURE__ */ s(de, { ...o, ref: t }) });
  }
);
F.displayName = p;
var ue = i.forwardRef(
  (e, t) => {
    const n = d(p, e.__scopeDialog), r = i.useRef(null), o = E(t, n.contentRef, r);
    return i.useEffect(() => {
      const a = r.current;
      if (a) return J(a);
    }, []), /* @__PURE__ */ s(
      W,
      {
        ...e,
        ref: o,
        trapFocus: n.open,
        disableOutsidePointerEvents: !0,
        onCloseAutoFocus: m(e.onCloseAutoFocus, (a) => {
          a.preventDefault(), n.triggerRef.current?.focus();
        }),
        onPointerDownOutside: m(e.onPointerDownOutside, (a) => {
          const c = a.detail.originalEvent, l = c.button === 0 && c.ctrlKey === !0;
          (c.button === 2 || l) && a.preventDefault();
        }),
        onFocusOutside: m(
          e.onFocusOutside,
          (a) => a.preventDefault()
        )
      }
    );
  }
), de = i.forwardRef(
  (e, t) => {
    const n = d(p, e.__scopeDialog), r = i.useRef(!1), o = i.useRef(!1);
    return /* @__PURE__ */ s(
      W,
      {
        ...e,
        ref: t,
        trapFocus: !1,
        disableOutsidePointerEvents: !1,
        onCloseAutoFocus: (a) => {
          e.onCloseAutoFocus?.(a), a.defaultPrevented || (r.current || n.triggerRef.current?.focus(), a.preventDefault()), r.current = !1, o.current = !1;
        },
        onInteractOutside: (a) => {
          e.onInteractOutside?.(a), a.defaultPrevented || (r.current = !0, a.detail.originalEvent.type === "pointerdown" && (o.current = !0));
          const c = a.target;
          n.triggerRef.current?.contains(c) && a.preventDefault(), a.detail.originalEvent.type === "focusin" && o.current && a.preventDefault();
        }
      }
    );
  }
), W = i.forwardRef(
  (e, t) => {
    const { __scopeDialog: n, trapFocus: r, onOpenAutoFocus: o, onCloseAutoFocus: a, ...c } = e, l = d(p, n), u = i.useRef(null), f = E(t, u);
    return Q(), /* @__PURE__ */ N(I, { children: [
      /* @__PURE__ */ s(
        X,
        {
          asChild: !0,
          loop: !0,
          trapped: r,
          onMountAutoFocus: o,
          onUnmountAutoFocus: a,
          children: /* @__PURE__ */ s(
            Y,
            {
              role: "dialog",
              id: l.contentId,
              "aria-describedby": l.descriptionId,
              "aria-labelledby": l.titleId,
              "data-state": O(l.open),
              ...c,
              ref: f,
              onDismiss: () => l.onOpenChange(!1)
            }
          )
        }
      ),
      /* @__PURE__ */ N(I, { children: [
        /* @__PURE__ */ s(fe, { titleId: l.titleId }),
        /* @__PURE__ */ s(ge, { contentRef: u, descriptionId: l.descriptionId })
      ] })
    ] });
  }
), P = "DialogTitle", L = i.forwardRef(
  (e, t) => {
    const { __scopeDialog: n, ...r } = e, o = d(P, n);
    return /* @__PURE__ */ s(D.h2, { id: o.titleId, ...r, ref: t });
  }
);
L.displayName = P;
var k = "DialogDescription", G = i.forwardRef(
  (e, t) => {
    const { __scopeDialog: n, ...r } = e, o = d(k, n);
    return /* @__PURE__ */ s(D.p, { id: o.descriptionId, ...r, ref: t });
  }
);
G.displayName = k;
var V = "DialogClose", $ = i.forwardRef(
  (e, t) => {
    const { __scopeDialog: n, ...r } = e, o = d(V, n);
    return /* @__PURE__ */ s(
      D.button,
      {
        type: "button",
        ...r,
        ref: t,
        onClick: m(e.onClick, () => o.onOpenChange(!1))
      }
    );
  }
);
$.displayName = V;
function O(e) {
  return e ? "open" : "closed";
}
var j = "DialogTitleWarning", [Ee, B] = Z(j, {
  contentName: p,
  titleName: P,
  docsSlug: "dialog"
}), fe = ({ titleId: e }) => {
  const t = B(j), n = `\`${t.contentName}\` requires a \`${t.titleName}\` for the component to be accessible for screen reader users.

If you want to hide the \`${t.titleName}\`, you can wrap it with our VisuallyHidden component.

For more information, see https://radix-ui.com/primitives/docs/components/${t.docsSlug}`;
  return i.useEffect(() => {
    e && (document.getElementById(e) || console.error(n));
  }, [n, e]), null;
}, pe = "DialogDescriptionWarning", ge = ({ contentRef: e, descriptionId: t }) => {
  const r = `Warning: Missing \`Description\` or \`aria-describedby={undefined}\` for {${B(pe).contentName}}.`;
  return i.useEffect(() => {
    const o = e.current?.getAttribute("aria-describedby");
    t && o && (document.getElementById(t) || console.warn(r));
  }, [r, e, t]), null;
}, _e = x, Pe = T, Oe = M, Ne = w, Ie = F, be = L, xe = G, Ae = $;
export {
  Ie as C,
  xe as D,
  Ne as O,
  Oe as P,
  _e as R,
  be as T,
  Ae as a,
  Pe as b
};
