// @btcv/ui/dist/backgrounds/Aurora.js
"use client";
import { jsx as A } from "react/jsx-runtime";
import { useRef as C, useEffect as R } from "react";
import { Renderer as S, Triangle as B, Color as h, Program as z, Mesh as E } from "ogl";
const _ = `#version 300 es
in vec2 position;
void main() {
  gl_Position = vec4(position, 0.0, 1.0);
}
`, b = `#version 300 es
precision highp float;
uniform float uTime;
uniform float uAmplitude;
uniform vec3 uColorStops[3];
uniform vec2 uResolution;
uniform float uBlend;
out vec4 fragColor;
vec3 permute(vec3 x) { return mod(((x * 34.0) + 1.0) * x, 289.0); }
float snoise(vec2 v){
  const vec4 C = vec4(0.211324865405187, 0.366025403784439, -0.577350269189626, 0.024390243902439);
  vec2 i = floor(v + dot(v, C.yy));
  vec2 x0 = v - i + dot(i, C.xx);
  vec2 i1 = (x0.x > x0.y) ? vec2(1.0, 0.0) : vec2(0.0, 1.0);
  vec4 x12 = x0.xyxy + C.xxzz;
  x12.xy -= i1;
  i = mod(i, 289.0);
  vec3 p = permute(permute(i.y + vec3(0.0, i1.y, 1.0)) + i.x + vec3(0.0, i1.x, 1.0));
  vec3 m = max(0.5 - vec3(dot(x0,x0), dot(x12.xy,x12.xy), dot(x12.zw,x12.zw)), 0.0);
  m = m * m; m = m * m;
  vec3 x = 2.0 * fract(p * C.www) - 1.0;
  vec3 h = abs(x) - 0.5;
  vec3 ox = floor(x + 0.5);
  vec3 a0 = x - ox;
  m *= 1.79284291400159 - 0.85373472095314 * (a0*a0 + h*h);
  vec3 g;
  g.x = a0.x * x0.x + h.x * x0.y;
  g.yz = a0.yz * x12.xz + h.yz * x12.yw;
  return 130.0 * dot(m, g);
}
struct ColorStop { vec3 color; float position; };
#define COLOR_RAMP(colors, factor, finalColor) { \\
  int index = 0; \\
  for (int i = 0; i < 2; i++) { \\
    ColorStop currentColor = colors[i]; \\
    bool isInBetween = currentColor.position <= factor; \\
    index = int(mix(float(index), float(i), float(isInBetween))); \\
  } \\
  ColorStop currentColor = colors[index]; \\
  ColorStop nextColor = colors[index + 1]; \\
  float range = nextColor.position - currentColor.position; \\
  float lerpFactor = (factor - currentColor.position) / range; \\
  finalColor = mix(currentColor.color, nextColor.color, lerpFactor); \\
}
void main() {
  vec2 uv = gl_FragCoord.xy / uResolution;
  ColorStop colors[3];
  colors[0] = ColorStop(uColorStops[0], 0.0);
  colors[1] = ColorStop(uColorStops[1], 0.5);
  colors[2] = ColorStop(uColorStops[2], 1.0);
  vec3 rampColor;
  COLOR_RAMP(colors, uv.x, rampColor);
  float height = snoise(vec2(uv.x * 2.0 + uTime * 0.1, uTime * 0.25)) * 0.5 * uAmplitude;
  height = exp(height);
  height = (uv.y * 2.0 - height + 0.2);
  float intensity = 0.6 * height;
  float midPoint = 0.20;
  float auroraAlpha = smoothstep(midPoint - uBlend * 0.5, midPoint + uBlend * 0.5, intensity);
  vec3 auroraColor = intensity * rampColor;
  fragColor = vec4(auroraColor * auroraAlpha, auroraAlpha);
}
`;
function T({
  colorStops: i = ["#E7BB1D", "#B8941A", "#E7BB1D"],
  amplitude: l = 1,
  blend: u = 0.5,
  speed: f = 1
}) {
  const x = C(null), p = C({ colorStops: i, amplitude: l, blend: u, speed: f });
  return p.current = { colorStops: i, amplitude: l, blend: u, speed: f }, R(() => {
    const r = x.current;
    if (!r) return;
    const a = new S({ alpha: !0, premultipliedAlpha: !0, antialias: !0 }), o = a.gl;
    o.clearColor(0, 0, 0, 0), o.enable(o.BLEND), o.blendFunc(o.ONE, o.ONE_MINUS_SRC_ALPHA), o.canvas.style.backgroundColor = "transparent";
    let t;
    function s() {
      if (!r) return;
      const n = r.offsetWidth, e = r.offsetHeight;
      a.setSize(n, e), t && (t.uniforms.uResolution.value = [n, e]);
    }
    window.addEventListener("resize", s);
    const c = new B(o);
    c.attributes.uv && delete c.attributes.uv;
    const g = i.map((n) => {
      const e = new h(n);
      return [e.r, e.g, e.b];
    });
    t = new z(o, {
      vertex: _,
      fragment: b,
      uniforms: {
        uTime: { value: 0 },
        uAmplitude: { value: l },
        uColorStops: { value: g },
        uResolution: { value: [r.offsetWidth, r.offsetHeight] },
        uBlend: { value: u }
      }
    });
    const y = new E(o, { geometry: c, program: t });
    r.appendChild(o.canvas);
    let m = 0;
    const d = (n) => {
      m = requestAnimationFrame(d);
      const e = p.current;
      t.uniforms.uTime.value = n * 0.01 * (e.speed ?? 1) * 0.1, t.uniforms.uAmplitude.value = e.amplitude ?? 1, t.uniforms.uBlend.value = e.blend ?? 0.5, t.uniforms.uColorStops.value = (e.colorStops ?? i).map((w) => {
        const v = new h(w);
        return [v.r, v.g, v.b];
      }), a.render({ scene: y });
    };
    return m = requestAnimationFrame(d), s(), () => {
      cancelAnimationFrame(m), window.removeEventListener("resize", s), r && o.canvas.parentNode === r && r.removeChild(o.canvas), o.getExtension("WEBGL_lose_context")?.loseContext();
    };
  }, [l]), /* @__PURE__ */ A("div", { ref: x, style: { width: "100%", height: "100%" } });
}
export {
  T as default
};
