// @btcv/ui/dist/backgrounds/Iridescence.js
"use client";
import { jsx as C } from "react/jsx-runtime";
import { useRef as g, useEffect as R } from "react";
import { Renderer as A, Triangle as E, Program as M, Color as v, Mesh as S } from "ogl";
const T = `
attribute vec2 uv;
attribute vec2 position;
varying vec2 vUv;
void main() { vUv = uv; gl_Position = vec4(position, 0, 1); }
`, F = `
precision highp float;
uniform float uTime;
uniform vec3 uColor;
uniform vec3 uResolution;
uniform vec2 uMouse;
uniform float uAmplitude;
uniform float uSpeed;
varying vec2 vUv;
void main() {
  float mr = min(uResolution.x, uResolution.y);
  vec2 uv = (vUv.xy * 2.0 - 1.0) * uResolution.xy / mr;
  uv += (uMouse - vec2(0.5)) * uAmplitude;
  float d = -uTime * 0.5 * uSpeed;
  float a = 0.0;
  for (float i = 0.0; i < 8.0; ++i) {
    a += cos(i - d - a * uv.x);
    d += sin(uv.y * i + a);
  }
  d += uTime * 0.5 * uSpeed;
  vec3 col = vec3(cos(uv * vec2(d, a)) * 0.6 + 0.4, cos(a + d) * 0.5 + 0.5);
  col = cos(col * cos(vec3(d, a, 2.5)) * 0.5 + 0.5) * uColor;
  gl_FragColor = vec4(col, 1.0);
}
`;
function _({ color: c = [0.91, 0.73, 0.11], speed: l = 1, amplitude: m = 0.1, mouseReact: u = !0 }) {
  const f = g(null);
  return g({ x: 0.5, y: 0.5 }), R(() => {
    const o = f.current;
    if (!o) return;
    const r = new A(), e = r.gl;
    e.clearColor(0.11, 0.09, 0.09, 1);
    let n;
    function a() {
      r.setSize(o.offsetWidth, o.offsetHeight), n && (n.uniforms.uResolution.value = new v(e.canvas.width, e.canvas.height, e.canvas.width / e.canvas.height));
    }
    window.addEventListener("resize", a), a();
    const p = new E(e);
    n = new M(e, {
      vertex: T,
      fragment: F,
      uniforms: {
        uTime: { value: 0 },
        uColor: { value: new v(...c) },
        uResolution: { value: new v(e.canvas.width, e.canvas.height, e.canvas.width / e.canvas.height) },
        uMouse: { value: new Float32Array([0.5, 0.5]) },
        uAmplitude: { value: m },
        uSpeed: { value: l }
      }
    });
    const w = new S(e, { geometry: p, program: n });
    let s = 0;
    const d = (t) => {
      s = requestAnimationFrame(d), n.uniforms.uTime.value = t * 1e-3, r.render({ scene: w });
    };
    s = requestAnimationFrame(d), o.appendChild(e.canvas);
    const h = (t) => {
      const i = o.getBoundingClientRect(), x = (t.clientX - i.left) / i.width, y = 1 - (t.clientY - i.top) / i.height;
      n.uniforms.uMouse.value[0] = x, n.uniforms.uMouse.value[1] = y;
    };
    return u && o.addEventListener("mousemove", h), () => {
      cancelAnimationFrame(s), window.removeEventListener("resize", a), u && o.removeEventListener("mousemove", h), o.contains(e.canvas) && o.removeChild(e.canvas), e.getExtension("WEBGL_lose_context")?.loseContext();
    };
  }, [c, l, m, u]), /* @__PURE__ */ C("div", { ref: f, style: { width: "100%", height: "100%" } });
}
export {
  _ as default
};
