// @btcv/ui/dist/backgrounds/LiquidChrome.js
"use client";
import { jsx as F } from "react/jsx-runtime";
import { useRef as R, useEffect as C } from "react";
import { Renderer as A, Triangle as E, Program as M, Mesh as T } from "ogl";
function B({ baseColor: n = [0.11, 0.09, 0.07], speed: v = 0.2, amplitude: c = 0.5, frequencyX: f = 3, frequencyY: m = 2, interactive: r = !0 }) {
  const d = R(null);
  return C(() => {
    const t = d.current;
    if (!t) return;
    const a = new A({ antialias: !0 }), e = a.gl;
    e.clearColor(n[0], n[1], n[2], 1);
    const p = `
      attribute vec2 position;
      attribute vec2 uv;
      varying vec2 vUv;
      void main() { vUv = uv; gl_Position = vec4(position, 0.0, 1.0); }
    `, y = `
      precision highp float;
      uniform float uTime;
      uniform vec3 uResolution;
      uniform vec3 uBaseColor;
      uniform float uAmplitude;
      uniform float uFrequencyX;
      uniform float uFrequencyY;
      uniform vec2 uMouse;
      varying vec2 vUv;
      vec4 renderImage(vec2 uvCoord) {
        vec2 fragCoord = uvCoord * uResolution.xy;
        vec2 uv = (2.0 * fragCoord - uResolution.xy) / min(uResolution.x, uResolution.y);
        for (float i = 1.0; i < 10.0; i++) {
          uv.x += uAmplitude / i * cos(i * uFrequencyX * uv.y + uTime + uMouse.x * 3.14159);
          uv.y += uAmplitude / i * cos(i * uFrequencyY * uv.x + uTime + uMouse.y * 3.14159);
        }
        vec2 diff = (uvCoord - uMouse);
        float dist = length(diff);
        float falloff = exp(-dist * 20.0);
        float ripple = sin(10.0 * dist - uTime * 2.0) * 0.03;
        uv += (diff / (dist + 0.0001)) * ripple * falloff;
        vec3 color = uBaseColor / abs(sin(uTime - uv.y - uv.x));
        return vec4(color, 1.0);
      }
      void main() {
        vec4 col = vec4(0.0);
        int samples = 0;
        for (int i = -1; i <= 1; i++) {
          for (int j = -1; j <= 1; j++) {
            vec2 offset = vec2(float(i), float(j)) * (1.0 / min(uResolution.x, uResolution.y));
            col += renderImage(vUv + offset);
            samples++;
          }
        }
        gl_FragColor = col / float(samples);
      }
    `, w = new E(e), i = new M(e, {
      vertex: p,
      fragment: y,
      uniforms: {
        uTime: { value: 0 },
        uResolution: { value: new Float32Array([e.canvas.width, e.canvas.height, e.canvas.width / e.canvas.height]) },
        uBaseColor: { value: new Float32Array(n) },
        uAmplitude: { value: c },
        uFrequencyX: { value: f },
        uFrequencyY: { value: m },
        uMouse: { value: new Float32Array([0, 0]) }
      }
    }), x = new T(e, { geometry: w, program: i });
    function s() {
      a.setSize(t.offsetWidth, t.offsetHeight);
      const o = i.uniforms.uResolution.value;
      o[0] = e.canvas.width, o[1] = e.canvas.height, o[2] = e.canvas.width / e.canvas.height;
    }
    window.addEventListener("resize", s), s();
    const h = (o) => {
      const u = t.getBoundingClientRect();
      i.uniforms.uMouse.value[0] = (o.clientX - u.left) / u.width, i.uniforms.uMouse.value[1] = 1 - (o.clientY - u.top) / u.height;
    };
    r && t.addEventListener("mousemove", h);
    let l = 0;
    const g = (o) => {
      l = requestAnimationFrame(g), i.uniforms.uTime.value = o * 1e-3 * v, a.render({ scene: x });
    };
    return l = requestAnimationFrame(g), t.appendChild(e.canvas), () => {
      cancelAnimationFrame(l), window.removeEventListener("resize", s), r && t.removeEventListener("mousemove", h), e.canvas.parentElement && e.canvas.parentElement.removeChild(e.canvas), e.getExtension("WEBGL_lose_context")?.loseContext();
    };
  }, [n, v, c, f, m, r]), /* @__PURE__ */ F("div", { ref: d, style: { width: "100%", height: "100%" } });
}
export {
  B as default
};
