// @btcv/ui/dist/backgrounds/Particles.js
"use client";
import { jsx as G } from "react/jsx-runtime";
import { useRef as L, useEffect as N } from "react";
import { Renderer as W, Camera as X, Geometry as Y, Program as k, Mesh as J } from "ogl";
const K = ["#ffffff", "#ffffff", "#ffffff"], Q = (e) => {
  e = e.replace(/^#/, ""), e.length === 3 && (e = e.split("").map((r) => r + r).join(""));
  const i = parseInt(e, 16);
  return [(i >> 16 & 255) / 255, (i >> 8 & 255) / 255, (i & 255) / 255];
}, U = (
  /* glsl */
  `
  attribute vec3 position;
  attribute vec4 random;
  attribute vec3 color;
  uniform mat4 modelMatrix;
  uniform mat4 viewMatrix;
  uniform mat4 projectionMatrix;
  uniform float uTime;
  uniform float uSpread;
  uniform float uBaseSize;
  uniform float uSizeRandomness;
  varying vec4 vRandom;
  varying vec3 vColor;
  void main() {
    vRandom = random;
    vColor = color;
    vec3 pos = position * uSpread;
    pos.z *= 10.0;
    vec4 mPos = modelMatrix * vec4(pos, 1.0);
    float t = uTime;
    mPos.x += sin(t * random.z + 6.28 * random.w) * mix(0.1, 1.5, random.x);
    mPos.y += sin(t * random.y + 6.28 * random.x) * mix(0.1, 1.5, random.w);
    mPos.z += sin(t * random.w + 6.28 * random.y) * mix(0.1, 1.5, random.z);
    vec4 mvPos = viewMatrix * mPos;
    if (uSizeRandomness == 0.0) {
      gl_PointSize = uBaseSize;
    } else {
      gl_PointSize = (uBaseSize * (1.0 + uSizeRandomness * (random.x - 0.5))) / length(mvPos.xyz);
    }
    gl_Position = projectionMatrix * mvPos;
  }
`
), V = (
  /* glsl */
  `
  precision highp float;
  uniform float uTime;
  uniform float uAlphaParticles;
  varying vec4 vRandom;
  varying vec3 vColor;
  void main() {
    vec2 uv = gl_PointCoord.xy;
    float d = length(uv - vec2(0.5));
    if(uAlphaParticles < 0.5) {
      if(d > 0.5) discard;
      gl_FragColor = vec4(vColor + 0.2 * sin(uv.yxx + uTime + vRandom.y * 6.28), 1.0);
    } else {
      float circle = smoothstep(0.5, 0.4, d) * 0.8;
      gl_FragColor = vec4(vColor + 0.2 * sin(uv.yxx + uTime + vRandom.y * 6.28), circle);
    }
  }
`
);
function O({
  particleCount: e = 200,
  particleSpread: i = 10,
  speed: r = 0.1,
  particleColors: m,
  moveParticlesOnHover: l = !1,
  particleHoverFactor: u = 1,
  alphaParticles: M = !1,
  particleBaseSize: z = 100,
  sizeRandomness: P = 1,
  cameraDistance: R = 20,
  disableRotation: S = !1,
  className: I = ""
}) {
  const T = L(null), v = L({ x: 0, y: 0 });
  return N(() => {
    const a = T.current;
    if (!a) return;
    const h = new W({ dpr: Math.min(window.devicePixelRatio, 2), depth: !1, alpha: !0 }), o = h.gl;
    a.appendChild(o.canvas), o.clearColor(0, 0, 0, 0);
    const p = new X(o, { fov: 15 });
    p.position.set(0, 0, R);
    const g = () => {
      h.setSize(a.clientWidth, a.clientHeight), p.perspective({ aspect: o.canvas.width / o.canvas.height });
    };
    window.addEventListener("resize", g, !1), g();
    const A = (t) => {
      const n = a.getBoundingClientRect();
      v.current = {
        x: (t.clientX - n.left) / n.width * 2 - 1,
        y: -((t.clientY - n.top) / n.height * 2 - 1)
      };
    };
    l && a.addEventListener("mousemove", A);
    const C = new Float32Array(e * 3), F = new Float32Array(e * 4), _ = new Float32Array(e * 3), E = m && m.length > 0 ? m : K;
    for (let t = 0; t < e; t++) {
      let n, f, c, x;
      do
        n = Math.random() * 2 - 1, f = Math.random() * 2 - 1, c = Math.random() * 2 - 1, x = n * n + f * f + c * c;
      while (x > 1 || x === 0);
      const y = Math.cbrt(Math.random());
      C.set([n * y, f * y, c * y], t * 3), F.set([Math.random(), Math.random(), Math.random(), Math.random()], t * 4), _.set(Q(E[Math.floor(Math.random() * E.length)]), t * 3);
    }
    const q = new Y(o, {
      position: { size: 3, data: C },
      random: { size: 4, data: F },
      color: { size: 3, data: _ }
    }), b = new k(o, {
      vertex: U,
      fragment: V,
      uniforms: {
        uTime: { value: 0 },
        uSpread: { value: i },
        uBaseSize: { value: z * Math.min(window.devicePixelRatio, 2) },
        uSizeRandomness: { value: P },
        uAlphaParticles: { value: M ? 1 : 0 }
      },
      transparent: !0,
      depthTest: !1
    }), s = new J(o, { mode: o.POINTS, geometry: q, program: b });
    let w = 0, j = performance.now(), d = 0;
    const B = (t) => {
      w = requestAnimationFrame(B);
      const n = t - j;
      j = t, d += n * r, b.uniforms.uTime.value = d * 1e-3, l && (s.position.x = -v.current.x * u, s.position.y = -v.current.y * u), S || (s.rotation.x = Math.sin(d * 2e-4) * 0.1, s.rotation.y = Math.cos(d * 5e-4) * 0.15, s.rotation.z += 0.01 * r), h.render({ scene: s, camera: p });
    };
    return w = requestAnimationFrame(B), () => {
      window.removeEventListener("resize", g), l && a.removeEventListener("mousemove", A), cancelAnimationFrame(w), a.contains(o.canvas) && a.removeChild(o.canvas);
    };
  }, [e, i, r, l, u, M, z, P, R, S, m]), /* @__PURE__ */ G("div", { ref: T, className: I, style: { position: "relative", width: "100%", height: "100%" } });
}
export {
  O as default
};
