// @btcv/ui/dist/backgrounds/PixelSnow.js
"use client";
import { jsx as x } from "react/jsx-runtime";
import { useRef as h, useEffect as y } from "react";
function A({
  snowColor: i = "#E7BB1D",
  count: o = 180,
  speed: d = 0.6,
  pixelSize: r = 3,
  wind: c = 0.3,
  opacity: u = 0.7
}) {
  const l = h(null), n = h(null), m = h([]);
  return y(() => {
    const e = l.current;
    if (!e) return;
    const a = e.getContext("2d");
    if (!a) return;
    const s = () => {
      e.width = e.offsetWidth, e.height = e.offsetHeight;
    };
    window.addEventListener("resize", s), s(), m.current = Array.from({ length: o }, () => ({
      x: Math.random() * e.width,
      y: Math.random() * e.height,
      size: r * (0.5 + Math.random() * 1),
      speed: d * (0.4 + Math.random() * 0.8),
      drift: (Math.random() - 0.5) * c,
      alpha: 0.3 + Math.random() * u * 0.7,
      phase: Math.random() * Math.PI * 2,
      phaseSpeed: 5e-3 + Math.random() * 0.01
    }));
    let M = 0;
    const w = () => {
      a.clearRect(0, 0, e.width, e.height), M++, m.current.forEach((t) => {
        t.y += t.speed, t.x += t.drift + Math.sin(t.phase + M * t.phaseSpeed) * 0.3, t.y > e.height + t.size && (t.y = -t.size, t.x = Math.random() * e.width), t.x > e.width + t.size && (t.x = -t.size), t.x < -t.size && (t.x = e.width + t.size);
        const g = Math.round(t.x / r) * r, p = Math.round(t.y / r) * r;
        a.globalAlpha = t.alpha, a.fillStyle = i, a.fillRect(g, p, Math.round(t.size), Math.round(t.size));
      }), a.globalAlpha = 1, n.current = requestAnimationFrame(w);
    };
    return n.current = requestAnimationFrame(w), () => {
      window.removeEventListener("resize", s), n.current && cancelAnimationFrame(n.current);
    };
  }, [i, o, d, r, c, u]), /* @__PURE__ */ x(
    "canvas",
    {
      ref: l,
      style: { width: "100%", height: "100%", border: "none", display: "block" }
    }
  );
}
export {
  A as default
};
