// @btcv/ui/dist/backgrounds/Silk.js
"use client";
import { jsxs as l, jsx as i } from "react/jsx-runtime";
import { forwardRef as u, useLayoutEffect as c, useRef as v, useMemo as f } from "react";
import { useThree as m, useFrame as x, Canvas as d } from "@react-three/fiber";
import { Color as p } from "three";
const g = (n) => {
  const e = n.replace("#", "");
  return [
    parseInt(e.slice(0, 2), 16) / 255,
    parseInt(e.slice(2, 4), 16) / 255,
    parseInt(e.slice(4, 6), 16) / 255
  ];
}, y = `
varying vec2 vUv;
varying vec3 vPosition;
void main() {
  vPosition = position;
  vUv = uv;
  gl_Position = projectionMatrix * modelViewMatrix * vec4(position, 1.0);
}
`, h = `
varying vec2 vUv;
varying vec3 vPosition;
uniform float uTime;
uniform vec3  uColor;
uniform float uSpeed;
uniform float uScale;
uniform float uRotation;
uniform float uNoiseIntensity;
const float e = 2.71828182845904523536;
float noise(vec2 texCoord) {
  float G = e;
  vec2  r = (G * sin(G * texCoord));
  return fract(r.x * r.y * (1.0 + texCoord.x));
}
vec2 rotateUvs(vec2 uv, float angle) {
  float c = cos(angle);
  float s = sin(angle);
  mat2  rot = mat2(c, -s, s, c);
  return rot * uv;
}
void main() {
  float rnd        = noise(gl_FragCoord.xy);
  vec2  uv         = rotateUvs(vUv * uScale, uRotation);
  vec2  tex        = uv * uScale;
  float tOffset    = uSpeed * uTime;
  tex.y += 0.03 * sin(8.0 * tex.x - tOffset);
  float pattern = 0.6 +
                  0.4 * sin(5.0 * (tex.x + tex.y +
                                   cos(3.0 * tex.x + 5.0 * tex.y) +
                                   0.02 * tOffset) +
                           sin(20.0 * (tex.x + tex.y - 0.1 * tOffset)));
  vec4 col = vec4(uColor, 1.0) * vec4(pattern) - rnd / 15.0 * uNoiseIntensity;
  col.a = 1.0;
  gl_FragColor = col;
}
`, S = u(function({ uniforms: e }, t) {
  const { viewport: o } = m();
  return c(() => {
    const r = t;
    r?.current && r.current.scale.set(o.width, o.height, 1);
  }, [t, o]), x((r, s) => {
    const a = t;
    a?.current && (a.current.material.uniforms.uTime.value += 0.1 * s);
  }), /* @__PURE__ */ l("mesh", { ref: t, children: [
    /* @__PURE__ */ i("planeGeometry", { args: [1, 1, 1, 1] }),
    /* @__PURE__ */ i("shaderMaterial", { uniforms: e, vertexShader: y, fragmentShader: h })
  ] });
});
function T({
  speed: n = 5,
  scale: e = 1,
  color: t = "#7B7481",
  noiseIntensity: o = 1.5,
  rotation: r = 0
}) {
  const s = v(null), a = f(
    () => ({
      uSpeed: { value: n },
      uScale: { value: e },
      uNoiseIntensity: { value: o },
      uColor: { value: new p(...g(t)) },
      uRotation: { value: r },
      uTime: { value: 0 }
    }),
    [n, e, o, t, r]
  );
  return /* @__PURE__ */ i(d, { dpr: [1, 2], frameloop: "always", children: /* @__PURE__ */ i(S, { ref: s, uniforms: a }) });
}
export {
  T as default
};
