// @btcv/ui/dist/backgrounds/Squares.js
"use client";
import { jsx as E } from "react/jsx-runtime";
import { useRef as i, useEffect as L } from "react";
function X({
  direction: u = "diagonal",
  speed: x = 0.5,
  borderColor: y = "#E7BB1D",
  squareSize: t = 40,
  hoverFillColor: g = "rgba(231, 187, 29, 0.08)",
  vignetteColor: v = "#1D1917"
}) {
  const w = i(null), d = i(null), r = i({ x: 0, y: 0 }), h = i(null);
  return L(() => {
    const n = w.current;
    if (!n) return;
    const c = n.getContext("2d");
    if (!c) return;
    const m = () => {
      n.width = n.offsetWidth, n.height = n.offsetHeight;
    };
    window.addEventListener("resize", m), m();
    const p = () => {
      c.clearRect(0, 0, n.width, n.height);
      const e = Math.floor(r.current.x / t) * t, f = Math.floor(r.current.y / t) * t;
      for (let o = e; o < n.width + t; o += t)
        for (let l = f; l < n.height + t; l += t) {
          const s = o - r.current.x % t, k = l - r.current.y % t;
          h.current && Math.floor((o - e) / t) === h.current.x && Math.floor((l - f) / t) === h.current.y && (c.fillStyle = g, c.fillRect(s, k, t, t)), c.strokeStyle = y, c.strokeRect(s, k, t, t);
        }
      const a = c.createRadialGradient(
        n.width / 2,
        n.height / 2,
        0,
        n.width / 2,
        n.height / 2,
        Math.sqrt(n.width ** 2 + n.height ** 2) / 2
      );
      a.addColorStop(0, "rgba(0, 0, 0, 0)"), a.addColorStop(1, v), c.fillStyle = a, c.fillRect(0, 0, n.width, n.height);
    }, M = () => {
      const e = Math.max(x, 0.1);
      switch (u) {
        case "right":
          r.current.x = (r.current.x - e + t) % t;
          break;
        case "left":
          r.current.x = (r.current.x + e + t) % t;
          break;
        case "up":
          r.current.y = (r.current.y + e + t) % t;
          break;
        case "down":
          r.current.y = (r.current.y - e + t) % t;
          break;
        case "diagonal":
          r.current.x = (r.current.x - e + t) % t, r.current.y = (r.current.y - e + t) % t;
          break;
      }
      p(), d.current = requestAnimationFrame(M);
    }, R = (e) => {
      const f = n.getBoundingClientRect(), a = e.clientX - f.left, o = e.clientY - f.top, l = Math.floor(r.current.x / t) * t, s = Math.floor(r.current.y / t) * t;
      h.current = {
        x: Math.floor((a + r.current.x - l) / t),
        y: Math.floor((o + r.current.y - s) / t)
      };
    }, b = () => {
      h.current = null;
    };
    return n.addEventListener("mousemove", R), n.addEventListener("mouseleave", b), d.current = requestAnimationFrame(M), () => {
      window.removeEventListener("resize", m), d.current && cancelAnimationFrame(d.current), n.removeEventListener("mousemove", R), n.removeEventListener("mouseleave", b);
    };
  }, [u, x, y, g, t, v]), /* @__PURE__ */ E("canvas", { ref: w, style: { width: "100%", height: "100%", border: "none", display: "block" } });
}
export {
  X as default
};
