// @btcv/ui/dist/backgrounds/Threads.js
"use client";
import { jsx as y } from "react/jsx-runtime";
import { useRef as w, useEffect as C } from "react";
import { Renderer as b, Triangle as A, Program as R, Color as h, Mesh as E } from "ogl";
const L = `
attribute vec2 position;
attribute vec2 uv;
varying vec2 vUv;
void main() { vUv = uv; gl_Position = vec4(position, 0.0, 1.0); }
`, z = `
precision highp float;
uniform float iTime;
uniform vec3 iResolution;
uniform vec3 uColor;
uniform float uAmplitude;
uniform float uDistance;
uniform vec2 uMouse;
#define PI 3.1415926538
const int u_line_count = 40;
const float u_line_width = 7.0;
const float u_line_blur = 10.0;
float Perlin2D(vec2 P) {
  vec2 Pi = floor(P);
  vec4 Pf_Pfmin1 = P.xyxy - vec4(Pi, Pi + 1.0);
  vec4 Pt = vec4(Pi.xy, Pi.xy + 1.0);
  Pt = Pt - floor(Pt * (1.0 / 71.0)) * 71.0;
  Pt += vec2(26.0, 161.0).xyxy;
  Pt *= Pt;
  Pt = Pt.xzxz * Pt.yyww;
  vec4 hash_x = fract(Pt * (1.0 / 951.135664));
  vec4 hash_y = fract(Pt * (1.0 / 642.949883));
  vec4 grad_x = hash_x - 0.49999;
  vec4 grad_y = hash_y - 0.49999;
  vec4 grad_results = inversesqrt(grad_x * grad_x + grad_y * grad_y) * (grad_x * Pf_Pfmin1.xzxz + grad_y * Pf_Pfmin1.yyww);
  grad_results *= 1.4142135623730950;
  vec2 blend = Pf_Pfmin1.xy * Pf_Pfmin1.xy * Pf_Pfmin1.xy * (Pf_Pfmin1.xy * (Pf_Pfmin1.xy * 6.0 - 15.0) + 10.0);
  vec4 blend2 = vec4(blend, vec2(1.0 - blend));
  return dot(grad_results, blend2.zxzx * blend2.wwyy);
}
float pixel(float count, vec2 resolution) { return (1.0 / max(resolution.x, resolution.y)) * count; }
float lineFn(vec2 st, float width, float perc, float offset, vec2 mouse, float time, float amplitude, float distance) {
  float split_offset = (perc * 0.4);
  float split_point = 0.1 + split_offset;
  float amplitude_normal = smoothstep(split_point, 0.7, st.x);
  float finalAmplitude = amplitude_normal * 0.5 * amplitude * (1.0 + (mouse.y - 0.5) * 0.2);
  float time_scaled = time / 10.0 + (mouse.x - 0.5) * 1.0;
  float blur = smoothstep(split_point, split_point + 0.05, st.x) * perc;
  float xnoise = mix(Perlin2D(vec2(time_scaled, st.x + perc) * 2.5), Perlin2D(vec2(time_scaled, st.x + time_scaled) * 3.5) / 1.5, st.x * 0.3);
  float y = 0.5 + (perc - 0.5) * distance + xnoise / 2.0 * finalAmplitude;
  float line_start = smoothstep(y + (width / 2.0) + (u_line_blur * pixel(1.0, iResolution.xy) * blur), y, st.y);
  float line_end = smoothstep(y, y - (width / 2.0) - (u_line_blur * pixel(1.0, iResolution.xy) * blur), st.y);
  return clamp((line_start - line_end) * (1.0 - smoothstep(0.0, 1.0, pow(perc, 0.3))), 0.0, 1.0);
}
void mainImage(out vec4 fragColor, in vec2 fragCoord) {
  vec2 uv = fragCoord / iResolution.xy;
  float line_strength = 1.0;
  for (int i = 0; i < u_line_count; i++) {
    float p = float(i) / float(u_line_count);
    line_strength *= (1.0 - lineFn(uv, u_line_width * pixel(1.0, iResolution.xy) * (1.0 - p), p, (PI * 1.0) * p, uMouse, iTime, uAmplitude, uDistance));
  }
  float colorVal = 1.0 - line_strength;
  fragColor = vec4(uColor * colorVal, colorVal);
}
void main() { mainImage(gl_FragColor, gl_FragCoord.xy); }
`;
function S({ color: c = [0.91, 0.73, 0.11], amplitude: m = 1, distance: v = 0, enableMouseInteraction: s = !0 }) {
  const d = w(null);
  return C(() => {
    const t = d.current;
    if (!t) return;
    const a = new b({ alpha: !0 }), e = a.gl;
    e.clearColor(0, 0, 0, 0), e.enable(e.BLEND), e.blendFunc(e.SRC_ALPHA, e.ONE_MINUS_SRC_ALPHA), t.appendChild(e.canvas);
    const g = new A(e), n = new R(e, {
      vertex: L,
      fragment: z,
      uniforms: {
        iTime: { value: 0 },
        iResolution: { value: new h(e.canvas.width, e.canvas.height, e.canvas.width / e.canvas.height) },
        uColor: { value: new h(...c) },
        uAmplitude: { value: m },
        uDistance: { value: v },
        uMouse: { value: new Float32Array([0.5, 0.5]) }
      }
    }), P = new E(e, { geometry: g, program: n });
    function u() {
      a.setSize(t.offsetWidth, t.offsetHeight);
      const o = n.uniforms.iResolution.value;
      o.r = t.offsetWidth, o.g = t.offsetHeight, o.b = t.offsetWidth / t.offsetHeight;
    }
    window.addEventListener("resize", u), u();
    let i = [0.5, 0.5], l = [0.5, 0.5];
    const _ = (o) => {
      const r = t.getBoundingClientRect();
      l = [(o.clientX - r.left) / r.width, 1 - (o.clientY - r.top) / r.height];
    }, p = () => {
      l = [0.5, 0.5];
    };
    s && (t.addEventListener("mousemove", _), t.addEventListener("mouseleave", p));
    let f = 0;
    const x = (o) => {
      f = requestAnimationFrame(x), i[0] += 0.05 * (l[0] - i[0]), i[1] += 0.05 * (l[1] - i[1]), n.uniforms.uMouse.value[0] = i[0], n.uniforms.uMouse.value[1] = i[1], n.uniforms.iTime.value = o * 1e-3, a.render({ scene: P });
    };
    return f = requestAnimationFrame(x), () => {
      cancelAnimationFrame(f), window.removeEventListener("resize", u), s && (t.removeEventListener("mousemove", _), t.removeEventListener("mouseleave", p)), t.contains(e.canvas) && t.removeChild(e.canvas), e.getExtension("WEBGL_lose_context")?.loseContext();
    };
  }, [c, m, v, s]), /* @__PURE__ */ y("div", { ref: d, style: { width: "100%", height: "100%", position: "relative" } });
}
export {
  S as default
};
