// @btcv/ui/dist/backgrounds/Waves.js
"use client";
import { jsx as J } from "react/jsx-runtime";
import { useRef as l, useEffect as K } from "react";
class u {
  constructor(r, s, i) {
    this.x = r, this.y = s, this.z = i;
  }
  dot2(r, s) {
    return this.x * r + this.y * s;
  }
}
class V {
  constructor(r = 0) {
    this.grad3 = [
      new u(1, 1, 0),
      new u(-1, 1, 0),
      new u(1, -1, 0),
      new u(-1, -1, 0),
      new u(1, 0, 1),
      new u(-1, 0, 1),
      new u(1, 0, -1),
      new u(-1, 0, -1),
      new u(0, 1, 1),
      new u(0, -1, 1),
      new u(0, 1, -1),
      new u(0, -1, -1)
    ], this.p = [151, 160, 137, 91, 90, 15, 131, 13, 201, 95, 96, 53, 194, 233, 7, 225, 140, 36, 103, 30, 69, 142, 8, 99, 37, 240, 21, 10, 23, 190, 6, 148, 247, 120, 234, 75, 0, 26, 197, 62, 94, 252, 219, 203, 117, 35, 11, 32, 57, 177, 33, 88, 237, 149, 56, 87, 174, 20, 125, 136, 171, 168, 68, 175, 74, 165, 71, 134, 139, 48, 27, 166, 77, 146, 158, 231, 83, 111, 229, 122, 60, 211, 133, 230, 220, 105, 92, 41, 55, 46, 245, 40, 244, 102, 143, 54, 65, 25, 63, 161, 1, 216, 80, 73, 209, 76, 132, 187, 208, 89, 18, 169, 200, 196, 135, 130, 116, 188, 159, 86, 164, 100, 109, 198, 173, 186, 3, 64, 52, 217, 226, 250, 124, 123, 5, 202, 38, 147, 118, 126, 255, 82, 85, 212, 207, 206, 59, 227, 47, 16, 58, 17, 182, 189, 28, 42, 223, 183, 170, 213, 119, 248, 152, 2, 44, 154, 163, 70, 221, 153, 101, 155, 167, 43, 172, 9, 129, 22, 39, 253, 19, 98, 108, 110, 79, 113, 224, 232, 178, 185, 112, 104, 218, 246, 97, 228, 251, 34, 242, 193, 238, 210, 144, 12, 191, 179, 162, 241, 81, 51, 145, 235, 249, 14, 239, 107, 49, 192, 214, 31, 181, 199, 106, 157, 184, 84, 204, 176, 115, 121, 50, 45, 127, 4, 150, 254, 138, 236, 205, 93, 222, 114, 67, 29, 24, 72, 243, 141, 128, 195, 78, 66, 215, 61, 156, 180], this.perm = new Array(512), this.gradP = new Array(512), this.seed(r);
  }
  seed(r) {
    r > 0 && r < 1 && (r *= 65536), r = Math.floor(r), r < 256 && (r |= r << 8);
    for (let s = 0; s < 256; s++) {
      let i = s & 1 ? this.p[s] ^ r & 255 : this.p[s] ^ r >> 8 & 255;
      this.perm[s] = this.perm[s + 256] = i, this.gradP[s] = this.gradP[s + 256] = this.grad3[i % 12];
    }
  }
  fade(r) {
    return r * r * r * (r * (r * 6 - 15) + 10);
  }
  lerp(r, s, i) {
    return (1 - i) * r + i * s;
  }
  perlin2(r, s) {
    let i = Math.floor(r), h = Math.floor(s);
    r -= i, s -= h, i &= 255, h &= 255;
    const g = this.gradP[i + this.perm[h]].dot2(r, s), M = this.gradP[i + this.perm[h + 1]].dot2(r, s - 1), R = this.gradP[i + 1 + this.perm[h]].dot2(r - 1, s), E = this.gradP[i + 1 + this.perm[h + 1]].dot2(r - 1, s - 1), y = this.fade(r);
    return this.lerp(this.lerp(g, R, y), this.lerp(M, E, y), this.fade(s));
  }
}
function Z({
  lineColor: w = "#E7BB1D",
  backgroundColor: r = "transparent",
  waveSpeedX: s = 0.0125,
  waveSpeedY: i = 5e-3,
  waveAmpX: h = 32,
  waveAmpY: g = 16,
  xGap: M = 10,
  yGap: R = 32,
  friction: E = 0.925,
  tension: y = 5e-3,
  maxCursorMove: F = 100
}) {
  const q = l(null), C = l(null), S = l(null), L = l({ width: 0, height: 0, left: 0, top: 0 }), O = l(new V(Math.random())), b = l([]), T = l({ x: -10, y: 0, lx: 0, ly: 0, sx: 0, sy: 0, v: 0, vs: 0, a: 0, set: !1 }), z = l({ lineColor: w, waveSpeedX: s, waveSpeedY: i, waveAmpX: h, waveAmpY: g, friction: E, tension: y, maxCursorMove: F, xGap: M, yGap: R }), k = l(null);
  return K(() => {
    z.current = { lineColor: w, waveSpeedX: s, waveSpeedY: i, waveAmpX: h, waveAmpY: g, friction: E, tension: y, maxCursorMove: F, xGap: M, yGap: R };
  }, [w, s, i, h, g, E, y, F, M, R]), K(() => {
    const A = C.current, W = q.current;
    if (!A || !W) return;
    S.current = A.getContext("2d");
    function D() {
      const e = W.getBoundingClientRect();
      L.current = { width: e.width, height: e.height, left: e.left, top: e.top }, A.width = e.width, A.height = e.height;
    }
    function G() {
      const { width: e, height: t } = L.current;
      b.current = [];
      const c = e + 200, o = t + 30, { xGap: a, yGap: n } = z.current, f = Math.ceil(c / a), x = Math.ceil(o / n), d = (e - a * f) / 2, P = (t - n * x) / 2;
      for (let v = 0; v <= f; v++) {
        const B = [];
        for (let m = 0; m <= x; m++)
          B.push({ x: d + a * v, y: P + n * m, wave: { x: 0, y: 0 }, cursor: { x: 0, y: 0, vx: 0, vy: 0 } });
        b.current.push(B);
      }
    }
    function Q(e) {
      const t = T.current, c = O.current, o = z.current;
      b.current.forEach((a) => {
        a.forEach((n) => {
          const f = c.perlin2((n.x + e * o.waveSpeedX) * 2e-3, (n.y + e * o.waveSpeedY) * 15e-4) * 12;
          n.wave.x = Math.cos(f) * o.waveAmpX, n.wave.y = Math.sin(f) * o.waveAmpY;
          const x = n.x - t.sx, d = n.y - t.sy, P = Math.hypot(x, d), v = Math.max(175, t.vs);
          if (P < v) {
            const B = 1 - P / v, m = Math.cos(P * 1e-3) * B;
            n.cursor.vx += Math.cos(t.a) * m * v * t.vs * 65e-5, n.cursor.vy += Math.sin(t.a) * m * v * t.vs * 65e-5;
          }
          n.cursor.vx += (0 - n.cursor.x) * o.tension, n.cursor.vy += (0 - n.cursor.y) * o.tension, n.cursor.vx *= o.friction, n.cursor.vy *= o.friction, n.cursor.x += n.cursor.vx * 2, n.cursor.y += n.cursor.vy * 2, n.cursor.x = Math.min(o.maxCursorMove, Math.max(-o.maxCursorMove, n.cursor.x)), n.cursor.y = Math.min(o.maxCursorMove, Math.max(-o.maxCursorMove, n.cursor.y));
        });
      });
    }
    function j(e, t = !0) {
      return {
        x: Math.round((e.x + e.wave.x + (t ? e.cursor.x : 0)) * 10) / 10,
        y: Math.round((e.y + e.wave.y + (t ? e.cursor.y : 0)) * 10) / 10
      };
    }
    function U() {
      const e = S.current;
      if (!e) return;
      const { width: t, height: c } = L.current;
      e.clearRect(0, 0, t, c), e.beginPath(), e.strokeStyle = z.current.lineColor, b.current.forEach((o) => {
        let a = j(o[0], !1);
        e.moveTo(a.x, a.y), o.forEach((n, f) => {
          const x = f === o.length - 1;
          a = j(n, !x);
          const d = j(o[f + 1] || o[o.length - 1], !x);
          e.lineTo(a.x, a.y), x && e.moveTo(d.x, d.y);
        });
      }), e.stroke();
    }
    function H(e) {
      const t = T.current;
      t.sx += (t.x - t.sx) * 0.1, t.sy += (t.y - t.sy) * 0.1;
      const c = t.x - t.lx, o = t.y - t.ly;
      t.v = Math.hypot(c, o), t.vs += (t.v - t.vs) * 0.1, t.vs = Math.min(100, t.vs), t.lx = t.x, t.ly = t.y, t.a = Math.atan2(o, c), Q(e), U(), k.current = requestAnimationFrame(H);
    }
    function I() {
      D(), G();
    }
    function N(e) {
      const t = L.current, c = T.current;
      c.x = e.clientX - t.left, c.y = e.clientY - t.top, c.set || (c.sx = c.x, c.sy = c.y, c.lx = c.x, c.ly = c.y, c.set = !0);
    }
    return D(), G(), k.current = requestAnimationFrame(H), window.addEventListener("resize", I), window.addEventListener("mousemove", N), () => {
      window.removeEventListener("resize", I), window.removeEventListener("mousemove", N), k.current !== null && cancelAnimationFrame(k.current);
    };
  }, []), /* @__PURE__ */ J("div", { ref: q, style: { position: "absolute", inset: 0, overflow: "hidden", backgroundColor: r }, children: /* @__PURE__ */ J("canvas", { ref: C, style: { display: "block", width: "100%", height: "100%" } }) });
}
export {
  Z as default
};
