// @btcv/ui/dist/Alert.js
import { jsx as e, jsxs as m } from "react/jsx-runtime";
import { c as r } from "./_shared/utils-eGXXUFl7.js";
import { Card as f } from "./Card.js";
const d = {
  primary: "text-primary",
  info: "text-info",
  success: "text-success",
  destructive: "text-destructive",
  warning: "text-warning"
}, x = {
  primary: "bg-primary/15",
  info: "bg-info/15",
  success: "bg-success/15",
  destructive: "bg-destructive/15",
  warning: "bg-warning/15"
};
function N({
  icon: t,
  variant: s,
  title: c,
  children: i,
  className: a,
  filled: n = !0,
  ...l
}) {
  const o = s && !n, u = c && !i;
  return /* @__PURE__ */ e(
    f,
    {
      accent: n ? s : void 0,
      className: r(
        "!p-4 !shadow-none",
        a
      ),
      role: "alert",
      ...l,
      children: /* @__PURE__ */ m("div", { className: r("flex gap-3", u ? "items-center" : "items-start"), children: [
        t && o && /* @__PURE__ */ e("div", { className: r("shrink-0 w-8 h-8 rounded-full flex items-center justify-center", x[s]), children: /* @__PURE__ */ e(t, { size: 16, className: d[s] }) }),
        t && !o && /* @__PURE__ */ e(
          "div",
          {
            className: r(
              "shrink-0 h-5 flex items-center",
              s ? d[s] : "text-muted-foreground"
            ),
            children: /* @__PURE__ */ e(t, { size: 16 })
          }
        ),
        /* @__PURE__ */ m("div", { className: "flex-1 min-w-0 flex flex-col gap-1", children: [
          c && /* @__PURE__ */ e("div", { className: "text-sm font-semibold text-foreground", children: c }),
          i && /* @__PURE__ */ e("div", { className: "text-sm text-muted-foreground", children: i })
        ] })
      ] })
    }
  );
}
export {
  N as Alert
};
