// @btcv/ui/dist/Avatar.js
"use client";
import { jsx as e } from "react/jsx-runtime";
import * as r from "react";
import { c as f } from "./_shared/utils-eGXXUFl7.js";
const u = {
  sm: "size-7 text-xs",
  md: "size-9 text-sm",
  lg: "size-11 text-base",
  xl: "size-14 text-lg"
}, x = r.forwardRef(
  ({ src: t, alt: o, fallback: s = "?", size: i = "md", className: m, ...n }, a) => {
    const [l, c] = r.useState(!1), d = t && !l;
    return /* @__PURE__ */ e(
      "div",
      {
        ref: a,
        className: f(
          "relative inline-flex items-center justify-center shrink-0 rounded-full overflow-hidden",
          "bg-muted text-muted-foreground font-medium select-none",
          u[i],
          m
        ),
        ...n,
        children: d ? /* @__PURE__ */ e(
          "img",
          {
            src: t,
            alt: o || s,
            onError: () => c(!0),
            className: "size-full object-cover"
          }
        ) : /* @__PURE__ */ e("span", { children: s })
      }
    );
  }
);
x.displayName = "Avatar";
export {
  x as Avatar
};
