// @btcv/ui/dist/Badge.js
import { jsx as s, jsxs as u } from "react/jsx-runtime";
import { c as g } from "./_shared/index-uPOYJZpG.js";
import { c as o } from "./_shared/utils-eGXXUFl7.js";
const p = g(
  "inline-flex items-center font-semibold rounded-full transition-colors",
  {
    variants: {
      variant: {
        default: "bg-primary/15 text-primary border border-primary/20",
        success: "bg-success/15 text-success border border-success/20",
        destructive: "bg-destructive/15 text-destructive border border-destructive/20",
        warning: "bg-warning/15 text-warning border border-warning/20",
        info: "bg-info/15 text-info border border-info/20",
        secondary: "bg-muted text-muted-foreground border border-border",
        outline: "bg-transparent text-foreground border border-border"
      },
      size: {
        sm: "text-[10px] px-1.5 py-0.5 gap-1",
        md: "text-xs px-2.5 py-0.5 gap-1.5",
        lg: "text-sm px-3 py-1 gap-2"
      }
    },
    defaultVariants: {
      variant: "default",
      size: "md"
    }
  }
);
function c({ children: t, variant: n = "default", size: r = "md", color: e, className: a, style: d, ...i }) {
  return e ? /* @__PURE__ */ s(
    "span",
    {
      className: o(
        "inline-flex items-center font-semibold rounded-full border",
        r === "sm" ? "text-[10px] px-1.5 py-0.5 gap-1" : r === "lg" ? "text-sm px-3 py-1 gap-2" : "text-xs px-2.5 py-0.5 gap-1.5",
        a
      ),
      style: {
        color: e,
        borderColor: `${e}33`,
        backgroundColor: `${e}1A`,
        ...d
      },
      ...i,
      children: t
    }
  ) : /* @__PURE__ */ s("span", { className: o(p({ variant: n, size: r }), a), style: d, ...i, children: t });
}
const b = {
  online: "bg-success",
  offline: "bg-destructive",
  loading: "bg-warning",
  idle: "bg-muted-foreground"
};
function m({ status: t = "offline", label: n, pulse: r = !1, className: e }) {
  return /* @__PURE__ */ u("span", { className: o("inline-flex items-center gap-2 text-sm text-muted-foreground", e), children: [
    /* @__PURE__ */ s(
      "span",
      {
        className: o(
          "w-2 h-2 rounded-full shrink-0",
          b[t],
          r && "animate-pulse"
        )
      }
    ),
    n && /* @__PURE__ */ s("span", { children: n })
  ] });
}
export {
  c as Badge,
  m as StatusDot,
  p as badgeVariants
};
