// @btcv/ui/dist/Button.js
import { jsx as s, jsxs as u } from "react/jsx-runtime";
import { S as c } from "./_shared/index-BrLJJgkl.js";
import { c as g } from "./_shared/index-uPOYJZpG.js";
import { Loader2 as l } from "lucide-react";
import { c as o } from "./_shared/utils-eGXXUFl7.js";
const d = g(
  "relative inline-flex items-center justify-center gap-2 whitespace-nowrap font-medium border border-transparent cursor-pointer transition-all select-none outline-none focus-visible:border-ring focus-visible:ring-3 focus-visible:ring-ring/50 disabled:opacity-40 disabled:cursor-not-allowed disabled:pointer-events-none [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0",
  {
    variants: {
      variant: {
        primary: "bg-primary text-primary-foreground border-primary hover:not-disabled:bg-primary-light hover:not-disabled:border-primary-light hover:not-disabled:shadow-[0_0_20px_rgba(231,187,29,0.15)]",
        secondary: "bg-secondary text-foreground border-secondary hover:not-disabled:bg-muted hover:not-disabled:border-secondary focus-visible:border-foreground/30 focus-visible:ring-foreground/15",
        outline: "bg-input/30 border-border hover:not-disabled:bg-input/50 focus-visible:border-foreground/30 focus-visible:ring-foreground/15",
        destructive: "bg-destructive text-destructive-foreground border-transparent hover:not-disabled:bg-destructive-light focus-visible:border-destructive focus-visible:ring-destructive/50",
        success: "bg-success text-success-foreground border-transparent hover:not-disabled:bg-success-light focus-visible:border-success focus-visible:ring-success/50",
        warning: "bg-warning text-warning-foreground border-transparent hover:not-disabled:bg-warning-light focus-visible:border-warning focus-visible:ring-warning/50",
        info: "bg-info text-info-foreground border-transparent hover:not-disabled:bg-info-light focus-visible:border-info focus-visible:ring-info/50",
        ghost: "bg-transparent text-muted-foreground border-transparent hover:not-disabled:bg-secondary/50 hover:not-disabled:text-foreground focus-visible:border-foreground/30 focus-visible:ring-foreground/15",
        "ghost-primary": "bg-transparent text-primary border-transparent hover:not-disabled:bg-primary/15",
        "ghost-destructive": "bg-transparent text-destructive border-transparent hover:not-disabled:bg-destructive/15 focus-visible:border-destructive focus-visible:ring-destructive/30",
        "ghost-success": "bg-transparent text-success border-transparent hover:not-disabled:bg-success/15 focus-visible:border-success focus-visible:ring-success/30",
        "ghost-warning": "bg-transparent text-warning border-transparent hover:not-disabled:bg-warning/15 focus-visible:border-warning focus-visible:ring-warning/30",
        "ghost-info": "bg-transparent text-info border-transparent hover:not-disabled:bg-info/15 focus-visible:border-info focus-visible:ring-info/30",
        semi: "bg-foreground/15 text-foreground border-transparent hover:not-disabled:bg-foreground/25 focus-visible:border-foreground/30 focus-visible:ring-foreground/15",
        "semi-primary": "bg-primary/15 text-primary border-transparent hover:not-disabled:bg-primary/25",
        "semi-destructive": "bg-destructive/15 text-destructive border-transparent hover:not-disabled:bg-destructive/25 focus-visible:border-destructive focus-visible:ring-destructive/30",
        "semi-success": "bg-success/15 text-success border-transparent hover:not-disabled:bg-success/25 focus-visible:border-success focus-visible:ring-success/30",
        "semi-warning": "bg-warning/15 text-warning border-transparent hover:not-disabled:bg-warning/25 focus-visible:border-warning focus-visible:ring-warning/30",
        "semi-info": "bg-info/15 text-info border-transparent hover:not-disabled:bg-info/25 focus-visible:border-info focus-visible:ring-info/30",
        link: "bg-transparent text-primary border-transparent underline-offset-4 hover:not-disabled:underline"
      },
      size: {
        sm: "text-xs px-3 py-1 rounded-xl h-7",
        md: "text-sm px-4 py-2 rounded-2xl h-9",
        lg: "text-base px-6 py-3 rounded-3xl h-11",
        icon: "size-9 rounded-2xl"
      }
    },
    defaultVariants: {
      variant: "primary",
      size: "md"
    }
  }
);
function x({ className: n, variant: e, size: r, loading: i = !1, asChild: b = !1, children: a, ...t }) {
  return b ? /* @__PURE__ */ s(
    c,
    {
      "data-slot": "button",
      "data-variant": e,
      "data-size": r,
      className: o(d({ variant: e, size: r, className: n })),
      ...t,
      children: a
    }
  ) : /* @__PURE__ */ u(
    "button",
    {
      "data-slot": "button",
      "data-variant": e,
      "data-size": r,
      disabled: i || t.disabled,
      className: o(d({ variant: e, size: r, className: n })),
      ...t,
      children: [
        i && /* @__PURE__ */ s("span", { className: "absolute inset-0 flex items-center justify-center", children: /* @__PURE__ */ s(l, { className: "animate-spin" }) }),
        /* @__PURE__ */ s("span", { className: o("inline-flex items-center justify-center gap-2", i && "invisible"), children: a })
      ]
    }
  );
}
export {
  x as Button,
  d as buttonVariants
};
