// @btcv/ui/dist/Card.js
import { jsx as d } from "react/jsx-runtime";
import { forwardRef as s } from "react";
import { c as o } from "./_shared/utils-eGXXUFl7.js";
const t = {
  primary: "bg-primary/10 dark:bg-primary/10 backdrop-blur-md",
  success: "bg-success/10 dark:bg-success/10 backdrop-blur-md",
  info: "bg-info/10 dark:bg-info/10 backdrop-blur-md",
  warning: "bg-warning/10 dark:bg-warning/10 backdrop-blur-md",
  destructive: "bg-destructive/10 dark:bg-destructive/10 backdrop-blur-md"
}, c = s(({ className: a, accent: r, ...e }, i) => /* @__PURE__ */ d(
  "div",
  {
    ref: i,
    className: o(
      "rounded-4xl flex flex-col gap-6 p-6",
      r ? t[r] : "bg-glass shadow-md ring-1 ring-foreground/5 dark:ring-foreground/10",
      a
    ),
    ...e
  }
));
c.displayName = "Card";
const n = s(({ className: a, ...r }, e) => /* @__PURE__ */ d(
  "div",
  {
    ref: e,
    className: o("flex flex-col space-y-1.5", a),
    ...r
  }
));
n.displayName = "CardHeader";
const l = s(({ className: a, ...r }, e) => /* @__PURE__ */ d(
  "div",
  {
    ref: e,
    className: o("text-base font-semibold leading-none tracking-tight", a),
    ...r
  }
));
l.displayName = "CardTitle";
const m = s(({ className: a, ...r }, e) => /* @__PURE__ */ d(
  "div",
  {
    ref: e,
    className: o("text-sm text-muted-foreground", a),
    ...r
  }
));
m.displayName = "CardDescription";
const g = s(({ className: a, ...r }, e) => /* @__PURE__ */ d(
  "div",
  {
    ref: e,
    className: o(a),
    ...r
  }
));
g.displayName = "CardContent";
const p = s(({ className: a, ...r }, e) => /* @__PURE__ */ d(
  "div",
  {
    ref: e,
    className: o("flex items-center gap-2", a),
    ...r
  }
));
p.displayName = "CardFooter";
export {
  c as Card,
  g as CardContent,
  m as CardDescription,
  p as CardFooter,
  n as CardHeader,
  l as CardTitle
};
