// @btcv/ui/dist/CodeBlock.js
"use client";
import { jsxs as h, Fragment as C, jsx as s } from "react/jsx-runtime";
import { useState as v, useCallback as w } from "react";
import { c as y } from "./_shared/utils-eGXXUFl7.js";
import { Card as S } from "./Card.js";
import { File as j, Check as k, Copy as N } from "lucide-react";
const x = {
  js: [
    { pattern: /(\/\/.*$)/gm, type: "comment" },
    { pattern: /(\/\*[\s\S]*?\*\/)/gm, type: "comment" },
    { pattern: /("(?:[^"\\]|\\.)*"|'(?:[^'\\]|\\.)*'|`(?:[^`\\]|\\.)*`)/g, type: "string" },
    { pattern: /\b(\d+(?:\.\d+)?)\b/g, type: "number" },
    { pattern: /\b(const|let|var|function|return|if|else|for|while|do|switch|case|break|continue|new|delete|typeof|instanceof|in|of|class|extends|super|import|export|from|default|async|await|try|catch|finally|throw|yield|this|null|undefined|true|false|void)\b/g, type: "keyword" },
    { pattern: /\b(console|window|document|Array|Object|String|Number|Boolean|Promise|Map|Set|RegExp|Error|JSON|Math)\b/g, type: "builtin" },
    { pattern: /([a-zA-Z_$][\w$]*)\s*(?=\()/g, type: "function" },
    { pattern: /(&lt;\/?)([a-zA-Z][\w.]*)/g, type: "tag" }
  ],
  jsx: "js",
  ts: "js",
  tsx: "js",
  typescript: "js",
  javascript: "js",
  css: [
    { pattern: /(\/\*[\s\S]*?\*\/)/gm, type: "comment" },
    { pattern: /("(?:[^"\\]|\\.)*"|'(?:[^'\\]|\\.)*')/g, type: "string" },
    { pattern: /(\d+(?:\.\d+)?(?:px|em|rem|%|vh|vw|ch|ex|deg|s|ms)?)\b/g, type: "number" },
    { pattern: /([@#.:][\w-]+)/g, type: "selector" },
    { pattern: /\b(var|calc|rgb|rgba|hsl|hsla|url|linear-gradient|radial-gradient)\b/g, type: "builtin" },
    { pattern: /(--[\w-]+)/g, type: "property" },
    { pattern: /\b(!important)\b/g, type: "keyword" }
  ],
  html: [
    { pattern: /(&lt;!--[\s\S]*?--&gt;)/g, type: "comment" },
    { pattern: /("(?:[^"\\]|\\.)*"|'(?:[^'\\]|\\.)*')/g, type: "string" },
    { pattern: /(&lt;\/?)(\w[\w-]*)/g, type: "tag" },
    { pattern: /([\w-]+)(?==)/g, type: "property" }
  ],
  json: [
    { pattern: /("(?:[^"\\]|\\.)*")\s*:/g, type: "property" },
    { pattern: /:\s*("(?:[^"\\]|\\.)*")/g, type: "string" },
    { pattern: /\b(\d+(?:\.\d+)?)\b/g, type: "number" },
    { pattern: /\b(true|false|null)\b/g, type: "keyword" }
  ],
  bash: [
    { pattern: /(#.*$)/gm, type: "comment" },
    { pattern: /("(?:[^"\\]|\\.)*"|'(?:[^'\\]|\\.)*')/g, type: "string" },
    { pattern: /\b(if|then|else|elif|fi|for|while|do|done|case|esac|function|return|in|select|until|export|source|alias|unalias|readonly)\b/g, type: "keyword" },
    { pattern: /(\$[\w{][\w}]*)/g, type: "builtin" },
    { pattern: /^(\s*[\w.-]+)/gm, type: "function" }
  ],
  sh: "bash",
  shell: "bash",
  python: [
    { pattern: /(#.*$)/gm, type: "comment" },
    { pattern: /("""[\s\S]*?"""|'''[\s\S]*?''')/g, type: "string" },
    { pattern: /("(?:[^"\\]|\\.)*"|'(?:[^'\\]|\\.)*')/g, type: "string" },
    { pattern: /\b(\d+(?:\.\d+)?)\b/g, type: "number" },
    { pattern: /\b(def|class|if|elif|else|for|while|return|import|from|as|try|except|finally|raise|with|yield|lambda|pass|break|continue|and|or|not|is|in|True|False|None|self|async|await)\b/g, type: "keyword" },
    { pattern: /\b(print|len|range|type|int|str|float|list|dict|set|tuple|bool|input|open|super|isinstance|hasattr|getattr|setattr)\b/g, type: "builtin" },
    { pattern: /([a-zA-Z_]\w*)\s*(?=\()/g, type: "function" }
  ],
  py: "python"
}, T = {
  comment: "var(--codeblock-comment, var(--muted-foreground))",
  string: "var(--codeblock-string, var(--success))",
  number: "var(--codeblock-number, var(--info))",
  keyword: "var(--codeblock-keyword, var(--primary))",
  builtin: "var(--codeblock-builtin, var(--warning))",
  function: "var(--codeblock-function, var(--info))",
  tag: "var(--codeblock-tag, var(--destructive))",
  selector: "var(--codeblock-selector, var(--primary))",
  property: "var(--codeblock-property, var(--foreground))"
};
function L(a) {
  return a.replace(/&/g, "&amp;").replace(/</g, "&lt;").replace(/>/g, "&gt;");
}
function $(a) {
  let n = x[a];
  for (; typeof n == "string"; )
    n = x[n];
  return n || null;
}
function E(a, n) {
  const r = L(a);
  if (!n) return r;
  const i = $(n);
  if (!i) return r;
  const p = [];
  for (const t of i) {
    const m = new RegExp(t.pattern.source, t.pattern.flags);
    let u;
    for (; (u = m.exec(r)) !== null; ) {
      const e = u.length > 1 ? 1 : 0, g = u[e], f = u.index + (e > 0 ? u[0].indexOf(g) : 0);
      p.push({
        start: f,
        end: f + g.length,
        text: g,
        type: t.type
      });
    }
  }
  p.sort((t, m) => t.start - m.start || m.end - t.end);
  const o = [];
  let d = 0;
  for (const t of p)
    t.start >= d && (o.push(t), d = t.end);
  let c = "", l = 0;
  for (const t of o)
    t.start > l && (c += r.slice(l, t.start)), c += `<span style="color:${T[t.type]}">${t.text}</span>`, l = t.end;
  return l < r.length && (c += r.slice(l)), c;
}
const z = {
  js: "JavaScript",
  jsx: "JSX",
  ts: "TypeScript",
  tsx: "TSX",
  javascript: "JavaScript",
  typescript: "TypeScript",
  css: "CSS",
  html: "HTML",
  json: "JSON",
  bash: "Bash",
  sh: "Shell",
  shell: "Shell",
  python: "Python",
  py: "Python"
};
function O({ text: a, className: n }) {
  const [r, i] = v(!1), p = w(async () => {
    try {
      await navigator.clipboard.writeText(a), i(!0), setTimeout(() => i(!1), 2e3);
    } catch {
      const o = document.createElement("textarea");
      o.value = a, o.style.position = "fixed", o.style.opacity = "0", document.body.appendChild(o), o.select(), document.execCommand("copy"), document.body.removeChild(o), i(!0), setTimeout(() => i(!1), 2e3);
    }
  }, [a]);
  return /* @__PURE__ */ s(
    "button",
    {
      onClick: p,
      className: y(
        "inline-flex items-center justify-center rounded-xl p-1.5",
        "text-muted-foreground hover:text-foreground",
        "bg-transparent hover:bg-secondary",
        "transition-all duration-150",
        "focus-visible:outline-none focus-visible:border-ring focus-visible:ring-3 focus-visible:ring-ring/50",
        n
      ),
      "aria-label": r ? "Copied" : "Copy code",
      children: r ? /* @__PURE__ */ s(k, { className: "size-3.5 text-success" }) : /* @__PURE__ */ s(N, { className: "size-3.5" })
    }
  );
}
function A({
  number: a,
  content: n,
  rawText: r,
  highlighted: i,
  showLineNumbers: p,
  diff: o
}) {
  const d = o === "add", c = o === "remove", [l, t] = v(!1), m = w(async (u) => {
    u.stopPropagation();
    try {
      await navigator.clipboard.writeText(r);
    } catch {
      const e = document.createElement("textarea");
      e.value = r, e.style.position = "fixed", e.style.opacity = "0", document.body.appendChild(e), e.select(), document.execCommand("copy"), document.body.removeChild(e);
    }
    t(!0), setTimeout(() => t(!1), 1500);
  }, [r]);
  return /* @__PURE__ */ h(
    "div",
    {
      className: y(
        "group/line flex min-h-[1.625rem] leading-relaxed relative",
        "transition-colors duration-75",
        i && "bg-primary/8",
        d && "bg-success/15",
        c && "bg-destructive/15",
        !i && !d && !c && "hover:bg-secondary/50"
      ),
      children: [
        o !== void 0 && /* @__PURE__ */ s(
          "span",
          {
            className: y(
              "select-none w-5 shrink-0 text-center text-xs leading-relaxed",
              d && "text-success",
              c && "text-destructive"
            ),
            children: d ? "+" : c ? "-" : " "
          }
        ),
        p && /* @__PURE__ */ s(
          "span",
          {
            className: y(
              "select-none shrink-0 w-10 pr-3 text-right text-xs leading-relaxed tabular-nums",
              "text-muted-foreground",
              i && "text-primary/60"
            ),
            children: a
          }
        ),
        /* @__PURE__ */ s(
          "span",
          {
            className: "flex-1 pr-8 pl-3",
            dangerouslySetInnerHTML: { __html: n }
          }
        ),
        /* @__PURE__ */ s(
          "button",
          {
            onClick: m,
            className: y(
              "absolute right-1 top-1/2 -translate-y-1/2 p-0.5 rounded",
              "text-muted-foreground hover:text-foreground",
              "opacity-0 group-hover/line:opacity-100",
              "transition-opacity duration-100 cursor-pointer",
              l && "opacity-100"
            ),
            "aria-label": "Copy line",
            children: l ? /* @__PURE__ */ s(k, { className: "size-3 text-success" }) : /* @__PURE__ */ s(N, { className: "size-3" })
          }
        )
      ]
    }
  );
}
function P({
  children: a,
  filename: n,
  language: r,
  lineNumbers: i = !0,
  highlightLines: p = [],
  showCopyButton: o = !0,
  className: d,
  "aria-label": c
}) {
  const l = typeof a == "string" ? a.replace(/\n$/, "") : "", t = l.split(`
`), m = t.some(
    (e) => /^[+-][^+-]/.test(e) || /^[+-]$/.test(e)
  ), u = t.map((e, g) => {
    let f, b = e;
    return m && (e.startsWith("+") ? (f = "add", b = e.slice(1)) : e.startsWith("-") ? (f = "remove", b = e.slice(1)) : e.startsWith(" ") && (b = e.slice(1))), {
      number: g + 1,
      raw: b,
      highlighted: p.includes(g + 1),
      diff: f,
      html: E(b, r)
    };
  });
  return /* @__PURE__ */ h(
    S,
    {
      className: y(
        "overflow-hidden !p-0 !gap-0",
        "text-sm font-mono",
        d
      ),
      role: "region",
      "aria-label": c || (n ? `Code: ${n}` : "Code block"),
      children: [
        (n || o) && /* @__PURE__ */ h(
          "div",
          {
            className: y(
              "flex items-center justify-between gap-2",
              "px-3 py-2 border-b border-border",
              "bg-secondary/30"
            ),
            children: [
              /* @__PURE__ */ h("div", { className: "flex items-center gap-2 min-w-0 text-xs text-muted-foreground", children: [
                n && /* @__PURE__ */ h(C, { children: [
                  /* @__PURE__ */ s(j, { className: "size-3.5 text-muted-foreground shrink-0" }),
                  /* @__PURE__ */ s("span", { className: "truncate", children: n })
                ] }),
                n && r && /* @__PURE__ */ s("span", { children: " — " }),
                r && /* @__PURE__ */ s("span", { children: z[r] || r })
              ] }),
              o && /* @__PURE__ */ s(O, { text: l })
            ]
          }
        ),
        /* @__PURE__ */ s("div", { className: "overflow-x-auto", children: /* @__PURE__ */ s("pre", { className: "m-0 py-3 text-[13px] leading-relaxed", children: /* @__PURE__ */ s("code", { children: u.map((e) => /* @__PURE__ */ s(
          A,
          {
            number: e.number,
            content: e.html,
            rawText: e.raw,
            highlighted: e.highlighted,
            showLineNumbers: i,
            diff: e.diff
          },
          e.number
        )) }) }) })
      ]
    }
  );
}
export {
  P as CodeBlock,
  O as CopyButton
};
