// @btcv/ui/dist/Collapse.js
"use client";
import { jsxs as k, jsx as e } from "react/jsx-runtime";
import { createContext as H, useState as v, useRef as j, useId as q, useContext as z, useCallback as I, useEffect as w } from "react";
import { ChevronDown as F } from "lucide-react";
import { c as b } from "./_shared/utils-eGXXUFl7.js";
const y = H(null);
function G({
  title: x,
  children: g,
  icon: s,
  defaultOpen: r = !1,
  open: n,
  onOpenChange: i,
  className: p,
  ...l
}) {
  const [a, h] = v(r), C = j(null), [A, c] = v(r ? "auto" : "0px"), d = q(), o = z(y), u = n !== void 0, f = o !== null && !u, m = u ? n : f ? o.openId === d : a, E = I(() => {
    u ? i?.(!n) : f ? o.toggle(d) : h((t) => !t);
  }, [u, n, i, f, o, d]);
  return w(() => {
    f && r && o.openId === null && o.toggle(d);
  }, []), w(() => {
    const t = C.current;
    if (t)
      if (m) {
        c(`${t.scrollHeight}px`);
        const N = () => c("auto");
        return t.addEventListener("transitionend", N, { once: !0 }), () => t.removeEventListener("transitionend", N);
      } else
        c(`${t.scrollHeight}px`), requestAnimationFrame(() => {
          requestAnimationFrame(() => {
            c("0px");
          });
        });
  }, [m]), /* @__PURE__ */ k(
    "div",
    {
      className: b(
        "bg-glass border border-border rounded-2xl overflow-hidden transition-colors hover:border-secondary ring-1 ring-foreground/5 dark:ring-foreground/10",
        p
      ),
      ...l,
      children: [
        /* @__PURE__ */ k(
          "button",
          {
            type: "button",
            onClick: E,
            "aria-expanded": m,
            className: "flex items-center gap-3 w-full px-4 py-3 text-left text-sm font-medium text-foreground cursor-pointer",
            children: [
              s && /* @__PURE__ */ e("span", { className: "text-muted-foreground shrink-0 inline-flex items-center", children: /* @__PURE__ */ e(s, { size: 16 }) }),
              /* @__PURE__ */ e("span", { className: "flex-1 min-w-0", children: x }),
              /* @__PURE__ */ e(
                F,
                {
                  size: 16,
                  className: b(
                    "text-muted-foreground shrink-0 transition-transform duration-200",
                    m && "rotate-180"
                  )
                }
              )
            ]
          }
        ),
        /* @__PURE__ */ e(
          "div",
          {
            ref: C,
            style: { height: A },
            className: "overflow-hidden transition-[height] duration-200 ease-in-out",
            children: /* @__PURE__ */ e("div", { className: "px-4 pb-4 text-sm text-muted-foreground", children: g })
          }
        )
      ]
    }
  );
}
function P({ children: x, accordion: g = !1, className: s, ...r }) {
  const [n, i] = v(null), p = I((a) => {
    i((h) => h === a ? null : a);
  }, []), l = /* @__PURE__ */ e("div", { className: b("flex flex-col gap-2", s), ...r, children: x });
  return g ? /* @__PURE__ */ e(y.Provider, { value: { openId: n, toggle: p }, children: l }) : l;
}
export {
  G as Collapse,
  P as CollapseGroup
};
