// @btcv/ui/dist/Combobox.js
"use client";
import { jsxs as $, jsx as S } from "react/jsx-runtime";
import * as l from "react";
import { R as Se, P as Ie, O as Re, C as Ae } from "./_shared/index-DlSuDb9N.js";
import { P as A } from "./_shared/index-V-Ajw7Ac.js";
import { u as j } from "./_shared/index-BC7vfx-u.js";
import { c as L } from "./_shared/index-DLcqcWxM.js";
import { R as Me, T as Ne, P as $e, C as De } from "./_shared/index-C3aZemLI.js";
import { ChevronsUpDown as Pe, Search as ze, Check as Oe } from "lucide-react";
import { c as U } from "./_shared/utils-eGXXUFl7.js";
var pe = 1, je = 0.9, Fe = 0.8, Ke = 0.17, te = 0.1, re = 0.999, qe = 0.9999, Le = 0.99, Ve = /[\\\/_+.#"@\[\(\{&]/, Be = /[\\\/_+.#"@\[\(\{&]/g, Te = /[\s-]/, ge = /[\s-]/g;
function le(t, n, r, f, u, i, s) {
  if (i === n.length) return u === t.length ? pe : Le;
  var m = `${u},${i}`;
  if (s[m] !== void 0) return s[m];
  for (var h = f.charAt(i), o = r.indexOf(h, u), d = 0, g, C, w, I; o >= 0; ) g = le(t, n, r, f, o + 1, i + 1, s), g > d && (o === u ? g *= pe : Ve.test(t.charAt(o - 1)) ? (g *= Fe, w = t.slice(u, o - 1).match(Be), w && u > 0 && (g *= Math.pow(re, w.length))) : Te.test(t.charAt(o - 1)) ? (g *= je, I = t.slice(u, o - 1).match(ge), I && u > 0 && (g *= Math.pow(re, I.length))) : (g *= Ke, u > 0 && (g *= Math.pow(re, o - u))), t.charAt(o) !== n.charAt(i) && (g *= qe)), (g < te && r.charAt(o - 1) === f.charAt(i + 1) || f.charAt(i + 1) === f.charAt(i) && r.charAt(o - 1) !== f.charAt(i)) && (C = le(t, n, r, f, o + 1, i + 2, s), C * te > g && (g = C * te)), g > d && (d = g), o = r.indexOf(h, o + 1);
  return s[m] = d, d;
}
function ve(t) {
  return t.toLowerCase().replace(ge, " ");
}
function Ue(t, n, r) {
  return t = r && r.length > 0 ? `${t + " " + r.join(" ")}` : t, le(t, n, ve(t), ve(n), 0, 0, {});
}
var K = '[cmdk-group=""]', ne = '[cmdk-group-items=""]', He = '[cmdk-group-heading=""]', be = '[cmdk-item=""]', he = `${be}:not([aria-disabled="true"])`, ae = "cmdk-item-select", z = "data-value", We = (t, n, r) => Ue(t, n, r), xe = l.createContext(void 0), V = () => l.useContext(xe), ye = l.createContext(void 0), ue = () => l.useContext(ye), we = l.createContext(void 0), ke = l.forwardRef((t, n) => {
  let r = O(() => {
    var e, c;
    return { search: "", value: (c = (e = t.value) != null ? e : t.defaultValue) != null ? c : "", selectedItemId: void 0, filtered: { count: 0, items: /* @__PURE__ */ new Map(), groups: /* @__PURE__ */ new Set() } };
  }), f = O(() => /* @__PURE__ */ new Set()), u = O(() => /* @__PURE__ */ new Map()), i = O(() => /* @__PURE__ */ new Map()), s = O(() => /* @__PURE__ */ new Set()), m = Ee(t), { label: h, children: o, value: d, onValueChange: g, filter: C, shouldFilter: w, loop: I, disablePointerSelection: W = !1, vimBindings: M = !0, ...B } = t, G = j(), oe = j(), J = j(), N = l.useRef(null), x = nt();
  D(() => {
    if (d !== void 0) {
      let e = d.trim();
      r.current.value = e, k.emit();
    }
  }, [d]), D(() => {
    x(6, ce);
  }, []);
  let k = l.useMemo(() => ({ subscribe: (e) => (s.current.add(e), () => s.current.delete(e)), snapshot: () => r.current, setState: (e, c, p) => {
    var a, v, b, E;
    if (!Object.is(r.current[e], c)) {
      if (r.current[e] = c, e === "search") Z(), Y(), x(1, Q);
      else if (e === "value") {
        if (document.activeElement.hasAttribute("cmdk-input") || document.activeElement.hasAttribute("cmdk-root")) {
          let y = document.getElementById(J);
          y ? y.focus() : (a = document.getElementById(G)) == null || a.focus();
        }
        if (x(7, () => {
          var y;
          r.current.selectedItemId = (y = P()) == null ? void 0 : y.id, k.emit();
        }), p || x(5, ce), ((v = m.current) == null ? void 0 : v.value) !== void 0) {
          let y = c ?? "";
          (E = (b = m.current).onValueChange) == null || E.call(b, y);
          return;
        }
      }
      k.emit();
    }
  }, emit: () => {
    s.current.forEach((e) => e());
  } }), []), X = l.useMemo(() => ({ value: (e, c, p) => {
    var a;
    c !== ((a = i.current.get(e)) == null ? void 0 : a.value) && (i.current.set(e, { value: c, keywords: p }), r.current.filtered.items.set(e, ie(c, p)), x(2, () => {
      Y(), k.emit();
    }));
  }, item: (e, c) => (f.current.add(e), c && (u.current.has(c) ? u.current.get(c).add(e) : u.current.set(c, /* @__PURE__ */ new Set([e]))), x(3, () => {
    Z(), Y(), r.current.value || Q(), k.emit();
  }), () => {
    i.current.delete(e), f.current.delete(e), r.current.filtered.items.delete(e);
    let p = P();
    x(4, () => {
      Z(), p?.getAttribute("id") === e && Q(), k.emit();
    });
  }), group: (e) => (u.current.has(e) || u.current.set(e, /* @__PURE__ */ new Set()), () => {
    i.current.delete(e), u.current.delete(e);
  }), filter: () => m.current.shouldFilter, label: h || t["aria-label"], getDisablePointerSelection: () => m.current.disablePointerSelection, listId: G, inputId: J, labelId: oe, listInnerRef: N }), []);
  function ie(e, c) {
    var p, a;
    let v = (a = (p = m.current) == null ? void 0 : p.filter) != null ? a : We;
    return e ? v(e, r.current.search, c) : 0;
  }
  function Y() {
    if (!r.current.search || m.current.shouldFilter === !1) return;
    let e = r.current.filtered.items, c = [];
    r.current.filtered.groups.forEach((a) => {
      let v = u.current.get(a), b = 0;
      v.forEach((E) => {
        let y = e.get(E);
        b = Math.max(y, b);
      }), c.push([a, b]);
    });
    let p = N.current;
    F().sort((a, v) => {
      var b, E;
      let y = a.getAttribute("id"), T = v.getAttribute("id");
      return ((b = e.get(T)) != null ? b : 0) - ((E = e.get(y)) != null ? E : 0);
    }).forEach((a) => {
      let v = a.closest(ne);
      v ? v.appendChild(a.parentElement === v ? a : a.closest(`${ne} > *`)) : p.appendChild(a.parentElement === p ? a : a.closest(`${ne} > *`));
    }), c.sort((a, v) => v[1] - a[1]).forEach((a) => {
      var v;
      let b = (v = N.current) == null ? void 0 : v.querySelector(`${K}[${z}="${encodeURIComponent(a[0])}"]`);
      b?.parentElement.appendChild(b);
    });
  }
  function Q() {
    let e = F().find((p) => p.getAttribute("aria-disabled") !== "true"), c = e?.getAttribute(z);
    k.setState("value", c || void 0);
  }
  function Z() {
    var e, c, p, a;
    if (!r.current.search || m.current.shouldFilter === !1) {
      r.current.filtered.count = f.current.size;
      return;
    }
    r.current.filtered.groups = /* @__PURE__ */ new Set();
    let v = 0;
    for (let b of f.current) {
      let E = (c = (e = i.current.get(b)) == null ? void 0 : e.value) != null ? c : "", y = (a = (p = i.current.get(b)) == null ? void 0 : p.keywords) != null ? a : [], T = ie(E, y);
      r.current.filtered.items.set(b, T), T > 0 && v++;
    }
    for (let [b, E] of u.current) for (let y of E) if (r.current.filtered.items.get(y) > 0) {
      r.current.filtered.groups.add(b);
      break;
    }
    r.current.filtered.count = v;
  }
  function ce() {
    var e, c, p;
    let a = P();
    a && (((e = a.parentElement) == null ? void 0 : e.firstChild) === a && ((p = (c = a.closest(K)) == null ? void 0 : c.querySelector(He)) == null || p.scrollIntoView({ block: "nearest" })), a.scrollIntoView({ block: "nearest" }));
  }
  function P() {
    var e;
    return (e = N.current) == null ? void 0 : e.querySelector(`${be}[aria-selected="true"]`);
  }
  function F() {
    var e;
    return Array.from(((e = N.current) == null ? void 0 : e.querySelectorAll(he)) || []);
  }
  function _(e) {
    let c = F()[e];
    c && k.setState("value", c.getAttribute(z));
  }
  function ee(e) {
    var c;
    let p = P(), a = F(), v = a.findIndex((E) => E === p), b = a[v + e];
    (c = m.current) != null && c.loop && (b = v + e < 0 ? a[a.length - 1] : v + e === a.length ? a[0] : a[v + e]), b && k.setState("value", b.getAttribute(z));
  }
  function se(e) {
    let c = P(), p = c?.closest(K), a;
    for (; p && !a; ) p = e > 0 ? tt(p, K) : rt(p, K), a = p?.querySelector(he);
    a ? k.setState("value", a.getAttribute(z)) : ee(e);
  }
  let de = () => _(F().length - 1), fe = (e) => {
    e.preventDefault(), e.metaKey ? de() : e.altKey ? se(1) : ee(1);
  }, me = (e) => {
    e.preventDefault(), e.metaKey ? _(0) : e.altKey ? se(-1) : ee(-1);
  };
  return l.createElement(A.div, { ref: n, tabIndex: -1, ...B, "cmdk-root": "", onKeyDown: (e) => {
    var c;
    (c = B.onKeyDown) == null || c.call(B, e);
    let p = e.nativeEvent.isComposing || e.keyCode === 229;
    if (!(e.defaultPrevented || p)) switch (e.key) {
      case "n":
      case "j": {
        M && e.ctrlKey && fe(e);
        break;
      }
      case "ArrowDown": {
        fe(e);
        break;
      }
      case "p":
      case "k": {
        M && e.ctrlKey && me(e);
        break;
      }
      case "ArrowUp": {
        me(e);
        break;
      }
      case "Home": {
        e.preventDefault(), _(0);
        break;
      }
      case "End": {
        e.preventDefault(), de();
        break;
      }
      case "Enter": {
        e.preventDefault();
        let a = P();
        if (a) {
          let v = new Event(ae);
          a.dispatchEvent(v);
        }
      }
    }
  } }, l.createElement("label", { "cmdk-label": "", htmlFor: X.inputId, id: X.labelId, style: at }, h), H(t, (e) => l.createElement(ye.Provider, { value: k }, l.createElement(xe.Provider, { value: X }, e))));
}), Ge = l.forwardRef((t, n) => {
  var r, f;
  let u = j(), i = l.useRef(null), s = l.useContext(we), m = V(), h = Ee(t), o = (f = (r = h.current) == null ? void 0 : r.forceMount) != null ? f : s?.forceMount;
  D(() => {
    if (!o) return m.item(u, s?.id);
  }, [o]);
  let d = Ce(u, i, [t.value, t.children, i], t.keywords), g = ue(), C = R((x) => x.value && x.value === d.current), w = R((x) => o || m.filter() === !1 ? !0 : x.search ? x.filtered.items.get(u) > 0 : !0);
  l.useEffect(() => {
    let x = i.current;
    if (!(!x || t.disabled)) return x.addEventListener(ae, I), () => x.removeEventListener(ae, I);
  }, [w, t.onSelect, t.disabled]);
  function I() {
    var x, k;
    W(), (k = (x = h.current).onSelect) == null || k.call(x, d.current);
  }
  function W() {
    g.setState("value", d.current, !0);
  }
  if (!w) return null;
  let { disabled: M, value: B, onSelect: G, forceMount: oe, keywords: J, ...N } = t;
  return l.createElement(A.div, { ref: L(i, n), ...N, id: u, "cmdk-item": "", role: "option", "aria-disabled": !!M, "aria-selected": !!C, "data-disabled": !!M, "data-selected": !!C, onPointerMove: M || m.getDisablePointerSelection() ? void 0 : W, onClick: M ? void 0 : I }, t.children);
}), Je = l.forwardRef((t, n) => {
  let { heading: r, children: f, forceMount: u, ...i } = t, s = j(), m = l.useRef(null), h = l.useRef(null), o = j(), d = V(), g = R((w) => u || d.filter() === !1 ? !0 : w.search ? w.filtered.groups.has(s) : !0);
  D(() => d.group(s), []), Ce(s, m, [t.value, t.heading, h]);
  let C = l.useMemo(() => ({ id: s, forceMount: u }), [u]);
  return l.createElement(A.div, { ref: L(m, n), ...i, "cmdk-group": "", role: "presentation", hidden: g ? void 0 : !0 }, r && l.createElement("div", { ref: h, "cmdk-group-heading": "", "aria-hidden": !0, id: o }, r), H(t, (w) => l.createElement("div", { "cmdk-group-items": "", role: "group", "aria-labelledby": r ? o : void 0 }, l.createElement(we.Provider, { value: C }, w))));
}), Xe = l.forwardRef((t, n) => {
  let { alwaysRender: r, ...f } = t, u = l.useRef(null), i = R((s) => !s.search);
  return !r && !i ? null : l.createElement(A.div, { ref: L(u, n), ...f, "cmdk-separator": "", role: "separator" });
}), Ye = l.forwardRef((t, n) => {
  let { onValueChange: r, ...f } = t, u = t.value != null, i = ue(), s = R((o) => o.search), m = R((o) => o.selectedItemId), h = V();
  return l.useEffect(() => {
    t.value != null && i.setState("search", t.value);
  }, [t.value]), l.createElement(A.input, { ref: n, ...f, "cmdk-input": "", autoComplete: "off", autoCorrect: "off", spellCheck: !1, "aria-autocomplete": "list", role: "combobox", "aria-expanded": !0, "aria-controls": h.listId, "aria-labelledby": h.labelId, "aria-activedescendant": m, id: h.inputId, type: "text", value: u ? t.value : s, onChange: (o) => {
    u || i.setState("search", o.target.value), r?.(o.target.value);
  } });
}), Qe = l.forwardRef((t, n) => {
  let { children: r, label: f = "Suggestions", ...u } = t, i = l.useRef(null), s = l.useRef(null), m = R((o) => o.selectedItemId), h = V();
  return l.useEffect(() => {
    if (s.current && i.current) {
      let o = s.current, d = i.current, g, C = new ResizeObserver(() => {
        g = requestAnimationFrame(() => {
          let w = o.offsetHeight;
          d.style.setProperty("--cmdk-list-height", w.toFixed(1) + "px");
        });
      });
      return C.observe(o), () => {
        cancelAnimationFrame(g), C.unobserve(o);
      };
    }
  }, []), l.createElement(A.div, { ref: L(i, n), ...u, "cmdk-list": "", role: "listbox", tabIndex: -1, "aria-activedescendant": m, "aria-label": f, id: h.listId }, H(t, (o) => l.createElement("div", { ref: L(s, h.listInnerRef), "cmdk-list-sizer": "" }, o)));
}), Ze = l.forwardRef((t, n) => {
  let { open: r, onOpenChange: f, overlayClassName: u, contentClassName: i, container: s, ...m } = t;
  return l.createElement(Se, { open: r, onOpenChange: f }, l.createElement(Ie, { container: s }, l.createElement(Re, { "cmdk-overlay": "", className: u }), l.createElement(Ae, { "aria-label": t.label, "cmdk-dialog": "", className: i }, l.createElement(ke, { ref: n, ...m }))));
}), _e = l.forwardRef((t, n) => R((r) => r.filtered.count === 0) ? l.createElement(A.div, { ref: n, ...t, "cmdk-empty": "", role: "presentation" }) : null), et = l.forwardRef((t, n) => {
  let { progress: r, children: f, label: u = "Loading...", ...i } = t;
  return l.createElement(A.div, { ref: n, ...i, "cmdk-loading": "", role: "progressbar", "aria-valuenow": r, "aria-valuemin": 0, "aria-valuemax": 100, "aria-label": u }, H(t, (s) => l.createElement("div", { "aria-hidden": !0 }, s)));
}), q = Object.assign(ke, { List: Qe, Item: Ge, Input: Ye, Group: Je, Separator: Xe, Dialog: Ze, Empty: _e, Loading: et });
function tt(t, n) {
  let r = t.nextElementSibling;
  for (; r; ) {
    if (r.matches(n)) return r;
    r = r.nextElementSibling;
  }
}
function rt(t, n) {
  let r = t.previousElementSibling;
  for (; r; ) {
    if (r.matches(n)) return r;
    r = r.previousElementSibling;
  }
}
function Ee(t) {
  let n = l.useRef(t);
  return D(() => {
    n.current = t;
  }), n;
}
var D = typeof window > "u" ? l.useEffect : l.useLayoutEffect;
function O(t) {
  let n = l.useRef();
  return n.current === void 0 && (n.current = t()), n;
}
function R(t) {
  let n = ue(), r = () => t(n.snapshot());
  return l.useSyncExternalStore(n.subscribe, r, r);
}
function Ce(t, n, r, f = []) {
  let u = l.useRef(), i = V();
  return D(() => {
    var s;
    let m = (() => {
      var o;
      for (let d of r) {
        if (typeof d == "string") return d.trim();
        if (typeof d == "object" && "current" in d) return d.current ? (o = d.current.textContent) == null ? void 0 : o.trim() : u.current;
      }
    })(), h = f.map((o) => o.trim());
    i.value(t, m, h), (s = n.current) == null || s.setAttribute(z, m), u.current = m;
  }), u;
}
var nt = () => {
  let [t, n] = l.useState(), r = O(() => /* @__PURE__ */ new Map());
  return D(() => {
    r.current.forEach((f) => f()), r.current = /* @__PURE__ */ new Map();
  }, [t]), (f, u) => {
    r.current.set(f, u), n({});
  };
};
function lt(t) {
  let n = t.type;
  return typeof n == "function" ? n(t.props) : "render" in n ? n.render(t.props) : t;
}
function H({ asChild: t, children: n }, r) {
  return t && l.isValidElement(n) ? l.cloneElement(lt(n), { ref: n.ref }, r(n.props.children)) : r(n);
}
var at = { position: "absolute", width: "1px", height: "1px", padding: "0", margin: "-1px", overflow: "hidden", clip: "rect(0, 0, 0, 0)", whiteSpace: "nowrap", borderWidth: "0" };
function pt({
  options: t,
  value: n,
  onChange: r,
  placeholder: f = "Rechercher...",
  emptyMessage: u = "Aucun resultat.",
  disabled: i = !1,
  className: s
}) {
  const [m, h] = l.useState(!1), o = t.find((d) => d.value === n);
  return /* @__PURE__ */ $(Me, { open: m, onOpenChange: h, children: [
    /* @__PURE__ */ S(Ne, { asChild: !0, children: /* @__PURE__ */ $(
      "button",
      {
        type: "button",
        role: "combobox",
        "aria-expanded": m,
        disabled: i,
        className: U(
          "flex items-center justify-between gap-2 w-full bg-input/50 border border-transparent rounded-2xl px-3 py-2 text-sm text-foreground transition-[color,box-shadow,border-color] cursor-pointer",
          "focus:border-ring focus:ring-3 focus:ring-ring/50 focus:outline-none",
          "disabled:opacity-50 disabled:cursor-not-allowed",
          !n && "text-muted-foreground",
          s
        ),
        children: [
          /* @__PURE__ */ $("span", { className: "flex items-center gap-2 truncate", children: [
            o?.icon && /* @__PURE__ */ S(o.icon, { className: "size-4 shrink-0" }),
            o ? o.label : f
          ] }),
          /* @__PURE__ */ S(Pe, { size: 16, className: "text-muted-foreground shrink-0" })
        ]
      }
    ) }),
    /* @__PURE__ */ S($e, { children: /* @__PURE__ */ S(
      De,
      {
        sideOffset: 4,
        align: "start",
        className: U(
          "z-50 w-[var(--radix-popover-trigger-width)] overflow-hidden rounded-2xl border border-border bg-glass shadow-lg",
          "animate-in fade-in-0 zoom-in-95"
        ),
        children: /* @__PURE__ */ $(q, { className: "flex flex-col", children: [
          /* @__PURE__ */ $("div", { className: "flex items-center gap-2 border-b border-border px-3", children: [
            /* @__PURE__ */ S(ze, { size: 16, className: "text-muted-foreground shrink-0" }),
            /* @__PURE__ */ S(
              q.Input,
              {
                placeholder: f,
                className: "flex-1 bg-transparent py-2 text-sm text-foreground placeholder:text-muted-foreground outline-none"
              }
            )
          ] }),
          /* @__PURE__ */ $(q.List, { className: "max-h-[300px] overflow-y-auto p-1", children: [
            /* @__PURE__ */ S(q.Empty, { className: "px-2 py-6 text-center text-sm text-muted-foreground", children: u }),
            t.map((d) => /* @__PURE__ */ $(
              q.Item,
              {
                value: d.label,
                onSelect: () => {
                  r(d.value === n ? "" : d.value), h(!1);
                },
                className: U(
                  "flex items-center gap-2 rounded-xl px-2 py-1.5 text-sm text-foreground cursor-pointer select-none outline-none transition-colors",
                  "data-[selected=true]:bg-primary/10 data-[selected=true]:text-foreground"
                ),
                children: [
                  d.icon && /* @__PURE__ */ S(d.icon, { className: "size-4 shrink-0" }),
                  /* @__PURE__ */ S("span", { className: "flex-1 truncate", children: d.label }),
                  /* @__PURE__ */ S(
                    Oe,
                    {
                      size: 14,
                      className: U(
                        "shrink-0 text-primary",
                        n === d.value ? "opacity-100" : "opacity-0"
                      )
                    }
                  )
                ]
              },
              d.value
            ))
          ] })
        ] })
      }
    ) })
  ] });
}
export {
  pt as Combobox
};
