// @btcv/ui/dist/DarkMode.js
"use client";
import { jsx as o } from "react/jsx-runtime";
import { useState as m, useEffect as f } from "react";
import { c as g } from "./_shared/utils-eGXXUFl7.js";
import { Sun as h, Moon as p } from "lucide-react";
const i = "btcv-theme";
function c() {
  return typeof window > "u" ? "dark" : localStorage.getItem(i) || "dark";
}
function d(e) {
  typeof document > "u" || (document.documentElement.className = e);
}
function b(e) {
  typeof window < "u" && localStorage.setItem(i, e), d(e);
}
function x() {
  const n = c() === "dark" ? "light" : "dark";
  return b(n), n;
}
function E({ size: e = "md", onChange: n, className: u, ...l }) {
  const [a, r] = m("dark");
  f(() => {
    const t = c();
    r(t), d(t);
  }, []);
  const s = e === "sm" ? 14 : e === "lg" ? 18 : 16;
  return /* @__PURE__ */ o(
    "button",
    {
      type: "button",
      onClick: () => {
        const t = x();
        r(t), n?.(t);
      },
      className: g(
        "inline-flex items-center justify-center cursor-pointer transition-all select-none outline-none",
        "bg-transparent text-muted-foreground border border-transparent",
        "hover:bg-accent hover:text-foreground",
        "focus-visible:border-ring focus-visible:ring-3 focus-visible:ring-ring/50",
        {
          sm: "h-7 w-7 rounded-xl",
          md: "h-9 w-9 rounded-xl",
          lg: "h-11 w-11 rounded-xl"
        }[e],
        u
      ),
      ...l,
      children: a === "dark" ? /* @__PURE__ */ o(h, { size: s }) : /* @__PURE__ */ o(p, { size: s })
    }
  );
}
export {
  E as ThemeToggle,
  b as setTheme,
  x as toggleTheme
};
