// @btcv/ui/dist/DatePicker.js
"use client";
import { jsx as V, jsxs as qe } from "react/jsx-runtime";
import * as Wn from "react";
import h, { createContext as Sn, useContext as Cn, useCallback as $, useRef as Se, useLayoutEffect as Nn, useState as Qe, useEffect as xn, useMemo as Ce } from "react";
import { R as Yn, T as Tn, P as Pn, C as En } from "./_shared/index-C3aZemLI.js";
import { Calendar as _n, ChevronLeft as Fn, ChevronRight as Bn } from "lucide-react";
import { c as Ne } from "./_shared/utils-eGXXUFl7.js";
function In(e, t, r = "long") {
  return new Intl.DateTimeFormat("en-US", {
    // Enforces engine to render the time. Without the option JavaScriptCore omits it.
    hour: "numeric",
    timeZone: e,
    timeZoneName: r
  }).format(t).split(/\s/g).slice(2).join(" ");
}
const Hn = {}, ge = {};
function ie(e, t) {
  try {
    const n = (Hn[e] ||= new Intl.DateTimeFormat("en-US", {
      timeZone: e,
      timeZoneName: "longOffset"
    }).format)(t).split("GMT")[1];
    return n in ge ? ge[n] : ct(n, n.split(":"));
  } catch {
    if (e in ge) return ge[e];
    const r = e?.match(An);
    return r ? ct(e, r.slice(1)) : NaN;
  }
}
const An = /([+-]\d\d):?(\d\d)?/;
function ct(e, t) {
  const r = +(t[0] || 0), n = +(t[1] || 0), o = +(t[2] || 0) / 60;
  return ge[e] = r * 60 + n > 0 ? r * 60 + n + o : r * 60 - n - o;
}
class J extends Date {
  //#region static
  constructor(...t) {
    super(), t.length > 1 && typeof t[t.length - 1] == "string" && (this.timeZone = t.pop()), this.internal = /* @__PURE__ */ new Date(), isNaN(ie(this.timeZone, this)) ? this.setTime(NaN) : t.length ? typeof t[0] == "number" && (t.length === 1 || t.length === 2 && typeof t[1] != "number") ? this.setTime(t[0]) : typeof t[0] == "string" ? this.setTime(+new Date(t[0])) : t[0] instanceof Date ? this.setTime(+t[0]) : (this.setTime(+new Date(...t)), Dt(this), Xe(this)) : this.setTime(Date.now());
  }
  static tz(t, ...r) {
    return r.length ? new J(...r, t) : new J(Date.now(), t);
  }
  //#endregion
  //#region time zone
  withTimeZone(t) {
    return new J(+this, t);
  }
  getTimezoneOffset() {
    const t = -ie(this.timeZone, this);
    return t > 0 ? Math.floor(t) : Math.ceil(t);
  }
  //#endregion
  //#region time
  setTime(t) {
    return Date.prototype.setTime.apply(this, arguments), Xe(this), +this;
  }
  //#endregion
  //#region date-fns integration
  [/* @__PURE__ */ Symbol.for("constructDateFrom")](t) {
    return new J(+new Date(t), this.timeZone);
  }
  //#endregion
}
const ut = /^(get|set)(?!UTC)/;
Object.getOwnPropertyNames(Date.prototype).forEach((e) => {
  if (!ut.test(e)) return;
  const t = e.replace(ut, "$1UTC");
  J.prototype[t] && (e.startsWith("get") ? J.prototype[e] = function() {
    return this.internal[t]();
  } : (J.prototype[e] = function() {
    return Date.prototype[t].apply(this.internal, arguments), qn(this), +this;
  }, J.prototype[t] = function() {
    return Date.prototype[t].apply(this, arguments), Xe(this), +this;
  }));
});
function Xe(e) {
  e.internal.setTime(+e), e.internal.setUTCSeconds(e.internal.getUTCSeconds() - Math.round(-ie(e.timeZone, e) * 60));
}
function qn(e) {
  Date.prototype.setFullYear.call(e, e.internal.getUTCFullYear(), e.internal.getUTCMonth(), e.internal.getUTCDate()), Date.prototype.setHours.call(e, e.internal.getUTCHours(), e.internal.getUTCMinutes(), e.internal.getUTCSeconds(), e.internal.getUTCMilliseconds()), Dt(e);
}
function Dt(e) {
  const t = ie(e.timeZone, e), r = t > 0 ? Math.floor(t) : Math.ceil(t), n = /* @__PURE__ */ new Date(+e);
  n.setUTCHours(n.getUTCHours() - 1);
  const o = -(/* @__PURE__ */ new Date(+e)).getTimezoneOffset(), a = -(/* @__PURE__ */ new Date(+n)).getTimezoneOffset(), i = o - a, s = Date.prototype.getHours.apply(e) !== e.internal.getUTCHours();
  i && s && e.internal.setUTCMinutes(e.internal.getUTCMinutes() + i);
  const c = o - r;
  c && Date.prototype.setUTCMinutes.call(e, Date.prototype.getUTCMinutes.call(e) + c);
  const u = /* @__PURE__ */ new Date(+e);
  u.setUTCSeconds(0);
  const f = o > 0 ? u.getSeconds() : (u.getSeconds() - 60) % 60, l = Math.round(-(ie(e.timeZone, e) * 60)) % 60;
  (l || f) && (e.internal.setUTCSeconds(e.internal.getUTCSeconds() + l), Date.prototype.setUTCSeconds.call(e, Date.prototype.getUTCSeconds.call(e) + l + f));
  const m = ie(e.timeZone, e), g = m > 0 ? Math.floor(m) : Math.ceil(m), C = -(/* @__PURE__ */ new Date(+e)).getTimezoneOffset() - g, O = g !== r, Y = C - c;
  if (O && Y) {
    Date.prototype.setUTCMinutes.call(e, Date.prototype.getUTCMinutes.call(e) + Y);
    const M = ie(e.timeZone, e), p = M > 0 ? Math.floor(M) : Math.ceil(M), D = g - p;
    D && (e.internal.setUTCMinutes(e.internal.getUTCMinutes() + D), Date.prototype.setUTCMinutes.call(e, Date.prototype.getUTCMinutes.call(e) + D));
  }
}
class H extends J {
  //#region static
  static tz(t, ...r) {
    return r.length ? new H(...r, t) : new H(Date.now(), t);
  }
  //#endregion
  //#region representation
  toISOString() {
    const [t, r, n] = this.tzComponents(), o = `${t}${r}:${n}`;
    return this.internal.toISOString().slice(0, -1) + o;
  }
  toString() {
    return `${this.toDateString()} ${this.toTimeString()}`;
  }
  toDateString() {
    const [t, r, n, o] = this.internal.toUTCString().split(" ");
    return `${t?.slice(0, -1)} ${n} ${r} ${o}`;
  }
  toTimeString() {
    const t = this.internal.toUTCString().split(" ")[4], [r, n, o] = this.tzComponents();
    return `${t} GMT${r}${n}${o} (${In(this.timeZone, this)})`;
  }
  toLocaleString(t, r) {
    return Date.prototype.toLocaleString.call(this, t, {
      ...r,
      timeZone: r?.timeZone || this.timeZone
    });
  }
  toLocaleDateString(t, r) {
    return Date.prototype.toLocaleDateString.call(this, t, {
      ...r,
      timeZone: r?.timeZone || this.timeZone
    });
  }
  toLocaleTimeString(t, r) {
    return Date.prototype.toLocaleTimeString.call(this, t, {
      ...r,
      timeZone: r?.timeZone || this.timeZone
    });
  }
  //#endregion
  //#region private
  tzComponents() {
    const t = this.getTimezoneOffset(), r = t > 0 ? "-" : "+", n = String(Math.floor(Math.abs(t) / 60)).padStart(2, "0"), o = String(Math.abs(t) % 60).padStart(2, "0");
    return [r, n, o];
  }
  //#endregion
  withTimeZone(t) {
    return new H(+this, t);
  }
  //#region date-fns integration
  [/* @__PURE__ */ Symbol.for("constructDateFrom")](t) {
    return new H(+new Date(t), this.timeZone);
  }
  //#endregion
}
const kt = 6048e5, jn = 864e5, lt = /* @__PURE__ */ Symbol.for("constructDateFrom");
function B(e, t) {
  return typeof e == "function" ? e(t) : e && typeof e == "object" && lt in e ? e[lt](t) : e instanceof Date ? new e.constructor(t) : new Date(t);
}
function P(e, t) {
  return B(t || e, e);
}
function Ot(e, t, r) {
  const n = P(e, r?.in);
  return isNaN(t) ? B(e, NaN) : (t && n.setDate(n.getDate() + t), n);
}
function vt(e, t, r) {
  const n = P(e, r?.in);
  if (isNaN(t)) return B(e, NaN);
  if (!t)
    return n;
  const o = n.getDate(), a = B(e, n.getTime());
  a.setMonth(n.getMonth() + t + 1, 0);
  const i = a.getDate();
  return o >= i ? a : (n.setFullYear(
    a.getFullYear(),
    a.getMonth(),
    o
  ), n);
}
let Gn = {};
function Me() {
  return Gn;
}
function de(e, t) {
  const r = Me(), n = t?.weekStartsOn ?? t?.locale?.options?.weekStartsOn ?? r.weekStartsOn ?? r.locale?.options?.weekStartsOn ?? 0, o = P(e, t?.in), a = o.getDay(), i = (a < n ? 7 : 0) + a - n;
  return o.setDate(o.getDate() - i), o.setHours(0, 0, 0, 0), o;
}
function pe(e, t) {
  return de(e, { ...t, weekStartsOn: 1 });
}
function Wt(e, t) {
  const r = P(e, t?.in), n = r.getFullYear(), o = B(r, 0);
  o.setFullYear(n + 1, 0, 4), o.setHours(0, 0, 0, 0);
  const a = pe(o), i = B(r, 0);
  i.setFullYear(n, 0, 4), i.setHours(0, 0, 0, 0);
  const s = pe(i);
  return r.getTime() >= a.getTime() ? n + 1 : r.getTime() >= s.getTime() ? n : n - 1;
}
function ft(e) {
  const t = P(e), r = new Date(
    Date.UTC(
      t.getFullYear(),
      t.getMonth(),
      t.getDate(),
      t.getHours(),
      t.getMinutes(),
      t.getSeconds(),
      t.getMilliseconds()
    )
  );
  return r.setUTCFullYear(t.getFullYear()), +e - +r;
}
function he(e, ...t) {
  const r = B.bind(
    null,
    t.find((n) => typeof n == "object")
  );
  return t.map(r);
}
function we(e, t) {
  const r = P(e, t?.in);
  return r.setHours(0, 0, 0, 0), r;
}
function Ue(e, t, r) {
  const [n, o] = he(
    r?.in,
    e,
    t
  ), a = we(n), i = we(o), s = +a - ft(a), c = +i - ft(i);
  return Math.round((s - c) / jn);
}
function Rn(e, t) {
  const r = Wt(e, t), n = B(e, 0);
  return n.setFullYear(r, 0, 4), n.setHours(0, 0, 0, 0), pe(n);
}
function $n(e, t, r) {
  return Ot(e, t * 7, r);
}
function zn(e, t, r) {
  return vt(e, t * 12, r);
}
function Qn(e, t) {
  let r, n = t?.in;
  return e.forEach((o) => {
    !n && typeof o == "object" && (n = B.bind(null, o));
    const a = P(o, n);
    (!r || r < a || isNaN(+a)) && (r = a);
  }), B(n, r || NaN);
}
function Xn(e, t) {
  let r, n = t?.in;
  return e.forEach((o) => {
    !n && typeof o == "object" && (n = B.bind(null, o));
    const a = P(o, n);
    (!r || r > a || isNaN(+a)) && (r = a);
  }), B(n, r || NaN);
}
function Un(e, t, r) {
  const [n, o] = he(
    r?.in,
    e,
    t
  );
  return +we(n) == +we(o);
}
function St(e) {
  return e instanceof Date || typeof e == "object" && Object.prototype.toString.call(e) === "[object Date]";
}
function Vn(e) {
  return !(!St(e) && typeof e != "number" || isNaN(+P(e)));
}
function Ct(e, t, r) {
  const [n, o] = he(
    r?.in,
    e,
    t
  ), a = n.getFullYear() - o.getFullYear(), i = n.getMonth() - o.getMonth();
  return a * 12 + i;
}
function Zn(e, t) {
  const r = P(e, t?.in), n = r.getMonth();
  return r.setFullYear(r.getFullYear(), n + 1, 0), r.setHours(23, 59, 59, 999), r;
}
function Nt(e, t) {
  const [r, n] = he(e, t.start, t.end);
  return { start: r, end: n };
}
function Kn(e, t) {
  const { start: r, end: n } = Nt(t?.in, e);
  let o = +r > +n;
  const a = o ? +r : +n, i = o ? n : r;
  i.setHours(0, 0, 0, 0), i.setDate(1);
  let s = 1;
  const c = [];
  for (; +i <= a; )
    c.push(B(r, i)), i.setMonth(i.getMonth() + s);
  return o ? c.reverse() : c;
}
function Jn(e, t) {
  const r = P(e, t?.in);
  return r.setDate(1), r.setHours(0, 0, 0, 0), r;
}
function Ln(e, t) {
  const r = P(e, t?.in), n = r.getFullYear();
  return r.setFullYear(n + 1, 0, 0), r.setHours(23, 59, 59, 999), r;
}
function xt(e, t) {
  const r = P(e, t?.in);
  return r.setFullYear(r.getFullYear(), 0, 1), r.setHours(0, 0, 0, 0), r;
}
function er(e, t) {
  const { start: r, end: n } = Nt(t?.in, e);
  let o = +r > +n;
  const a = o ? +r : +n, i = o ? n : r;
  i.setHours(0, 0, 0, 0), i.setMonth(0, 1);
  let s = 1;
  const c = [];
  for (; +i <= a; )
    c.push(B(r, i)), i.setFullYear(i.getFullYear() + s);
  return o ? c.reverse() : c;
}
function Yt(e, t) {
  const r = Me(), n = t?.weekStartsOn ?? t?.locale?.options?.weekStartsOn ?? r.weekStartsOn ?? r.locale?.options?.weekStartsOn ?? 0, o = P(e, t?.in), a = o.getDay(), i = (a < n ? -7 : 0) + 6 - (a - n);
  return o.setDate(o.getDate() + i), o.setHours(23, 59, 59, 999), o;
}
function tr(e, t) {
  return Yt(e, { ...t, weekStartsOn: 1 });
}
const nr = {
  lessThanXSeconds: {
    one: "less than a second",
    other: "less than {{count}} seconds"
  },
  xSeconds: {
    one: "1 second",
    other: "{{count}} seconds"
  },
  halfAMinute: "half a minute",
  lessThanXMinutes: {
    one: "less than a minute",
    other: "less than {{count}} minutes"
  },
  xMinutes: {
    one: "1 minute",
    other: "{{count}} minutes"
  },
  aboutXHours: {
    one: "about 1 hour",
    other: "about {{count}} hours"
  },
  xHours: {
    one: "1 hour",
    other: "{{count}} hours"
  },
  xDays: {
    one: "1 day",
    other: "{{count}} days"
  },
  aboutXWeeks: {
    one: "about 1 week",
    other: "about {{count}} weeks"
  },
  xWeeks: {
    one: "1 week",
    other: "{{count}} weeks"
  },
  aboutXMonths: {
    one: "about 1 month",
    other: "about {{count}} months"
  },
  xMonths: {
    one: "1 month",
    other: "{{count}} months"
  },
  aboutXYears: {
    one: "about 1 year",
    other: "about {{count}} years"
  },
  xYears: {
    one: "1 year",
    other: "{{count}} years"
  },
  overXYears: {
    one: "over 1 year",
    other: "over {{count}} years"
  },
  almostXYears: {
    one: "almost 1 year",
    other: "almost {{count}} years"
  }
}, rr = (e, t, r) => {
  let n;
  const o = nr[e];
  return typeof o == "string" ? n = o : t === 1 ? n = o.one : n = o.other.replace("{{count}}", t.toString()), r?.addSuffix ? r.comparison && r.comparison > 0 ? "in " + n : n + " ago" : n;
};
function je(e) {
  return (t = {}) => {
    const r = t.width ? String(t.width) : e.defaultWidth;
    return e.formats[r] || e.formats[e.defaultWidth];
  };
}
const or = {
  full: "EEEE, MMMM do, y",
  long: "MMMM do, y",
  medium: "MMM d, y",
  short: "MM/dd/yyyy"
}, sr = {
  full: "h:mm:ss a zzzz",
  long: "h:mm:ss a z",
  medium: "h:mm:ss a",
  short: "h:mm a"
}, ar = {
  full: "{{date}} 'at' {{time}}",
  long: "{{date}} 'at' {{time}}",
  medium: "{{date}}, {{time}}",
  short: "{{date}}, {{time}}"
}, ir = {
  date: je({
    formats: or,
    defaultWidth: "full"
  }),
  time: je({
    formats: sr,
    defaultWidth: "full"
  }),
  dateTime: je({
    formats: ar,
    defaultWidth: "full"
  })
}, cr = {
  lastWeek: "'last' eeee 'at' p",
  yesterday: "'yesterday at' p",
  today: "'today at' p",
  tomorrow: "'tomorrow at' p",
  nextWeek: "eeee 'at' p",
  other: "P"
}, ur = (e, t, r, n) => cr[e];
function me(e) {
  return (t, r) => {
    const n = r?.context ? String(r.context) : "standalone";
    let o;
    if (n === "formatting" && e.formattingValues) {
      const i = e.defaultFormattingWidth || e.defaultWidth, s = r?.width ? String(r.width) : i;
      o = e.formattingValues[s] || e.formattingValues[i];
    } else {
      const i = e.defaultWidth, s = r?.width ? String(r.width) : e.defaultWidth;
      o = e.values[s] || e.values[i];
    }
    const a = e.argumentCallback ? e.argumentCallback(t) : t;
    return o[a];
  };
}
const lr = {
  narrow: ["B", "A"],
  abbreviated: ["BC", "AD"],
  wide: ["Before Christ", "Anno Domini"]
}, fr = {
  narrow: ["1", "2", "3", "4"],
  abbreviated: ["Q1", "Q2", "Q3", "Q4"],
  wide: ["1st quarter", "2nd quarter", "3rd quarter", "4th quarter"]
}, dr = {
  narrow: ["J", "F", "M", "A", "M", "J", "J", "A", "S", "O", "N", "D"],
  abbreviated: [
    "Jan",
    "Feb",
    "Mar",
    "Apr",
    "May",
    "Jun",
    "Jul",
    "Aug",
    "Sep",
    "Oct",
    "Nov",
    "Dec"
  ],
  wide: [
    "January",
    "February",
    "March",
    "April",
    "May",
    "June",
    "July",
    "August",
    "September",
    "October",
    "November",
    "December"
  ]
}, hr = {
  narrow: ["S", "M", "T", "W", "T", "F", "S"],
  short: ["Su", "Mo", "Tu", "We", "Th", "Fr", "Sa"],
  abbreviated: ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"],
  wide: [
    "Sunday",
    "Monday",
    "Tuesday",
    "Wednesday",
    "Thursday",
    "Friday",
    "Saturday"
  ]
}, mr = {
  narrow: {
    am: "a",
    pm: "p",
    midnight: "mi",
    noon: "n",
    morning: "morning",
    afternoon: "afternoon",
    evening: "evening",
    night: "night"
  },
  abbreviated: {
    am: "AM",
    pm: "PM",
    midnight: "midnight",
    noon: "noon",
    morning: "morning",
    afternoon: "afternoon",
    evening: "evening",
    night: "night"
  },
  wide: {
    am: "a.m.",
    pm: "p.m.",
    midnight: "midnight",
    noon: "noon",
    morning: "morning",
    afternoon: "afternoon",
    evening: "evening",
    night: "night"
  }
}, yr = {
  narrow: {
    am: "a",
    pm: "p",
    midnight: "mi",
    noon: "n",
    morning: "in the morning",
    afternoon: "in the afternoon",
    evening: "in the evening",
    night: "at night"
  },
  abbreviated: {
    am: "AM",
    pm: "PM",
    midnight: "midnight",
    noon: "noon",
    morning: "in the morning",
    afternoon: "in the afternoon",
    evening: "in the evening",
    night: "at night"
  },
  wide: {
    am: "a.m.",
    pm: "p.m.",
    midnight: "midnight",
    noon: "noon",
    morning: "in the morning",
    afternoon: "in the afternoon",
    evening: "in the evening",
    night: "at night"
  }
}, gr = (e, t) => {
  const r = Number(e), n = r % 100;
  if (n > 20 || n < 10)
    switch (n % 10) {
      case 1:
        return r + "st";
      case 2:
        return r + "nd";
      case 3:
        return r + "rd";
    }
  return r + "th";
}, br = {
  ordinalNumber: gr,
  era: me({
    values: lr,
    defaultWidth: "wide"
  }),
  quarter: me({
    values: fr,
    defaultWidth: "wide",
    argumentCallback: (e) => e - 1
  }),
  month: me({
    values: dr,
    defaultWidth: "wide"
  }),
  day: me({
    values: hr,
    defaultWidth: "wide"
  }),
  dayPeriod: me({
    values: mr,
    defaultWidth: "wide",
    formattingValues: yr,
    defaultFormattingWidth: "wide"
  })
};
function ye(e) {
  return (t, r = {}) => {
    const n = r.width, o = n && e.matchPatterns[n] || e.matchPatterns[e.defaultMatchWidth], a = t.match(o);
    if (!a)
      return null;
    const i = a[0], s = n && e.parsePatterns[n] || e.parsePatterns[e.defaultParseWidth], c = Array.isArray(s) ? wr(s, (l) => l.test(i)) : (
      // [TODO] -- I challenge you to fix the type
      pr(s, (l) => l.test(i))
    );
    let u;
    u = e.valueCallback ? e.valueCallback(c) : c, u = r.valueCallback ? (
      // [TODO] -- I challenge you to fix the type
      r.valueCallback(u)
    ) : u;
    const f = t.slice(i.length);
    return { value: u, rest: f };
  };
}
function pr(e, t) {
  for (const r in e)
    if (Object.prototype.hasOwnProperty.call(e, r) && t(e[r]))
      return r;
}
function wr(e, t) {
  for (let r = 0; r < e.length; r++)
    if (t(e[r]))
      return r;
}
function Mr(e) {
  return (t, r = {}) => {
    const n = t.match(e.matchPattern);
    if (!n) return null;
    const o = n[0], a = t.match(e.parsePattern);
    if (!a) return null;
    let i = e.valueCallback ? e.valueCallback(a[0]) : a[0];
    i = r.valueCallback ? r.valueCallback(i) : i;
    const s = t.slice(o.length);
    return { value: i, rest: s };
  };
}
const Dr = /^(\d+)(th|st|nd|rd)?/i, kr = /\d+/i, Or = {
  narrow: /^(b|a)/i,
  abbreviated: /^(b\.?\s?c\.?|b\.?\s?c\.?\s?e\.?|a\.?\s?d\.?|c\.?\s?e\.?)/i,
  wide: /^(before christ|before common era|anno domini|common era)/i
}, vr = {
  any: [/^b/i, /^(a|c)/i]
}, Wr = {
  narrow: /^[1234]/i,
  abbreviated: /^q[1234]/i,
  wide: /^[1234](th|st|nd|rd)? quarter/i
}, Sr = {
  any: [/1/i, /2/i, /3/i, /4/i]
}, Cr = {
  narrow: /^[jfmasond]/i,
  abbreviated: /^(jan|feb|mar|apr|may|jun|jul|aug|sep|oct|nov|dec)/i,
  wide: /^(january|february|march|april|may|june|july|august|september|october|november|december)/i
}, Nr = {
  narrow: [
    /^j/i,
    /^f/i,
    /^m/i,
    /^a/i,
    /^m/i,
    /^j/i,
    /^j/i,
    /^a/i,
    /^s/i,
    /^o/i,
    /^n/i,
    /^d/i
  ],
  any: [
    /^ja/i,
    /^f/i,
    /^mar/i,
    /^ap/i,
    /^may/i,
    /^jun/i,
    /^jul/i,
    /^au/i,
    /^s/i,
    /^o/i,
    /^n/i,
    /^d/i
  ]
}, xr = {
  narrow: /^[smtwf]/i,
  short: /^(su|mo|tu|we|th|fr|sa)/i,
  abbreviated: /^(sun|mon|tue|wed|thu|fri|sat)/i,
  wide: /^(sunday|monday|tuesday|wednesday|thursday|friday|saturday)/i
}, Yr = {
  narrow: [/^s/i, /^m/i, /^t/i, /^w/i, /^t/i, /^f/i, /^s/i],
  any: [/^su/i, /^m/i, /^tu/i, /^w/i, /^th/i, /^f/i, /^sa/i]
}, Tr = {
  narrow: /^(a|p|mi|n|(in the|at) (morning|afternoon|evening|night))/i,
  any: /^([ap]\.?\s?m\.?|midnight|noon|(in the|at) (morning|afternoon|evening|night))/i
}, Pr = {
  any: {
    am: /^a/i,
    pm: /^p/i,
    midnight: /^mi/i,
    noon: /^no/i,
    morning: /morning/i,
    afternoon: /afternoon/i,
    evening: /evening/i,
    night: /night/i
  }
}, Er = {
  ordinalNumber: Mr({
    matchPattern: Dr,
    parsePattern: kr,
    valueCallback: (e) => parseInt(e, 10)
  }),
  era: ye({
    matchPatterns: Or,
    defaultMatchWidth: "wide",
    parsePatterns: vr,
    defaultParseWidth: "any"
  }),
  quarter: ye({
    matchPatterns: Wr,
    defaultMatchWidth: "wide",
    parsePatterns: Sr,
    defaultParseWidth: "any",
    valueCallback: (e) => e + 1
  }),
  month: ye({
    matchPatterns: Cr,
    defaultMatchWidth: "wide",
    parsePatterns: Nr,
    defaultParseWidth: "any"
  }),
  day: ye({
    matchPatterns: xr,
    defaultMatchWidth: "wide",
    parsePatterns: Yr,
    defaultParseWidth: "any"
  }),
  dayPeriod: ye({
    matchPatterns: Tr,
    defaultMatchWidth: "any",
    parsePatterns: Pr,
    defaultParseWidth: "any"
  })
}, fe = {
  code: "en-US",
  formatDistance: rr,
  formatLong: ir,
  formatRelative: ur,
  localize: br,
  match: Er,
  options: {
    weekStartsOn: 0,
    firstWeekContainsDate: 1
  }
};
function _r(e, t) {
  const r = P(e, t?.in);
  return Ue(r, xt(r)) + 1;
}
function Ve(e, t) {
  const r = P(e, t?.in), n = +pe(r) - +Rn(r);
  return Math.round(n / kt) + 1;
}
function Tt(e, t) {
  const r = P(e, t?.in), n = r.getFullYear(), o = Me(), a = t?.firstWeekContainsDate ?? t?.locale?.options?.firstWeekContainsDate ?? o.firstWeekContainsDate ?? o.locale?.options?.firstWeekContainsDate ?? 1, i = B(t?.in || e, 0);
  i.setFullYear(n + 1, 0, a), i.setHours(0, 0, 0, 0);
  const s = de(i, t), c = B(t?.in || e, 0);
  c.setFullYear(n, 0, a), c.setHours(0, 0, 0, 0);
  const u = de(c, t);
  return +r >= +s ? n + 1 : +r >= +u ? n : n - 1;
}
function Fr(e, t) {
  const r = Me(), n = t?.firstWeekContainsDate ?? t?.locale?.options?.firstWeekContainsDate ?? r.firstWeekContainsDate ?? r.locale?.options?.firstWeekContainsDate ?? 1, o = Tt(e, t), a = B(t?.in || e, 0);
  return a.setFullYear(o, 0, n), a.setHours(0, 0, 0, 0), de(a, t);
}
function Ze(e, t) {
  const r = P(e, t?.in), n = +de(r, t) - +Fr(r, t);
  return Math.round(n / kt) + 1;
}
function T(e, t) {
  const r = e < 0 ? "-" : "", n = Math.abs(e).toString().padStart(t, "0");
  return r + n;
}
const se = {
  // Year
  y(e, t) {
    const r = e.getFullYear(), n = r > 0 ? r : 1 - r;
    return T(t === "yy" ? n % 100 : n, t.length);
  },
  // Month
  M(e, t) {
    const r = e.getMonth();
    return t === "M" ? String(r + 1) : T(r + 1, 2);
  },
  // Day of the month
  d(e, t) {
    return T(e.getDate(), t.length);
  },
  // AM or PM
  a(e, t) {
    const r = e.getHours() / 12 >= 1 ? "pm" : "am";
    switch (t) {
      case "a":
      case "aa":
        return r.toUpperCase();
      case "aaa":
        return r;
      case "aaaaa":
        return r[0];
      default:
        return r === "am" ? "a.m." : "p.m.";
    }
  },
  // Hour [1-12]
  h(e, t) {
    return T(e.getHours() % 12 || 12, t.length);
  },
  // Hour [0-23]
  H(e, t) {
    return T(e.getHours(), t.length);
  },
  // Minute
  m(e, t) {
    return T(e.getMinutes(), t.length);
  },
  // Second
  s(e, t) {
    return T(e.getSeconds(), t.length);
  },
  // Fraction of second
  S(e, t) {
    const r = t.length, n = e.getMilliseconds(), o = Math.trunc(
      n * Math.pow(10, r - 3)
    );
    return T(o, t.length);
  }
}, ue = {
  midnight: "midnight",
  noon: "noon",
  morning: "morning",
  afternoon: "afternoon",
  evening: "evening",
  night: "night"
}, dt = {
  // Era
  G: function(e, t, r) {
    const n = e.getFullYear() > 0 ? 1 : 0;
    switch (t) {
      // AD, BC
      case "G":
      case "GG":
      case "GGG":
        return r.era(n, { width: "abbreviated" });
      // A, B
      case "GGGGG":
        return r.era(n, { width: "narrow" });
      default:
        return r.era(n, { width: "wide" });
    }
  },
  // Year
  y: function(e, t, r) {
    if (t === "yo") {
      const n = e.getFullYear(), o = n > 0 ? n : 1 - n;
      return r.ordinalNumber(o, { unit: "year" });
    }
    return se.y(e, t);
  },
  // Local week-numbering year
  Y: function(e, t, r, n) {
    const o = Tt(e, n), a = o > 0 ? o : 1 - o;
    if (t === "YY") {
      const i = a % 100;
      return T(i, 2);
    }
    return t === "Yo" ? r.ordinalNumber(a, { unit: "year" }) : T(a, t.length);
  },
  // ISO week-numbering year
  R: function(e, t) {
    const r = Wt(e);
    return T(r, t.length);
  },
  // Extended year. This is a single number designating the year of this calendar system.
  // The main difference between `y` and `u` localizers are B.C. years:
  // | Year | `y` | `u` |
  // |------|-----|-----|
  // | AC 1 |   1 |   1 |
  // | BC 1 |   1 |   0 |
  // | BC 2 |   2 |  -1 |
  // Also `yy` always returns the last two digits of a year,
  // while `uu` pads single digit years to 2 characters and returns other years unchanged.
  u: function(e, t) {
    const r = e.getFullYear();
    return T(r, t.length);
  },
  // Quarter
  Q: function(e, t, r) {
    const n = Math.ceil((e.getMonth() + 1) / 3);
    switch (t) {
      // 1, 2, 3, 4
      case "Q":
        return String(n);
      // 01, 02, 03, 04
      case "QQ":
        return T(n, 2);
      // 1st, 2nd, 3rd, 4th
      case "Qo":
        return r.ordinalNumber(n, { unit: "quarter" });
      // Q1, Q2, Q3, Q4
      case "QQQ":
        return r.quarter(n, {
          width: "abbreviated",
          context: "formatting"
        });
      // 1, 2, 3, 4 (narrow quarter; could be not numerical)
      case "QQQQQ":
        return r.quarter(n, {
          width: "narrow",
          context: "formatting"
        });
      default:
        return r.quarter(n, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  // Stand-alone quarter
  q: function(e, t, r) {
    const n = Math.ceil((e.getMonth() + 1) / 3);
    switch (t) {
      // 1, 2, 3, 4
      case "q":
        return String(n);
      // 01, 02, 03, 04
      case "qq":
        return T(n, 2);
      // 1st, 2nd, 3rd, 4th
      case "qo":
        return r.ordinalNumber(n, { unit: "quarter" });
      // Q1, Q2, Q3, Q4
      case "qqq":
        return r.quarter(n, {
          width: "abbreviated",
          context: "standalone"
        });
      // 1, 2, 3, 4 (narrow quarter; could be not numerical)
      case "qqqqq":
        return r.quarter(n, {
          width: "narrow",
          context: "standalone"
        });
      default:
        return r.quarter(n, {
          width: "wide",
          context: "standalone"
        });
    }
  },
  // Month
  M: function(e, t, r) {
    const n = e.getMonth();
    switch (t) {
      case "M":
      case "MM":
        return se.M(e, t);
      // 1st, 2nd, ..., 12th
      case "Mo":
        return r.ordinalNumber(n + 1, { unit: "month" });
      // Jan, Feb, ..., Dec
      case "MMM":
        return r.month(n, {
          width: "abbreviated",
          context: "formatting"
        });
      // J, F, ..., D
      case "MMMMM":
        return r.month(n, {
          width: "narrow",
          context: "formatting"
        });
      default:
        return r.month(n, { width: "wide", context: "formatting" });
    }
  },
  // Stand-alone month
  L: function(e, t, r) {
    const n = e.getMonth();
    switch (t) {
      // 1, 2, ..., 12
      case "L":
        return String(n + 1);
      // 01, 02, ..., 12
      case "LL":
        return T(n + 1, 2);
      // 1st, 2nd, ..., 12th
      case "Lo":
        return r.ordinalNumber(n + 1, { unit: "month" });
      // Jan, Feb, ..., Dec
      case "LLL":
        return r.month(n, {
          width: "abbreviated",
          context: "standalone"
        });
      // J, F, ..., D
      case "LLLLL":
        return r.month(n, {
          width: "narrow",
          context: "standalone"
        });
      default:
        return r.month(n, { width: "wide", context: "standalone" });
    }
  },
  // Local week of year
  w: function(e, t, r, n) {
    const o = Ze(e, n);
    return t === "wo" ? r.ordinalNumber(o, { unit: "week" }) : T(o, t.length);
  },
  // ISO week of year
  I: function(e, t, r) {
    const n = Ve(e);
    return t === "Io" ? r.ordinalNumber(n, { unit: "week" }) : T(n, t.length);
  },
  // Day of the month
  d: function(e, t, r) {
    return t === "do" ? r.ordinalNumber(e.getDate(), { unit: "date" }) : se.d(e, t);
  },
  // Day of year
  D: function(e, t, r) {
    const n = _r(e);
    return t === "Do" ? r.ordinalNumber(n, { unit: "dayOfYear" }) : T(n, t.length);
  },
  // Day of week
  E: function(e, t, r) {
    const n = e.getDay();
    switch (t) {
      // Tue
      case "E":
      case "EE":
      case "EEE":
        return r.day(n, {
          width: "abbreviated",
          context: "formatting"
        });
      // T
      case "EEEEE":
        return r.day(n, {
          width: "narrow",
          context: "formatting"
        });
      // Tu
      case "EEEEEE":
        return r.day(n, {
          width: "short",
          context: "formatting"
        });
      default:
        return r.day(n, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  // Local day of week
  e: function(e, t, r, n) {
    const o = e.getDay(), a = (o - n.weekStartsOn + 8) % 7 || 7;
    switch (t) {
      // Numerical value (Nth day of week with current locale or weekStartsOn)
      case "e":
        return String(a);
      // Padded numerical value
      case "ee":
        return T(a, 2);
      // 1st, 2nd, ..., 7th
      case "eo":
        return r.ordinalNumber(a, { unit: "day" });
      case "eee":
        return r.day(o, {
          width: "abbreviated",
          context: "formatting"
        });
      // T
      case "eeeee":
        return r.day(o, {
          width: "narrow",
          context: "formatting"
        });
      // Tu
      case "eeeeee":
        return r.day(o, {
          width: "short",
          context: "formatting"
        });
      default:
        return r.day(o, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  // Stand-alone local day of week
  c: function(e, t, r, n) {
    const o = e.getDay(), a = (o - n.weekStartsOn + 8) % 7 || 7;
    switch (t) {
      // Numerical value (same as in `e`)
      case "c":
        return String(a);
      // Padded numerical value
      case "cc":
        return T(a, t.length);
      // 1st, 2nd, ..., 7th
      case "co":
        return r.ordinalNumber(a, { unit: "day" });
      case "ccc":
        return r.day(o, {
          width: "abbreviated",
          context: "standalone"
        });
      // T
      case "ccccc":
        return r.day(o, {
          width: "narrow",
          context: "standalone"
        });
      // Tu
      case "cccccc":
        return r.day(o, {
          width: "short",
          context: "standalone"
        });
      default:
        return r.day(o, {
          width: "wide",
          context: "standalone"
        });
    }
  },
  // ISO day of week
  i: function(e, t, r) {
    const n = e.getDay(), o = n === 0 ? 7 : n;
    switch (t) {
      // 2
      case "i":
        return String(o);
      // 02
      case "ii":
        return T(o, t.length);
      // 2nd
      case "io":
        return r.ordinalNumber(o, { unit: "day" });
      // Tue
      case "iii":
        return r.day(n, {
          width: "abbreviated",
          context: "formatting"
        });
      // T
      case "iiiii":
        return r.day(n, {
          width: "narrow",
          context: "formatting"
        });
      // Tu
      case "iiiiii":
        return r.day(n, {
          width: "short",
          context: "formatting"
        });
      default:
        return r.day(n, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  // AM or PM
  a: function(e, t, r) {
    const o = e.getHours() / 12 >= 1 ? "pm" : "am";
    switch (t) {
      case "a":
      case "aa":
        return r.dayPeriod(o, {
          width: "abbreviated",
          context: "formatting"
        });
      case "aaa":
        return r.dayPeriod(o, {
          width: "abbreviated",
          context: "formatting"
        }).toLowerCase();
      case "aaaaa":
        return r.dayPeriod(o, {
          width: "narrow",
          context: "formatting"
        });
      default:
        return r.dayPeriod(o, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  // AM, PM, midnight, noon
  b: function(e, t, r) {
    const n = e.getHours();
    let o;
    switch (n === 12 ? o = ue.noon : n === 0 ? o = ue.midnight : o = n / 12 >= 1 ? "pm" : "am", t) {
      case "b":
      case "bb":
        return r.dayPeriod(o, {
          width: "abbreviated",
          context: "formatting"
        });
      case "bbb":
        return r.dayPeriod(o, {
          width: "abbreviated",
          context: "formatting"
        }).toLowerCase();
      case "bbbbb":
        return r.dayPeriod(o, {
          width: "narrow",
          context: "formatting"
        });
      default:
        return r.dayPeriod(o, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  // in the morning, in the afternoon, in the evening, at night
  B: function(e, t, r) {
    const n = e.getHours();
    let o;
    switch (n >= 17 ? o = ue.evening : n >= 12 ? o = ue.afternoon : n >= 4 ? o = ue.morning : o = ue.night, t) {
      case "B":
      case "BB":
      case "BBB":
        return r.dayPeriod(o, {
          width: "abbreviated",
          context: "formatting"
        });
      case "BBBBB":
        return r.dayPeriod(o, {
          width: "narrow",
          context: "formatting"
        });
      default:
        return r.dayPeriod(o, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  // Hour [1-12]
  h: function(e, t, r) {
    if (t === "ho") {
      let n = e.getHours() % 12;
      return n === 0 && (n = 12), r.ordinalNumber(n, { unit: "hour" });
    }
    return se.h(e, t);
  },
  // Hour [0-23]
  H: function(e, t, r) {
    return t === "Ho" ? r.ordinalNumber(e.getHours(), { unit: "hour" }) : se.H(e, t);
  },
  // Hour [0-11]
  K: function(e, t, r) {
    const n = e.getHours() % 12;
    return t === "Ko" ? r.ordinalNumber(n, { unit: "hour" }) : T(n, t.length);
  },
  // Hour [1-24]
  k: function(e, t, r) {
    let n = e.getHours();
    return n === 0 && (n = 24), t === "ko" ? r.ordinalNumber(n, { unit: "hour" }) : T(n, t.length);
  },
  // Minute
  m: function(e, t, r) {
    return t === "mo" ? r.ordinalNumber(e.getMinutes(), { unit: "minute" }) : se.m(e, t);
  },
  // Second
  s: function(e, t, r) {
    return t === "so" ? r.ordinalNumber(e.getSeconds(), { unit: "second" }) : se.s(e, t);
  },
  // Fraction of second
  S: function(e, t) {
    return se.S(e, t);
  },
  // Timezone (ISO-8601. If offset is 0, output is always `'Z'`)
  X: function(e, t, r) {
    const n = e.getTimezoneOffset();
    if (n === 0)
      return "Z";
    switch (t) {
      // Hours and optional minutes
      case "X":
        return mt(n);
      // Hours, minutes and optional seconds without `:` delimiter
      // Note: neither ISO-8601 nor JavaScript supports seconds in timezone offsets
      // so this token always has the same output as `XX`
      case "XXXX":
      case "XX":
        return ae(n);
      // Hours and minutes with `:` delimiter
      default:
        return ae(n, ":");
    }
  },
  // Timezone (ISO-8601. If offset is 0, output is `'+00:00'` or equivalent)
  x: function(e, t, r) {
    const n = e.getTimezoneOffset();
    switch (t) {
      // Hours and optional minutes
      case "x":
        return mt(n);
      // Hours, minutes and optional seconds without `:` delimiter
      // Note: neither ISO-8601 nor JavaScript supports seconds in timezone offsets
      // so this token always has the same output as `xx`
      case "xxxx":
      case "xx":
        return ae(n);
      // Hours and minutes with `:` delimiter
      default:
        return ae(n, ":");
    }
  },
  // Timezone (GMT)
  O: function(e, t, r) {
    const n = e.getTimezoneOffset();
    switch (t) {
      // Short
      case "O":
      case "OO":
      case "OOO":
        return "GMT" + ht(n, ":");
      default:
        return "GMT" + ae(n, ":");
    }
  },
  // Timezone (specific non-location)
  z: function(e, t, r) {
    const n = e.getTimezoneOffset();
    switch (t) {
      // Short
      case "z":
      case "zz":
      case "zzz":
        return "GMT" + ht(n, ":");
      default:
        return "GMT" + ae(n, ":");
    }
  },
  // Seconds timestamp
  t: function(e, t, r) {
    const n = Math.trunc(+e / 1e3);
    return T(n, t.length);
  },
  // Milliseconds timestamp
  T: function(e, t, r) {
    return T(+e, t.length);
  }
};
function ht(e, t = "") {
  const r = e > 0 ? "-" : "+", n = Math.abs(e), o = Math.trunc(n / 60), a = n % 60;
  return a === 0 ? r + String(o) : r + String(o) + t + T(a, 2);
}
function mt(e, t) {
  return e % 60 === 0 ? (e > 0 ? "-" : "+") + T(Math.abs(e) / 60, 2) : ae(e, t);
}
function ae(e, t = "") {
  const r = e > 0 ? "-" : "+", n = Math.abs(e), o = T(Math.trunc(n / 60), 2), a = T(n % 60, 2);
  return r + o + t + a;
}
const yt = (e, t) => {
  switch (e) {
    case "P":
      return t.date({ width: "short" });
    case "PP":
      return t.date({ width: "medium" });
    case "PPP":
      return t.date({ width: "long" });
    default:
      return t.date({ width: "full" });
  }
}, Pt = (e, t) => {
  switch (e) {
    case "p":
      return t.time({ width: "short" });
    case "pp":
      return t.time({ width: "medium" });
    case "ppp":
      return t.time({ width: "long" });
    default:
      return t.time({ width: "full" });
  }
}, Br = (e, t) => {
  const r = e.match(/(P+)(p+)?/) || [], n = r[1], o = r[2];
  if (!o)
    return yt(e, t);
  let a;
  switch (n) {
    case "P":
      a = t.dateTime({ width: "short" });
      break;
    case "PP":
      a = t.dateTime({ width: "medium" });
      break;
    case "PPP":
      a = t.dateTime({ width: "long" });
      break;
    default:
      a = t.dateTime({ width: "full" });
      break;
  }
  return a.replace("{{date}}", yt(n, t)).replace("{{time}}", Pt(o, t));
}, Ir = {
  p: Pt,
  P: Br
}, Hr = /^D+$/, Ar = /^Y+$/, qr = ["D", "DD", "YY", "YYYY"];
function jr(e) {
  return Hr.test(e);
}
function Gr(e) {
  return Ar.test(e);
}
function Rr(e, t, r) {
  const n = $r(e, t, r);
  if (console.warn(n), qr.includes(e)) throw new RangeError(n);
}
function $r(e, t, r) {
  const n = e[0] === "Y" ? "years" : "days of the month";
  return `Use \`${e.toLowerCase()}\` instead of \`${e}\` (in \`${t}\`) for formatting ${n} to the input \`${r}\`; see: https://github.com/date-fns/date-fns/blob/master/docs/unicodeTokens.md`;
}
const zr = /[yYQqMLwIdDecihHKkms]o|(\w)\1*|''|'(''|[^'])+('|$)|./g, Qr = /P+p+|P+|p+|''|'(''|[^'])+('|$)|./g, Xr = /^'([^]*?)'?$/, Ur = /''/g, Vr = /[a-zA-Z]/;
function be(e, t, r) {
  const n = Me(), o = r?.locale ?? n.locale ?? fe, a = r?.firstWeekContainsDate ?? r?.locale?.options?.firstWeekContainsDate ?? n.firstWeekContainsDate ?? n.locale?.options?.firstWeekContainsDate ?? 1, i = r?.weekStartsOn ?? r?.locale?.options?.weekStartsOn ?? n.weekStartsOn ?? n.locale?.options?.weekStartsOn ?? 0, s = P(e, r?.in);
  if (!Vn(s))
    throw new RangeError("Invalid time value");
  let c = t.match(Qr).map((f) => {
    const l = f[0];
    if (l === "p" || l === "P") {
      const m = Ir[l];
      return m(f, o.formatLong);
    }
    return f;
  }).join("").match(zr).map((f) => {
    if (f === "''")
      return { isToken: !1, value: "'" };
    const l = f[0];
    if (l === "'")
      return { isToken: !1, value: Zr(f) };
    if (dt[l])
      return { isToken: !0, value: f };
    if (l.match(Vr))
      throw new RangeError(
        "Format string contains an unescaped latin alphabet character `" + l + "`"
      );
    return { isToken: !1, value: f };
  });
  o.localize.preprocessor && (c = o.localize.preprocessor(s, c));
  const u = {
    firstWeekContainsDate: a,
    weekStartsOn: i,
    locale: o
  };
  return c.map((f) => {
    if (!f.isToken) return f.value;
    const l = f.value;
    (!r?.useAdditionalWeekYearTokens && Gr(l) || !r?.useAdditionalDayOfYearTokens && jr(l)) && Rr(l, t, String(e));
    const m = dt[l[0]];
    return m(s, l, o.localize, u);
  }).join("");
}
function Zr(e) {
  const t = e.match(Xr);
  return t ? t[1].replace(Ur, "'") : e;
}
function Kr(e, t) {
  const r = P(e, t?.in), n = r.getFullYear(), o = r.getMonth(), a = B(r, 0);
  return a.setFullYear(n, o + 1, 0), a.setHours(0, 0, 0, 0), a.getDate();
}
function Jr(e, t) {
  return P(e, t?.in).getMonth();
}
function Lr(e, t) {
  return P(e, t?.in).getFullYear();
}
function eo(e, t) {
  return +P(e) > +P(t);
}
function to(e, t) {
  return +P(e) < +P(t);
}
function no(e, t, r) {
  const [n, o] = he(
    r?.in,
    e,
    t
  );
  return n.getFullYear() === o.getFullYear() && n.getMonth() === o.getMonth();
}
function ro(e, t, r) {
  const [n, o] = he(
    r?.in,
    e,
    t
  );
  return n.getFullYear() === o.getFullYear();
}
function oo(e, t, r) {
  const n = P(e, r?.in), o = n.getFullYear(), a = n.getDate(), i = B(e, 0);
  i.setFullYear(o, t, 15), i.setHours(0, 0, 0, 0);
  const s = Kr(i);
  return n.setMonth(t, Math.min(a, s)), n;
}
function so(e, t, r) {
  const n = P(e, r?.in);
  return isNaN(+n) ? B(e, NaN) : (n.setFullYear(t), n);
}
const gt = 5, ao = 4;
function io(e, t) {
  const r = t.startOfMonth(e), n = r.getDay() > 0 ? r.getDay() : 7, o = t.addDays(e, -n + 1), a = t.addDays(o, gt * 7 - 1);
  return t.getMonth(e) === t.getMonth(a) ? gt : ao;
}
function Et(e, t) {
  const r = t.startOfMonth(e), n = r.getDay();
  return n === 1 ? r : n === 0 ? t.addDays(r, -6) : t.addDays(r, -1 * (n - 1));
}
function co(e, t) {
  const r = Et(e, t), n = io(e, t);
  return t.addDays(r, n * 7 - 1);
}
const _t = {
  ...fe,
  labels: {
    labelDayButton: (e, t, r, n) => {
      let o;
      n && typeof n.format == "function" ? o = n.format.bind(n) : o = (i, s) => be(i, s, { locale: fe, ...r });
      let a = o(e, "PPPP");
      return t.today && (a = `Today, ${a}`), t.selected && (a = `${a}, selected`), a;
    },
    labelMonthDropdown: "Choose the Month",
    labelNext: "Go to the Next Month",
    labelPrevious: "Go to the Previous Month",
    labelWeekNumber: (e) => `Week ${e}`,
    labelYearDropdown: "Choose the Year",
    labelGrid: (e, t, r) => {
      let n;
      return r && typeof r.format == "function" ? n = r.format.bind(r) : n = (o, a) => be(o, a, { locale: fe, ...t }), n(e, "LLLL yyyy");
    },
    labelGridcell: (e, t, r, n) => {
      let o;
      n && typeof n.format == "function" ? o = n.format.bind(n) : o = (i, s) => be(i, s, { locale: fe, ...r });
      let a = o(e, "PPPP");
      return t?.today && (a = `Today, ${a}`), a;
    },
    labelNav: "Navigation bar",
    labelWeekNumberHeader: "Week Number",
    labelWeekday: (e, t, r) => {
      let n;
      return r && typeof r.format == "function" ? n = r.format.bind(r) : n = (o, a) => be(o, a, { locale: fe, ...t }), n(e, "cccc");
    }
  }
};
class G {
  /**
   * Creates an instance of `DateLib`.
   *
   * @param options Configuration options for the date library.
   * @param overrides Custom overrides for the date library functions.
   */
  constructor(t, r) {
    this.Date = Date, this.today = () => this.overrides?.today ? this.overrides.today() : this.options.timeZone ? H.tz(this.options.timeZone) : new this.Date(), this.newDate = (n, o, a) => this.overrides?.newDate ? this.overrides.newDate(n, o, a) : this.options.timeZone ? new H(n, o, a, this.options.timeZone) : new Date(n, o, a), this.addDays = (n, o) => this.overrides?.addDays ? this.overrides.addDays(n, o) : Ot(n, o), this.addMonths = (n, o) => this.overrides?.addMonths ? this.overrides.addMonths(n, o) : vt(n, o), this.addWeeks = (n, o) => this.overrides?.addWeeks ? this.overrides.addWeeks(n, o) : $n(n, o), this.addYears = (n, o) => this.overrides?.addYears ? this.overrides.addYears(n, o) : zn(n, o), this.differenceInCalendarDays = (n, o) => this.overrides?.differenceInCalendarDays ? this.overrides.differenceInCalendarDays(n, o) : Ue(n, o), this.differenceInCalendarMonths = (n, o) => this.overrides?.differenceInCalendarMonths ? this.overrides.differenceInCalendarMonths(n, o) : Ct(n, o), this.eachMonthOfInterval = (n) => this.overrides?.eachMonthOfInterval ? this.overrides.eachMonthOfInterval(n) : Kn(n), this.eachYearOfInterval = (n) => {
      const o = this.overrides?.eachYearOfInterval ? this.overrides.eachYearOfInterval(n) : er(n), a = new Set(o.map((s) => this.getYear(s)));
      if (a.size === o.length)
        return o;
      const i = [];
      return a.forEach((s) => {
        i.push(new Date(s, 0, 1));
      }), i;
    }, this.endOfBroadcastWeek = (n) => this.overrides?.endOfBroadcastWeek ? this.overrides.endOfBroadcastWeek(n) : co(n, this), this.endOfISOWeek = (n) => this.overrides?.endOfISOWeek ? this.overrides.endOfISOWeek(n) : tr(n), this.endOfMonth = (n) => this.overrides?.endOfMonth ? this.overrides.endOfMonth(n) : Zn(n), this.endOfWeek = (n, o) => this.overrides?.endOfWeek ? this.overrides.endOfWeek(n, o) : Yt(n, this.options), this.endOfYear = (n) => this.overrides?.endOfYear ? this.overrides.endOfYear(n) : Ln(n), this.format = (n, o, a) => {
      const i = this.overrides?.format ? this.overrides.format(n, o, this.options) : be(n, o, this.options);
      return this.options.numerals && this.options.numerals !== "latn" ? this.replaceDigits(i) : i;
    }, this.getISOWeek = (n) => this.overrides?.getISOWeek ? this.overrides.getISOWeek(n) : Ve(n), this.getMonth = (n, o) => this.overrides?.getMonth ? this.overrides.getMonth(n, this.options) : Jr(n, this.options), this.getYear = (n, o) => this.overrides?.getYear ? this.overrides.getYear(n, this.options) : Lr(n, this.options), this.getWeek = (n, o) => this.overrides?.getWeek ? this.overrides.getWeek(n, this.options) : Ze(n, this.options), this.isAfter = (n, o) => this.overrides?.isAfter ? this.overrides.isAfter(n, o) : eo(n, o), this.isBefore = (n, o) => this.overrides?.isBefore ? this.overrides.isBefore(n, o) : to(n, o), this.isDate = (n) => this.overrides?.isDate ? this.overrides.isDate(n) : St(n), this.isSameDay = (n, o) => this.overrides?.isSameDay ? this.overrides.isSameDay(n, o) : Un(n, o), this.isSameMonth = (n, o) => this.overrides?.isSameMonth ? this.overrides.isSameMonth(n, o) : no(n, o), this.isSameYear = (n, o) => this.overrides?.isSameYear ? this.overrides.isSameYear(n, o) : ro(n, o), this.max = (n) => this.overrides?.max ? this.overrides.max(n) : Qn(n), this.min = (n) => this.overrides?.min ? this.overrides.min(n) : Xn(n), this.setMonth = (n, o) => this.overrides?.setMonth ? this.overrides.setMonth(n, o) : oo(n, o), this.setYear = (n, o) => this.overrides?.setYear ? this.overrides.setYear(n, o) : so(n, o), this.startOfBroadcastWeek = (n, o) => this.overrides?.startOfBroadcastWeek ? this.overrides.startOfBroadcastWeek(n, this) : Et(n, this), this.startOfDay = (n) => this.overrides?.startOfDay ? this.overrides.startOfDay(n) : we(n), this.startOfISOWeek = (n) => this.overrides?.startOfISOWeek ? this.overrides.startOfISOWeek(n) : pe(n), this.startOfMonth = (n) => this.overrides?.startOfMonth ? this.overrides.startOfMonth(n) : Jn(n), this.startOfWeek = (n, o) => this.overrides?.startOfWeek ? this.overrides.startOfWeek(n, this.options) : de(n, this.options), this.startOfYear = (n) => this.overrides?.startOfYear ? this.overrides.startOfYear(n) : xt(n), this.options = { locale: _t, ...t }, this.overrides = r;
  }
  /**
   * Generates a mapping of Arabic digits (0-9) to the target numbering system
   * digits.
   *
   * @since 9.5.0
   * @returns A record mapping Arabic digits to the target numerals.
   */
  getDigitMap() {
    const { numerals: t = "latn" } = this.options, r = new Intl.NumberFormat("en-US", {
      numberingSystem: t
    }), n = {};
    for (let o = 0; o < 10; o++)
      n[o.toString()] = r.format(o);
    return n;
  }
  /**
   * Replaces Arabic digits in a string with the target numbering system digits.
   *
   * @since 9.5.0
   * @param input The string containing Arabic digits.
   * @returns The string with digits replaced.
   */
  replaceDigits(t) {
    const r = this.getDigitMap();
    return t.replace(/\d/g, (n) => r[n] || n);
  }
  /**
   * Formats a number using the configured numbering system.
   *
   * @since 9.5.0
   * @param value The number to format.
   * @returns The formatted number as a string.
   */
  formatNumber(t) {
    return this.replaceDigits(t.toString());
  }
  /**
   * Returns the preferred ordering for month and year labels for the current
   * locale.
   */
  getMonthYearOrder() {
    const t = this.options.locale?.code;
    return t && G.yearFirstLocales.has(t) ? "year-first" : "month-first";
  }
  /**
   * Formats the month/year pair respecting locale conventions.
   *
   * @since 9.11.0
   */
  formatMonthYear(t) {
    const { locale: r, timeZone: n, numerals: o } = this.options, a = r?.code;
    if (a && G.yearFirstLocales.has(a))
      try {
        return new Intl.DateTimeFormat(a, {
          month: "long",
          year: "numeric",
          timeZone: n,
          numberingSystem: o
        }).format(t);
      } catch {
      }
    const i = this.getMonthYearOrder() === "year-first" ? "y LLLL" : "LLLL y";
    return this.format(t, i);
  }
}
G.yearFirstLocales = /* @__PURE__ */ new Set([
  "eu",
  "hu",
  "ja",
  "ja-Hira",
  "ja-JP",
  "ko",
  "ko-KR",
  "lt",
  "lt-LT",
  "lv",
  "lv-LV",
  "mn",
  "mn-MN",
  "zh",
  "zh-CN",
  "zh-HK",
  "zh-TW"
]);
const L = new G();
class Ft {
  constructor(t, r, n = L) {
    this.date = t, this.displayMonth = r, this.outside = !!(r && !n.isSameMonth(t, r)), this.dateLib = n, this.isoDate = n.format(t, "yyyy-MM-dd"), this.displayMonthId = n.format(r, "yyyy-MM"), this.dateMonthId = n.format(t, "yyyy-MM");
  }
  /**
   * Checks if this day is equal to another `CalendarDay`, considering both the
   * date and the displayed month.
   *
   * @param day The `CalendarDay` to compare with.
   * @returns `true` if the days are equal, otherwise `false`.
   */
  isEqualTo(t) {
    return this.dateLib.isSameDay(t.date, this.date) && this.dateLib.isSameMonth(t.displayMonth, this.displayMonth);
  }
}
class uo {
  constructor(t, r) {
    this.date = t, this.weeks = r;
  }
}
class lo {
  constructor(t, r) {
    this.days = r, this.weekNumber = t;
  }
}
function fo(e) {
  return h.createElement("button", { ...e });
}
function ho(e) {
  return h.createElement("span", { ...e });
}
function mo(e) {
  const { size: t = 24, orientation: r = "left", className: n } = e;
  return (
    // biome-ignore lint/a11y/noSvgWithoutTitle: handled by the parent component
    h.createElement(
      "svg",
      { className: n, width: t, height: t, viewBox: "0 0 24 24" },
      r === "up" && h.createElement("polygon", { points: "6.77 17 12.5 11.43 18.24 17 20 15.28 12.5 8 5 15.28" }),
      r === "down" && h.createElement("polygon", { points: "6.77 8 12.5 13.57 18.24 8 20 9.72 12.5 17 5 9.72" }),
      r === "left" && h.createElement("polygon", { points: "16 18.112 9.81111111 12 16 5.87733333 14.0888889 4 6 12 14.0888889 20" }),
      r === "right" && h.createElement("polygon", { points: "8 18.112 14.18888889 12 8 5.87733333 9.91111111 4 18 12 9.91111111 20" })
    )
  );
}
function yo(e) {
  const { day: t, modifiers: r, ...n } = e;
  return h.createElement("td", { ...n });
}
function go(e) {
  const { day: t, modifiers: r, ...n } = e, o = h.useRef(null);
  return h.useEffect(() => {
    r.focused && o.current?.focus();
  }, [r.focused]), h.createElement("button", { ref: o, ...n });
}
var d;
(function(e) {
  e.Root = "root", e.Chevron = "chevron", e.Day = "day", e.DayButton = "day_button", e.CaptionLabel = "caption_label", e.Dropdowns = "dropdowns", e.Dropdown = "dropdown", e.DropdownRoot = "dropdown_root", e.Footer = "footer", e.MonthGrid = "month_grid", e.MonthCaption = "month_caption", e.MonthsDropdown = "months_dropdown", e.Month = "month", e.Months = "months", e.Nav = "nav", e.NextMonthButton = "button_next", e.PreviousMonthButton = "button_previous", e.Week = "week", e.Weeks = "weeks", e.Weekday = "weekday", e.Weekdays = "weekdays", e.WeekNumber = "week_number", e.WeekNumberHeader = "week_number_header", e.YearsDropdown = "years_dropdown";
})(d || (d = {}));
var E;
(function(e) {
  e.disabled = "disabled", e.hidden = "hidden", e.outside = "outside", e.focused = "focused", e.today = "today";
})(E || (E = {}));
var A;
(function(e) {
  e.range_end = "range_end", e.range_middle = "range_middle", e.range_start = "range_start", e.selected = "selected";
})(A || (A = {}));
var j;
(function(e) {
  e.weeks_before_enter = "weeks_before_enter", e.weeks_before_exit = "weeks_before_exit", e.weeks_after_enter = "weeks_after_enter", e.weeks_after_exit = "weeks_after_exit", e.caption_after_enter = "caption_after_enter", e.caption_after_exit = "caption_after_exit", e.caption_before_enter = "caption_before_enter", e.caption_before_exit = "caption_before_exit";
})(j || (j = {}));
function bo(e) {
  const { options: t, className: r, components: n, classNames: o, ...a } = e, i = [o[d.Dropdown], r].join(" "), s = t?.find(({ value: c }) => c === a.value);
  return h.createElement(
    "span",
    { "data-disabled": a.disabled, className: o[d.DropdownRoot] },
    h.createElement(n.Select, { className: i, ...a }, t?.map(({ value: c, label: u, disabled: f }) => h.createElement(n.Option, { key: c, value: c, disabled: f }, u))),
    h.createElement(
      "span",
      { className: o[d.CaptionLabel], "aria-hidden": !0 },
      s?.label,
      h.createElement(n.Chevron, { orientation: "down", size: 18, className: o[d.Chevron] })
    )
  );
}
function po(e) {
  return h.createElement("div", { ...e });
}
function wo(e) {
  return h.createElement("div", { ...e });
}
function Mo(e) {
  const { calendarMonth: t, displayIndex: r, ...n } = e;
  return h.createElement("div", { ...n }, e.children);
}
function Do(e) {
  const { calendarMonth: t, displayIndex: r, ...n } = e;
  return h.createElement("div", { ...n });
}
function ko(e) {
  return h.createElement("table", { ...e });
}
function Oo(e) {
  return h.createElement("div", { ...e });
}
const Bt = Sn(void 0);
function De() {
  const e = Cn(Bt);
  if (e === void 0)
    throw new Error("useDayPicker() must be used within a custom component.");
  return e;
}
function vo(e) {
  const { components: t } = De();
  return h.createElement(t.Dropdown, { ...e });
}
function Wo(e) {
  const { onPreviousClick: t, onNextClick: r, previousMonth: n, nextMonth: o, ...a } = e, { components: i, classNames: s, labels: { labelPrevious: c, labelNext: u } } = De(), f = $((m) => {
    o && r?.(m);
  }, [o, r]), l = $((m) => {
    n && t?.(m);
  }, [n, t]);
  return h.createElement(
    "nav",
    { ...a },
    h.createElement(
      i.PreviousMonthButton,
      { type: "button", className: s[d.PreviousMonthButton], tabIndex: n ? void 0 : -1, "aria-disabled": n ? void 0 : !0, "aria-label": c(n), onClick: l },
      h.createElement(i.Chevron, { disabled: n ? void 0 : !0, className: s[d.Chevron], orientation: "left" })
    ),
    h.createElement(
      i.NextMonthButton,
      { type: "button", className: s[d.NextMonthButton], tabIndex: o ? void 0 : -1, "aria-disabled": o ? void 0 : !0, "aria-label": u(o), onClick: f },
      h.createElement(i.Chevron, { disabled: o ? void 0 : !0, orientation: "right", className: s[d.Chevron] })
    )
  );
}
function So(e) {
  const { components: t } = De();
  return h.createElement(t.Button, { ...e });
}
function Co(e) {
  return h.createElement("option", { ...e });
}
function No(e) {
  const { components: t } = De();
  return h.createElement(t.Button, { ...e });
}
function xo(e) {
  const { rootRef: t, ...r } = e;
  return h.createElement("div", { ...r, ref: t });
}
function Yo(e) {
  return h.createElement("select", { ...e });
}
function To(e) {
  const { week: t, ...r } = e;
  return h.createElement("tr", { ...r });
}
function Po(e) {
  return h.createElement("th", { ...e });
}
function Eo(e) {
  return h.createElement(
    "thead",
    { "aria-hidden": !0 },
    h.createElement("tr", { ...e })
  );
}
function _o(e) {
  const { week: t, ...r } = e;
  return h.createElement("th", { ...r });
}
function Fo(e) {
  return h.createElement("th", { ...e });
}
function Bo(e) {
  return h.createElement("tbody", { ...e });
}
function Io(e) {
  const { components: t } = De();
  return h.createElement(t.Dropdown, { ...e });
}
const Ho = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  Button: fo,
  CaptionLabel: ho,
  Chevron: mo,
  Day: yo,
  DayButton: go,
  Dropdown: bo,
  DropdownNav: po,
  Footer: wo,
  Month: Mo,
  MonthCaption: Do,
  MonthGrid: ko,
  Months: Oo,
  MonthsDropdown: vo,
  Nav: Wo,
  NextMonthButton: So,
  Option: Co,
  PreviousMonthButton: No,
  Root: xo,
  Select: Yo,
  Week: To,
  WeekNumber: _o,
  WeekNumberHeader: Fo,
  Weekday: Po,
  Weekdays: Eo,
  Weeks: Bo,
  YearsDropdown: Io
}, Symbol.toStringTag, { value: "Module" }));
function ne(e, t, r = !1, n = L) {
  let { from: o, to: a } = e;
  const { differenceInCalendarDays: i, isSameDay: s } = n;
  return o && a ? (i(a, o) < 0 && ([o, a] = [a, o]), i(t, o) >= (r ? 1 : 0) && i(a, t) >= (r ? 1 : 0)) : !r && a ? s(a, t) : !r && o ? s(o, t) : !1;
}
function Ke(e) {
  return !!(e && typeof e == "object" && "before" in e && "after" in e);
}
function xe(e) {
  return !!(e && typeof e == "object" && "from" in e);
}
function Je(e) {
  return !!(e && typeof e == "object" && "after" in e);
}
function Le(e) {
  return !!(e && typeof e == "object" && "before" in e);
}
function It(e) {
  return !!(e && typeof e == "object" && "dayOfWeek" in e);
}
function Ht(e, t) {
  return Array.isArray(e) && e.every(t.isDate);
}
function re(e, t, r = L) {
  const n = Array.isArray(t) ? t : [t], { isSameDay: o, differenceInCalendarDays: a, isAfter: i } = r;
  return n.some((s) => {
    if (typeof s == "boolean")
      return s;
    if (r.isDate(s))
      return o(e, s);
    if (Ht(s, r))
      return s.some((c) => o(e, c));
    if (xe(s))
      return ne(s, e, !1, r);
    if (It(s))
      return Array.isArray(s.dayOfWeek) ? s.dayOfWeek.includes(e.getDay()) : s.dayOfWeek === e.getDay();
    if (Ke(s)) {
      const c = a(s.before, e), u = a(s.after, e), f = c > 0, l = u < 0;
      return i(s.before, s.after) ? l && f : f || l;
    }
    return Je(s) ? a(e, s.after) > 0 : Le(s) ? a(s.before, e) > 0 : typeof s == "function" ? s(e) : !1;
  });
}
function Ao(e, t, r, n, o) {
  const { disabled: a, hidden: i, modifiers: s, showOutsideDays: c, broadcastCalendar: u, today: f = o.today() } = t, { isSameDay: l, isSameMonth: m, startOfMonth: g, isBefore: v, endOfMonth: C, isAfter: O } = o, Y = r && g(r), M = n && C(n), p = {
    [E.focused]: [],
    [E.outside]: [],
    [E.disabled]: [],
    [E.hidden]: [],
    [E.today]: []
  }, D = {};
  for (const b of e) {
    const { date: y, displayMonth: W } = b, _ = !!(W && !m(y, W)), I = !!(Y && v(y, Y)), F = !!(M && O(y, M)), Z = !!(a && re(y, a, o)), oe = !!(i && re(y, i, o)) || I || F || // Broadcast calendar will show outside days as default
    !u && !c && _ || u && c === !1 && _, ee = l(y, f);
    _ && p.outside.push(b), Z && p.disabled.push(b), oe && p.hidden.push(b), ee && p.today.push(b), s && Object.keys(s).forEach((R) => {
      const ce = s?.[R];
      ce && re(y, ce, o) && (D[R] ? D[R].push(b) : D[R] = [b]);
    });
  }
  return (b) => {
    const y = {
      [E.focused]: !1,
      [E.disabled]: !1,
      [E.hidden]: !1,
      [E.outside]: !1,
      [E.today]: !1
    }, W = {};
    for (const _ in p) {
      const I = p[_];
      y[_] = I.some((F) => F === b);
    }
    for (const _ in D)
      W[_] = D[_].some((I) => I === b);
    return {
      ...y,
      // custom modifiers should override all the previous ones
      ...W
    };
  };
}
function qo(e, t, r = {}) {
  return Object.entries(e).filter(([, o]) => o === !0).reduce((o, [a]) => (r[a] ? o.push(r[a]) : t[E[a]] ? o.push(t[E[a]]) : t[A[a]] && o.push(t[A[a]]), o), [t[d.Day]]);
}
function jo(e) {
  return {
    ...Ho,
    ...e
  };
}
function Go(e) {
  const t = {
    "data-mode": e.mode ?? void 0,
    "data-required": "required" in e ? e.required : void 0,
    "data-multiple-months": e.numberOfMonths && e.numberOfMonths > 1 || void 0,
    "data-week-numbers": e.showWeekNumber || void 0,
    "data-broadcast-calendar": e.broadcastCalendar || void 0,
    "data-nav-layout": e.navLayout || void 0
  };
  return Object.entries(e).forEach(([r, n]) => {
    r.startsWith("data-") && (t[r] = n);
  }), t;
}
function Ro() {
  const e = {};
  for (const t in d)
    e[d[t]] = `rdp-${d[t]}`;
  for (const t in E)
    e[E[t]] = `rdp-${E[t]}`;
  for (const t in A)
    e[A[t]] = `rdp-${A[t]}`;
  for (const t in j)
    e[j[t]] = `rdp-${j[t]}`;
  return e;
}
function At(e, t, r) {
  return (r ?? new G(t)).formatMonthYear(e);
}
const $o = At;
function zo(e, t, r) {
  return (r ?? new G(t)).format(e, "d");
}
function Qo(e, t = L) {
  return t.format(e, "LLLL");
}
function Xo(e, t, r) {
  return (r ?? new G(t)).format(e, "cccccc");
}
function Uo(e, t = L) {
  return e < 10 ? t.formatNumber(`0${e.toLocaleString()}`) : t.formatNumber(`${e.toLocaleString()}`);
}
function Vo() {
  return "";
}
function qt(e, t = L) {
  return t.format(e, "yyyy");
}
const Zo = qt, Ko = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  formatCaption: At,
  formatDay: zo,
  formatMonthCaption: $o,
  formatMonthDropdown: Qo,
  formatWeekNumber: Uo,
  formatWeekNumberHeader: Vo,
  formatWeekdayName: Xo,
  formatYearCaption: Zo,
  formatYearDropdown: qt
}, Symbol.toStringTag, { value: "Module" }));
function Jo(e) {
  return e?.formatMonthCaption && !e.formatCaption && (e.formatCaption = e.formatMonthCaption), e?.formatYearCaption && !e.formatYearDropdown && (e.formatYearDropdown = e.formatYearCaption), {
    ...Ko,
    ...e
  };
}
function et(e, t, r, n) {
  let o = (n ?? new G(r)).format(e, "PPPP");
  return t.today && (o = `Today, ${o}`), t.selected && (o = `${o}, selected`), o;
}
const Lo = et;
function tt(e, t, r) {
  return (r ?? new G(t)).formatMonthYear(e);
}
const es = tt;
function jt(e, t, r, n) {
  let o = (n ?? new G(r)).format(e, "PPPP");
  return t?.today && (o = `Today, ${o}`), o;
}
function Gt(e) {
  return "Choose the Month";
}
function Rt() {
  return "";
}
const ts = "Go to the Next Month";
function $t(e, t) {
  return ts;
}
function zt(e) {
  return "Go to the Previous Month";
}
function Qt(e, t, r) {
  return (r ?? new G(t)).format(e, "cccc");
}
function Xt(e, t) {
  return `Week ${e}`;
}
function Ut(e) {
  return "Week Number";
}
function Vt(e) {
  return "Choose the Year";
}
const ns = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  labelCaption: es,
  labelDay: Lo,
  labelDayButton: et,
  labelGrid: tt,
  labelGridcell: jt,
  labelMonthDropdown: Gt,
  labelNav: Rt,
  labelNext: $t,
  labelPrevious: zt,
  labelWeekNumber: Xt,
  labelWeekNumberHeader: Ut,
  labelWeekday: Qt,
  labelYearDropdown: Vt
}, Symbol.toStringTag, { value: "Module" })), U = (e, t, r) => t || (r ? typeof r == "function" ? r : (...n) => r : e);
function rs(e, t) {
  const r = t.locale?.labels ?? {};
  return {
    ...ns,
    ...e ?? {},
    labelDayButton: U(et, e?.labelDayButton, r.labelDayButton),
    labelMonthDropdown: U(Gt, e?.labelMonthDropdown, r.labelMonthDropdown),
    labelNext: U($t, e?.labelNext, r.labelNext),
    labelPrevious: U(zt, e?.labelPrevious, r.labelPrevious),
    labelWeekNumber: U(Xt, e?.labelWeekNumber, r.labelWeekNumber),
    labelYearDropdown: U(Vt, e?.labelYearDropdown, r.labelYearDropdown),
    labelGrid: U(tt, e?.labelGrid, r.labelGrid),
    labelGridcell: U(jt, e?.labelGridcell, r.labelGridcell),
    labelNav: U(Rt, e?.labelNav, r.labelNav),
    labelWeekNumberHeader: U(Ut, e?.labelWeekNumberHeader, r.labelWeekNumberHeader),
    labelWeekday: U(Qt, e?.labelWeekday, r.labelWeekday)
  };
}
function os(e, t, r, n, o) {
  const { startOfMonth: a, startOfYear: i, endOfYear: s, eachMonthOfInterval: c, getMonth: u } = o;
  return c({
    start: i(e),
    end: s(e)
  }).map((m) => {
    const g = n.formatMonthDropdown(m, o), v = u(m), C = t && m < a(t) || r && m > a(r) || !1;
    return { value: v, label: g, disabled: C };
  });
}
function ss(e, t = {}, r = {}) {
  let n = { ...t?.[d.Day] };
  return Object.entries(e).filter(([, o]) => o === !0).forEach(([o]) => {
    n = {
      ...n,
      ...r?.[o]
    };
  }), n;
}
function as(e, t, r, n) {
  const o = n ?? e.today(), a = r ? e.startOfBroadcastWeek(o, e) : t ? e.startOfISOWeek(o) : e.startOfWeek(o), i = [];
  for (let s = 0; s < 7; s++) {
    const c = e.addDays(a, s);
    i.push(c);
  }
  return i;
}
function is(e, t, r, n, o = !1) {
  if (!e || !t)
    return;
  const { startOfYear: a, endOfYear: i, eachYearOfInterval: s, getYear: c } = n, u = a(e), f = i(t), l = s({ start: u, end: f });
  return o && l.reverse(), l.map((m) => {
    const g = r.formatYearDropdown(m, n);
    return {
      value: c(m),
      label: g,
      disabled: !1
    };
  });
}
function cs(e, t = {}) {
  const { weekStartsOn: r, locale: n } = t, o = r ?? n?.options?.weekStartsOn ?? 0, a = (s) => {
    const c = typeof s == "number" || typeof s == "string" ? new Date(s) : s;
    return new H(c.getFullYear(), c.getMonth(), c.getDate(), 12, 0, 0, e);
  }, i = (s) => {
    const c = a(s);
    return new Date(c.getFullYear(), c.getMonth(), c.getDate(), 0, 0, 0, 0);
  };
  return {
    today: () => a(H.tz(e)),
    newDate: (s, c, u) => new H(s, c, u, 12, 0, 0, e),
    startOfDay: (s) => a(s),
    startOfWeek: (s, c) => {
      const u = a(s), f = c?.weekStartsOn ?? o, l = (u.getDay() - f + 7) % 7;
      return u.setDate(u.getDate() - l), u;
    },
    startOfISOWeek: (s) => {
      const c = a(s), u = (c.getDay() - 1 + 7) % 7;
      return c.setDate(c.getDate() - u), c;
    },
    startOfMonth: (s) => {
      const c = a(s);
      return c.setDate(1), c;
    },
    startOfYear: (s) => {
      const c = a(s);
      return c.setMonth(0, 1), c;
    },
    endOfWeek: (s, c) => {
      const u = a(s), m = (((c?.weekStartsOn ?? o) + 6) % 7 - u.getDay() + 7) % 7;
      return u.setDate(u.getDate() + m), u;
    },
    endOfISOWeek: (s) => {
      const c = a(s), u = (7 - c.getDay()) % 7;
      return c.setDate(c.getDate() + u), c;
    },
    endOfMonth: (s) => {
      const c = a(s);
      return c.setMonth(c.getMonth() + 1, 0), c;
    },
    endOfYear: (s) => {
      const c = a(s);
      return c.setMonth(11, 31), c;
    },
    eachMonthOfInterval: (s) => {
      const c = a(s.start), u = a(s.end), f = [], l = new H(c.getFullYear(), c.getMonth(), 1, 12, 0, 0, e), m = u.getFullYear() * 12 + u.getMonth();
      for (; l.getFullYear() * 12 + l.getMonth() <= m; )
        f.push(new H(l, e)), l.setMonth(l.getMonth() + 1, 1);
      return f;
    },
    // Normalize to noon once before arithmetic (avoid DST/midnight edge cases),
    // mutate the same TZDate, and return it.
    addDays: (s, c) => {
      const u = a(s);
      return u.setDate(u.getDate() + c), u;
    },
    addWeeks: (s, c) => {
      const u = a(s);
      return u.setDate(u.getDate() + c * 7), u;
    },
    addMonths: (s, c) => {
      const u = a(s);
      return u.setMonth(u.getMonth() + c), u;
    },
    addYears: (s, c) => {
      const u = a(s);
      return u.setFullYear(u.getFullYear() + c), u;
    },
    eachYearOfInterval: (s) => {
      const c = a(s.start), u = a(s.end), f = [], l = new H(c.getFullYear(), 0, 1, 12, 0, 0, e);
      for (; l.getFullYear() <= u.getFullYear(); )
        f.push(new H(l, e)), l.setFullYear(l.getFullYear() + 1, 0, 1);
      return f;
    },
    getWeek: (s, c) => {
      const u = i(s);
      return Ze(u, {
        weekStartsOn: c?.weekStartsOn ?? o,
        firstWeekContainsDate: c?.firstWeekContainsDate ?? n?.options?.firstWeekContainsDate ?? 1
      });
    },
    getISOWeek: (s) => {
      const c = i(s);
      return Ve(c);
    },
    differenceInCalendarDays: (s, c) => {
      const u = i(s), f = i(c);
      return Ue(u, f);
    },
    differenceInCalendarMonths: (s, c) => {
      const u = i(s), f = i(c);
      return Ct(u, f);
    }
  };
}
const ke = (e) => e instanceof HTMLElement ? e : null, Ge = (e) => [
  ...e.querySelectorAll("[data-animated-month]") ?? []
], us = (e) => ke(e.querySelector("[data-animated-month]")), Re = (e) => ke(e.querySelector("[data-animated-caption]")), $e = (e) => ke(e.querySelector("[data-animated-weeks]")), ls = (e) => ke(e.querySelector("[data-animated-nav]")), fs = (e) => ke(e.querySelector("[data-animated-weekdays]"));
function ds(e, t, { classNames: r, months: n, focused: o, dateLib: a }) {
  const i = Se(null), s = Se(n), c = Se(!1);
  Nn(() => {
    const u = s.current;
    if (s.current = n, !t || !e.current || // safety check because the ref can be set to anything by consumers
    !(e.current instanceof HTMLElement) || // validation required for the animation to work as expected
    n.length === 0 || u.length === 0 || n.length !== u.length)
      return;
    const f = a.isSameMonth(n[0].date, u[0].date), l = a.isAfter(n[0].date, u[0].date), m = l ? r[j.caption_after_enter] : r[j.caption_before_enter], g = l ? r[j.weeks_after_enter] : r[j.weeks_before_enter], v = i.current, C = e.current.cloneNode(!0);
    if (C instanceof HTMLElement ? (Ge(C).forEach((p) => {
      if (!(p instanceof HTMLElement))
        return;
      const D = us(p);
      D && p.contains(D) && p.removeChild(D);
      const b = Re(p);
      b && b.classList.remove(m);
      const y = $e(p);
      y && y.classList.remove(g);
    }), i.current = C) : i.current = null, c.current || f || // skip animation if a day is focused because it can cause issues to the animation and is better for a11y
    o)
      return;
    const O = v instanceof HTMLElement ? Ge(v) : [], Y = Ge(e.current);
    if (Y?.every((M) => M instanceof HTMLElement) && O && O.every((M) => M instanceof HTMLElement)) {
      c.current = !0, e.current.style.isolation = "isolate";
      const M = ls(e.current);
      M && (M.style.zIndex = "1"), Y.forEach((p, D) => {
        const b = O[D];
        if (!b)
          return;
        p.style.position = "relative", p.style.overflow = "hidden";
        const y = Re(p);
        y && y.classList.add(m);
        const W = $e(p);
        W && W.classList.add(g);
        const _ = () => {
          c.current = !1, e.current && (e.current.style.isolation = ""), M && (M.style.zIndex = ""), y && y.classList.remove(m), W && W.classList.remove(g), p.style.position = "", p.style.overflow = "", p.contains(b) && p.removeChild(b);
        };
        b.style.pointerEvents = "none", b.style.position = "absolute", b.style.overflow = "hidden", b.setAttribute("aria-hidden", "true");
        const I = fs(b);
        I && (I.style.opacity = "0");
        const F = Re(b);
        F && (F.classList.add(l ? r[j.caption_before_exit] : r[j.caption_after_exit]), F.addEventListener("animationend", _));
        const Z = $e(b);
        Z && Z.classList.add(l ? r[j.weeks_before_exit] : r[j.weeks_after_exit]), p.insertBefore(b, p.firstChild);
      });
    }
  });
}
function hs(e, t, r, n) {
  const o = e[0], a = e[e.length - 1], { ISOWeek: i, fixedWeeks: s, broadcastCalendar: c } = r ?? {}, { addDays: u, differenceInCalendarDays: f, differenceInCalendarMonths: l, endOfBroadcastWeek: m, endOfISOWeek: g, endOfMonth: v, endOfWeek: C, isAfter: O, startOfBroadcastWeek: Y, startOfISOWeek: M, startOfWeek: p } = n, D = c ? Y(o, n) : i ? M(o) : p(o), b = c ? m(a) : i ? g(v(a)) : C(v(a)), y = t && (c ? m(t) : i ? g(t) : C(t)), W = y && O(b, y) ? y : b, _ = f(W, D), I = l(a, o) + 1, F = [];
  for (let ee = 0; ee <= _; ee++) {
    const R = u(D, ee);
    F.push(R);
  }
  const oe = (c ? 35 : 42) * I;
  if (s && F.length < oe) {
    const ee = oe - F.length;
    for (let R = 0; R < ee; R++) {
      const ce = u(F[F.length - 1], 1);
      F.push(ce);
    }
  }
  return F;
}
function ms(e) {
  const t = [];
  return e.reduce((r, n) => {
    const o = n.weeks.reduce((a, i) => a.concat(i.days.slice()), t.slice());
    return r.concat(o.slice());
  }, t.slice());
}
function ys(e, t, r, n) {
  const { numberOfMonths: o = 1 } = r, a = [];
  for (let i = 0; i < o; i++) {
    const s = n.addMonths(e, i);
    if (t && s > t)
      break;
    a.push(s);
  }
  return a;
}
function bt(e, t, r, n) {
  const { month: o, defaultMonth: a, today: i = n.today(), numberOfMonths: s = 1 } = e;
  let c = o || a || i;
  const { differenceInCalendarMonths: u, addMonths: f, startOfMonth: l } = n;
  if (r && u(r, c) < s - 1) {
    const m = -1 * (s - 1);
    c = f(r, m);
  }
  return t && u(c, t) < 0 && (c = t), l(c);
}
function gs(e, t, r, n) {
  const { addDays: o, endOfBroadcastWeek: a, endOfISOWeek: i, endOfMonth: s, endOfWeek: c, getISOWeek: u, getWeek: f, startOfBroadcastWeek: l, startOfISOWeek: m, startOfWeek: g } = n, v = e.reduce((C, O) => {
    const Y = r.broadcastCalendar ? l(O, n) : r.ISOWeek ? m(O) : g(O), M = r.broadcastCalendar ? a(O) : r.ISOWeek ? i(s(O)) : c(s(O)), p = t.filter((W) => W >= Y && W <= M), D = r.broadcastCalendar ? 35 : 42;
    if (r.fixedWeeks && p.length < D) {
      const W = t.filter((_) => {
        const I = D - p.length;
        return _ > M && _ <= o(M, I);
      });
      p.push(...W);
    }
    const b = p.reduce((W, _) => {
      const I = r.ISOWeek ? u(_) : f(_), F = W.find((oe) => oe.weekNumber === I), Z = new Ft(_, O, n);
      return F ? F.days.push(Z) : W.push(new lo(I, [Z])), W;
    }, []), y = new uo(O, b);
    return C.push(y), C;
  }, []);
  return r.reverseMonths ? v.reverse() : v;
}
function bs(e, t) {
  let { startMonth: r, endMonth: n } = e;
  const { startOfYear: o, startOfDay: a, startOfMonth: i, endOfMonth: s, addYears: c, endOfYear: u, newDate: f, today: l } = t, { fromYear: m, toYear: g, fromMonth: v, toMonth: C } = e;
  !r && v && (r = v), !r && m && (r = t.newDate(m, 0, 1)), !n && C && (n = C), !n && g && (n = f(g, 11, 31));
  const O = e.captionLayout === "dropdown" || e.captionLayout === "dropdown-years";
  return r ? r = i(r) : m ? r = f(m, 0, 1) : !r && O && (r = o(c(e.today ?? l(), -100))), n ? n = s(n) : g ? n = f(g, 11, 31) : !n && O && (n = u(e.today ?? l())), [
    r && a(r),
    n && a(n)
  ];
}
function ps(e, t, r, n) {
  if (r.disableNavigation)
    return;
  const { pagedNavigation: o, numberOfMonths: a = 1 } = r, { startOfMonth: i, addMonths: s, differenceInCalendarMonths: c } = n, u = o ? a : 1, f = i(e);
  if (!t)
    return s(f, u);
  if (!(c(t, e) < a))
    return s(f, u);
}
function ws(e, t, r, n) {
  if (r.disableNavigation)
    return;
  const { pagedNavigation: o, numberOfMonths: a } = r, { startOfMonth: i, addMonths: s, differenceInCalendarMonths: c } = n, u = o ? a ?? 1 : 1, f = i(e);
  if (!t)
    return s(f, -u);
  if (!(c(f, t) <= 0))
    return s(f, -u);
}
function Ms(e) {
  const t = [];
  return e.reduce((r, n) => r.concat(n.weeks.slice()), t.slice());
}
function Ye(e, t) {
  const [r, n] = Qe(e);
  return [t === void 0 ? r : t, n];
}
function Ds(e, t) {
  const [r, n] = bs(e, t), { startOfMonth: o, endOfMonth: a } = t, i = bt(e, r, n, t), [s, c] = Ye(
    i,
    // initialMonth is always computed from props.month if provided
    e.month ? i : void 0
  );
  xn(() => {
    const D = bt(e, r, n, t);
    c(D);
  }, [e.timeZone]);
  const { months: u, weeks: f, days: l, previousMonth: m, nextMonth: g } = Ce(() => {
    const D = ys(s, n, { numberOfMonths: e.numberOfMonths }, t), b = hs(D, e.endMonth ? a(e.endMonth) : void 0, {
      ISOWeek: e.ISOWeek,
      fixedWeeks: e.fixedWeeks,
      broadcastCalendar: e.broadcastCalendar
    }, t), y = gs(D, b, {
      broadcastCalendar: e.broadcastCalendar,
      fixedWeeks: e.fixedWeeks,
      ISOWeek: e.ISOWeek,
      reverseMonths: e.reverseMonths
    }, t), W = Ms(y), _ = ms(y), I = ws(s, r, e, t), F = ps(s, n, e, t);
    return {
      months: y,
      weeks: W,
      days: _,
      previousMonth: I,
      nextMonth: F
    };
  }, [
    t,
    s.getTime(),
    n?.getTime(),
    r?.getTime(),
    e.disableNavigation,
    e.broadcastCalendar,
    e.endMonth?.getTime(),
    e.fixedWeeks,
    e.ISOWeek,
    e.numberOfMonths,
    e.pagedNavigation,
    e.reverseMonths
  ]), { disableNavigation: v, onMonthChange: C } = e, O = (D) => f.some((b) => b.days.some((y) => y.isEqualTo(D))), Y = (D) => {
    if (v)
      return;
    let b = o(D);
    r && b < o(r) && (b = o(r)), n && b > o(n) && (b = o(n)), c(b), C?.(b);
  };
  return {
    months: u,
    weeks: f,
    days: l,
    navStart: r,
    navEnd: n,
    previousMonth: m,
    nextMonth: g,
    goToMonth: Y,
    goToDay: (D) => {
      O(D) || Y(D.date);
    }
  };
}
var K;
(function(e) {
  e[e.Today = 0] = "Today", e[e.Selected = 1] = "Selected", e[e.LastFocused = 2] = "LastFocused", e[e.FocusedModifier = 3] = "FocusedModifier";
})(K || (K = {}));
function pt(e) {
  return !e[E.disabled] && !e[E.hidden] && !e[E.outside];
}
function ks(e, t, r, n) {
  let o, a = -1;
  for (const i of e) {
    const s = t(i);
    pt(s) && (s[E.focused] && a < K.FocusedModifier ? (o = i, a = K.FocusedModifier) : n?.isEqualTo(i) && a < K.LastFocused ? (o = i, a = K.LastFocused) : r(i.date) && a < K.Selected ? (o = i, a = K.Selected) : s[E.today] && a < K.Today && (o = i, a = K.Today));
  }
  return o || (o = e.find((i) => pt(t(i)))), o;
}
function Os(e, t, r, n, o, a, i) {
  const { ISOWeek: s, broadcastCalendar: c } = a, { addDays: u, addMonths: f, addWeeks: l, addYears: m, endOfBroadcastWeek: g, endOfISOWeek: v, endOfWeek: C, max: O, min: Y, startOfBroadcastWeek: M, startOfISOWeek: p, startOfWeek: D } = i;
  let y = {
    day: u,
    week: l,
    month: f,
    year: m,
    startOfWeek: (W) => c ? M(W, i) : s ? p(W) : D(W),
    endOfWeek: (W) => c ? g(W) : s ? v(W) : C(W)
  }[e](r, t === "after" ? 1 : -1);
  return t === "before" && n ? y = O([n, y]) : t === "after" && o && (y = Y([o, y])), y;
}
function Zt(e, t, r, n, o, a, i, s = 0) {
  if (s > 365)
    return;
  const c = Os(e, t, r.date, n, o, a, i), u = !!(a.disabled && re(c, a.disabled, i)), f = !!(a.hidden && re(c, a.hidden, i)), l = c, m = new Ft(c, l, i);
  return !u && !f ? m : Zt(e, t, m, n, o, a, i, s + 1);
}
function vs(e, t, r, n, o) {
  const { autoFocus: a } = e, [i, s] = Qe(), c = ks(t.days, r, n || (() => !1), i), [u, f] = Qe(a ? c : void 0);
  return {
    isFocusTarget: (C) => !!c?.isEqualTo(C),
    setFocused: f,
    focused: u,
    blur: () => {
      s(u), f(void 0);
    },
    moveFocus: (C, O) => {
      if (!u)
        return;
      const Y = Zt(C, O, u, t.navStart, t.navEnd, e, o);
      Y && (e.disableNavigation && !t.days.some((p) => p.isEqualTo(Y)) || (t.goToDay(Y), f(Y)));
    }
  };
}
function Ws(e, t) {
  const { selected: r, required: n, onSelect: o } = e, [a, i] = Ye(r, o ? r : void 0), s = o ? r : a, { isSameDay: c } = t, u = (g) => s?.some((v) => c(v, g)) ?? !1, { min: f, max: l } = e;
  return {
    selected: s,
    select: (g, v, C) => {
      let O = [...s ?? []];
      if (u(g)) {
        if (s?.length === f || n && s?.length === 1)
          return;
        O = s?.filter((Y) => !c(Y, g));
      } else
        s?.length === l ? O = [g] : O = [...O, g];
      return o || i(O), o?.(O, g, v, C), O;
    },
    isSelected: u
  };
}
function Ss(e, t, r = 0, n = 0, o = !1, a = L) {
  const { from: i, to: s } = t || {}, { isSameDay: c, isAfter: u, isBefore: f } = a;
  let l;
  if (!i && !s)
    l = { from: e, to: r > 0 ? void 0 : e };
  else if (i && !s)
    c(i, e) ? r === 0 ? l = { from: i, to: e } : o ? l = { from: i, to: void 0 } : l = void 0 : f(e, i) ? l = { from: e, to: i } : l = { from: i, to: e };
  else if (i && s)
    if (c(i, e) && c(s, e))
      o ? l = { from: i, to: s } : l = void 0;
    else if (c(i, e))
      l = { from: i, to: r > 0 ? void 0 : e };
    else if (c(s, e))
      l = { from: e, to: r > 0 ? void 0 : e };
    else if (f(e, i))
      l = { from: e, to: s };
    else if (u(e, i))
      l = { from: i, to: e };
    else if (u(e, s))
      l = { from: i, to: e };
    else
      throw new Error("Invalid range");
  if (l?.from && l?.to) {
    const m = a.differenceInCalendarDays(l.to, l.from);
    n > 0 && m > n ? l = { from: e, to: void 0 } : r > 1 && m < r && (l = { from: e, to: void 0 });
  }
  return l;
}
function Cs(e, t, r = L) {
  const n = Array.isArray(t) ? t : [t];
  let o = e.from;
  const a = r.differenceInCalendarDays(e.to, e.from), i = Math.min(a, 6);
  for (let s = 0; s <= i; s++) {
    if (n.includes(o.getDay()))
      return !0;
    o = r.addDays(o, 1);
  }
  return !1;
}
function wt(e, t, r = L) {
  return ne(e, t.from, !1, r) || ne(e, t.to, !1, r) || ne(t, e.from, !1, r) || ne(t, e.to, !1, r);
}
function Ns(e, t, r = L) {
  const n = Array.isArray(t) ? t : [t];
  if (n.filter((s) => typeof s != "function").some((s) => typeof s == "boolean" ? s : r.isDate(s) ? ne(e, s, !1, r) : Ht(s, r) ? s.some((c) => ne(e, c, !1, r)) : xe(s) ? s.from && s.to ? wt(e, { from: s.from, to: s.to }, r) : !1 : It(s) ? Cs(e, s.dayOfWeek, r) : Ke(s) ? r.isAfter(s.before, s.after) ? wt(e, {
    from: r.addDays(s.after, 1),
    to: r.addDays(s.before, -1)
  }, r) : re(e.from, s, r) || re(e.to, s, r) : Je(s) || Le(s) ? re(e.from, s, r) || re(e.to, s, r) : !1))
    return !0;
  const i = n.filter((s) => typeof s == "function");
  if (i.length) {
    let s = e.from;
    const c = r.differenceInCalendarDays(e.to, e.from);
    for (let u = 0; u <= c; u++) {
      if (i.some((f) => f(s)))
        return !0;
      s = r.addDays(s, 1);
    }
  }
  return !1;
}
function xs(e, t) {
  const { disabled: r, excludeDisabled: n, resetOnSelect: o, selected: a, required: i, onSelect: s } = e, [c, u] = Ye(a, s ? a : void 0), f = s ? a : c;
  return {
    selected: f,
    select: (g, v, C) => {
      const { min: O, max: Y } = e;
      let M;
      if (g) {
        const p = f?.from, D = f?.to, b = !!p && !!D, y = !!p && !!D && t.isSameDay(p, D) && t.isSameDay(g, p);
        o && (b || !f?.from) ? !i && y ? M = void 0 : M = { from: g, to: void 0 } : M = Ss(g, f, O, Y, i, t);
      }
      return n && r && M?.from && M.to && Ns({ from: M.from, to: M.to }, r, t) && (M.from = g, M.to = void 0), s || u(M), s?.(M, g, v, C), M;
    },
    isSelected: (g) => f && ne(f, g, !1, t)
  };
}
function Ys(e, t) {
  const { selected: r, required: n, onSelect: o } = e, [a, i] = Ye(r, o ? r : void 0), s = o ? r : a, { isSameDay: c } = t;
  return {
    selected: s,
    select: (l, m, g) => {
      let v = l;
      return !n && s && s && c(l, s) && (v = void 0), o || i(v), o?.(v, l, m, g), v;
    },
    isSelected: (l) => s ? c(s, l) : !1
  };
}
function Ts(e, t) {
  const r = Ys(e, t), n = Ws(e, t), o = xs(e, t);
  switch (e.mode) {
    case "single":
      return r;
    case "multiple":
      return n;
    case "range":
      return o;
    default:
      return;
  }
}
function z(e, t) {
  return e instanceof H && e.timeZone === t ? e : new H(e, t);
}
function le(e, t, r) {
  return z(e, t);
}
function Mt(e, t, r) {
  return typeof e == "boolean" || typeof e == "function" ? e : e instanceof Date ? le(e, t) : Array.isArray(e) ? e.map((n) => n instanceof Date ? le(n, t) : n) : xe(e) ? {
    ...e,
    from: e.from ? z(e.from, t) : e.from,
    to: e.to ? z(e.to, t) : e.to
  } : Ke(e) ? {
    before: le(e.before, t),
    after: le(e.after, t)
  } : Je(e) ? {
    after: le(e.after, t)
  } : Le(e) ? {
    before: le(e.before, t)
  } : e;
}
function ze(e, t, r) {
  return e && (Array.isArray(e) ? e.map((n) => Mt(n, t)) : Mt(e, t));
}
function Ps(e) {
  let t = e;
  const r = t.timeZone;
  if (r && (t = {
    ...e,
    timeZone: r
  }, t.today && (t.today = z(t.today, r)), t.month && (t.month = z(t.month, r)), t.defaultMonth && (t.defaultMonth = z(t.defaultMonth, r)), t.startMonth && (t.startMonth = z(t.startMonth, r)), t.endMonth && (t.endMonth = z(t.endMonth, r)), t.mode === "single" && t.selected ? t.selected = z(t.selected, r) : t.mode === "multiple" && t.selected ? t.selected = t.selected?.map((k) => z(k, r)) : t.mode === "range" && t.selected && (t.selected = {
    from: t.selected.from ? z(t.selected.from, r) : t.selected.from,
    to: t.selected.to ? z(t.selected.to, r) : t.selected.to
  }), t.disabled !== void 0 && (t.disabled = ze(t.disabled, r)), t.hidden !== void 0 && (t.hidden = ze(t.hidden, r)), t.modifiers)) {
    const k = {};
    Object.keys(t.modifiers).forEach((x) => {
      k[x] = ze(t.modifiers?.[x], r);
    }), t.modifiers = k;
  }
  const { components: n, formatters: o, labels: a, dateLib: i, locale: s, classNames: c } = Ce(() => {
    const k = { ..._t, ...t.locale }, x = t.broadcastCalendar ? 1 : t.weekStartsOn, w = t.noonSafe && t.timeZone ? cs(t.timeZone, {
      weekStartsOn: x,
      locale: k
    }) : void 0, N = t.dateLib && w ? { ...w, ...t.dateLib } : t.dateLib ?? w, q = new G({
      locale: k,
      weekStartsOn: x,
      firstWeekContainsDate: t.firstWeekContainsDate,
      useAdditionalWeekYearTokens: t.useAdditionalWeekYearTokens,
      useAdditionalDayOfYearTokens: t.useAdditionalDayOfYearTokens,
      timeZone: t.timeZone,
      numerals: t.numerals
    }, N);
    return {
      dateLib: q,
      components: jo(t.components),
      formatters: Jo(t.formatters),
      labels: rs(t.labels, q.options),
      locale: k,
      classNames: { ...Ro(), ...t.classNames }
    };
  }, [
    t.locale,
    t.broadcastCalendar,
    t.weekStartsOn,
    t.firstWeekContainsDate,
    t.useAdditionalWeekYearTokens,
    t.useAdditionalDayOfYearTokens,
    t.timeZone,
    t.numerals,
    t.dateLib,
    t.noonSafe,
    t.components,
    t.formatters,
    t.labels,
    t.classNames
  ]);
  t.today || (t = { ...t, today: i.today() });
  const { captionLayout: u, mode: f, navLayout: l, numberOfMonths: m = 1, onDayBlur: g, onDayClick: v, onDayFocus: C, onDayKeyDown: O, onDayMouseEnter: Y, onDayMouseLeave: M, onNextClick: p, onPrevClick: D, showWeekNumber: b, styles: y } = t, { formatCaption: W, formatDay: _, formatMonthDropdown: I, formatWeekNumber: F, formatWeekNumberHeader: Z, formatWeekdayName: oe, formatYearDropdown: ee } = o, R = Ds(t, i), { days: ce, months: Oe, navStart: Te, navEnd: Pe, previousMonth: Q, nextMonth: X, goToMonth: te } = R, Ee = Ao(ce, t, Te, Pe, i), { isSelected: _e, select: Fe, selected: ve } = Ts(t, i) ?? {}, { blur: nt, focused: rt, isFocusTarget: Kt, moveFocus: ot, setFocused: We } = vs(t, R, Ee, _e ?? (() => !1), i), { labelDayButton: Jt, labelGridcell: Lt, labelGrid: en, labelMonthDropdown: tn, labelNav: st, labelPrevious: nn, labelNext: rn, labelWeekday: on, labelWeekNumber: sn, labelWeekNumberHeader: an, labelYearDropdown: cn } = a, un = Ce(() => as(i, t.ISOWeek, t.broadcastCalendar, t.today), [i, t.ISOWeek, t.broadcastCalendar, t.today]), at = f !== void 0 || v !== void 0, Be = $(() => {
    Q && (te(Q), D?.(Q));
  }, [Q, te, D]), Ie = $(() => {
    X && (te(X), p?.(X));
  }, [te, X, p]), ln = $((k, x) => (w) => {
    w.preventDefault(), w.stopPropagation(), We(k), !x.disabled && (Fe?.(k.date, x, w), v?.(k.date, x, w));
  }, [Fe, v, We]), fn = $((k, x) => (w) => {
    We(k), C?.(k.date, x, w);
  }, [C, We]), dn = $((k, x) => (w) => {
    nt(), g?.(k.date, x, w);
  }, [nt, g]), hn = $((k, x) => (w) => {
    const N = {
      ArrowLeft: [
        w.shiftKey ? "month" : "day",
        t.dir === "rtl" ? "after" : "before"
      ],
      ArrowRight: [
        w.shiftKey ? "month" : "day",
        t.dir === "rtl" ? "before" : "after"
      ],
      ArrowDown: [w.shiftKey ? "year" : "week", "after"],
      ArrowUp: [w.shiftKey ? "year" : "week", "before"],
      PageUp: [w.shiftKey ? "year" : "month", "before"],
      PageDown: [w.shiftKey ? "year" : "month", "after"],
      Home: ["startOfWeek", "before"],
      End: ["endOfWeek", "after"]
    };
    if (N[w.key]) {
      w.preventDefault(), w.stopPropagation();
      const [q, S] = N[w.key];
      ot(q, S);
    }
    O?.(k.date, x, w);
  }, [ot, O, t.dir]), mn = $((k, x) => (w) => {
    Y?.(k.date, x, w);
  }, [Y]), yn = $((k, x) => (w) => {
    M?.(k.date, x, w);
  }, [M]), gn = $((k) => (x) => {
    const w = Number(x.target.value), N = i.setMonth(i.startOfMonth(k), w);
    te(N);
  }, [i, te]), bn = $((k) => (x) => {
    const w = Number(x.target.value), N = i.setYear(i.startOfMonth(k), w);
    te(N);
  }, [i, te]), { className: pn, style: wn } = Ce(() => ({
    className: [c[d.Root], t.className].filter(Boolean).join(" "),
    style: { ...y?.[d.Root], ...t.style }
  }), [c, t.className, t.style, y]), Mn = Go(t), it = Se(null);
  ds(it, !!t.animate, {
    classNames: c,
    months: Oe,
    focused: rt,
    dateLib: i
  });
  const Dn = {
    dayPickerProps: t,
    selected: ve,
    select: Fe,
    isSelected: _e,
    months: Oe,
    nextMonth: X,
    previousMonth: Q,
    goToMonth: te,
    getModifiers: Ee,
    components: n,
    classNames: c,
    styles: y,
    labels: a,
    formatters: o
  };
  return h.createElement(
    Bt.Provider,
    { value: Dn },
    h.createElement(
      n.Root,
      { rootRef: t.animate ? it : void 0, className: pn, style: wn, dir: t.dir, id: t.id, lang: t.lang ?? s.code, nonce: t.nonce, title: t.title, role: t.role, "aria-label": t["aria-label"], "aria-labelledby": t["aria-labelledby"], ...Mn },
      h.createElement(
        n.Months,
        { className: c[d.Months], style: y?.[d.Months] },
        !t.hideNavigation && !l && h.createElement(n.Nav, { "data-animated-nav": t.animate ? "true" : void 0, className: c[d.Nav], style: y?.[d.Nav], "aria-label": st(), onPreviousClick: Be, onNextClick: Ie, previousMonth: Q, nextMonth: X }),
        Oe.map((k, x) => h.createElement(
          n.Month,
          {
            "data-animated-month": t.animate ? "true" : void 0,
            className: c[d.Month],
            style: y?.[d.Month],
            // biome-ignore lint/suspicious/noArrayIndexKey: breaks animation
            key: x,
            displayIndex: x,
            calendarMonth: k
          },
          l === "around" && !t.hideNavigation && x === 0 && h.createElement(
            n.PreviousMonthButton,
            { type: "button", className: c[d.PreviousMonthButton], tabIndex: Q ? void 0 : -1, "aria-disabled": Q ? void 0 : !0, "aria-label": nn(Q), onClick: Be, "data-animated-button": t.animate ? "true" : void 0 },
            h.createElement(n.Chevron, { disabled: Q ? void 0 : !0, className: c[d.Chevron], orientation: t.dir === "rtl" ? "right" : "left" })
          ),
          h.createElement(n.MonthCaption, { "data-animated-caption": t.animate ? "true" : void 0, className: c[d.MonthCaption], style: y?.[d.MonthCaption], calendarMonth: k, displayIndex: x }, u?.startsWith("dropdown") ? h.createElement(
            n.DropdownNav,
            { className: c[d.Dropdowns], style: y?.[d.Dropdowns] },
            (() => {
              const w = u === "dropdown" || u === "dropdown-months" ? h.createElement(n.MonthsDropdown, { key: "month", className: c[d.MonthsDropdown], "aria-label": tn(), classNames: c, components: n, disabled: !!t.disableNavigation, onChange: gn(k.date), options: os(k.date, Te, Pe, o, i), style: y?.[d.Dropdown], value: i.getMonth(k.date) }) : h.createElement("span", { key: "month" }, I(k.date, i)), N = u === "dropdown" || u === "dropdown-years" ? h.createElement(n.YearsDropdown, { key: "year", className: c[d.YearsDropdown], "aria-label": cn(i.options), classNames: c, components: n, disabled: !!t.disableNavigation, onChange: bn(k.date), options: is(Te, Pe, o, i, !!t.reverseYears), style: y?.[d.Dropdown], value: i.getYear(k.date) }) : h.createElement("span", { key: "year" }, ee(k.date, i));
              return i.getMonthYearOrder() === "year-first" ? [N, w] : [w, N];
            })(),
            h.createElement("span", { role: "status", "aria-live": "polite", style: {
              border: 0,
              clip: "rect(0 0 0 0)",
              height: "1px",
              margin: "-1px",
              overflow: "hidden",
              padding: 0,
              position: "absolute",
              width: "1px",
              whiteSpace: "nowrap",
              wordWrap: "normal"
            } }, W(k.date, i.options, i))
          ) : h.createElement(n.CaptionLabel, { className: c[d.CaptionLabel], role: "status", "aria-live": "polite" }, W(k.date, i.options, i))),
          l === "around" && !t.hideNavigation && x === m - 1 && h.createElement(
            n.NextMonthButton,
            { type: "button", className: c[d.NextMonthButton], tabIndex: X ? void 0 : -1, "aria-disabled": X ? void 0 : !0, "aria-label": rn(X), onClick: Ie, "data-animated-button": t.animate ? "true" : void 0 },
            h.createElement(n.Chevron, { disabled: X ? void 0 : !0, className: c[d.Chevron], orientation: t.dir === "rtl" ? "left" : "right" })
          ),
          x === m - 1 && l === "after" && !t.hideNavigation && h.createElement(n.Nav, { "data-animated-nav": t.animate ? "true" : void 0, className: c[d.Nav], style: y?.[d.Nav], "aria-label": st(), onPreviousClick: Be, onNextClick: Ie, previousMonth: Q, nextMonth: X }),
          h.createElement(
            n.MonthGrid,
            { role: "grid", "aria-multiselectable": f === "multiple" || f === "range", "aria-label": en(k.date, i.options, i) || void 0, className: c[d.MonthGrid], style: y?.[d.MonthGrid] },
            !t.hideWeekdays && h.createElement(
              n.Weekdays,
              { "data-animated-weekdays": t.animate ? "true" : void 0, className: c[d.Weekdays], style: y?.[d.Weekdays] },
              b && h.createElement(n.WeekNumberHeader, { "aria-label": an(i.options), className: c[d.WeekNumberHeader], style: y?.[d.WeekNumberHeader], scope: "col" }, Z()),
              un.map((w) => h.createElement(n.Weekday, { "aria-label": on(w, i.options, i), className: c[d.Weekday], key: String(w), style: y?.[d.Weekday], scope: "col" }, oe(w, i.options, i)))
            ),
            h.createElement(n.Weeks, { "data-animated-weeks": t.animate ? "true" : void 0, className: c[d.Weeks], style: y?.[d.Weeks] }, k.weeks.map((w) => h.createElement(
              n.Week,
              { className: c[d.Week], key: w.weekNumber, style: y?.[d.Week], week: w },
              b && h.createElement(n.WeekNumber, { week: w, style: y?.[d.WeekNumber], "aria-label": sn(w.weekNumber, {
                locale: s
              }), className: c[d.WeekNumber], scope: "row", role: "rowheader" }, F(w.weekNumber, i)),
              w.days.map((N) => {
                const { date: q } = N, S = Ee(N);
                if (S[E.focused] = !S.hidden && !!rt?.isEqualTo(N), S[A.selected] = _e?.(q) || S.selected, xe(ve)) {
                  const { from: He, to: Ae } = ve;
                  S[A.range_start] = !!(He && Ae && i.isSameDay(q, He)), S[A.range_end] = !!(He && Ae && i.isSameDay(q, Ae)), S[A.range_middle] = ne(ve, q, !0, i);
                }
                const kn = ss(S, y, t.modifiersStyles), On = qo(S, c, t.modifiersClassNames), vn = !at && !S.hidden ? Lt(q, S, i.options, i) : void 0;
                return h.createElement(n.Day, { key: `${N.isoDate}_${N.displayMonthId}`, day: N, modifiers: S, className: On.join(" "), style: kn, role: "gridcell", "aria-selected": S.selected || void 0, "aria-label": vn, "data-day": N.isoDate, "data-month": N.outside ? N.dateMonthId : void 0, "data-selected": S.selected || void 0, "data-disabled": S.disabled || void 0, "data-hidden": S.hidden || void 0, "data-outside": N.outside || void 0, "data-focused": S.focused || void 0, "data-today": S.today || void 0 }, !S.hidden && at ? h.createElement(n.DayButton, { className: c[d.DayButton], style: y?.[d.DayButton], type: "button", day: N, modifiers: S, disabled: !S.focused && S.disabled || void 0, "aria-disabled": S.focused && S.disabled || void 0, tabIndex: Kt(N) ? 0 : -1, "aria-label": Jt(q, S, i.options, i), onClick: ln(N, S), onBlur: dn(N, S), onFocus: fn(N, S), onKeyDown: hn(N, S), onMouseEnter: mn(N, S), onMouseLeave: yn(N, S) }, _(q, i.options, i)) : !S.hidden && _(N.date, i.options, i));
              })
            )))
          )
        ))
      ),
      t.footer && h.createElement(n.Footer, { className: c[d.Footer], style: y?.[d.Footer], role: "status", "aria-live": "polite" }, t.footer)
    )
  );
}
const Es = {
  [d.Months]: "relative flex flex-col gap-4 sm:flex-row",
  [d.Month]: "flex w-full flex-col gap-4",
  [d.MonthCaption]: "flex h-8 w-full items-center justify-center",
  [d.CaptionLabel]: "text-sm font-medium text-foreground select-none",
  [d.Nav]: "absolute inset-x-0 top-0 flex w-full items-center justify-between gap-1",
  [d.PreviousMonthButton]: "inline-flex items-center justify-center h-8 w-8 rounded-4xl p-0 select-none cursor-pointer bg-transparent text-muted-foreground hover:bg-muted hover:text-foreground transition-colors aria-disabled:opacity-50",
  [d.NextMonthButton]: "inline-flex items-center justify-center h-8 w-8 rounded-4xl p-0 select-none cursor-pointer bg-transparent text-muted-foreground hover:bg-muted hover:text-foreground transition-colors aria-disabled:opacity-50",
  [d.MonthGrid]: "w-full border-collapse",
  [d.Weekdays]: "flex",
  [d.Weekday]: "flex-1 text-[0.8rem] font-normal text-muted-foreground text-center select-none w-8",
  [d.Week]: "flex w-full mt-1",
  [d.Day]: "relative aspect-square h-full w-full p-0 text-center select-none",
  [d.DayButton]: "inline-flex items-center justify-center h-8 w-8 rounded-4xl text-sm font-normal transition-colors cursor-pointer border-none bg-transparent text-foreground hover:bg-muted",
  // Leave selection states empty — styled inline by the Day component below
  [A.selected]: "",
  [A.range_start]: "",
  [A.range_end]: "",
  [A.range_middle]: "",
  [E.today]: "bg-muted rounded-4xl",
  [E.outside]: "text-muted-foreground opacity-50",
  [E.disabled]: "text-muted-foreground opacity-50 cursor-not-allowed"
};
function _s({ className: e, modifiers: t, ...r }) {
  let n = "";
  return t.range_middle ? n = "bg-primary/10 rounded-none [&>button]:text-foreground [&>button]:hover:bg-primary/20" : t.range_start ? n = "bg-primary/10 rounded-l-4xl rounded-r-none [&>button]:bg-primary [&>button]:text-primary-foreground [&>button]:hover:bg-primary" : t.range_end ? n = "bg-primary/10 rounded-r-4xl rounded-l-none [&>button]:bg-primary [&>button]:text-primary-foreground [&>button]:hover:bg-primary" : t.selected && (n = "bg-primary rounded-4xl [&>button]:text-primary-foreground [&>button]:hover:bg-primary"), /* @__PURE__ */ V(
    "td",
    {
      ...r,
      className: Ne(e, n)
    }
  );
}
function Fs({ orientation: e }) {
  return e === "left" ? /* @__PURE__ */ V(Fn, { size: 16 }) : /* @__PURE__ */ V(Bn, { size: 16 });
}
function Bs(e) {
  return new Intl.DateTimeFormat("fr-FR", {
    day: "numeric",
    month: "long",
    year: "numeric"
  }).format(e);
}
function Is(e) {
  const { className: t, ...r } = e;
  return /* @__PURE__ */ V(
    Ps,
    {
      showOutsideDays: !0,
      ...r,
      classNames: Es,
      className: Ne("p-3", t),
      components: {
        Chevron: Fs,
        Day: _s
      }
    }
  );
}
function Rs({
  value: e,
  onChange: t,
  placeholder: r = "Choisir une date",
  label: n,
  className: o
}) {
  const [a, i] = Wn.useState(!1), s = (c) => {
    t?.(c), c && i(!1);
  };
  return /* @__PURE__ */ qe("div", { className: "flex flex-col gap-1", children: [
    n && /* @__PURE__ */ V("span", { className: "text-xs font-semibold uppercase tracking-wider text-muted-foreground mb-0.5 block", children: n }),
    /* @__PURE__ */ qe(Yn, { open: a, onOpenChange: i, children: [
      /* @__PURE__ */ V(Tn, { asChild: !0, children: /* @__PURE__ */ qe(
        "button",
        {
          type: "button",
          className: Ne(
            "flex items-center gap-2 bg-input/50 border border-transparent rounded-3xl px-3 py-2 text-sm transition-[color,box-shadow,border-color] cursor-pointer h-9",
            "focus-visible:border-ring focus-visible:ring-3 focus-visible:ring-ring/50",
            e ? "text-foreground" : "text-muted-foreground",
            o
          ),
          children: [
            /* @__PURE__ */ V(_n, { size: 16, className: "text-muted-foreground shrink-0" }),
            /* @__PURE__ */ V("span", { className: "flex-1 text-left truncate", children: e ? Bs(e) : r })
          ]
        }
      ) }),
      /* @__PURE__ */ V(Pn, { children: /* @__PURE__ */ V(
        En,
        {
          align: "start",
          sideOffset: 4,
          className: Ne(
            "z-50 rounded-3xl bg-glass shadow-lg ring-1 ring-foreground/5 dark:ring-foreground/10",
            "data-[state=open]:animate-in data-[state=open]:fade-in-0 data-[state=open]:zoom-in-95",
            "data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=closed]:zoom-out-95"
          ),
          children: /* @__PURE__ */ V(
            Is,
            {
              mode: "single",
              selected: e,
              onSelect: s
            }
          )
        }
      ) })
    ] })
  ] });
}
export {
  Is as Calendar,
  Rs as DatePicker
};
