// @btcv/ui/dist/Dialog.js
"use client";
import { jsx as t, jsxs as f } from "react/jsx-runtime";
import * as d from "react";
import { O as K, P as V, C as q, a as z, T as H, D as L, R as X, b as G } from "./_shared/index-DlSuDb9N.js";
import { X as J } from "lucide-react";
import { c as h } from "./_shared/utils-eGXXUFl7.js";
import { Button as M } from "./Button.js";
import { EasyForm as Q } from "./EasyForm.js";
const P = X, oe = G, ae = z, U = V, I = d.forwardRef(({ className: n, ...e }, r) => /* @__PURE__ */ t(
  K,
  {
    ref: r,
    className: h(
      "fixed inset-0 z-50 bg-black/30 backdrop-blur-sm",
      "data-[state=open]:animate-in data-[state=open]:fade-in-0",
      "data-[state=closed]:animate-out data-[state=closed]:fade-out-0",
      n
    ),
    ...e
  }
));
I.displayName = "DialogOverlay";
const N = d.forwardRef(({ className: n, children: e, hideClose: r, ...i }, a) => /* @__PURE__ */ f(U, { children: [
  /* @__PURE__ */ t(I, {}),
  /* @__PURE__ */ t("div", { className: "fixed inset-x-0 top-0 z-50 h-[100dvh] flex items-center justify-center overflow-y-auto p-4 pointer-events-none", children: /* @__PURE__ */ f(
    q,
    {
      ref: a,
      className: h(
        "relative pointer-events-auto my-auto",
        "w-[90%] max-w-[480px] max-h-[85dvh]",
        "bg-card text-card-foreground backdrop-blur-xl dark:bg-card/50 border border-border rounded-4xl shadow-xl ring-1 ring-foreground/5 dark:ring-foreground/10",
        "data-[state=open]:animate-in data-[state=open]:fade-in-0 data-[state=open]:zoom-in-95",
        "data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=closed]:zoom-out-95",
        "focus:outline-none",
        "flex flex-col gap-5 py-5",
        n
      ),
      ...i,
      children: [
        e,
        !r && /* @__PURE__ */ t(z, { className: "absolute right-4 top-4 w-7 h-7 rounded-xl flex items-center justify-center text-muted-foreground hover:text-foreground hover:bg-secondary transition-colors cursor-pointer outline-none focus-visible:border-ring focus-visible:ring-3 focus-visible:ring-ring/50", children: /* @__PURE__ */ t(J, { size: 18 }) })
      ]
    }
  ) })
] }));
N.displayName = "DialogContent";
function k({ className: n, ...e }) {
  return /* @__PURE__ */ t(
    "div",
    {
      className: h("flex flex-col gap-0.5 px-5", n),
      ...e
    }
  );
}
function E({ className: n, ...e }) {
  return /* @__PURE__ */ t(
    "div",
    {
      className: h(
        "flex items-center justify-end gap-2 py-4 px-5 -mb-5 border-t border-border",
        n
      ),
      ...e
    }
  );
}
const O = d.forwardRef(({ className: n, ...e }, r) => /* @__PURE__ */ t(
  H,
  {
    ref: r,
    className: h("text-base font-semibold text-foreground", n),
    ...e
  }
));
O.displayName = "DialogTitle";
const R = d.forwardRef(({ className: n, ...e }, r) => /* @__PURE__ */ t(
  L,
  {
    ref: r,
    className: h("text-sm text-muted-foreground", n),
    ...e
  }
));
R.displayName = "DialogDescription";
function S({ className: n, ...e }) {
  return /* @__PURE__ */ t(
    "div",
    {
      className: h("px-5 text-sm text-foreground", n),
      ...e
    }
  );
}
function v(n, e, r, i, a) {
  return d.isValidElement(n) ? d.cloneElement(n, {
    onClick: r,
    loading: i,
    disabled: a
  }) : /* @__PURE__ */ t(M, { variant: e, onClick: r, loading: i, disabled: a, children: n });
}
function A(n) {
  return new Promise((e) => {
    const r = document.createElement("div");
    document.body.appendChild(r), import("react-dom/client").then(({ createRoot: i }) => {
      const a = i(r);
      a.render(
        n((c) => {
          a.unmount(), r.remove(), e(c);
        })
      );
    });
  });
}
function W({
  title: n,
  description: e,
  message: r,
  button: i = "OK",
  onDone: a
}) {
  const [c, g] = d.useState(!0), u = () => {
    g(!1), setTimeout(() => a(), 200);
  };
  return /* @__PURE__ */ t(P, { open: c, onOpenChange: (x) => {
    x || u();
  }, children: /* @__PURE__ */ f(N, { children: [
    /* @__PURE__ */ f(k, { children: [
      /* @__PURE__ */ t(O, { children: n }),
      e && /* @__PURE__ */ t(R, { children: e })
    ] }),
    r && /* @__PURE__ */ t(S, { children: r }),
    /* @__PURE__ */ t(E, { children: v(i, "primary", u, !1, !1) })
  ] }) });
}
function ie(n) {
  return A((e) => /* @__PURE__ */ t(W, { ...n, onDone: e }));
}
function Y({
  title: n,
  description: e,
  message: r,
  confirmButton: i = "Confirmer",
  cancelButton: a = "Annuler",
  closable: c = !0,
  onSuccess: g,
  onCancel: u,
  onDone: x
}) {
  const [b, T] = d.useState(!0), [p, D] = d.useState(null), s = p !== null, m = (o) => {
    T(!1), setTimeout(() => x(o), 200);
  }, y = async () => {
    if (!s)
      if (g) {
        D("confirm");
        try {
          await g(), m(!0);
        } catch {
          D(null);
        }
      } else
        m(!0);
  }, C = async () => {
    if (!s)
      if (u) {
        D("cancel");
        try {
          await u(), m(!1);
        } catch {
          D(null);
        }
      } else
        m(!1);
  };
  return /* @__PURE__ */ t(
    P,
    {
      open: b,
      onOpenChange: (o) => {
        !o && !s && c && C();
      },
      children: /* @__PURE__ */ f(
        N,
        {
          hideClose: !c || s,
          onEscapeKeyDown: (o) => {
            (!c || s) && o.preventDefault();
          },
          onPointerDownOutside: (o) => {
            (!c || s) && o.preventDefault();
          },
          onInteractOutside: (o) => {
            (!c || s) && o.preventDefault();
          },
          children: [
            /* @__PURE__ */ f(k, { children: [
              /* @__PURE__ */ t(O, { children: n }),
              e && /* @__PURE__ */ t(R, { children: e })
            ] }),
            r && /* @__PURE__ */ t(S, { children: r }),
            /* @__PURE__ */ f(E, { children: [
              v(a, "secondary", C, p === "cancel", p === "confirm"),
              v(i, "primary", y, p === "confirm", p === "cancel")
            ] })
          ]
        }
      )
    }
  );
}
function le(n) {
  return A((e) => /* @__PURE__ */ t(Y, { ...n, onDone: e }));
}
function Z({
  title: n,
  description: e,
  message: r,
  fields: i,
  defaultValues: a,
  confirmButton: c = "Confirmer",
  cancelButton: g = "Annuler",
  closable: u = !0,
  onSuccess: x,
  onCancel: b,
  onDone: T
}) {
  const [p, D] = d.useState(!0), [s, m] = d.useState(null), y = d.useRef(null), C = d.useRef(null), o = s !== null, w = (l) => {
    D(!1), setTimeout(() => T(l), 200);
  }, B = async (l) => {
    if (!o)
      if (C.current = l, x) {
        m("confirm");
        try {
          await x(l), w(l);
        } catch {
          m(null);
        }
      } else
        w(l);
  }, F = () => {
    o || y.current && y.current.requestSubmit();
  }, j = async () => {
    if (!o)
      if (b) {
        m("cancel");
        try {
          await b(), w(null);
        } catch {
          m(null);
        }
      } else
        w(null);
  };
  return /* @__PURE__ */ t(
    P,
    {
      open: p,
      onOpenChange: (l) => {
        !l && !o && u && j();
      },
      children: /* @__PURE__ */ f(
        N,
        {
          hideClose: !u || o,
          onEscapeKeyDown: (l) => {
            (!u || o) && l.preventDefault();
          },
          onPointerDownOutside: (l) => {
            (!u || o) && l.preventDefault();
          },
          onInteractOutside: (l) => {
            (!u || o) && l.preventDefault();
          },
          className: "max-h-[85dvh] flex flex-col",
          children: [
            /* @__PURE__ */ f(k, { children: [
              /* @__PURE__ */ t(O, { children: n }),
              e && /* @__PURE__ */ t(R, { children: e })
            ] }),
            /* @__PURE__ */ f(S, { className: "overflow-y-auto flex flex-col gap-4", children: [
              r && /* @__PURE__ */ t("div", { children: r }),
              /* @__PURE__ */ t(
                Q,
                {
                  ref: y,
                  fields: i,
                  defaultValues: a,
                  onSubmit: B
                }
              )
            ] }),
            /* @__PURE__ */ f(E, { children: [
              v(g, "secondary", j, s === "cancel", s === "confirm"),
              v(c, "primary", F, s === "confirm", s === "cancel")
            ] })
          ]
        }
      )
    }
  );
}
function ce(n) {
  return new Promise((e) => {
    const r = document.createElement("div");
    document.body.appendChild(r), import("react-dom/client").then(({ createRoot: i }) => {
      const a = i(r);
      a.render(
        /* @__PURE__ */ t(
          Z,
          {
            ...n,
            onDone: (c) => {
              a.unmount(), r.remove(), e(c);
            }
          }
        )
      );
    });
  });
}
export {
  P as Dialog,
  S as DialogBody,
  ae as DialogClose,
  N as DialogContent,
  R as DialogDescription,
  E as DialogFooter,
  k as DialogHeader,
  I as DialogOverlay,
  U as DialogPortal,
  O as DialogTitle,
  oe as DialogTrigger,
  ie as alertDialog,
  le as confirmDialog,
  ce as promptDialog
};
