// @btcv/ui/dist/Drawer.js
"use client";
import { jsx as J, jsxs as Wt } from "react/jsx-runtime";
import * as mt from "react";
import r, { useMemo as ye, useLayoutEffect as be, useEffect as Se } from "react";
import { O as Te, C as De, T as Re, D as Ee, P as xe, a as Oe, b as Ce, R as $e } from "./_shared/index-DlSuDb9N.js";
import { c as ot } from "./_shared/utils-eGXXUFl7.js";
function Ae(t) {
  if (typeof document > "u") return;
  let n = document.head || document.getElementsByTagName("head")[0], e = document.createElement("style");
  e.type = "text/css", n.appendChild(e), e.styleSheet ? e.styleSheet.cssText = t : e.appendChild(document.createTextNode(t));
}
const qt = r.createContext({
  drawerRef: {
    current: null
  },
  overlayRef: {
    current: null
  },
  onPress: () => {
  },
  onRelease: () => {
  },
  onDrag: () => {
  },
  onNestedDrag: () => {
  },
  onNestedOpenChange: () => {
  },
  onNestedRelease: () => {
  },
  openProp: void 0,
  dismissible: !1,
  isOpen: !1,
  isDragging: !1,
  keyboardIsOpen: {
    current: !1
  },
  snapPointsOffset: null,
  snapPoints: null,
  handleOnly: !1,
  modal: !1,
  shouldFade: !1,
  activeSnapPoint: null,
  onOpenChange: () => {
  },
  setActiveSnapPoint: () => {
  },
  closeDrawer: () => {
  },
  direction: "bottom",
  shouldAnimate: {
    current: !0
  },
  shouldScaleBackground: !1,
  setBackgroundColorOnScale: !0,
  noBodyStyles: !1,
  container: null,
  autoFocus: !1
}), pt = () => {
  const t = r.useContext(qt);
  if (!t)
    throw new Error("useDrawerContext must be used within a Drawer.Root");
  return t;
};
Ae(`[data-vaul-drawer]{touch-action:none;will-change:transform;transition:transform .5s cubic-bezier(.32, .72, 0, 1);animation-duration:.5s;animation-timing-function:cubic-bezier(0.32,0.72,0,1)}[data-vaul-drawer][data-vaul-snap-points=false][data-vaul-drawer-direction=bottom][data-state=open]{animation-name:slideFromBottom}[data-vaul-drawer][data-vaul-snap-points=false][data-vaul-drawer-direction=bottom][data-state=closed]{animation-name:slideToBottom}[data-vaul-drawer][data-vaul-snap-points=false][data-vaul-drawer-direction=top][data-state=open]{animation-name:slideFromTop}[data-vaul-drawer][data-vaul-snap-points=false][data-vaul-drawer-direction=top][data-state=closed]{animation-name:slideToTop}[data-vaul-drawer][data-vaul-snap-points=false][data-vaul-drawer-direction=left][data-state=open]{animation-name:slideFromLeft}[data-vaul-drawer][data-vaul-snap-points=false][data-vaul-drawer-direction=left][data-state=closed]{animation-name:slideToLeft}[data-vaul-drawer][data-vaul-snap-points=false][data-vaul-drawer-direction=right][data-state=open]{animation-name:slideFromRight}[data-vaul-drawer][data-vaul-snap-points=false][data-vaul-drawer-direction=right][data-state=closed]{animation-name:slideToRight}[data-vaul-drawer][data-vaul-snap-points=true][data-vaul-drawer-direction=bottom]{transform:translate3d(0,var(--initial-transform,100%),0)}[data-vaul-drawer][data-vaul-snap-points=true][data-vaul-drawer-direction=top]{transform:translate3d(0,calc(var(--initial-transform,100%) * -1),0)}[data-vaul-drawer][data-vaul-snap-points=true][data-vaul-drawer-direction=left]{transform:translate3d(calc(var(--initial-transform,100%) * -1),0,0)}[data-vaul-drawer][data-vaul-snap-points=true][data-vaul-drawer-direction=right]{transform:translate3d(var(--initial-transform,100%),0,0)}[data-vaul-drawer][data-vaul-delayed-snap-points=true][data-vaul-drawer-direction=top]{transform:translate3d(0,var(--snap-point-height,0),0)}[data-vaul-drawer][data-vaul-delayed-snap-points=true][data-vaul-drawer-direction=bottom]{transform:translate3d(0,var(--snap-point-height,0),0)}[data-vaul-drawer][data-vaul-delayed-snap-points=true][data-vaul-drawer-direction=left]{transform:translate3d(var(--snap-point-height,0),0,0)}[data-vaul-drawer][data-vaul-delayed-snap-points=true][data-vaul-drawer-direction=right]{transform:translate3d(var(--snap-point-height,0),0,0)}[data-vaul-overlay][data-vaul-snap-points=false]{animation-duration:.5s;animation-timing-function:cubic-bezier(0.32,0.72,0,1)}[data-vaul-overlay][data-vaul-snap-points=false][data-state=open]{animation-name:fadeIn}[data-vaul-overlay][data-state=closed]{animation-name:fadeOut}[data-vaul-animate=false]{animation:none!important}[data-vaul-overlay][data-vaul-snap-points=true]{opacity:0;transition:opacity .5s cubic-bezier(.32, .72, 0, 1)}[data-vaul-overlay][data-vaul-snap-points=true]{opacity:1}[data-vaul-drawer]:not([data-vaul-custom-container=true])::after{content:'';position:absolute;background:inherit;background-color:inherit}[data-vaul-drawer][data-vaul-drawer-direction=top]::after{top:initial;bottom:100%;left:0;right:0;height:200%}[data-vaul-drawer][data-vaul-drawer-direction=bottom]::after{top:100%;bottom:initial;left:0;right:0;height:200%}[data-vaul-drawer][data-vaul-drawer-direction=left]::after{left:initial;right:100%;top:0;bottom:0;width:200%}[data-vaul-drawer][data-vaul-drawer-direction=right]::after{left:100%;right:initial;top:0;bottom:0;width:200%}[data-vaul-overlay][data-vaul-snap-points=true]:not([data-vaul-snap-points-overlay=true]):not(
[data-state=closed]
){opacity:0}[data-vaul-overlay][data-vaul-snap-points-overlay=true]{opacity:1}[data-vaul-handle]{display:block;position:relative;opacity:.7;background:#e2e2e4;margin-left:auto;margin-right:auto;height:5px;width:32px;border-radius:1rem;touch-action:pan-y}[data-vaul-handle]:active,[data-vaul-handle]:hover{opacity:1}[data-vaul-handle-hitarea]{position:absolute;left:50%;top:50%;transform:translate(-50%,-50%);width:max(100%,2.75rem);height:max(100%,2.75rem);touch-action:inherit}@media (hover:hover) and (pointer:fine){[data-vaul-drawer]{user-select:none}}@media (pointer:fine){[data-vaul-handle-hitarea]:{width:100%;height:100%}}@keyframes fadeIn{from{opacity:0}to{opacity:1}}@keyframes fadeOut{to{opacity:0}}@keyframes slideFromBottom{from{transform:translate3d(0,var(--initial-transform,100%),0)}to{transform:translate3d(0,0,0)}}@keyframes slideToBottom{to{transform:translate3d(0,var(--initial-transform,100%),0)}}@keyframes slideFromTop{from{transform:translate3d(0,calc(var(--initial-transform,100%) * -1),0)}to{transform:translate3d(0,0,0)}}@keyframes slideToTop{to{transform:translate3d(0,calc(var(--initial-transform,100%) * -1),0)}}@keyframes slideFromLeft{from{transform:translate3d(calc(var(--initial-transform,100%) * -1),0,0)}to{transform:translate3d(0,0,0)}}@keyframes slideToLeft{to{transform:translate3d(calc(var(--initial-transform,100%) * -1),0,0)}}@keyframes slideFromRight{from{transform:translate3d(var(--initial-transform,100%),0,0)}to{transform:translate3d(0,0,0)}}@keyframes slideToRight{to{transform:translate3d(var(--initial-transform,100%),0,0)}}`);
function Me() {
  const t = navigator.userAgent;
  return typeof window < "u" && (/Firefox/.test(t) && /Mobile/.test(t) || // Android Firefox
  /FxiOS/.test(t));
}
function Pe() {
  return Mt(/^Mac/);
}
function Ne() {
  return Mt(/^iPhone/);
}
function zt() {
  return /^((?!chrome|android).)*safari/i.test(navigator.userAgent);
}
function Ie() {
  return Mt(/^iPad/) || // iPadOS 13 lies and says it's a Mac, but we can distinguish by detecting touch support.
  Pe() && navigator.maxTouchPoints > 1;
}
function Xt() {
  return Ne() || Ie();
}
function Mt(t) {
  return typeof window < "u" && window.navigator != null ? t.test(window.navigator.platform) : void 0;
}
const _e = 24, He = typeof window < "u" ? be : Se;
function Vt(...t) {
  return (...n) => {
    for (let e of t)
      typeof e == "function" && e(...n);
  };
}
const Et = typeof document < "u" && window.visualViewport;
function jt(t) {
  let n = window.getComputedStyle(t);
  return /(auto|scroll)/.test(n.overflow + n.overflowX + n.overflowY);
}
function Kt(t) {
  for (jt(t) && (t = t.parentElement); t && !jt(t); )
    t = t.parentElement;
  return t || document.scrollingElement || document.documentElement;
}
const Le = /* @__PURE__ */ new Set([
  "checkbox",
  "radio",
  "range",
  "color",
  "file",
  "image",
  "button",
  "submit",
  "reset"
]);
let vt = 0, xt;
function ke(t = {}) {
  let { isDisabled: n } = t;
  He(() => {
    if (!n)
      return vt++, vt === 1 && Xt() && (xt = Be()), () => {
        vt--, vt === 0 && xt?.();
      };
  }, [
    n
  ]);
}
function Be() {
  let t, n = 0, e = (f) => {
    t = Kt(f.target), !(t === document.documentElement && t === document.body) && (n = f.changedTouches[0].pageY);
  }, a = (f) => {
    if (!t || t === document.documentElement || t === document.body) {
      f.preventDefault();
      return;
    }
    let m = f.changedTouches[0].pageY, B = t.scrollTop, L = t.scrollHeight - t.clientHeight;
    L !== 0 && ((B <= 0 && m > n || B >= L && m < n) && f.preventDefault(), n = m);
  }, i = (f) => {
    let m = f.target;
    $t(m) && m !== document.activeElement && (f.preventDefault(), m.style.transform = "translateY(-2000px)", m.focus(), requestAnimationFrame(() => {
      m.style.transform = "";
    }));
  }, o = (f) => {
    let m = f.target;
    $t(m) && (m.style.transform = "translateY(-2000px)", requestAnimationFrame(() => {
      m.style.transform = "", Et && (Et.height < window.innerHeight ? requestAnimationFrame(() => {
        Yt(m);
      }) : Et.addEventListener("resize", () => Yt(m), {
        once: !0
      }));
    }));
  }, v = () => {
    window.scrollTo(0, 0);
  }, w = window.pageXOffset, T = window.pageYOffset, R = Vt(Ue(document.documentElement, "paddingRight", `${window.innerWidth - document.documentElement.clientWidth}px`));
  window.scrollTo(0, 0);
  let h = Vt(dt(document, "touchstart", e, {
    passive: !1,
    capture: !0
  }), dt(document, "touchmove", a, {
    passive: !1,
    capture: !0
  }), dt(document, "touchend", i, {
    passive: !1,
    capture: !0
  }), dt(document, "focus", o, !0), dt(window, "scroll", v));
  return () => {
    R(), h(), window.scrollTo(w, T);
  };
}
function Ue(t, n, e) {
  let a = t.style[n];
  return t.style[n] = e, () => {
    t.style[n] = a;
  };
}
function dt(t, n, e, a) {
  return t.addEventListener(n, e, a), () => {
    t.removeEventListener(n, e, a);
  };
}
function Yt(t) {
  let n = document.scrollingElement || document.documentElement;
  for (; t && t !== n; ) {
    let e = Kt(t);
    if (e !== document.documentElement && e !== document.body && e !== t) {
      let a = e.getBoundingClientRect().top, i = t.getBoundingClientRect().top, o = t.getBoundingClientRect().bottom;
      const v = e.getBoundingClientRect().bottom + _e;
      o > v && (e.scrollTop += i - a);
    }
    t = e.parentElement;
  }
}
function $t(t) {
  return t instanceof HTMLInputElement && !Le.has(t.type) || t instanceof HTMLTextAreaElement || t instanceof HTMLElement && t.isContentEditable;
}
function Fe(t, n) {
  typeof t == "function" ? t(n) : t != null && (t.current = n);
}
function We(...t) {
  return (n) => t.forEach((e) => Fe(e, n));
}
function Gt(...t) {
  return mt.useCallback(We(...t), t);
}
const Jt = /* @__PURE__ */ new WeakMap();
function $(t, n, e = !1) {
  if (!t || !(t instanceof HTMLElement)) return;
  let a = {};
  Object.entries(n).forEach(([i, o]) => {
    if (i.startsWith("--")) {
      t.style.setProperty(i, o);
      return;
    }
    a[i] = t.style[i], t.style[i] = o;
  }), !e && Jt.set(t, a);
}
function ze(t, n) {
  if (!t || !(t instanceof HTMLElement)) return;
  let e = Jt.get(t);
  e && (t.style[n] = e[n]);
}
const C = (t) => {
  switch (t) {
    case "top":
    case "bottom":
      return !0;
    case "left":
    case "right":
      return !1;
    default:
      return t;
  }
};
function yt(t, n) {
  if (!t)
    return null;
  const e = window.getComputedStyle(t), a = (
    // @ts-ignore
    e.transform || e.webkitTransform || e.mozTransform
  );
  let i = a.match(/^matrix3d\((.+)\)$/);
  return i ? parseFloat(i[1].split(", ")[C(n) ? 13 : 12]) : (i = a.match(/^matrix\((.+)\)$/), i ? parseFloat(i[1].split(", ")[C(n) ? 5 : 4]) : null);
}
function Ve(t) {
  return 8 * (Math.log(t + 1) - 2);
}
function Ot(t, n) {
  if (!t) return () => {
  };
  const e = t.style.cssText;
  return Object.assign(t.style, n), () => {
    t.style.cssText = e;
  };
}
function je(...t) {
  return (...n) => {
    for (const e of t)
      typeof e == "function" && e(...n);
  };
}
const x = {
  DURATION: 0.5,
  EASE: [
    0.32,
    0.72,
    0,
    1
  ]
}, Qt = 0.4, Ye = 0.25, qe = 100, Zt = 8, et = 16, At = 26, Ct = "vaul-dragging";
function te(t) {
  const n = r.useRef(t);
  return r.useEffect(() => {
    n.current = t;
  }), r.useMemo(() => (...e) => n.current == null ? void 0 : n.current.call(n, ...e), []);
}
function Xe({ defaultProp: t, onChange: n }) {
  const e = r.useState(t), [a] = e, i = r.useRef(a), o = te(n);
  return r.useEffect(() => {
    i.current !== a && (o(a), i.current = a);
  }, [
    a,
    i,
    o
  ]), e;
}
function ee({ prop: t, defaultProp: n, onChange: e = () => {
} }) {
  const [a, i] = Xe({
    defaultProp: n,
    onChange: e
  }), o = t !== void 0, v = o ? t : a, w = te(e), T = r.useCallback((R) => {
    if (o) {
      const f = typeof R == "function" ? R(t) : R;
      f !== t && w(f);
    } else
      i(R);
  }, [
    o,
    t,
    i,
    w
  ]);
  return [
    v,
    T
  ];
}
function Ke({ activeSnapPointProp: t, setActiveSnapPointProp: n, snapPoints: e, drawerRef: a, overlayRef: i, fadeFromIndex: o, onSnapPointChange: v, direction: w = "bottom", container: T, snapToSequentialPoint: R }) {
  const [h, f] = ee({
    prop: t,
    defaultProp: e?.[0],
    onChange: n
  }), [m, B] = r.useState(typeof window < "u" ? {
    innerWidth: window.innerWidth,
    innerHeight: window.innerHeight
  } : void 0);
  r.useEffect(() => {
    function u() {
      B({
        innerWidth: window.innerWidth,
        innerHeight: window.innerHeight
      });
    }
    return window.addEventListener("resize", u), () => window.removeEventListener("resize", u);
  }, []);
  const L = r.useMemo(() => h === e?.[e.length - 1] || null, [
    e,
    h
  ]), E = r.useMemo(() => {
    var u;
    return (u = e?.findIndex((b) => b === h)) != null ? u : null;
  }, [
    e,
    h
  ]), U = e && e.length > 0 && (o || o === 0) && !Number.isNaN(o) && e[o] === h || !e, p = r.useMemo(() => {
    const u = T ? {
      width: T.getBoundingClientRect().width,
      height: T.getBoundingClientRect().height
    } : typeof window < "u" ? {
      width: window.innerWidth,
      height: window.innerHeight
    } : {
      width: 0,
      height: 0
    };
    var b;
    return (b = e?.map((y) => {
      const I = typeof y == "string";
      let P = 0;
      if (I && (P = parseInt(y, 10)), C(w)) {
        const l = I ? P : m ? y * u.height : 0;
        return m ? w === "bottom" ? u.height - l : -u.height + l : l;
      }
      const W = I ? P : m ? y * u.width : 0;
      return m ? w === "right" ? u.width - W : -u.width + W : W;
    })) != null ? b : [];
  }, [
    e,
    m,
    T
  ]), M = r.useMemo(() => E !== null ? p?.[E] : null, [
    p,
    E
  ]), A = r.useCallback((u) => {
    var b;
    const y = (b = p?.findIndex((I) => I === u)) != null ? b : null;
    v(y), $(a.current, {
      transition: `transform ${x.DURATION}s cubic-bezier(${x.EASE.join(",")})`,
      transform: C(w) ? `translate3d(0, ${u}px, 0)` : `translate3d(${u}px, 0, 0)`
    }), p && y !== p.length - 1 && o !== void 0 && y !== o && y < o ? $(i.current, {
      transition: `opacity ${x.DURATION}s cubic-bezier(${x.EASE.join(",")})`,
      opacity: "0"
    }) : $(i.current, {
      transition: `opacity ${x.DURATION}s cubic-bezier(${x.EASE.join(",")})`,
      opacity: "1"
    }), f(e?.[Math.max(y, 0)]);
  }, [
    a.current,
    e,
    p,
    o,
    i,
    f
  ]);
  r.useEffect(() => {
    if (h || t) {
      var u;
      const b = (u = e?.findIndex((y) => y === t || y === h)) != null ? u : -1;
      p && b !== -1 && typeof p[b] == "number" && A(p[b]);
    }
  }, [
    h,
    t,
    e,
    p,
    A
  ]);
  function c({ draggedDistance: u, closeDrawer: b, velocity: y, dismissible: I }) {
    if (o === void 0) return;
    const P = w === "bottom" || w === "right" ? (M ?? 0) - u : (M ?? 0) + u, W = E === o - 1, l = E === 0, F = u > 0;
    if (W && $(i.current, {
      transition: `opacity ${x.DURATION}s cubic-bezier(${x.EASE.join(",")})`
    }), !R && y > 2 && !F) {
      I ? b() : A(p[0]);
      return;
    }
    if (!R && y > 2 && F && p && e) {
      A(p[e.length - 1]);
      return;
    }
    const _ = p?.reduce((N, q) => typeof N != "number" || typeof q != "number" ? N : Math.abs(q - P) < Math.abs(N - P) ? q : N), z = C(w) ? window.innerHeight : window.innerWidth;
    if (y > Qt && Math.abs(u) < z * 0.4) {
      const N = F ? 1 : -1;
      if (N > 0 && L && e) {
        A(p[e.length - 1]);
        return;
      }
      if (l && N < 0 && I && b(), E === null) return;
      A(p[E + N]);
      return;
    }
    A(_);
  }
  function j({ draggedDistance: u }) {
    if (M === null) return;
    const b = w === "bottom" || w === "right" ? M - u : M + u;
    (w === "bottom" || w === "right") && b < p[p.length - 1] || (w === "top" || w === "left") && b > p[p.length - 1] || $(a.current, {
      transform: C(w) ? `translate3d(0, ${b}px, 0)` : `translate3d(${b}px, 0, 0)`
    });
  }
  function Z(u, b) {
    if (!e || typeof E != "number" || !p || o === void 0) return null;
    const y = E === o - 1;
    if (E >= o && b)
      return 0;
    if (y && !b) return 1;
    if (!U && !y) return null;
    const P = y ? E + 1 : E - 1, W = y ? p[P] - p[P - 1] : p[P + 1] - p[P], l = u / Math.abs(W);
    return y ? 1 - l : l;
  }
  return {
    isLastSnapPoint: L,
    activeSnapPoint: h,
    shouldFade: U,
    getPercentageDragged: Z,
    setActiveSnapPoint: f,
    activeSnapPointIndex: E,
    onRelease: c,
    onDrag: j,
    snapPointsOffset: p
  };
}
const Ge = () => () => {
};
function Je() {
  const { direction: t, isOpen: n, shouldScaleBackground: e, setBackgroundColorOnScale: a, noBodyStyles: i } = pt(), o = r.useRef(null), v = ye(() => document.body.style.backgroundColor, []);
  function w() {
    return (window.innerWidth - At) / window.innerWidth;
  }
  r.useEffect(() => {
    if (n && e) {
      o.current && clearTimeout(o.current);
      const T = document.querySelector("[data-vaul-drawer-wrapper]") || document.querySelector("[vaul-drawer-wrapper]");
      if (!T) return;
      je(a && !i ? Ot(document.body, {
        background: "black"
      }) : Ge, Ot(T, {
        transformOrigin: C(t) ? "top" : "left",
        transitionProperty: "transform, border-radius",
        transitionDuration: `${x.DURATION}s`,
        transitionTimingFunction: `cubic-bezier(${x.EASE.join(",")})`
      }));
      const R = Ot(T, {
        borderRadius: `${Zt}px`,
        overflow: "hidden",
        ...C(t) ? {
          transform: `scale(${w()}) translate3d(0, calc(env(safe-area-inset-top) + 14px), 0)`
        } : {
          transform: `scale(${w()}) translate3d(calc(env(safe-area-inset-top) + 14px), 0, 0)`
        }
      });
      return () => {
        R(), o.current = window.setTimeout(() => {
          v ? document.body.style.background = v : document.body.style.removeProperty("background");
        }, x.DURATION * 1e3);
      };
    }
  }, [
    n,
    e,
    v
  ]);
}
let ft = null;
function Qe({ isOpen: t, modal: n, nested: e, hasBeenOpened: a, preventScrollRestoration: i, noBodyStyles: o }) {
  const [v, w] = r.useState(() => typeof window < "u" ? window.location.href : ""), T = r.useRef(0), R = r.useCallback(() => {
    if (zt() && ft === null && t && !o) {
      ft = {
        position: document.body.style.position,
        top: document.body.style.top,
        left: document.body.style.left,
        height: document.body.style.height,
        right: "unset"
      };
      const { scrollX: f, innerHeight: m } = window;
      document.body.style.setProperty("position", "fixed", "important"), Object.assign(document.body.style, {
        top: `${-T.current}px`,
        left: `${-f}px`,
        right: "0px",
        height: "auto"
      }), window.setTimeout(() => window.requestAnimationFrame(() => {
        const B = m - window.innerHeight;
        B && T.current >= m && (document.body.style.top = `${-(T.current + B)}px`);
      }), 300);
    }
  }, [
    t
  ]), h = r.useCallback(() => {
    if (zt() && ft !== null && !o) {
      const f = -parseInt(document.body.style.top, 10), m = -parseInt(document.body.style.left, 10);
      Object.assign(document.body.style, ft), window.requestAnimationFrame(() => {
        if (i && v !== window.location.href) {
          w(window.location.href);
          return;
        }
        window.scrollTo(m, f);
      }), ft = null;
    }
  }, [
    v
  ]);
  return r.useEffect(() => {
    function f() {
      T.current = window.scrollY;
    }
    return f(), window.addEventListener("scroll", f), () => {
      window.removeEventListener("scroll", f);
    };
  }, []), r.useEffect(() => {
    if (n)
      return () => {
        typeof document > "u" || document.querySelector("[data-vaul-drawer]") || h();
      };
  }, [
    n,
    h
  ]), r.useEffect(() => {
    e || !a || (t ? (!window.matchMedia("(display-mode: standalone)").matches && R(), n || window.setTimeout(() => {
      h();
    }, 500)) : h());
  }, [
    t,
    a,
    v,
    n,
    e,
    R,
    h
  ]), {
    restorePositionSetting: h
  };
}
function Ze({ open: t, onOpenChange: n, children: e, onDrag: a, onRelease: i, snapPoints: o, shouldScaleBackground: v = !1, setBackgroundColorOnScale: w = !0, closeThreshold: T = Ye, scrollLockTimeout: R = qe, dismissible: h = !0, handleOnly: f = !1, fadeFromIndex: m = o && o.length - 1, activeSnapPoint: B, setActiveSnapPoint: L, fixed: E, modal: U = !0, onClose: p, nested: M, noBodyStyles: A = !1, direction: c = "bottom", defaultOpen: j = !1, disablePreventScroll: Z = !0, snapToSequentialPoint: u = !1, preventScrollRestoration: b = !1, repositionInputs: y = !0, onAnimationEnd: I, container: P, autoFocus: W = !1 }) {
  var l, F;
  const [_ = !1, z] = ee({
    defaultProp: j,
    prop: t,
    onChange: (s) => {
      n?.(s), !s && !M && de(), setTimeout(() => {
        I?.(s);
      }, x.DURATION * 1e3), s && !U && typeof window < "u" && window.requestAnimationFrame(() => {
        document.body.style.pointerEvents = "auto";
      }), s || (document.body.style.pointerEvents = "auto");
    }
  }), [N, q] = r.useState(!1), [X, it] = r.useState(!1), [oe, Pt] = r.useState(!1), nt = r.useRef(null), wt = r.useRef(null), bt = r.useRef(null), St = r.useRef(null), st = r.useRef(null), lt = r.useRef(!1), Tt = r.useRef(null), Dt = r.useRef(0), rt = r.useRef(!1), Nt = r.useRef(!j), It = r.useRef(0), d = r.useRef(null), _t = r.useRef(((l = d.current) == null ? void 0 : l.getBoundingClientRect().height) || 0), Ht = r.useRef(((F = d.current) == null ? void 0 : F.getBoundingClientRect().width) || 0), Rt = r.useRef(0), ie = r.useCallback((s) => {
    o && s === ut.length - 1 && (wt.current = /* @__PURE__ */ new Date());
  }, []), { activeSnapPoint: se, activeSnapPointIndex: at, setActiveSnapPoint: Lt, onRelease: le, snapPointsOffset: ut, onDrag: ue, shouldFade: kt, getPercentageDragged: ce } = Ke({
    snapPoints: o,
    activeSnapPointProp: B,
    setActiveSnapPointProp: L,
    drawerRef: d,
    fadeFromIndex: m,
    overlayRef: nt,
    onSnapPointChange: ie,
    direction: c,
    container: P,
    snapToSequentialPoint: u
  });
  ke({
    isDisabled: !_ || X || !U || oe || !N || !y || !Z
  });
  const { restorePositionSetting: de } = Qe({
    isOpen: _,
    modal: U,
    nested: M ?? !1,
    hasBeenOpened: N,
    preventScrollRestoration: b,
    noBodyStyles: A
  });
  function ht() {
    return (window.innerWidth - At) / window.innerWidth;
  }
  function fe(s) {
    var S, D;
    !h && !o || d.current && !d.current.contains(s.target) || (_t.current = ((S = d.current) == null ? void 0 : S.getBoundingClientRect().height) || 0, Ht.current = ((D = d.current) == null ? void 0 : D.getBoundingClientRect().width) || 0, it(!0), bt.current = /* @__PURE__ */ new Date(), Xt() && window.addEventListener("touchend", () => lt.current = !1, {
      once: !0
    }), s.target.setPointerCapture(s.pointerId), Dt.current = C(c) ? s.pageY : s.pageX);
  }
  function Bt(s, S) {
    var D;
    let g = s;
    const O = (D = window.getSelection()) == null ? void 0 : D.toString(), k = d.current ? yt(d.current, c) : null, H = /* @__PURE__ */ new Date();
    if (g.tagName === "SELECT" || g.hasAttribute("data-vaul-no-drag") || g.closest("[data-vaul-no-drag]"))
      return !1;
    if (c === "right" || c === "left")
      return !0;
    if (wt.current && H.getTime() - wt.current.getTime() < 500)
      return !1;
    if (k !== null && (c === "bottom" ? k > 0 : k < 0))
      return !0;
    if (O && O.length > 0)
      return !1;
    if (st.current && H.getTime() - st.current.getTime() < R && k === 0 || S)
      return st.current = H, !1;
    for (; g; ) {
      if (g.scrollHeight > g.clientHeight) {
        if (g.scrollTop !== 0)
          return st.current = /* @__PURE__ */ new Date(), !1;
        if (g.getAttribute("role") === "dialog")
          return !0;
      }
      g = g.parentNode;
    }
    return !0;
  }
  function me(s) {
    if (d.current && X) {
      const S = c === "bottom" || c === "right" ? 1 : -1, D = (Dt.current - (C(c) ? s.pageY : s.pageX)) * S, g = D > 0, O = o && !h && !g;
      if (O && at === 0) return;
      const k = Math.abs(D), H = document.querySelector("[data-vaul-drawer-wrapper]"), K = c === "bottom" || c === "top" ? _t.current : Ht.current;
      let V = k / K;
      const tt = ce(k, g);
      if (tt !== null && (V = tt), O && V >= 1 || !lt.current && !Bt(s.target, g)) return;
      if (d.current.classList.add(Ct), lt.current = !0, $(d.current, {
        transition: "none"
      }), $(nt.current, {
        transition: "none"
      }), o && ue({
        draggedDistance: D
      }), g && !o) {
        const Y = Ve(D), gt = Math.min(Y * -1, 0) * S;
        $(d.current, {
          transform: C(c) ? `translate3d(0, ${gt}px, 0)` : `translate3d(${gt}px, 0, 0)`
        });
        return;
      }
      const G = 1 - V;
      if ((kt || m && at === m - 1) && (a?.(s, V), $(nt.current, {
        opacity: `${G}`,
        transition: "none"
      }, !0)), H && nt.current && v) {
        const Y = Math.min(ht() + V * (1 - ht()), 1), gt = 8 - V * 8, Ft = Math.max(0, 14 - V * 14);
        $(H, {
          borderRadius: `${gt}px`,
          transform: C(c) ? `scale(${Y}) translate3d(0, ${Ft}px, 0)` : `scale(${Y}) translate3d(${Ft}px, 0, 0)`,
          transition: "none"
        }, !0);
      }
      if (!o) {
        const Y = k * S;
        $(d.current, {
          transform: C(c) ? `translate3d(0, ${Y}px, 0)` : `translate3d(${Y}px, 0, 0)`
        });
      }
    }
  }
  r.useEffect(() => {
    window.requestAnimationFrame(() => {
      Nt.current = !0;
    });
  }, []), r.useEffect(() => {
    var s;
    function S() {
      if (!d.current || !y) return;
      const D = document.activeElement;
      if ($t(D) || rt.current) {
        var g;
        const O = ((g = window.visualViewport) == null ? void 0 : g.height) || 0, k = window.innerHeight;
        let H = k - O;
        const K = d.current.getBoundingClientRect().height || 0, V = K > k * 0.8;
        Rt.current || (Rt.current = K);
        const tt = d.current.getBoundingClientRect().top;
        if (Math.abs(It.current - H) > 60 && (rt.current = !rt.current), o && o.length > 0 && ut && at) {
          const G = ut[at] || 0;
          H += G;
        }
        if (It.current = H, K > O || rt.current) {
          const G = d.current.getBoundingClientRect().height;
          let Y = G;
          G > O && (Y = O - (V ? tt : At)), E ? d.current.style.height = `${G - Math.max(H, 0)}px` : d.current.style.height = `${Math.max(Y, O - tt)}px`;
        } else Me() || (d.current.style.height = `${Rt.current}px`);
        o && o.length > 0 && !rt.current ? d.current.style.bottom = "0px" : d.current.style.bottom = `${Math.max(H, 0)}px`;
      }
    }
    return (s = window.visualViewport) == null || s.addEventListener("resize", S), () => {
      var D;
      return (D = window.visualViewport) == null ? void 0 : D.removeEventListener("resize", S);
    };
  }, [
    at,
    o,
    ut
  ]);
  function ct(s) {
    pe(), p?.(), s || z(!1), setTimeout(() => {
      o && Lt(o[0]);
    }, x.DURATION * 1e3);
  }
  function Ut() {
    if (!d.current) return;
    const s = document.querySelector("[data-vaul-drawer-wrapper]"), S = yt(d.current, c);
    $(d.current, {
      transform: "translate3d(0, 0, 0)",
      transition: `transform ${x.DURATION}s cubic-bezier(${x.EASE.join(",")})`
    }), $(nt.current, {
      transition: `opacity ${x.DURATION}s cubic-bezier(${x.EASE.join(",")})`,
      opacity: "1"
    }), v && S && S > 0 && _ && $(s, {
      borderRadius: `${Zt}px`,
      overflow: "hidden",
      ...C(c) ? {
        transform: `scale(${ht()}) translate3d(0, calc(env(safe-area-inset-top) + 14px), 0)`,
        transformOrigin: "top"
      } : {
        transform: `scale(${ht()}) translate3d(calc(env(safe-area-inset-top) + 14px), 0, 0)`,
        transformOrigin: "left"
      },
      transitionProperty: "transform, border-radius",
      transitionDuration: `${x.DURATION}s`,
      transitionTimingFunction: `cubic-bezier(${x.EASE.join(",")})`
    }, !0);
  }
  function pe() {
    !X || !d.current || (d.current.classList.remove(Ct), lt.current = !1, it(!1), St.current = /* @__PURE__ */ new Date());
  }
  function we(s) {
    if (!X || !d.current) return;
    d.current.classList.remove(Ct), lt.current = !1, it(!1), St.current = /* @__PURE__ */ new Date();
    const S = yt(d.current, c);
    if (!s || !Bt(s.target, !1) || !S || Number.isNaN(S) || bt.current === null) return;
    const D = St.current.getTime() - bt.current.getTime(), g = Dt.current - (C(c) ? s.pageY : s.pageX), O = Math.abs(g) / D;
    if (O > 0.05 && (Pt(!0), setTimeout(() => {
      Pt(!1);
    }, 200)), o) {
      le({
        draggedDistance: g * (c === "bottom" || c === "right" ? 1 : -1),
        closeDrawer: ct,
        velocity: O,
        dismissible: h
      }), i?.(s, !0);
      return;
    }
    if (c === "bottom" || c === "right" ? g > 0 : g < 0) {
      Ut(), i?.(s, !0);
      return;
    }
    if (O > Qt) {
      ct(), i?.(s, !1);
      return;
    }
    var k;
    const H = Math.min((k = d.current.getBoundingClientRect().height) != null ? k : 0, window.innerHeight);
    var K;
    const V = Math.min((K = d.current.getBoundingClientRect().width) != null ? K : 0, window.innerWidth), tt = c === "left" || c === "right";
    if (Math.abs(S) >= (tt ? V : H) * T) {
      ct(), i?.(s, !1);
      return;
    }
    i?.(s, !0), Ut();
  }
  r.useEffect(() => (_ && ($(document.documentElement, {
    scrollBehavior: "auto"
  }), wt.current = /* @__PURE__ */ new Date()), () => {
    ze(document.documentElement, "scrollBehavior");
  }), [
    _
  ]);
  function he(s) {
    const S = s ? (window.innerWidth - et) / window.innerWidth : 1, D = s ? -et : 0;
    Tt.current && window.clearTimeout(Tt.current), $(d.current, {
      transition: `transform ${x.DURATION}s cubic-bezier(${x.EASE.join(",")})`,
      transform: C(c) ? `scale(${S}) translate3d(0, ${D}px, 0)` : `scale(${S}) translate3d(${D}px, 0, 0)`
    }), !s && d.current && (Tt.current = setTimeout(() => {
      const g = yt(d.current, c);
      $(d.current, {
        transition: "none",
        transform: C(c) ? `translate3d(0, ${g}px, 0)` : `translate3d(${g}px, 0, 0)`
      });
    }, 500));
  }
  function ge(s, S) {
    if (S < 0) return;
    const D = (window.innerWidth - et) / window.innerWidth, g = D + S * (1 - D), O = -et + S * et;
    $(d.current, {
      transform: C(c) ? `scale(${g}) translate3d(0, ${O}px, 0)` : `scale(${g}) translate3d(${O}px, 0, 0)`,
      transition: "none"
    });
  }
  function ve(s, S) {
    const D = C(c) ? window.innerHeight : window.innerWidth, g = S ? (D - et) / D : 1, O = S ? -et : 0;
    S && $(d.current, {
      transition: `transform ${x.DURATION}s cubic-bezier(${x.EASE.join(",")})`,
      transform: C(c) ? `scale(${g}) translate3d(0, ${O}px, 0)` : `scale(${g}) translate3d(${O}px, 0, 0)`
    });
  }
  return r.useEffect(() => {
    U || window.requestAnimationFrame(() => {
      document.body.style.pointerEvents = "auto";
    });
  }, [
    U
  ]), /* @__PURE__ */ r.createElement($e, {
    defaultOpen: j,
    onOpenChange: (s) => {
      !h && !s || (s ? q(!0) : ct(!0), z(s));
    },
    open: _
  }, /* @__PURE__ */ r.createElement(qt.Provider, {
    value: {
      activeSnapPoint: se,
      snapPoints: o,
      setActiveSnapPoint: Lt,
      drawerRef: d,
      overlayRef: nt,
      onOpenChange: n,
      onPress: fe,
      onRelease: we,
      onDrag: me,
      dismissible: h,
      shouldAnimate: Nt,
      handleOnly: f,
      isOpen: _,
      isDragging: X,
      shouldFade: kt,
      closeDrawer: ct,
      onNestedDrag: ge,
      onNestedOpenChange: he,
      onNestedRelease: ve,
      keyboardIsOpen: rt,
      modal: U,
      snapPointsOffset: ut,
      activeSnapPointIndex: at,
      direction: c,
      shouldScaleBackground: v,
      setBackgroundColorOnScale: w,
      noBodyStyles: A,
      container: P,
      autoFocus: W
    }
  }, e));
}
const ne = /* @__PURE__ */ r.forwardRef(function({ ...t }, n) {
  const { overlayRef: e, snapPoints: a, onRelease: i, shouldFade: o, isOpen: v, modal: w, shouldAnimate: T } = pt(), R = Gt(n, e), h = a && a.length > 0;
  if (!w)
    return null;
  const f = r.useCallback((m) => i(m), [
    i
  ]);
  return /* @__PURE__ */ r.createElement(Te, {
    onMouseUp: f,
    ref: R,
    "data-vaul-overlay": "",
    "data-vaul-snap-points": v && h ? "true" : "false",
    "data-vaul-snap-points-overlay": v && o ? "true" : "false",
    "data-vaul-animate": T?.current ? "true" : "false",
    ...t
  });
});
ne.displayName = "Drawer.Overlay";
const re = /* @__PURE__ */ r.forwardRef(function({ onPointerDownOutside: t, style: n, onOpenAutoFocus: e, ...a }, i) {
  const { drawerRef: o, onPress: v, onRelease: w, onDrag: T, keyboardIsOpen: R, snapPointsOffset: h, activeSnapPointIndex: f, modal: m, isOpen: B, direction: L, snapPoints: E, container: U, handleOnly: p, shouldAnimate: M, autoFocus: A } = pt(), [c, j] = r.useState(!1), Z = Gt(i, o), u = r.useRef(null), b = r.useRef(null), y = r.useRef(!1), I = E && E.length > 0;
  Je();
  const P = (l, F, _ = 0) => {
    if (y.current) return !0;
    const z = Math.abs(l.y), N = Math.abs(l.x), q = N > z, X = [
      "bottom",
      "right"
    ].includes(F) ? 1 : -1;
    if (F === "left" || F === "right") {
      if (!(l.x * X < 0) && N >= 0 && N <= _)
        return q;
    } else if (!(l.y * X < 0) && z >= 0 && z <= _)
      return !q;
    return y.current = !0, !0;
  };
  r.useEffect(() => {
    I && window.requestAnimationFrame(() => {
      j(!0);
    });
  }, []);
  function W(l) {
    u.current = null, y.current = !1, w(l);
  }
  return /* @__PURE__ */ r.createElement(De, {
    "data-vaul-drawer-direction": L,
    "data-vaul-drawer": "",
    "data-vaul-delayed-snap-points": c ? "true" : "false",
    "data-vaul-snap-points": B && I ? "true" : "false",
    "data-vaul-custom-container": U ? "true" : "false",
    "data-vaul-animate": M?.current ? "true" : "false",
    ...a,
    ref: Z,
    style: h && h.length > 0 ? {
      "--snap-point-height": `${h[f ?? 0]}px`,
      ...n
    } : n,
    onPointerDown: (l) => {
      p || (a.onPointerDown == null || a.onPointerDown.call(a, l), u.current = {
        x: l.pageX,
        y: l.pageY
      }, v(l));
    },
    onOpenAutoFocus: (l) => {
      e?.(l), A || l.preventDefault();
    },
    onPointerDownOutside: (l) => {
      if (t?.(l), !m || l.defaultPrevented) {
        l.preventDefault();
        return;
      }
      R.current && (R.current = !1);
    },
    onFocusOutside: (l) => {
      if (!m) {
        l.preventDefault();
        return;
      }
    },
    onPointerMove: (l) => {
      if (b.current = l, p || (a.onPointerMove == null || a.onPointerMove.call(a, l), !u.current)) return;
      const F = l.pageY - u.current.y, _ = l.pageX - u.current.x, z = l.pointerType === "touch" ? 10 : 2;
      P({
        x: _,
        y: F
      }, L, z) ? T(l) : (Math.abs(_) > z || Math.abs(F) > z) && (u.current = null);
    },
    onPointerUp: (l) => {
      a.onPointerUp == null || a.onPointerUp.call(a, l), u.current = null, y.current = !1, w(l);
    },
    onPointerOut: (l) => {
      a.onPointerOut == null || a.onPointerOut.call(a, l), W(b.current);
    },
    onContextMenu: (l) => {
      a.onContextMenu == null || a.onContextMenu.call(a, l), b.current && W(b.current);
    }
  });
});
re.displayName = "Drawer.Content";
const tn = 250, en = 120, nn = /* @__PURE__ */ r.forwardRef(function({ preventCycle: t = !1, children: n, ...e }, a) {
  const { closeDrawer: i, isDragging: o, snapPoints: v, activeSnapPoint: w, setActiveSnapPoint: T, dismissible: R, handleOnly: h, isOpen: f, onPress: m, onDrag: B } = pt(), L = r.useRef(null), E = r.useRef(!1);
  function U() {
    if (E.current) {
      A();
      return;
    }
    window.setTimeout(() => {
      p();
    }, en);
  }
  function p() {
    if (o || t || E.current) {
      A();
      return;
    }
    if (A(), !v || v.length === 0) {
      R || i();
      return;
    }
    if (w === v[v.length - 1] && R) {
      i();
      return;
    }
    const j = v.findIndex((u) => u === w);
    if (j === -1) return;
    const Z = v[j + 1];
    T(Z);
  }
  function M() {
    L.current = window.setTimeout(() => {
      E.current = !0;
    }, tn);
  }
  function A() {
    L.current && window.clearTimeout(L.current), E.current = !1;
  }
  return /* @__PURE__ */ r.createElement("div", {
    onClick: U,
    onPointerCancel: A,
    onPointerDown: (c) => {
      h && m(c), M();
    },
    onPointerMove: (c) => {
      h && B(c);
    },
    // onPointerUp is already handled by the content component
    ref: a,
    "data-vaul-drawer-visible": f ? "true" : "false",
    "data-vaul-handle": "",
    "aria-hidden": "true",
    ...e
  }, /* @__PURE__ */ r.createElement("span", {
    "data-vaul-handle-hitarea": "",
    "aria-hidden": "true"
  }, n));
});
nn.displayName = "Drawer.Handle";
function rn(t) {
  const n = pt(), { container: e = n.container, ...a } = t;
  return /* @__PURE__ */ r.createElement(xe, {
    container: e,
    ...a
  });
}
const Q = {
  Root: Ze,
  Content: re,
  Overlay: ne,
  Trigger: Ce,
  Portal: rn,
  Close: Oe,
  Title: Re,
  Description: Ee
}, an = ({
  shouldScaleBackground: t = !0,
  ...n
}) => /* @__PURE__ */ J(
  Q.Root,
  {
    shouldScaleBackground: t,
    ...n
  }
);
an.displayName = "Drawer";
const pn = Q.Trigger, wn = Q.Close, on = Q.Portal, ae = mt.forwardRef(({ className: t, ...n }, e) => /* @__PURE__ */ J(
  Q.Overlay,
  {
    ref: e,
    className: ot("fixed inset-0 z-50 bg-black/30", t),
    ...n
  }
));
ae.displayName = "DrawerOverlay";
const sn = mt.forwardRef(({ className: t, children: n, ...e }, a) => /* @__PURE__ */ Wt(on, { children: [
  /* @__PURE__ */ J(ae, {}),
  /* @__PURE__ */ Wt(
    Q.Content,
    {
      ref: a,
      className: ot(
        "fixed inset-x-0 bottom-0 z-50 max-h-[85dvh] bg-glass border-t border-border rounded-t-xl",
        t
      ),
      ...e,
      children: [
        /* @__PURE__ */ J("div", { className: "mx-auto mt-3 h-1.5 w-12 rounded-full bg-muted-foreground/30" }),
        n
      ]
    }
  )
] }));
sn.displayName = "DrawerContent";
function hn({ className: t, ...n }) {
  return /* @__PURE__ */ J(
    "div",
    {
      className: ot("flex flex-col gap-1 px-4 py-3 text-center", t),
      ...n
    }
  );
}
function gn({ className: t, ...n }) {
  return /* @__PURE__ */ J(
    "div",
    {
      className: ot(
        "flex items-center gap-2 px-4 py-3 border-t border-border",
        t
      ),
      ...n
    }
  );
}
const ln = mt.forwardRef(({ className: t, ...n }, e) => /* @__PURE__ */ J(
  Q.Title,
  {
    ref: e,
    className: ot("text-base font-semibold text-foreground", t),
    ...n
  }
));
ln.displayName = "DrawerTitle";
const un = mt.forwardRef(({ className: t, ...n }, e) => /* @__PURE__ */ J(
  Q.Description,
  {
    ref: e,
    className: ot("text-sm text-muted-foreground", t),
    ...n
  }
));
un.displayName = "DrawerDescription";
export {
  an as Drawer,
  wn as DrawerClose,
  sn as DrawerContent,
  un as DrawerDescription,
  gn as DrawerFooter,
  hn as DrawerHeader,
  ae as DrawerOverlay,
  on as DrawerPortal,
  ln as DrawerTitle,
  pn as DrawerTrigger
};
