// @btcv/ui/dist/Dropdown.js
"use client";
import { jsx as t } from "react/jsx-runtime";
import * as p from "react";
import { c as m, a as P, u as O } from "./_shared/index-CXeb1OMI.js";
import { c as T } from "./_shared/index-DLcqcWxM.js";
import { P as G } from "./_shared/index-V-Ajw7Ac.js";
import { P as k, C as L, I as $, S as K, L as U, c as v, R as B, G as z, A as F, a as H, b as W, d as j, e as X, f as q, g as J, h as Q } from "./_shared/index-CukUn3R0.js";
import { u as D } from "./_shared/index-BC7vfx-u.js";
import { c as f } from "./_shared/utils-eGXXUFl7.js";
var M = "DropdownMenu", [V] = P(
  M,
  [v]
), s = v(), [Y, _] = V(M), b = (e) => {
  const {
    __scopeDropdownMenu: r,
    children: o,
    dir: n,
    open: a,
    defaultOpen: u,
    onOpenChange: i,
    modal: d = !0
  } = e, c = s(r), g = p.useRef(null), [l, w] = O({
    prop: a,
    defaultProp: u ?? !1,
    onChange: i,
    caller: M
  });
  return /* @__PURE__ */ t(
    Y,
    {
      scope: r,
      triggerId: D(),
      triggerRef: g,
      contentId: D(),
      open: l,
      onOpenChange: w,
      onOpenToggle: p.useCallback(() => w((E) => !E), [w]),
      modal: d,
      children: /* @__PURE__ */ t(B, { ...c, open: l, onOpenChange: w, dir: n, modal: d, children: o })
    }
  );
};
b.displayName = M;
var h = "DropdownMenuTrigger", R = p.forwardRef(
  (e, r) => {
    const { __scopeDropdownMenu: o, disabled: n = !1, ...a } = e, u = _(h, o), i = s(o);
    return /* @__PURE__ */ t(F, { asChild: !0, ...i, children: /* @__PURE__ */ t(
      G.button,
      {
        type: "button",
        id: u.triggerId,
        "aria-haspopup": "menu",
        "aria-expanded": u.open,
        "aria-controls": u.open ? u.contentId : void 0,
        "data-state": u.open ? "open" : "closed",
        "data-disabled": n ? "" : void 0,
        disabled: n,
        ...a,
        ref: T(r, u.triggerRef),
        onPointerDown: m(e.onPointerDown, (d) => {
          !n && d.button === 0 && d.ctrlKey === !1 && (u.onOpenToggle(), u.open || d.preventDefault());
        }),
        onKeyDown: m(e.onKeyDown, (d) => {
          n || (["Enter", " "].includes(d.key) && u.onOpenToggle(), d.key === "ArrowDown" && u.onOpenChange(!0), ["Enter", " ", "ArrowDown"].includes(d.key) && d.preventDefault());
        })
      }
    ) });
  }
);
R.displayName = h;
var Z = "DropdownMenuPortal", x = (e) => {
  const { __scopeDropdownMenu: r, ...o } = e, n = s(r);
  return /* @__PURE__ */ t(k, { ...n, ...o });
};
x.displayName = Z;
var N = "DropdownMenuContent", I = p.forwardRef(
  (e, r) => {
    const { __scopeDropdownMenu: o, ...n } = e, a = _(N, o), u = s(o), i = p.useRef(!1);
    return /* @__PURE__ */ t(
      L,
      {
        id: a.contentId,
        "aria-labelledby": a.triggerId,
        ...u,
        ...n,
        ref: r,
        onCloseAutoFocus: m(e.onCloseAutoFocus, (d) => {
          i.current || a.triggerRef.current?.focus(), i.current = !1, d.preventDefault();
        }),
        onInteractOutside: m(e.onInteractOutside, (d) => {
          const c = d.detail.originalEvent, g = c.button === 0 && c.ctrlKey === !0, l = c.button === 2 || g;
          (!a.modal || l) && (i.current = !0);
        }),
        style: {
          ...e.style,
          "--radix-dropdown-menu-content-transform-origin": "var(--radix-popper-transform-origin)",
          "--radix-dropdown-menu-content-available-width": "var(--radix-popper-available-width)",
          "--radix-dropdown-menu-content-available-height": "var(--radix-popper-available-height)",
          "--radix-dropdown-menu-trigger-width": "var(--radix-popper-anchor-width)",
          "--radix-dropdown-menu-trigger-height": "var(--radix-popper-anchor-height)"
        }
      }
    );
  }
);
I.displayName = N;
var ee = "DropdownMenuGroup", y = p.forwardRef(
  (e, r) => {
    const { __scopeDropdownMenu: o, ...n } = e, a = s(o);
    return /* @__PURE__ */ t(z, { ...a, ...n, ref: r });
  }
);
y.displayName = ee;
var oe = "DropdownMenuLabel", S = p.forwardRef(
  (e, r) => {
    const { __scopeDropdownMenu: o, ...n } = e, a = s(o);
    return /* @__PURE__ */ t(U, { ...a, ...n, ref: r });
  }
);
S.displayName = oe;
var re = "DropdownMenuItem", C = p.forwardRef(
  (e, r) => {
    const { __scopeDropdownMenu: o, ...n } = e, a = s(o);
    return /* @__PURE__ */ t($, { ...a, ...n, ref: r });
  }
);
C.displayName = re;
var ne = "DropdownMenuCheckboxItem", ae = p.forwardRef((e, r) => {
  const { __scopeDropdownMenu: o, ...n } = e, a = s(o);
  return /* @__PURE__ */ t(H, { ...a, ...n, ref: r });
});
ae.displayName = ne;
var te = "DropdownMenuRadioGroup", pe = p.forwardRef((e, r) => {
  const { __scopeDropdownMenu: o, ...n } = e, a = s(o);
  return /* @__PURE__ */ t(W, { ...a, ...n, ref: r });
});
pe.displayName = te;
var de = "DropdownMenuRadioItem", se = p.forwardRef((e, r) => {
  const { __scopeDropdownMenu: o, ...n } = e, a = s(o);
  return /* @__PURE__ */ t(j, { ...a, ...n, ref: r });
});
se.displayName = de;
var ue = "DropdownMenuItemIndicator", ie = p.forwardRef((e, r) => {
  const { __scopeDropdownMenu: o, ...n } = e, a = s(o);
  return /* @__PURE__ */ t(X, { ...a, ...n, ref: r });
});
ie.displayName = ue;
var ce = "DropdownMenuSeparator", A = p.forwardRef((e, r) => {
  const { __scopeDropdownMenu: o, ...n } = e, a = s(o);
  return /* @__PURE__ */ t(K, { ...a, ...n, ref: r });
});
A.displayName = ce;
var le = "DropdownMenuArrow", we = p.forwardRef(
  (e, r) => {
    const { __scopeDropdownMenu: o, ...n } = e, a = s(o);
    return /* @__PURE__ */ t(q, { ...a, ...n, ref: r });
  }
);
we.displayName = le;
var me = "DropdownMenuSubTrigger", fe = p.forwardRef((e, r) => {
  const { __scopeDropdownMenu: o, ...n } = e, a = s(o);
  return /* @__PURE__ */ t(J, { ...a, ...n, ref: r });
});
fe.displayName = me;
var Me = "DropdownMenuSubContent", ge = p.forwardRef((e, r) => {
  const { __scopeDropdownMenu: o, ...n } = e, a = s(o);
  return /* @__PURE__ */ t(
    Q,
    {
      ...a,
      ...n,
      ref: r,
      style: {
        ...e.style,
        "--radix-dropdown-menu-content-transform-origin": "var(--radix-popper-transform-origin)",
        "--radix-dropdown-menu-content-available-width": "var(--radix-popper-available-width)",
        "--radix-dropdown-menu-content-available-height": "var(--radix-popper-available-height)",
        "--radix-dropdown-menu-trigger-width": "var(--radix-popper-anchor-width)",
        "--radix-dropdown-menu-trigger-height": "var(--radix-popper-anchor-height)"
      }
    }
  );
});
ge.displayName = Me;
var De = b, ve = R, _e = x, be = I, he = y, Re = S, xe = C, Ne = A;
const Le = De, $e = ve, Ke = he, Ie = p.forwardRef(({ className: e, sideOffset: r = 4, container: o, ...n }, a) => /* @__PURE__ */ t(_e, { container: o, children: /* @__PURE__ */ t(
  be,
  {
    ref: a,
    sideOffset: r,
    className: f(
      "z-50 bg-glass border border-border rounded-3xl shadow-lg p-1 min-w-[180px]",
      "data-[state=open]:animate-in data-[state=open]:fade-in-0 data-[state=open]:zoom-in-95",
      "data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=closed]:zoom-out-95",
      "outline-none",
      e
    ),
    ...n
  }
) }));
Ie.displayName = "DropdownMenuContent";
const ye = p.forwardRef(({ className: e, inset: r, ...o }, n) => /* @__PURE__ */ t(
  xe,
  {
    ref: n,
    className: f(
      "flex items-center gap-2 px-3 py-2 text-sm rounded-2xl cursor-pointer outline-none",
      "text-foreground hover:bg-muted focus:bg-muted",
      "data-[disabled]:opacity-50 data-[disabled]:pointer-events-none",
      r && "pl-8",
      e
    ),
    ...o
  }
));
ye.displayName = "DropdownMenuItem";
const Se = p.forwardRef(({ className: e, ...r }, o) => /* @__PURE__ */ t(
  Ne,
  {
    ref: o,
    className: f("mx-1 my-1 h-px bg-border", e),
    ...r
  }
));
Se.displayName = "DropdownMenuSeparator";
const Ce = p.forwardRef(({ className: e, inset: r, ...o }, n) => /* @__PURE__ */ t(
  Re,
  {
    ref: n,
    className: f(
      "px-3 py-1.5 text-xs font-semibold uppercase tracking-wider text-muted-foreground",
      r && "pl-8",
      e
    ),
    ...o
  }
));
Ce.displayName = "DropdownMenuLabel";
export {
  Le as DropdownMenu,
  Ie as DropdownMenuContent,
  Ke as DropdownMenuGroup,
  ye as DropdownMenuItem,
  Ce as DropdownMenuLabel,
  Se as DropdownMenuSeparator,
  $e as DropdownMenuTrigger
};
