// @btcv/ui/dist/EasyForm.js
"use client";
import { jsxs as L, jsx as o } from "react/jsx-runtime";
import * as T from "react";
import { useState as N, useRef as _, useCallback as w } from "react";
import { Slider as z, RadioGroup as H, RadioGroupItem as J, Checkbox as K, Toggle as Q, Textarea as U, Select as $, SelectGroup as W, SelectItem as R, Input as X } from "./Input.js";
import { c as C } from "./_shared/utils-eGXXUFl7.js";
function V(e) {
  return "group" in e;
}
function Y(e) {
  if (e.type !== "row") {
    if ("defaultValue" in e && e.defaultValue !== void 0) return e.defaultValue;
    switch (e.type) {
      case "toggle":
      case "checkbox":
        return !1;
      case "slider":
        return [e.min ?? 0];
      case "number":
        return "";
      default:
        return "";
    }
  }
}
function y(e) {
  const n = [];
  for (const c of e)
    c.type === "row" ? n.push(...y(c.fields)) : n.push(c);
  return n;
}
function G(e, n) {
  const c = {};
  for (const t of y(e)) {
    const i = t.name;
    c[i] = n?.[i] !== void 0 ? n[i] : Y(t);
  }
  return c;
}
function h(e) {
  return typeof e == "object" && e !== null && "value" in e ? e.value : e;
}
function g(e, n) {
  return typeof e == "object" && e !== null && "message" in e ? e.message : n;
}
function M(e, n, c) {
  const t = e.validation;
  if (!t) return;
  if (t.required && (n == null || n === "" || Array.isArray(n) && n.length === 0))
    return typeof t.required == "string" ? t.required : `${e.label || e.name} est requis`;
  const i = typeof n == "string" ? n : "", a = typeof n == "number" ? n : Number(n);
  if (t.minLength !== void 0 && typeof n == "string") {
    const r = h(t.minLength);
    if (i.length < r)
      return g(t.minLength, `Minimum ${r} caracteres`);
  }
  if (t.maxLength !== void 0 && typeof n == "string") {
    const r = h(t.maxLength);
    if (i.length > r)
      return g(t.maxLength, `Maximum ${r} caracteres`);
  }
  if (t.min !== void 0) {
    const r = h(t.min), l = Array.isArray(n) ? n[0] : a;
    if (typeof l == "number" && l < r)
      return g(t.min, `Minimum ${r}`);
  }
  if (t.max !== void 0) {
    const r = h(t.max), l = Array.isArray(n) ? n[n.length - 1] : a;
    if (typeof l == "number" && l > r)
      return g(t.max, `Maximum ${r}`);
  }
  if (t.pattern !== void 0 && typeof n == "string" && !h(t.pattern).test(i))
    return g(t.pattern, "Format invalide");
  if (t.custom)
    return t.custom(n, c);
}
function Z({ field: e, value: n, error: c, onValueChange: t }) {
  const { name: i } = e;
  switch (e.type) {
    case "text":
    case "email":
    case "password":
    case "number":
      return /* @__PURE__ */ o(
        X,
        {
          type: e.type,
          label: e.label,
          placeholder: e.placeholder,
          disabled: e.disabled,
          icon: e.icon,
          iconPosition: e.iconPosition,
          prefix: e.prefix,
          suffix: e.suffix,
          value: n,
          onChange: (a) => t(i, a.target.value),
          error: c
        }
      );
    case "select": {
      const a = e.options.some(V);
      return e.native ? /* @__PURE__ */ o(
        $,
        {
          label: e.label,
          placeholder: e.placeholder,
          disabled: e.disabled,
          native: !0,
          value: n,
          onValueChange: (r) => t(i, r),
          children: e.options.map(
            (r, l) => V(r) ? /* @__PURE__ */ o("optgroup", { label: r.group, children: r.options.map((u) => /* @__PURE__ */ o("option", { value: u.value, disabled: u.disabled, children: u.label }, u.value)) }, l) : /* @__PURE__ */ o("option", { value: r.value, disabled: r.disabled, children: r.label }, r.value)
          )
        }
      ) : /* @__PURE__ */ o(
        $,
        {
          label: e.label,
          placeholder: e.placeholder,
          disabled: e.disabled,
          value: n,
          onValueChange: (r) => t(i, r),
          children: a ? e.options.map(
            (r, l) => V(r) ? /* @__PURE__ */ o(W, { label: r.group, children: r.options.map((u) => /* @__PURE__ */ o(R, { value: u.value, disabled: u.disabled, children: u.label }, u.value)) }, l) : /* @__PURE__ */ o(R, { value: r.value, disabled: r.disabled, children: r.label }, r.value)
          ) : e.options.map((r) => /* @__PURE__ */ o(R, { value: r.value, disabled: r.disabled, children: r.label }, r.value))
        }
      );
    }
    case "textarea":
      return /* @__PURE__ */ o(
        U,
        {
          label: e.label,
          placeholder: e.placeholder,
          disabled: e.disabled,
          rows: e.rows,
          icon: e.icon,
          iconPosition: e.iconPosition,
          value: n,
          onChange: (a) => t(i, a.target.value),
          error: c
        }
      );
    case "toggle":
      return /* @__PURE__ */ o(
        Q,
        {
          label: e.label,
          description: e.description,
          disabled: e.disabled,
          labelPosition: e.labelPosition,
          checked: n,
          onChange: (a) => t(i, a)
        }
      );
    case "checkbox":
      return /* @__PURE__ */ o(
        K,
        {
          label: e.label,
          description: e.description,
          disabled: e.disabled,
          labelPosition: e.labelPosition,
          checked: n,
          onChange: (a) => t(i, a)
        }
      );
    case "radio":
      return /* @__PURE__ */ o(
        H,
        {
          label: e.label,
          disabled: e.disabled,
          value: n,
          onValueChange: (a) => t(i, a),
          children: e.options.map((a) => /* @__PURE__ */ o(
            J,
            {
              value: a.value,
              label: a.label,
              description: a.description,
              disabled: a.disabled
            },
            a.value
          ))
        }
      );
    case "slider":
      return /* @__PURE__ */ o(
        z,
        {
          label: e.label,
          disabled: e.disabled,
          min: e.min,
          max: e.max,
          step: e.step,
          renderValue: e.renderValue,
          value: n,
          onValueChange: (a) => t(i, a)
        }
      );
    default:
      return null;
  }
}
const ae = T.forwardRef(function({
  fields: n,
  onSubmit: c,
  onChange: t,
  defaultValues: i,
  submitButton: a,
  resetButton: r,
  gap: l = "gap-4",
  className: u
}, q) {
  const [p, P] = N(
    () => G(n, i)
  ), [I, f] = N({}), v = _(!1), D = w(
    (s, d) => {
      P((b) => {
        const m = { ...b, [s]: d };
        if (v.current) {
          const j = y(n).find((x) => x.name === s);
          if (j) {
            const x = M(j, d, m);
            f((A) => {
              if (x) return { ...A, [s]: x };
              const { [s]: B, ...O } = A;
              return O;
            });
          }
        }
        return t?.(m, s), m;
      });
    },
    [n, t]
  ), E = w(
    (s) => {
      s.preventDefault(), v.current = !0;
      const d = y(n), b = {};
      for (const m of d) {
        const k = M(m, p[m.name], p);
        k && (b[m.name] = k);
      }
      f(b), Object.keys(b).length === 0 && c?.(p);
    },
    [n, p, c]
  ), F = w(() => {
    const s = G(n, i);
    P(s), f({}), v.current = !1;
  }, [n, i]);
  function S(s, d) {
    return s.type === "row" ? /* @__PURE__ */ o("div", { className: C("grid gap-4", `grid-cols-${s.fields.length}`), children: s.fields.map((b, m) => S(b, m)) }, d) : /* @__PURE__ */ o(
      Z,
      {
        field: s,
        value: p[s.name],
        error: I[s.name],
        onValueChange: D
      },
      s.name
    );
  }
  return /* @__PURE__ */ L(
    "form",
    {
      ref: q,
      onSubmit: E,
      onReset: (s) => {
        s.preventDefault(), F();
      },
      className: C("flex flex-col", l, u),
      noValidate: !0,
      children: [
        n.map((s, d) => S(s, d)),
        (a || r) && /* @__PURE__ */ L("div", { className: "flex items-center justify-end gap-2 mt-2", children: [
          r && /* @__PURE__ */ o("span", { onClick: F, className: "cursor-pointer", children: r }),
          a && /* @__PURE__ */ o("button", { type: "submit", className: "contents", children: a })
        ] })
      ]
    }
  );
});
export {
  ae as EasyForm
};
