// @btcv/ui/dist/HoverCard.js
"use client";
import { jsx as i } from "react/jsx-runtime";
import * as n from "react";
import { P as H, c as d, a as k, u as F } from "./_shared/index-CXeb1OMI.js";
import { u as I } from "./_shared/index-DLcqcWxM.js";
import { C as U, c as T, R as $, A as W, a as z } from "./_shared/index-uu9PT5Nu.js";
import { P as B, D as G } from "./_shared/index-B03TCNO5.js";
import { P as K } from "./_shared/index-V-Ajw7Ac.js";
import { c as j } from "./_shared/utils-eGXXUFl7.js";
var x, S = "HoverCard", [O] = k(S, [
  T
]), b = T(), [V, g] = O(S), y = (e) => {
  const {
    __scopeHoverCard: o,
    children: t,
    open: s,
    defaultOpen: a,
    onOpenChange: c,
    openDelay: f = 700,
    closeDelay: v = 300
  } = e, l = b(o), C = n.useRef(0), u = n.useRef(0), h = n.useRef(!1), p = n.useRef(!1), [m, r] = F({
    prop: s,
    defaultProp: a ?? !1,
    onChange: c,
    caller: S
  }), P = n.useCallback(() => {
    clearTimeout(u.current), C.current = window.setTimeout(() => r(!0), f);
  }, [f, r]), L = n.useCallback(() => {
    clearTimeout(C.current), !h.current && !p.current && (u.current = window.setTimeout(() => r(!1), v));
  }, [v, r]), M = n.useCallback(() => r(!1), [r]);
  return n.useEffect(() => () => {
    clearTimeout(C.current), clearTimeout(u.current);
  }, []), /* @__PURE__ */ i(
    V,
    {
      scope: o,
      open: m,
      onOpenChange: r,
      onOpen: P,
      onClose: L,
      onDismiss: M,
      hasSelectionRef: h,
      isPointerDownOnContentRef: p,
      children: /* @__PURE__ */ i($, { ...l, children: t })
    }
  );
};
y.displayName = S;
var N = "HoverCardTrigger", _ = n.forwardRef(
  (e, o) => {
    const { __scopeHoverCard: t, ...s } = e, a = g(N, t), c = b(t);
    return /* @__PURE__ */ i(W, { asChild: !0, ...c, children: /* @__PURE__ */ i(
      K.a,
      {
        "data-state": a.open ? "open" : "closed",
        ...s,
        ref: o,
        onPointerEnter: d(e.onPointerEnter, w(a.onOpen)),
        onPointerLeave: d(e.onPointerLeave, w(a.onClose)),
        onFocus: d(e.onFocus, a.onOpen),
        onBlur: d(e.onBlur, a.onClose),
        onTouchStart: d(e.onTouchStart, (f) => f.preventDefault())
      }
    ) });
  }
);
_.displayName = N;
var E = "HoverCardPortal", [q, J] = O(E, {
  forceMount: void 0
}), D = (e) => {
  const { __scopeHoverCard: o, forceMount: t, children: s, container: a } = e, c = g(E, o);
  return /* @__PURE__ */ i(q, { scope: o, forceMount: t, children: /* @__PURE__ */ i(H, { present: t || c.open, children: /* @__PURE__ */ i(B, { asChild: !0, container: a, children: s }) }) });
};
D.displayName = E;
var R = "HoverCardContent", A = n.forwardRef(
  (e, o) => {
    const t = J(R, e.__scopeHoverCard), { forceMount: s = t.forceMount, ...a } = e, c = g(R, e.__scopeHoverCard);
    return /* @__PURE__ */ i(H, { present: s || c.open, children: /* @__PURE__ */ i(
      Q,
      {
        "data-state": c.open ? "open" : "closed",
        ...a,
        onPointerEnter: d(e.onPointerEnter, w(c.onOpen)),
        onPointerLeave: d(e.onPointerLeave, w(c.onClose)),
        ref: o
      }
    ) });
  }
);
A.displayName = R;
var Q = n.forwardRef((e, o) => {
  const {
    __scopeHoverCard: t,
    onEscapeKeyDown: s,
    onPointerDownOutside: a,
    onFocusOutside: c,
    onInteractOutside: f,
    ...v
  } = e, l = g(R, t), C = b(t), u = n.useRef(null), h = I(o, u), [p, m] = n.useState(!1);
  return n.useEffect(() => {
    if (p) {
      const r = document.body;
      return x = r.style.userSelect || r.style.webkitUserSelect, r.style.userSelect = "none", r.style.webkitUserSelect = "none", () => {
        r.style.userSelect = x, r.style.webkitUserSelect = x;
      };
    }
  }, [p]), n.useEffect(() => {
    if (u.current) {
      const r = () => {
        m(!1), l.isPointerDownOnContentRef.current = !1, setTimeout(() => {
          document.getSelection()?.toString() !== "" && (l.hasSelectionRef.current = !0);
        });
      };
      return document.addEventListener("pointerup", r), () => {
        document.removeEventListener("pointerup", r), l.hasSelectionRef.current = !1, l.isPointerDownOnContentRef.current = !1;
      };
    }
  }, [l.isPointerDownOnContentRef, l.hasSelectionRef]), n.useEffect(() => {
    u.current && Z(u.current).forEach((P) => P.setAttribute("tabindex", "-1"));
  }), /* @__PURE__ */ i(
    G,
    {
      asChild: !0,
      disableOutsidePointerEvents: !1,
      onInteractOutside: f,
      onEscapeKeyDown: s,
      onPointerDownOutside: a,
      onFocusOutside: d(c, (r) => {
        r.preventDefault();
      }),
      onDismiss: l.onDismiss,
      children: /* @__PURE__ */ i(
        U,
        {
          ...C,
          ...v,
          onPointerDown: d(v.onPointerDown, (r) => {
            r.currentTarget.contains(r.target) && m(!0), l.hasSelectionRef.current = !1, l.isPointerDownOnContentRef.current = !0;
          }),
          ref: h,
          style: {
            ...v.style,
            userSelect: p ? "text" : void 0,
            // Safari requires prefix
            WebkitUserSelect: p ? "text" : void 0,
            "--radix-hover-card-content-transform-origin": "var(--radix-popper-transform-origin)",
            "--radix-hover-card-content-available-width": "var(--radix-popper-available-width)",
            "--radix-hover-card-content-available-height": "var(--radix-popper-available-height)",
            "--radix-hover-card-trigger-width": "var(--radix-popper-anchor-width)",
            "--radix-hover-card-trigger-height": "var(--radix-popper-anchor-height)"
          }
        }
      )
    }
  );
}), X = "HoverCardArrow", Y = n.forwardRef(
  (e, o) => {
    const { __scopeHoverCard: t, ...s } = e, a = b(t);
    return /* @__PURE__ */ i(z, { ...a, ...s, ref: o });
  }
);
Y.displayName = X;
function w(e) {
  return (o) => o.pointerType === "touch" ? void 0 : e();
}
function Z(e) {
  const o = [], t = document.createTreeWalker(e, NodeFilter.SHOW_ELEMENT, {
    acceptNode: (s) => s.tabIndex >= 0 ? NodeFilter.FILTER_ACCEPT : NodeFilter.FILTER_SKIP
  });
  for (; t.nextNode(); ) o.push(t.currentNode);
  return o;
}
var ee = y, re = _, oe = D, te = A;
const pe = ee, fe = re, ne = n.forwardRef(({ className: e, align: o = "center", sideOffset: t = 4, container: s, ...a }, c) => /* @__PURE__ */ i(oe, { container: s, children: /* @__PURE__ */ i(
  te,
  {
    ref: c,
    align: o,
    sideOffset: t,
    className: j(
      "z-50 bg-glass border border-border rounded-3xl shadow-lg p-4 w-64",
      "data-[state=open]:animate-in data-[state=open]:fade-in-0 data-[state=open]:zoom-in-95",
      "data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=closed]:zoom-out-95",
      "outline-none",
      e
    ),
    ...a
  }
) }));
ne.displayName = "HoverCardContent";
export {
  pe as HoverCard,
  ne as HoverCardContent,
  fe as HoverCardTrigger
};
