// @btcv/ui/dist/Input.js
"use client";
import { jsx as i, jsxs as R, Fragment as ve } from "react/jsx-runtime";
import * as l from "react";
import * as qe from "react-dom";
import { c as ho } from "./_shared/index-BrLJJgkl.js";
import { u as se, c as N, a as ie, b as Z, d as go, P as Ye } from "./_shared/index-CXeb1OMI.js";
import { u as k, c as vo } from "./_shared/index-DLcqcWxM.js";
import { u as be, R as bo, A as xo, c as Xe, C as So, a as wo } from "./_shared/index-uu9PT5Nu.js";
import { P as E } from "./_shared/index-V-Ajw7Ac.js";
import { u as ze, a as Ze, R as yo, I as Co, c as Je } from "./_shared/index-C0gNQvxa.js";
import { P as Ro, D as No } from "./_shared/index-B03TCNO5.js";
import { h as Po, u as Eo, R as Io, F as _o } from "./_shared/Combination-D_l4PLF_.js";
import { u as Fe } from "./_shared/index-BC7vfx-u.js";
import { V as ko } from "./_shared/index-B1f-hyuh.js";
import { Check as Qe, X as je, ChevronDown as Me, ChevronUp as To, EyeOff as Mo, Eye as Do } from "lucide-react";
import { c as P } from "./_shared/utils-eGXXUFl7.js";
var Ao = [
  "a",
  "button",
  "div",
  "form",
  "h2",
  "h3",
  "img",
  "input",
  "label",
  "li",
  "nav",
  "ol",
  "p",
  "select",
  "span",
  "svg",
  "ul"
], Oo = Ao.reduce((e, o) => {
  const t = ho(`Primitive.${o}`), r = l.forwardRef((n, a) => {
    const { asChild: s, ...c } = n, d = s ? t : o;
    return typeof window < "u" && (window[/* @__PURE__ */ Symbol.for("radix-ui")] = !0), /* @__PURE__ */ i(d, { ...c, ref: a });
  });
  return r.displayName = `Primitive.${o}`, { ...e, [o]: r };
}, {}), Bo = "Label", et = l.forwardRef((e, o) => /* @__PURE__ */ i(
  Oo.label,
  {
    ...e,
    ref: o,
    onMouseDown: (t) => {
      t.target.closest("button, input, select, textarea") || (e.onMouseDown?.(t), !t.defaultPrevented && t.detail > 1 && t.preventDefault());
    }
  }
));
et.displayName = Bo;
var Lo = et;
function fe(e) {
  const o = l.useRef({ value: e, previous: e });
  return l.useMemo(() => (o.current.value !== e && (o.current.previous = o.current.value, o.current.value = e), o.current.previous), [e]);
}
var xe = "Switch", [Vo] = ie(xe), [Ho, Ko] = Vo(xe), tt = l.forwardRef(
  (e, o) => {
    const {
      __scopeSwitch: t,
      name: r,
      checked: n,
      defaultChecked: a,
      required: s,
      disabled: c,
      value: d = "on",
      onCheckedChange: u,
      form: m,
      ...f
    } = e, [h, v] = l.useState(null), b = k(o, (w) => v(w)), p = l.useRef(!1), g = h ? m || !!h.closest("form") : !0, [x, S] = se({
      prop: n,
      defaultProp: a ?? !1,
      onChange: u,
      caller: xe
    });
    return /* @__PURE__ */ R(Ho, { scope: t, checked: x, disabled: c, children: [
      /* @__PURE__ */ i(
        E.button,
        {
          type: "button",
          role: "switch",
          "aria-checked": x,
          "aria-required": s,
          "data-state": st(x),
          "data-disabled": c ? "" : void 0,
          disabled: c,
          value: d,
          ...f,
          ref: b,
          onClick: N(e.onClick, (w) => {
            S((C) => !C), g && (p.current = w.isPropagationStopped(), p.current || w.stopPropagation());
          })
        }
      ),
      g && /* @__PURE__ */ i(
        rt,
        {
          control: h,
          bubbles: !p.current,
          name: r,
          value: d,
          checked: x,
          required: s,
          disabled: c,
          form: m,
          style: { transform: "translateX(-100%)" }
        }
      )
    ] });
  }
);
tt.displayName = xe;
var ot = "SwitchThumb", nt = l.forwardRef(
  (e, o) => {
    const { __scopeSwitch: t, ...r } = e, n = Ko(ot, t);
    return /* @__PURE__ */ i(
      E.span,
      {
        "data-state": st(n.checked),
        "data-disabled": n.disabled ? "" : void 0,
        ...r,
        ref: o
      }
    );
  }
);
nt.displayName = ot;
var zo = "SwitchBubbleInput", rt = l.forwardRef(
  ({
    __scopeSwitch: e,
    control: o,
    checked: t,
    bubbles: r = !0,
    ...n
  }, a) => {
    const s = l.useRef(null), c = k(s, a), d = fe(t), u = be(o);
    return l.useEffect(() => {
      const m = s.current;
      if (!m) return;
      const f = window.HTMLInputElement.prototype, v = Object.getOwnPropertyDescriptor(
        f,
        "checked"
      ).set;
      if (d !== t && v) {
        const b = new Event("click", { bubbles: r });
        v.call(m, t), m.dispatchEvent(b);
      }
    }, [d, t, r]), /* @__PURE__ */ i(
      "input",
      {
        type: "checkbox",
        "aria-hidden": !0,
        defaultChecked: t,
        ...n,
        tabIndex: -1,
        ref: c,
        style: {
          ...n.style,
          ...u,
          position: "absolute",
          pointerEvents: "none",
          opacity: 0,
          margin: 0
        }
      }
    );
  }
);
rt.displayName = zo;
function st(e) {
  return e ? "checked" : "unchecked";
}
var Fo = tt, $o = nt;
function he(e, [o, t]) {
  return Math.min(t, Math.max(o, e));
}
// @__NO_SIDE_EFFECTS__
function Uo(e) {
  const o = /* @__PURE__ */ Go(e), t = l.forwardRef((r, n) => {
    const { children: a, ...s } = r, c = l.Children.toArray(a), d = c.find(jo);
    if (d) {
      const u = d.props.children, m = c.map((f) => f === d ? l.Children.count(u) > 1 ? l.Children.only(null) : l.isValidElement(u) ? u.props.children : null : f);
      return /* @__PURE__ */ i(o, { ...s, ref: n, children: l.isValidElement(u) ? l.cloneElement(u, void 0, m) : null });
    }
    return /* @__PURE__ */ i(o, { ...s, ref: n, children: a });
  });
  return t.displayName = `${e}.Slot`, t;
}
// @__NO_SIDE_EFFECTS__
function Go(e) {
  const o = l.forwardRef((t, r) => {
    const { children: n, ...a } = t;
    if (l.isValidElement(n)) {
      const s = Yo(n), c = qo(a, n.props);
      return n.type !== l.Fragment && (c.ref = r ? vo(r, s) : s), l.cloneElement(n, c);
    }
    return l.Children.count(n) > 1 ? l.Children.only(null) : null;
  });
  return o.displayName = `${e}.SlotClone`, o;
}
var Wo = /* @__PURE__ */ Symbol("radix.slottable");
function jo(e) {
  return l.isValidElement(e) && typeof e.type == "function" && "__radixId" in e.type && e.type.__radixId === Wo;
}
function qo(e, o) {
  const t = { ...o };
  for (const r in o) {
    const n = e[r], a = o[r];
    /^on[A-Z]/.test(r) ? n && a ? t[r] = (...c) => {
      const d = a(...c);
      return n(...c), d;
    } : n && (t[r] = n) : r === "style" ? t[r] = { ...n, ...a } : r === "className" && (t[r] = [n, a].filter(Boolean).join(" "));
  }
  return { ...e, ...t };
}
function Yo(e) {
  let o = Object.getOwnPropertyDescriptor(e.props, "ref")?.get, t = o && "isReactWarning" in o && o.isReactWarning;
  return t ? e.ref : (o = Object.getOwnPropertyDescriptor(e, "ref")?.get, t = o && "isReactWarning" in o && o.isReactWarning, t ? e.props.ref : e.props.ref || e.ref);
}
var Xo = [" ", "Enter", "ArrowUp", "ArrowDown"], Zo = [" ", "Enter"], oe = "Select", [Se, we, Jo] = Ze(oe), [ce] = ie(oe, [
  Jo,
  Xe
]), ye = Xe(), [Qo, J] = ce(oe), [en, tn] = ce(oe), at = (e) => {
  const {
    __scopeSelect: o,
    children: t,
    open: r,
    defaultOpen: n,
    onOpenChange: a,
    value: s,
    defaultValue: c,
    onValueChange: d,
    dir: u,
    name: m,
    autoComplete: f,
    disabled: h,
    required: v,
    form: b
  } = e, p = ye(o), [g, x] = l.useState(null), [S, w] = l.useState(null), [C, M] = l.useState(!1), W = ze(u), [D, $] = se({
    prop: r,
    defaultProp: n ?? !1,
    onChange: a,
    caller: oe
  }), [Y, K] = se({
    prop: s,
    defaultProp: c,
    onChange: d,
    caller: oe
  }), I = l.useRef(null), L = g ? b || !!g.closest("form") : !0, [U, G] = l.useState(/* @__PURE__ */ new Set()), z = Array.from(U).map((A) => A.props.value).join(";");
  return /* @__PURE__ */ i(bo, { ...p, children: /* @__PURE__ */ R(
    Qo,
    {
      required: v,
      scope: o,
      trigger: g,
      onTriggerChange: x,
      valueNode: S,
      onValueNodeChange: w,
      valueNodeHasChildren: C,
      onValueNodeHasChildrenChange: M,
      contentId: Fe(),
      value: Y,
      onValueChange: K,
      open: D,
      onOpenChange: $,
      dir: W,
      triggerPointerDownPosRef: I,
      disabled: h,
      children: [
        /* @__PURE__ */ i(Se.Provider, { scope: o, children: /* @__PURE__ */ i(
          en,
          {
            scope: e.__scopeSelect,
            onNativeOptionAdd: l.useCallback((A) => {
              G((V) => new Set(V).add(A));
            }, []),
            onNativeOptionRemove: l.useCallback((A) => {
              G((V) => {
                const H = new Set(V);
                return H.delete(A), H;
              });
            }, []),
            children: t
          }
        ) }),
        L ? /* @__PURE__ */ R(
          kt,
          {
            "aria-hidden": !0,
            required: v,
            tabIndex: -1,
            name: m,
            autoComplete: f,
            value: Y,
            onChange: (A) => K(A.target.value),
            disabled: h,
            form: b,
            children: [
              Y === void 0 ? /* @__PURE__ */ i("option", { value: "" }) : null,
              Array.from(U)
            ]
          },
          z
        ) : null
      ]
    }
  ) });
};
at.displayName = oe;
var it = "SelectTrigger", ct = l.forwardRef(
  (e, o) => {
    const { __scopeSelect: t, disabled: r = !1, ...n } = e, a = ye(t), s = J(it, t), c = s.disabled || r, d = k(o, s.onTriggerChange), u = we(t), m = l.useRef("touch"), [f, h, v] = Mt((p) => {
      const g = u().filter((w) => !w.disabled), x = g.find((w) => w.value === s.value), S = Dt(g, p, x);
      S !== void 0 && s.onValueChange(S.value);
    }), b = (p) => {
      c || (s.onOpenChange(!0), v()), p && (s.triggerPointerDownPosRef.current = {
        x: Math.round(p.pageX),
        y: Math.round(p.pageY)
      });
    };
    return /* @__PURE__ */ i(xo, { asChild: !0, ...a, children: /* @__PURE__ */ i(
      E.button,
      {
        type: "button",
        role: "combobox",
        "aria-controls": s.contentId,
        "aria-expanded": s.open,
        "aria-required": s.required,
        "aria-autocomplete": "none",
        dir: s.dir,
        "data-state": s.open ? "open" : "closed",
        disabled: c,
        "data-disabled": c ? "" : void 0,
        "data-placeholder": Tt(s.value) ? "" : void 0,
        ...n,
        ref: d,
        onClick: N(n.onClick, (p) => {
          p.currentTarget.focus(), m.current !== "mouse" && b(p);
        }),
        onPointerDown: N(n.onPointerDown, (p) => {
          m.current = p.pointerType;
          const g = p.target;
          g.hasPointerCapture(p.pointerId) && g.releasePointerCapture(p.pointerId), p.button === 0 && p.ctrlKey === !1 && p.pointerType === "mouse" && (b(p), p.preventDefault());
        }),
        onKeyDown: N(n.onKeyDown, (p) => {
          const g = f.current !== "";
          !(p.ctrlKey || p.altKey || p.metaKey) && p.key.length === 1 && h(p.key), !(g && p.key === " ") && Xo.includes(p.key) && (b(), p.preventDefault());
        })
      }
    ) });
  }
);
ct.displayName = it;
var lt = "SelectValue", dt = l.forwardRef(
  (e, o) => {
    const { __scopeSelect: t, className: r, style: n, children: a, placeholder: s = "", ...c } = e, d = J(lt, t), { onValueNodeHasChildrenChange: u } = d, m = a !== void 0, f = k(o, d.onValueNodeChange);
    return Z(() => {
      u(m);
    }, [u, m]), /* @__PURE__ */ i(
      E.span,
      {
        ...c,
        ref: f,
        style: { pointerEvents: "none" },
        children: Tt(d.value) ? /* @__PURE__ */ i(ve, { children: s }) : a
      }
    );
  }
);
dt.displayName = lt;
var on = "SelectIcon", ut = l.forwardRef(
  (e, o) => {
    const { __scopeSelect: t, children: r, ...n } = e;
    return /* @__PURE__ */ i(E.span, { "aria-hidden": !0, ...n, ref: o, children: r || "▼" });
  }
);
ut.displayName = on;
var nn = "SelectPortal", ft = (e) => /* @__PURE__ */ i(Ro, { asChild: !0, ...e });
ft.displayName = nn;
var ne = "SelectContent", pt = l.forwardRef(
  (e, o) => {
    const t = J(ne, e.__scopeSelect), [r, n] = l.useState();
    if (Z(() => {
      n(new DocumentFragment());
    }, []), !t.open) {
      const a = r;
      return a ? qe.createPortal(
        /* @__PURE__ */ i(mt, { scope: e.__scopeSelect, children: /* @__PURE__ */ i(Se.Slot, { scope: e.__scopeSelect, children: /* @__PURE__ */ i("div", { children: e.children }) }) }),
        a
      ) : null;
    }
    return /* @__PURE__ */ i(ht, { ...e, ref: o });
  }
);
pt.displayName = ne;
var q = 10, [mt, Q] = ce(ne), rn = "SelectContentImpl", sn = /* @__PURE__ */ Uo("SelectContent.RemoveScroll"), ht = l.forwardRef(
  (e, o) => {
    const {
      __scopeSelect: t,
      position: r = "item-aligned",
      onCloseAutoFocus: n,
      onEscapeKeyDown: a,
      onPointerDownOutside: s,
      //
      // PopperContent props
      side: c,
      sideOffset: d,
      align: u,
      alignOffset: m,
      arrowPadding: f,
      collisionBoundary: h,
      collisionPadding: v,
      sticky: b,
      hideWhenDetached: p,
      avoidCollisions: g,
      //
      ...x
    } = e, S = J(ne, t), [w, C] = l.useState(null), [M, W] = l.useState(null), D = k(o, (y) => C(y)), [$, Y] = l.useState(null), [K, I] = l.useState(
      null
    ), L = we(t), [U, G] = l.useState(!1), z = l.useRef(!1);
    l.useEffect(() => {
      if (w) return Po(w);
    }, [w]), Eo();
    const A = l.useCallback(
      (y) => {
        const [O, ...F] = L().map((B) => B.ref.current), [_] = F.slice(-1), T = document.activeElement;
        for (const B of y)
          if (B === T || (B?.scrollIntoView({ block: "nearest" }), B === O && M && (M.scrollTop = 0), B === _ && M && (M.scrollTop = M.scrollHeight), B?.focus(), document.activeElement !== T)) return;
      },
      [L, M]
    ), V = l.useCallback(
      () => A([$, w]),
      [A, $, w]
    );
    l.useEffect(() => {
      U && V();
    }, [U, V]);
    const { onOpenChange: H, triggerPointerDownPosRef: j } = S;
    l.useEffect(() => {
      if (w) {
        let y = { x: 0, y: 0 };
        const O = (_) => {
          y = {
            x: Math.abs(Math.round(_.pageX) - (j.current?.x ?? 0)),
            y: Math.abs(Math.round(_.pageY) - (j.current?.y ?? 0))
          };
        }, F = (_) => {
          y.x <= 10 && y.y <= 10 ? _.preventDefault() : w.contains(_.target) || H(!1), document.removeEventListener("pointermove", O), j.current = null;
        };
        return j.current !== null && (document.addEventListener("pointermove", O), document.addEventListener("pointerup", F, { capture: !0, once: !0 })), () => {
          document.removeEventListener("pointermove", O), document.removeEventListener("pointerup", F, { capture: !0 });
        };
      }
    }, [w, H, j]), l.useEffect(() => {
      const y = () => H(!1);
      return window.addEventListener("blur", y), window.addEventListener("resize", y), () => {
        window.removeEventListener("blur", y), window.removeEventListener("resize", y);
      };
    }, [H]);
    const [Pe, pe] = Mt((y) => {
      const O = L().filter((T) => !T.disabled), F = O.find((T) => T.ref.current === document.activeElement), _ = Dt(O, y, F);
      _ && setTimeout(() => _.ref.current.focus());
    }), Ee = l.useCallback(
      (y, O, F) => {
        const _ = !z.current && !F;
        (S.value !== void 0 && S.value === O || _) && (Y(y), _ && (z.current = !0));
      },
      [S.value]
    ), Ie = l.useCallback(() => w?.focus(), [w]), re = l.useCallback(
      (y, O, F) => {
        const _ = !z.current && !F;
        (S.value !== void 0 && S.value === O || _) && I(y);
      },
      [S.value]
    ), me = r === "popper" ? De : gt, de = me === De ? {
      side: c,
      sideOffset: d,
      align: u,
      alignOffset: m,
      arrowPadding: f,
      collisionBoundary: h,
      collisionPadding: v,
      sticky: b,
      hideWhenDetached: p,
      avoidCollisions: g
    } : {};
    return /* @__PURE__ */ i(
      mt,
      {
        scope: t,
        content: w,
        viewport: M,
        onViewportChange: W,
        itemRefCallback: Ee,
        selectedItem: $,
        onItemLeave: Ie,
        itemTextRefCallback: re,
        focusSelectedItem: V,
        selectedItemText: K,
        position: r,
        isPositioned: U,
        searchRef: Pe,
        children: /* @__PURE__ */ i(Io, { as: sn, allowPinchZoom: !0, children: /* @__PURE__ */ i(
          _o,
          {
            asChild: !0,
            trapped: S.open,
            onMountAutoFocus: (y) => {
              y.preventDefault();
            },
            onUnmountAutoFocus: N(n, (y) => {
              S.trigger?.focus({ preventScroll: !0 }), y.preventDefault();
            }),
            children: /* @__PURE__ */ i(
              No,
              {
                asChild: !0,
                disableOutsidePointerEvents: !0,
                onEscapeKeyDown: a,
                onPointerDownOutside: s,
                onFocusOutside: (y) => y.preventDefault(),
                onDismiss: () => S.onOpenChange(!1),
                children: /* @__PURE__ */ i(
                  me,
                  {
                    role: "listbox",
                    id: S.contentId,
                    "data-state": S.open ? "open" : "closed",
                    dir: S.dir,
                    onContextMenu: (y) => y.preventDefault(),
                    ...x,
                    ...de,
                    onPlaced: () => G(!0),
                    ref: D,
                    style: {
                      // flex layout so we can place the scroll buttons properly
                      display: "flex",
                      flexDirection: "column",
                      // reset the outline by default as the content MAY get focused
                      outline: "none",
                      ...x.style
                    },
                    onKeyDown: N(x.onKeyDown, (y) => {
                      const O = y.ctrlKey || y.altKey || y.metaKey;
                      if (y.key === "Tab" && y.preventDefault(), !O && y.key.length === 1 && pe(y.key), ["ArrowUp", "ArrowDown", "Home", "End"].includes(y.key)) {
                        let _ = L().filter((T) => !T.disabled).map((T) => T.ref.current);
                        if (["ArrowUp", "End"].includes(y.key) && (_ = _.slice().reverse()), ["ArrowUp", "ArrowDown"].includes(y.key)) {
                          const T = y.target, B = _.indexOf(T);
                          _ = _.slice(B + 1);
                        }
                        setTimeout(() => A(_)), y.preventDefault();
                      }
                    })
                  }
                )
              }
            )
          }
        ) })
      }
    );
  }
);
ht.displayName = rn;
var an = "SelectItemAlignedPosition", gt = l.forwardRef((e, o) => {
  const { __scopeSelect: t, onPlaced: r, ...n } = e, a = J(ne, t), s = Q(ne, t), [c, d] = l.useState(null), [u, m] = l.useState(null), f = k(o, (D) => m(D)), h = we(t), v = l.useRef(!1), b = l.useRef(!0), { viewport: p, selectedItem: g, selectedItemText: x, focusSelectedItem: S } = s, w = l.useCallback(() => {
    if (a.trigger && a.valueNode && c && u && p && g && x) {
      const D = a.trigger.getBoundingClientRect(), $ = u.getBoundingClientRect(), Y = a.valueNode.getBoundingClientRect(), K = x.getBoundingClientRect();
      if (a.dir !== "rtl") {
        const T = K.left - $.left, B = Y.left - T, ee = D.left - B, te = D.width + ee, _e = Math.max(te, $.width), ke = window.innerWidth - q, Te = he(B, [
          q,
          // Prevents the content from going off the starting edge of the
          // viewport. It may still go off the ending edge, but this can be
          // controlled by the user since they may want to manage overflow in a
          // specific way.
          // https://github.com/radix-ui/primitives/issues/2049
          Math.max(q, ke - _e)
        ]);
        c.style.minWidth = te + "px", c.style.left = Te + "px";
      } else {
        const T = $.right - K.right, B = window.innerWidth - Y.right - T, ee = window.innerWidth - D.right - B, te = D.width + ee, _e = Math.max(te, $.width), ke = window.innerWidth - q, Te = he(B, [
          q,
          Math.max(q, ke - _e)
        ]);
        c.style.minWidth = te + "px", c.style.right = Te + "px";
      }
      const I = h(), L = window.innerHeight - q * 2, U = p.scrollHeight, G = window.getComputedStyle(u), z = parseInt(G.borderTopWidth, 10), A = parseInt(G.paddingTop, 10), V = parseInt(G.borderBottomWidth, 10), H = parseInt(G.paddingBottom, 10), j = z + A + U + H + V, Pe = Math.min(g.offsetHeight * 5, j), pe = window.getComputedStyle(p), Ee = parseInt(pe.paddingTop, 10), Ie = parseInt(pe.paddingBottom, 10), re = D.top + D.height / 2 - q, me = L - re, de = g.offsetHeight / 2, y = g.offsetTop + de, O = z + A + y, F = j - O;
      if (O <= re) {
        const T = I.length > 0 && g === I[I.length - 1].ref.current;
        c.style.bottom = "0px";
        const B = u.clientHeight - p.offsetTop - p.offsetHeight, ee = Math.max(
          me,
          de + // viewport might have padding bottom, include it to avoid a scrollable viewport
          (T ? Ie : 0) + B + V
        ), te = O + ee;
        c.style.height = te + "px";
      } else {
        const T = I.length > 0 && g === I[0].ref.current;
        c.style.top = "0px";
        const ee = Math.max(
          re,
          z + p.offsetTop + // viewport might have padding top, include it to avoid a scrollable viewport
          (T ? Ee : 0) + de
        ) + F;
        c.style.height = ee + "px", p.scrollTop = O - re + p.offsetTop;
      }
      c.style.margin = `${q}px 0`, c.style.minHeight = Pe + "px", c.style.maxHeight = L + "px", r?.(), requestAnimationFrame(() => v.current = !0);
    }
  }, [
    h,
    a.trigger,
    a.valueNode,
    c,
    u,
    p,
    g,
    x,
    a.dir,
    r
  ]);
  Z(() => w(), [w]);
  const [C, M] = l.useState();
  Z(() => {
    u && M(window.getComputedStyle(u).zIndex);
  }, [u]);
  const W = l.useCallback(
    (D) => {
      D && b.current === !0 && (w(), S?.(), b.current = !1);
    },
    [w, S]
  );
  return /* @__PURE__ */ i(
    ln,
    {
      scope: t,
      contentWrapper: c,
      shouldExpandOnScrollRef: v,
      onScrollButtonChange: W,
      children: /* @__PURE__ */ i(
        "div",
        {
          ref: d,
          style: {
            display: "flex",
            flexDirection: "column",
            position: "fixed",
            zIndex: C
          },
          children: /* @__PURE__ */ i(
            E.div,
            {
              ...n,
              ref: f,
              style: {
                // When we get the height of the content, it includes borders. If we were to set
                // the height without having `boxSizing: 'border-box'` it would be too big.
                boxSizing: "border-box",
                // We need to ensure the content doesn't get taller than the wrapper
                maxHeight: "100%",
                ...n.style
              }
            }
          )
        }
      )
    }
  );
});
gt.displayName = an;
var cn = "SelectPopperPosition", De = l.forwardRef((e, o) => {
  const {
    __scopeSelect: t,
    align: r = "start",
    collisionPadding: n = q,
    ...a
  } = e, s = ye(t);
  return /* @__PURE__ */ i(
    So,
    {
      ...s,
      ...a,
      ref: o,
      align: r,
      collisionPadding: n,
      style: {
        // Ensure border-box for floating-ui calculations
        boxSizing: "border-box",
        ...a.style,
        "--radix-select-content-transform-origin": "var(--radix-popper-transform-origin)",
        "--radix-select-content-available-width": "var(--radix-popper-available-width)",
        "--radix-select-content-available-height": "var(--radix-popper-available-height)",
        "--radix-select-trigger-width": "var(--radix-popper-anchor-width)",
        "--radix-select-trigger-height": "var(--radix-popper-anchor-height)"
      }
    }
  );
});
De.displayName = cn;
var [ln, $e] = ce(ne, {}), Ae = "SelectViewport", vt = l.forwardRef(
  (e, o) => {
    const { __scopeSelect: t, nonce: r, ...n } = e, a = Q(Ae, t), s = $e(Ae, t), c = k(o, a.onViewportChange), d = l.useRef(0);
    return /* @__PURE__ */ R(ve, { children: [
      /* @__PURE__ */ i(
        "style",
        {
          dangerouslySetInnerHTML: {
            __html: "[data-radix-select-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-select-viewport]::-webkit-scrollbar{display:none}"
          },
          nonce: r
        }
      ),
      /* @__PURE__ */ i(Se.Slot, { scope: t, children: /* @__PURE__ */ i(
        E.div,
        {
          "data-radix-select-viewport": "",
          role: "presentation",
          ...n,
          ref: c,
          style: {
            // we use position: 'relative' here on the `viewport` so that when we call
            // `selectedItem.offsetTop` in calculations, the offset is relative to the viewport
            // (independent of the scrollUpButton).
            position: "relative",
            flex: 1,
            // Viewport should only be scrollable in the vertical direction.
            // This won't work in vertical writing modes, so we'll need to
            // revisit this if/when that is supported
            // https://developer.chrome.com/blog/vertical-form-controls
            overflow: "hidden auto",
            ...n.style
          },
          onScroll: N(n.onScroll, (u) => {
            const m = u.currentTarget, { contentWrapper: f, shouldExpandOnScrollRef: h } = s;
            if (h?.current && f) {
              const v = Math.abs(d.current - m.scrollTop);
              if (v > 0) {
                const b = window.innerHeight - q * 2, p = parseFloat(f.style.minHeight), g = parseFloat(f.style.height), x = Math.max(p, g);
                if (x < b) {
                  const S = x + v, w = Math.min(b, S), C = S - w;
                  f.style.height = w + "px", f.style.bottom === "0px" && (m.scrollTop = C > 0 ? C : 0, f.style.justifyContent = "flex-end");
                }
              }
            }
            d.current = m.scrollTop;
          })
        }
      ) })
    ] });
  }
);
vt.displayName = Ae;
var bt = "SelectGroup", [dn, un] = ce(bt), xt = l.forwardRef(
  (e, o) => {
    const { __scopeSelect: t, ...r } = e, n = Fe();
    return /* @__PURE__ */ i(dn, { scope: t, id: n, children: /* @__PURE__ */ i(E.div, { role: "group", "aria-labelledby": n, ...r, ref: o }) });
  }
);
xt.displayName = bt;
var St = "SelectLabel", wt = l.forwardRef(
  (e, o) => {
    const { __scopeSelect: t, ...r } = e, n = un(St, t);
    return /* @__PURE__ */ i(E.div, { id: n.id, ...r, ref: o });
  }
);
wt.displayName = St;
var ge = "SelectItem", [fn, yt] = ce(ge), Ct = l.forwardRef(
  (e, o) => {
    const {
      __scopeSelect: t,
      value: r,
      disabled: n = !1,
      textValue: a,
      ...s
    } = e, c = J(ge, t), d = Q(ge, t), u = c.value === r, [m, f] = l.useState(a ?? ""), [h, v] = l.useState(!1), b = k(
      o,
      (S) => d.itemRefCallback?.(S, r, n)
    ), p = Fe(), g = l.useRef("touch"), x = () => {
      n || (c.onValueChange(r), c.onOpenChange(!1));
    };
    if (r === "")
      throw new Error(
        "A <Select.Item /> must have a value prop that is not an empty string. This is because the Select value can be set to an empty string to clear the selection and show the placeholder."
      );
    return /* @__PURE__ */ i(
      fn,
      {
        scope: t,
        value: r,
        disabled: n,
        textId: p,
        isSelected: u,
        onItemTextChange: l.useCallback((S) => {
          f((w) => w || (S?.textContent ?? "").trim());
        }, []),
        children: /* @__PURE__ */ i(
          Se.ItemSlot,
          {
            scope: t,
            value: r,
            disabled: n,
            textValue: m,
            children: /* @__PURE__ */ i(
              E.div,
              {
                role: "option",
                "aria-labelledby": p,
                "data-highlighted": h ? "" : void 0,
                "aria-selected": u && h,
                "data-state": u ? "checked" : "unchecked",
                "aria-disabled": n || void 0,
                "data-disabled": n ? "" : void 0,
                tabIndex: n ? void 0 : -1,
                ...s,
                ref: b,
                onFocus: N(s.onFocus, () => v(!0)),
                onBlur: N(s.onBlur, () => v(!1)),
                onClick: N(s.onClick, () => {
                  g.current !== "mouse" && x();
                }),
                onPointerUp: N(s.onPointerUp, () => {
                  g.current === "mouse" && x();
                }),
                onPointerDown: N(s.onPointerDown, (S) => {
                  g.current = S.pointerType;
                }),
                onPointerMove: N(s.onPointerMove, (S) => {
                  g.current = S.pointerType, n ? d.onItemLeave?.() : g.current === "mouse" && S.currentTarget.focus({ preventScroll: !0 });
                }),
                onPointerLeave: N(s.onPointerLeave, (S) => {
                  S.currentTarget === document.activeElement && d.onItemLeave?.();
                }),
                onKeyDown: N(s.onKeyDown, (S) => {
                  d.searchRef?.current !== "" && S.key === " " || (Zo.includes(S.key) && x(), S.key === " " && S.preventDefault());
                })
              }
            )
          }
        )
      }
    );
  }
);
Ct.displayName = ge;
var ue = "SelectItemText", Rt = l.forwardRef(
  (e, o) => {
    const { __scopeSelect: t, className: r, style: n, ...a } = e, s = J(ue, t), c = Q(ue, t), d = yt(ue, t), u = tn(ue, t), [m, f] = l.useState(null), h = k(
      o,
      (x) => f(x),
      d.onItemTextChange,
      (x) => c.itemTextRefCallback?.(x, d.value, d.disabled)
    ), v = m?.textContent, b = l.useMemo(
      () => /* @__PURE__ */ i("option", { value: d.value, disabled: d.disabled, children: v }, d.value),
      [d.disabled, d.value, v]
    ), { onNativeOptionAdd: p, onNativeOptionRemove: g } = u;
    return Z(() => (p(b), () => g(b)), [p, g, b]), /* @__PURE__ */ R(ve, { children: [
      /* @__PURE__ */ i(E.span, { id: d.textId, ...a, ref: h }),
      d.isSelected && s.valueNode && !s.valueNodeHasChildren ? qe.createPortal(a.children, s.valueNode) : null
    ] });
  }
);
Rt.displayName = ue;
var Nt = "SelectItemIndicator", Pt = l.forwardRef(
  (e, o) => {
    const { __scopeSelect: t, ...r } = e;
    return yt(Nt, t).isSelected ? /* @__PURE__ */ i(E.span, { "aria-hidden": !0, ...r, ref: o }) : null;
  }
);
Pt.displayName = Nt;
var Oe = "SelectScrollUpButton", Et = l.forwardRef((e, o) => {
  const t = Q(Oe, e.__scopeSelect), r = $e(Oe, e.__scopeSelect), [n, a] = l.useState(!1), s = k(o, r.onScrollButtonChange);
  return Z(() => {
    if (t.viewport && t.isPositioned) {
      let c = function() {
        const u = d.scrollTop > 0;
        a(u);
      };
      const d = t.viewport;
      return c(), d.addEventListener("scroll", c), () => d.removeEventListener("scroll", c);
    }
  }, [t.viewport, t.isPositioned]), n ? /* @__PURE__ */ i(
    _t,
    {
      ...e,
      ref: s,
      onAutoScroll: () => {
        const { viewport: c, selectedItem: d } = t;
        c && d && (c.scrollTop = c.scrollTop - d.offsetHeight);
      }
    }
  ) : null;
});
Et.displayName = Oe;
var Be = "SelectScrollDownButton", It = l.forwardRef((e, o) => {
  const t = Q(Be, e.__scopeSelect), r = $e(Be, e.__scopeSelect), [n, a] = l.useState(!1), s = k(o, r.onScrollButtonChange);
  return Z(() => {
    if (t.viewport && t.isPositioned) {
      let c = function() {
        const u = d.scrollHeight - d.clientHeight, m = Math.ceil(d.scrollTop) < u;
        a(m);
      };
      const d = t.viewport;
      return c(), d.addEventListener("scroll", c), () => d.removeEventListener("scroll", c);
    }
  }, [t.viewport, t.isPositioned]), n ? /* @__PURE__ */ i(
    _t,
    {
      ...e,
      ref: s,
      onAutoScroll: () => {
        const { viewport: c, selectedItem: d } = t;
        c && d && (c.scrollTop = c.scrollTop + d.offsetHeight);
      }
    }
  ) : null;
});
It.displayName = Be;
var _t = l.forwardRef((e, o) => {
  const { __scopeSelect: t, onAutoScroll: r, ...n } = e, a = Q("SelectScrollButton", t), s = l.useRef(null), c = we(t), d = l.useCallback(() => {
    s.current !== null && (window.clearInterval(s.current), s.current = null);
  }, []);
  return l.useEffect(() => () => d(), [d]), Z(() => {
    c().find((m) => m.ref.current === document.activeElement)?.ref.current?.scrollIntoView({ block: "nearest" });
  }, [c]), /* @__PURE__ */ i(
    E.div,
    {
      "aria-hidden": !0,
      ...n,
      ref: o,
      style: { flexShrink: 0, ...n.style },
      onPointerDown: N(n.onPointerDown, () => {
        s.current === null && (s.current = window.setInterval(r, 50));
      }),
      onPointerMove: N(n.onPointerMove, () => {
        a.onItemLeave?.(), s.current === null && (s.current = window.setInterval(r, 50));
      }),
      onPointerLeave: N(n.onPointerLeave, () => {
        d();
      })
    }
  );
}), pn = "SelectSeparator", mn = l.forwardRef(
  (e, o) => {
    const { __scopeSelect: t, ...r } = e;
    return /* @__PURE__ */ i(E.div, { "aria-hidden": !0, ...r, ref: o });
  }
);
mn.displayName = pn;
var Le = "SelectArrow", hn = l.forwardRef(
  (e, o) => {
    const { __scopeSelect: t, ...r } = e, n = ye(t), a = J(Le, t), s = Q(Le, t);
    return a.open && s.position === "popper" ? /* @__PURE__ */ i(wo, { ...n, ...r, ref: o }) : null;
  }
);
hn.displayName = Le;
var gn = "SelectBubbleInput", kt = l.forwardRef(
  ({ __scopeSelect: e, value: o, ...t }, r) => {
    const n = l.useRef(null), a = k(r, n), s = fe(o);
    return l.useEffect(() => {
      const c = n.current;
      if (!c) return;
      const d = window.HTMLSelectElement.prototype, m = Object.getOwnPropertyDescriptor(
        d,
        "value"
      ).set;
      if (s !== o && m) {
        const f = new Event("change", { bubbles: !0 });
        m.call(c, o), c.dispatchEvent(f);
      }
    }, [s, o]), /* @__PURE__ */ i(
      E.select,
      {
        ...t,
        style: { ...ko, ...t.style },
        ref: a,
        defaultValue: o
      }
    );
  }
);
kt.displayName = gn;
function Tt(e) {
  return e === "" || e === void 0;
}
function Mt(e) {
  const o = go(e), t = l.useRef(""), r = l.useRef(0), n = l.useCallback(
    (s) => {
      const c = t.current + s;
      o(c), (function d(u) {
        t.current = u, window.clearTimeout(r.current), u !== "" && (r.current = window.setTimeout(() => d(""), 1e3));
      })(c);
    },
    [o]
  ), a = l.useCallback(() => {
    t.current = "", window.clearTimeout(r.current);
  }, []);
  return l.useEffect(() => () => window.clearTimeout(r.current), []), [t, n, a];
}
function Dt(e, o, t) {
  const n = o.length > 1 && Array.from(o).every((u) => u === o[0]) ? o[0] : o, a = t ? e.indexOf(t) : -1;
  let s = vn(e, Math.max(a, 0));
  n.length === 1 && (s = s.filter((u) => u !== t));
  const d = s.find(
    (u) => u.textValue.toLowerCase().startsWith(n.toLowerCase())
  );
  return d !== t ? d : void 0;
}
function vn(e, o) {
  return e.map((t, r) => e[(o + r) % e.length]);
}
var bn = at, xn = ct, Sn = dt, wn = ut, yn = ft, Cn = pt, Rn = vt, Nn = xt, Pn = wt, En = Ct, In = Rt, _n = Pt, kn = Et, Tn = It, Ce = "Checkbox", [Mn] = ie(Ce), [Dn, Ue] = Mn(Ce);
function An(e) {
  const {
    __scopeCheckbox: o,
    checked: t,
    children: r,
    defaultChecked: n,
    disabled: a,
    form: s,
    name: c,
    onCheckedChange: d,
    required: u,
    value: m = "on",
    // @ts-expect-error
    internal_do_not_use_render: f
  } = e, [h, v] = se({
    prop: t,
    defaultProp: n ?? !1,
    onChange: d,
    caller: Ce
  }), [b, p] = l.useState(null), [g, x] = l.useState(null), S = l.useRef(!1), w = b ? !!s || !!b.closest("form") : (
    // We set this to true by default so that events bubble to forms without JS (SSR)
    !0
  ), C = {
    checked: h,
    disabled: a,
    setChecked: v,
    control: b,
    setControl: p,
    name: c,
    form: s,
    value: m,
    hasConsumerStoppedPropagationRef: S,
    required: u,
    defaultChecked: X(n) ? !1 : n,
    isFormControl: w,
    bubbleInput: g,
    setBubbleInput: x
  };
  return /* @__PURE__ */ i(
    Dn,
    {
      scope: o,
      ...C,
      children: On(f) ? f(C) : r
    }
  );
}
var At = "CheckboxTrigger", Ot = l.forwardRef(
  ({ __scopeCheckbox: e, onKeyDown: o, onClick: t, ...r }, n) => {
    const {
      control: a,
      value: s,
      disabled: c,
      checked: d,
      required: u,
      setControl: m,
      setChecked: f,
      hasConsumerStoppedPropagationRef: h,
      isFormControl: v,
      bubbleInput: b
    } = Ue(At, e), p = k(n, m), g = l.useRef(d);
    return l.useEffect(() => {
      const x = a?.form;
      if (x) {
        const S = () => f(g.current);
        return x.addEventListener("reset", S), () => x.removeEventListener("reset", S);
      }
    }, [a, f]), /* @__PURE__ */ i(
      E.button,
      {
        type: "button",
        role: "checkbox",
        "aria-checked": X(d) ? "mixed" : d,
        "aria-required": u,
        "data-state": zt(d),
        "data-disabled": c ? "" : void 0,
        disabled: c,
        value: s,
        ...r,
        ref: p,
        onKeyDown: N(o, (x) => {
          x.key === "Enter" && x.preventDefault();
        }),
        onClick: N(t, (x) => {
          f((S) => X(S) ? !0 : !S), b && v && (h.current = x.isPropagationStopped(), h.current || x.stopPropagation());
        })
      }
    );
  }
);
Ot.displayName = At;
var Bt = l.forwardRef(
  (e, o) => {
    const {
      __scopeCheckbox: t,
      name: r,
      checked: n,
      defaultChecked: a,
      required: s,
      disabled: c,
      value: d,
      onCheckedChange: u,
      form: m,
      ...f
    } = e;
    return /* @__PURE__ */ i(
      An,
      {
        __scopeCheckbox: t,
        checked: n,
        defaultChecked: a,
        disabled: c,
        required: s,
        onCheckedChange: u,
        name: r,
        form: m,
        value: d,
        internal_do_not_use_render: ({ isFormControl: h }) => /* @__PURE__ */ R(ve, { children: [
          /* @__PURE__ */ i(
            Ot,
            {
              ...f,
              ref: o,
              __scopeCheckbox: t
            }
          ),
          h && /* @__PURE__ */ i(
            Kt,
            {
              __scopeCheckbox: t
            }
          )
        ] })
      }
    );
  }
);
Bt.displayName = Ce;
var Lt = "CheckboxIndicator", Vt = l.forwardRef(
  (e, o) => {
    const { __scopeCheckbox: t, forceMount: r, ...n } = e, a = Ue(Lt, t);
    return /* @__PURE__ */ i(
      Ye,
      {
        present: r || X(a.checked) || a.checked === !0,
        children: /* @__PURE__ */ i(
          E.span,
          {
            "data-state": zt(a.checked),
            "data-disabled": a.disabled ? "" : void 0,
            ...n,
            ref: o,
            style: { pointerEvents: "none", ...e.style }
          }
        )
      }
    );
  }
);
Vt.displayName = Lt;
var Ht = "CheckboxBubbleInput", Kt = l.forwardRef(
  ({ __scopeCheckbox: e, ...o }, t) => {
    const {
      control: r,
      hasConsumerStoppedPropagationRef: n,
      checked: a,
      defaultChecked: s,
      required: c,
      disabled: d,
      name: u,
      value: m,
      form: f,
      bubbleInput: h,
      setBubbleInput: v
    } = Ue(Ht, e), b = k(t, v), p = fe(a), g = be(r);
    l.useEffect(() => {
      const S = h;
      if (!S) return;
      const w = window.HTMLInputElement.prototype, M = Object.getOwnPropertyDescriptor(
        w,
        "checked"
      ).set, W = !n.current;
      if (p !== a && M) {
        const D = new Event("click", { bubbles: W });
        S.indeterminate = X(a), M.call(S, X(a) ? !1 : a), S.dispatchEvent(D);
      }
    }, [h, p, a, n]);
    const x = l.useRef(X(a) ? !1 : a);
    return /* @__PURE__ */ i(
      E.input,
      {
        type: "checkbox",
        "aria-hidden": !0,
        defaultChecked: s ?? x.current,
        required: c,
        disabled: d,
        name: u,
        value: m,
        form: f,
        ...o,
        tabIndex: -1,
        ref: b,
        style: {
          ...o.style,
          ...g,
          position: "absolute",
          pointerEvents: "none",
          opacity: 0,
          margin: 0,
          // We transform because the input is absolutely positioned but we have
          // rendered it **after** the button. This pulls it back to sit on top
          // of the button.
          transform: "translateX(-100%)"
        }
      }
    );
  }
);
Kt.displayName = Ht;
function On(e) {
  return typeof e == "function";
}
function X(e) {
  return e === "indeterminate";
}
function zt(e) {
  return X(e) ? "indeterminate" : e ? "checked" : "unchecked";
}
var Ge = "Radio", [Bn, Ft] = ie(Ge), [Ln, Vn] = Bn(Ge), $t = l.forwardRef(
  (e, o) => {
    const {
      __scopeRadio: t,
      name: r,
      checked: n = !1,
      required: a,
      disabled: s,
      value: c = "on",
      onCheck: d,
      form: u,
      ...m
    } = e, [f, h] = l.useState(null), v = k(o, (g) => h(g)), b = l.useRef(!1), p = f ? u || !!f.closest("form") : !0;
    return /* @__PURE__ */ R(Ln, { scope: t, checked: n, disabled: s, children: [
      /* @__PURE__ */ i(
        E.button,
        {
          type: "button",
          role: "radio",
          "aria-checked": n,
          "data-state": jt(n),
          "data-disabled": s ? "" : void 0,
          disabled: s,
          value: c,
          ...m,
          ref: v,
          onClick: N(e.onClick, (g) => {
            n || d?.(), p && (b.current = g.isPropagationStopped(), b.current || g.stopPropagation());
          })
        }
      ),
      p && /* @__PURE__ */ i(
        Wt,
        {
          control: f,
          bubbles: !b.current,
          name: r,
          value: c,
          checked: n,
          required: a,
          disabled: s,
          form: u,
          style: { transform: "translateX(-100%)" }
        }
      )
    ] });
  }
);
$t.displayName = Ge;
var Ut = "RadioIndicator", Gt = l.forwardRef(
  (e, o) => {
    const { __scopeRadio: t, forceMount: r, ...n } = e, a = Vn(Ut, t);
    return /* @__PURE__ */ i(Ye, { present: r || a.checked, children: /* @__PURE__ */ i(
      E.span,
      {
        "data-state": jt(a.checked),
        "data-disabled": a.disabled ? "" : void 0,
        ...n,
        ref: o
      }
    ) });
  }
);
Gt.displayName = Ut;
var Hn = "RadioBubbleInput", Wt = l.forwardRef(
  ({
    __scopeRadio: e,
    control: o,
    checked: t,
    bubbles: r = !0,
    ...n
  }, a) => {
    const s = l.useRef(null), c = k(s, a), d = fe(t), u = be(o);
    return l.useEffect(() => {
      const m = s.current;
      if (!m) return;
      const f = window.HTMLInputElement.prototype, v = Object.getOwnPropertyDescriptor(
        f,
        "checked"
      ).set;
      if (d !== t && v) {
        const b = new Event("click", { bubbles: r });
        v.call(m, t), m.dispatchEvent(b);
      }
    }, [d, t, r]), /* @__PURE__ */ i(
      E.input,
      {
        type: "radio",
        "aria-hidden": !0,
        defaultChecked: t,
        ...n,
        tabIndex: -1,
        ref: c,
        style: {
          ...n.style,
          ...u,
          position: "absolute",
          pointerEvents: "none",
          opacity: 0,
          margin: 0
        }
      }
    );
  }
);
Wt.displayName = Hn;
function jt(e) {
  return e ? "checked" : "unchecked";
}
var Kn = ["ArrowUp", "ArrowDown", "ArrowLeft", "ArrowRight"], Re = "RadioGroup", [zn] = ie(Re, [
  Je,
  Ft
]), qt = Je(), Yt = Ft(), [Fn, $n] = zn(Re), Xt = l.forwardRef(
  (e, o) => {
    const {
      __scopeRadioGroup: t,
      name: r,
      defaultValue: n,
      value: a,
      required: s = !1,
      disabled: c = !1,
      orientation: d,
      dir: u,
      loop: m = !0,
      onValueChange: f,
      ...h
    } = e, v = qt(t), b = ze(u), [p, g] = se({
      prop: a,
      defaultProp: n ?? null,
      onChange: f,
      caller: Re
    });
    return /* @__PURE__ */ i(
      Fn,
      {
        scope: t,
        name: r,
        required: s,
        disabled: c,
        value: p,
        onValueChange: g,
        children: /* @__PURE__ */ i(
          yo,
          {
            asChild: !0,
            ...v,
            orientation: d,
            dir: b,
            loop: m,
            children: /* @__PURE__ */ i(
              E.div,
              {
                role: "radiogroup",
                "aria-required": s,
                "aria-orientation": d,
                "data-disabled": c ? "" : void 0,
                dir: b,
                ...h,
                ref: o
              }
            )
          }
        )
      }
    );
  }
);
Xt.displayName = Re;
var Zt = "RadioGroupItem", Jt = l.forwardRef(
  (e, o) => {
    const { __scopeRadioGroup: t, disabled: r, ...n } = e, a = $n(Zt, t), s = a.disabled || r, c = qt(t), d = Yt(t), u = l.useRef(null), m = k(o, u), f = a.value === n.value, h = l.useRef(!1);
    return l.useEffect(() => {
      const v = (p) => {
        Kn.includes(p.key) && (h.current = !0);
      }, b = () => h.current = !1;
      return document.addEventListener("keydown", v), document.addEventListener("keyup", b), () => {
        document.removeEventListener("keydown", v), document.removeEventListener("keyup", b);
      };
    }, []), /* @__PURE__ */ i(
      Co,
      {
        asChild: !0,
        ...c,
        focusable: !s,
        active: f,
        children: /* @__PURE__ */ i(
          $t,
          {
            disabled: s,
            required: a.required,
            checked: f,
            ...d,
            ...n,
            name: a.name,
            ref: m,
            onCheck: () => a.onValueChange(n.value),
            onKeyDown: N((v) => {
              v.key === "Enter" && v.preventDefault();
            }),
            onFocus: N(n.onFocus, () => {
              h.current && u.current?.click();
            })
          }
        )
      }
    );
  }
);
Jt.displayName = Zt;
var Un = "RadioGroupIndicator", Qt = l.forwardRef(
  (e, o) => {
    const { __scopeRadioGroup: t, ...r } = e, n = Yt(t);
    return /* @__PURE__ */ i(Gt, { ...n, ...r, ref: o });
  }
);
Qt.displayName = Un;
var Gn = Xt, Wn = Jt, jn = Qt, eo = ["PageUp", "PageDown"], to = ["ArrowUp", "ArrowDown", "ArrowLeft", "ArrowRight"], oo = {
  "from-left": ["Home", "PageDown", "ArrowDown", "ArrowLeft"],
  "from-right": ["Home", "PageDown", "ArrowDown", "ArrowRight"],
  "from-bottom": ["Home", "PageDown", "ArrowDown", "ArrowLeft"],
  "from-top": ["Home", "PageDown", "ArrowUp", "ArrowLeft"]
}, le = "Slider", [Ve, qn, Yn] = Ze(le), [no] = ie(le, [
  Yn
]), [Xn, Ne] = no(le), ro = l.forwardRef(
  (e, o) => {
    const {
      name: t,
      min: r = 0,
      max: n = 100,
      step: a = 1,
      orientation: s = "horizontal",
      disabled: c = !1,
      minStepsBetweenThumbs: d = 0,
      defaultValue: u = [r],
      value: m,
      onValueChange: f = () => {
      },
      onValueCommit: h = () => {
      },
      inverted: v = !1,
      form: b,
      ...p
    } = e, g = l.useRef(/* @__PURE__ */ new Set()), x = l.useRef(0), w = s === "horizontal" ? Zn : Jn, [C = [], M] = se({
      prop: m,
      defaultProp: u,
      onChange: (I) => {
        [...g.current][x.current]?.focus(), f(I);
      }
    }), W = l.useRef(C);
    function D(I) {
      const L = nr(C, I);
      K(I, L);
    }
    function $(I) {
      K(I, x.current);
    }
    function Y() {
      const I = W.current[x.current];
      C[x.current] !== I && h(C);
    }
    function K(I, L, { commit: U } = { commit: !1 }) {
      const G = ir(a), z = cr(Math.round((I - r) / a) * a + r, G), A = he(z, [r, n]);
      M((V = []) => {
        const H = tr(V, A, L);
        if (ar(H, d * a)) {
          x.current = H.indexOf(A);
          const j = String(H) !== String(V);
          return j && U && h(H), j ? H : V;
        } else
          return V;
      });
    }
    return /* @__PURE__ */ i(
      Xn,
      {
        scope: e.__scopeSlider,
        name: t,
        disabled: c,
        min: r,
        max: n,
        valueIndexToChangeRef: x,
        thumbs: g.current,
        values: C,
        orientation: s,
        form: b,
        children: /* @__PURE__ */ i(Ve.Provider, { scope: e.__scopeSlider, children: /* @__PURE__ */ i(Ve.Slot, { scope: e.__scopeSlider, children: /* @__PURE__ */ i(
          w,
          {
            "aria-disabled": c,
            "data-disabled": c ? "" : void 0,
            ...p,
            ref: o,
            onPointerDown: N(p.onPointerDown, () => {
              c || (W.current = C);
            }),
            min: r,
            max: n,
            inverted: v,
            onSlideStart: c ? void 0 : D,
            onSlideMove: c ? void 0 : $,
            onSlideEnd: c ? void 0 : Y,
            onHomeKeyDown: () => !c && K(r, 0, { commit: !0 }),
            onEndKeyDown: () => !c && K(n, C.length - 1, { commit: !0 }),
            onStepKeyDown: ({ event: I, direction: L }) => {
              if (!c) {
                const z = eo.includes(I.key) || I.shiftKey && to.includes(I.key) ? 10 : 1, A = x.current, V = C[A], H = a * z * L;
                K(V + H, A, { commit: !0 });
              }
            }
          }
        ) }) })
      }
    );
  }
);
ro.displayName = le;
var [so, ao] = no(le, {
  startEdge: "left",
  endEdge: "right",
  size: "width",
  direction: 1
}), Zn = l.forwardRef(
  (e, o) => {
    const {
      min: t,
      max: r,
      dir: n,
      inverted: a,
      onSlideStart: s,
      onSlideMove: c,
      onSlideEnd: d,
      onStepKeyDown: u,
      ...m
    } = e, [f, h] = l.useState(null), v = k(o, (w) => h(w)), b = l.useRef(void 0), p = ze(n), g = p === "ltr", x = g && !a || !g && a;
    function S(w) {
      const C = b.current || f.getBoundingClientRect(), M = [0, C.width], D = We(M, x ? [t, r] : [r, t]);
      return b.current = C, D(w - C.left);
    }
    return /* @__PURE__ */ i(
      so,
      {
        scope: e.__scopeSlider,
        startEdge: x ? "left" : "right",
        endEdge: x ? "right" : "left",
        direction: x ? 1 : -1,
        size: "width",
        children: /* @__PURE__ */ i(
          io,
          {
            dir: p,
            "data-orientation": "horizontal",
            ...m,
            ref: v,
            style: {
              ...m.style,
              "--radix-slider-thumb-transform": "translateX(-50%)"
            },
            onSlideStart: (w) => {
              const C = S(w.clientX);
              s?.(C);
            },
            onSlideMove: (w) => {
              const C = S(w.clientX);
              c?.(C);
            },
            onSlideEnd: () => {
              b.current = void 0, d?.();
            },
            onStepKeyDown: (w) => {
              const M = oo[x ? "from-left" : "from-right"].includes(w.key);
              u?.({ event: w, direction: M ? -1 : 1 });
            }
          }
        )
      }
    );
  }
), Jn = l.forwardRef(
  (e, o) => {
    const {
      min: t,
      max: r,
      inverted: n,
      onSlideStart: a,
      onSlideMove: s,
      onSlideEnd: c,
      onStepKeyDown: d,
      ...u
    } = e, m = l.useRef(null), f = k(o, m), h = l.useRef(void 0), v = !n;
    function b(p) {
      const g = h.current || m.current.getBoundingClientRect(), x = [0, g.height], w = We(x, v ? [r, t] : [t, r]);
      return h.current = g, w(p - g.top);
    }
    return /* @__PURE__ */ i(
      so,
      {
        scope: e.__scopeSlider,
        startEdge: v ? "bottom" : "top",
        endEdge: v ? "top" : "bottom",
        size: "height",
        direction: v ? 1 : -1,
        children: /* @__PURE__ */ i(
          io,
          {
            "data-orientation": "vertical",
            ...u,
            ref: f,
            style: {
              ...u.style,
              "--radix-slider-thumb-transform": "translateY(50%)"
            },
            onSlideStart: (p) => {
              const g = b(p.clientY);
              a?.(g);
            },
            onSlideMove: (p) => {
              const g = b(p.clientY);
              s?.(g);
            },
            onSlideEnd: () => {
              h.current = void 0, c?.();
            },
            onStepKeyDown: (p) => {
              const x = oo[v ? "from-bottom" : "from-top"].includes(p.key);
              d?.({ event: p, direction: x ? -1 : 1 });
            }
          }
        )
      }
    );
  }
), io = l.forwardRef(
  (e, o) => {
    const {
      __scopeSlider: t,
      onSlideStart: r,
      onSlideMove: n,
      onSlideEnd: a,
      onHomeKeyDown: s,
      onEndKeyDown: c,
      onStepKeyDown: d,
      ...u
    } = e, m = Ne(le, t);
    return /* @__PURE__ */ i(
      E.span,
      {
        ...u,
        ref: o,
        onKeyDown: N(e.onKeyDown, (f) => {
          f.key === "Home" ? (s(f), f.preventDefault()) : f.key === "End" ? (c(f), f.preventDefault()) : eo.concat(to).includes(f.key) && (d(f), f.preventDefault());
        }),
        onPointerDown: N(e.onPointerDown, (f) => {
          const h = f.target;
          h.setPointerCapture(f.pointerId), f.preventDefault(), m.thumbs.has(h) ? h.focus() : r(f);
        }),
        onPointerMove: N(e.onPointerMove, (f) => {
          f.target.hasPointerCapture(f.pointerId) && n(f);
        }),
        onPointerUp: N(e.onPointerUp, (f) => {
          const h = f.target;
          h.hasPointerCapture(f.pointerId) && (h.releasePointerCapture(f.pointerId), a(f));
        })
      }
    );
  }
), co = "SliderTrack", lo = l.forwardRef(
  (e, o) => {
    const { __scopeSlider: t, ...r } = e, n = Ne(co, t);
    return /* @__PURE__ */ i(
      E.span,
      {
        "data-disabled": n.disabled ? "" : void 0,
        "data-orientation": n.orientation,
        ...r,
        ref: o
      }
    );
  }
);
lo.displayName = co;
var He = "SliderRange", uo = l.forwardRef(
  (e, o) => {
    const { __scopeSlider: t, ...r } = e, n = Ne(He, t), a = ao(He, t), s = l.useRef(null), c = k(o, s), d = n.values.length, u = n.values.map(
      (h) => mo(h, n.min, n.max)
    ), m = d > 1 ? Math.min(...u) : 0, f = 100 - Math.max(...u);
    return /* @__PURE__ */ i(
      E.span,
      {
        "data-orientation": n.orientation,
        "data-disabled": n.disabled ? "" : void 0,
        ...r,
        ref: c,
        style: {
          ...e.style,
          [a.startEdge]: m + "%",
          [a.endEdge]: f + "%"
        }
      }
    );
  }
);
uo.displayName = He;
var Ke = "SliderThumb", fo = l.forwardRef(
  (e, o) => {
    const t = qn(e.__scopeSlider), [r, n] = l.useState(null), a = k(o, (c) => n(c)), s = l.useMemo(
      () => r ? t().findIndex((c) => c.ref.current === r) : -1,
      [t, r]
    );
    return /* @__PURE__ */ i(Qn, { ...e, ref: a, index: s });
  }
), Qn = l.forwardRef(
  (e, o) => {
    const { __scopeSlider: t, index: r, name: n, ...a } = e, s = Ne(Ke, t), c = ao(Ke, t), [d, u] = l.useState(null), m = k(o, (S) => u(S)), f = d ? s.form || !!d.closest("form") : !0, h = be(d), v = s.values[r], b = v === void 0 ? 0 : mo(v, s.min, s.max), p = or(r, s.values.length), g = h?.[c.size], x = g ? rr(g, b, c.direction) : 0;
    return l.useEffect(() => {
      if (d)
        return s.thumbs.add(d), () => {
          s.thumbs.delete(d);
        };
    }, [d, s.thumbs]), /* @__PURE__ */ R(
      "span",
      {
        style: {
          transform: "var(--radix-slider-thumb-transform)",
          position: "absolute",
          [c.startEdge]: `calc(${b}% + ${x}px)`
        },
        children: [
          /* @__PURE__ */ i(Ve.ItemSlot, { scope: e.__scopeSlider, children: /* @__PURE__ */ i(
            E.span,
            {
              role: "slider",
              "aria-label": e["aria-label"] || p,
              "aria-valuemin": s.min,
              "aria-valuenow": v,
              "aria-valuemax": s.max,
              "aria-orientation": s.orientation,
              "data-orientation": s.orientation,
              "data-disabled": s.disabled ? "" : void 0,
              tabIndex: s.disabled ? void 0 : 0,
              ...a,
              ref: m,
              style: v === void 0 ? { display: "none" } : e.style,
              onFocus: N(e.onFocus, () => {
                s.valueIndexToChangeRef.current = r;
              })
            }
          ) }),
          f && /* @__PURE__ */ i(
            po,
            {
              name: n ?? (s.name ? s.name + (s.values.length > 1 ? "[]" : "") : void 0),
              form: s.form,
              value: v
            },
            r
          )
        ]
      }
    );
  }
);
fo.displayName = Ke;
var er = "RadioBubbleInput", po = l.forwardRef(
  ({ __scopeSlider: e, value: o, ...t }, r) => {
    const n = l.useRef(null), a = k(n, r), s = fe(o);
    return l.useEffect(() => {
      const c = n.current;
      if (!c) return;
      const d = window.HTMLInputElement.prototype, m = Object.getOwnPropertyDescriptor(d, "value").set;
      if (s !== o && m) {
        const f = new Event("input", { bubbles: !0 });
        m.call(c, o), c.dispatchEvent(f);
      }
    }, [s, o]), /* @__PURE__ */ i(
      E.input,
      {
        style: { display: "none" },
        ...t,
        ref: a,
        defaultValue: o
      }
    );
  }
);
po.displayName = er;
function tr(e = [], o, t) {
  const r = [...e];
  return r[t] = o, r.sort((n, a) => n - a);
}
function mo(e, o, t) {
  const a = 100 / (t - o) * (e - o);
  return he(a, [0, 100]);
}
function or(e, o) {
  return o > 2 ? `Value ${e + 1} of ${o}` : o === 2 ? ["Minimum", "Maximum"][e] : void 0;
}
function nr(e, o) {
  if (e.length === 1) return 0;
  const t = e.map((n) => Math.abs(n - o)), r = Math.min(...t);
  return t.indexOf(r);
}
function rr(e, o, t) {
  const r = e / 2, a = We([0, 50], [0, r]);
  return (r - a(o) * t) * t;
}
function sr(e) {
  return e.slice(0, -1).map((o, t) => e[t + 1] - o);
}
function ar(e, o) {
  if (o > 0) {
    const t = sr(e);
    return Math.min(...t) >= o;
  }
  return !0;
}
function We(e, o) {
  return (t) => {
    if (e[0] === e[1] || o[0] === o[1]) return o[0];
    const r = (o[1] - o[0]) / (e[1] - e[0]);
    return o[0] + r * (t - e[0]);
  };
}
function ir(e) {
  return (String(e).split(".")[1] || "").length;
}
function cr(e, o) {
  const t = Math.pow(10, o);
  return Math.round(e * t) / t;
}
var lr = ro, dr = lo, ur = uo, fr = fo;
function ae({ className: e, ...o }) {
  return /* @__PURE__ */ i(
    Lo,
    {
      className: P(
        "text-xs font-semibold uppercase tracking-wider text-muted-foreground mb-1.5 block",
        e
      ),
      ...o
    }
  );
}
function Pr({ label: e, error: o, icon: t, iconPosition: r = "left", prefix: n, suffix: a, className: s, type: c, ...d }) {
  const u = c === "password", [m, f] = l.useState(!1), h = u ? m ? "text" : "password" : c, v = t && r === "left", b = t && r === "right";
  return /* @__PURE__ */ R("div", { className: P("flex flex-col gap-1"), children: [
    e && /* @__PURE__ */ i(ae, { children: e }),
    /* @__PURE__ */ R(
      "div",
      {
        className: P(
          "flex items-center gap-2 bg-input/50 border rounded-2xl px-3 text-sm transition-[color,box-shadow,border-color,background-color]",
          "focus-within:border-ring focus-within:ring-3 focus-within:ring-ring/50",
          "has-[input:disabled]:bg-muted-foreground/10 has-[input:disabled]:cursor-not-allowed has-[input:disabled]:border-dashed has-[input:disabled]:border-muted-foreground/30 has-[input:disabled]:[&_input]:text-muted-foreground has-[input:disabled]:[&_*]:text-muted-foreground",
          o ? "border-destructive" : "border-transparent"
        ),
        children: [
          v && /* @__PURE__ */ i(t, { size: 16, className: "text-muted-foreground shrink-0" }),
          n && /* @__PURE__ */ i("span", { className: "text-muted-foreground shrink-0", children: n }),
          /* @__PURE__ */ i(
            "input",
            {
              type: h,
              className: P(
                "flex-1 min-w-0 bg-transparent py-2 text-foreground placeholder:text-muted-foreground outline-none",
                s
              ),
              ...d
            }
          ),
          a && !u && /* @__PURE__ */ i("span", { className: "text-muted-foreground shrink-0", children: a }),
          b && !u && /* @__PURE__ */ i(t, { size: 16, className: "text-muted-foreground shrink-0" }),
          u && /* @__PURE__ */ i(
            "button",
            {
              type: "button",
              onClick: () => f(!m),
              className: "text-muted-foreground hover:text-foreground cursor-pointer shrink-0 flex items-center",
              children: m ? /* @__PURE__ */ i(Mo, { size: 16 }) : /* @__PURE__ */ i(Do, { size: 16 })
            }
          )
        ]
      }
    ),
    o && /* @__PURE__ */ i("span", { className: "text-xs text-destructive", children: o })
  ] });
}
function Er({ label: e, placeholder: o, children: t, className: r, native: n, clearable: a, ...s }) {
  const { value: c, onValueChange: d, disabled: u } = s, [m, f] = l.useState(0), h = c !== void 0 && c !== "", v = (b) => {
    b.stopPropagation(), d?.(""), f((p) => p + 1);
  };
  if (n) {
    const { defaultValue: b } = s;
    return /* @__PURE__ */ R("div", { className: "flex flex-col gap-1", children: [
      e && /* @__PURE__ */ i(ae, { children: e }),
      /* @__PURE__ */ R("div", { className: P(
        "flex items-center gap-2 bg-input/50 border border-transparent rounded-2xl px-3 text-sm transition-[color,box-shadow,border-color]",
        "focus-within:border-ring focus-within:ring-3 focus-within:ring-ring/50",
        r
      ), children: [
        /* @__PURE__ */ R(
          "select",
          {
            value: c,
            defaultValue: b,
            onChange: d ? (p) => d(p.target.value) : void 0,
            disabled: u,
            className: "flex-1 min-w-0 bg-transparent py-2 text-foreground outline-none cursor-pointer appearance-none",
            children: [
              o && /* @__PURE__ */ i("option", { value: "", disabled: !a, hidden: !a, children: o }),
              t
            ]
          }
        ),
        a && h ? /* @__PURE__ */ i("button", { type: "button", onClick: v, className: "text-muted-foreground hover:text-foreground cursor-pointer bg-transparent border-none p-0 shrink-0", children: /* @__PURE__ */ i(je, { size: 14 }) }) : /* @__PURE__ */ i(Me, { size: 16, className: "text-muted-foreground shrink-0 pointer-events-none" })
      ] })
    ] });
  }
  return /* @__PURE__ */ R("div", { className: "flex flex-col gap-1", children: [
    e && /* @__PURE__ */ i(ae, { children: e }),
    /* @__PURE__ */ R("div", { className: "relative", children: [
      /* @__PURE__ */ R(bn, { ...s, value: h ? c : void 0, children: [
        /* @__PURE__ */ R(
          xn,
          {
            className: P(
              "flex w-full items-center justify-between gap-1.5 rounded-2xl border border-transparent bg-input/50 px-3 py-2 text-sm whitespace-nowrap transition-[color,box-shadow,background-color] outline-none h-9",
              "focus-visible:border-ring focus-visible:ring-3 focus-visible:ring-ring/50",
              "disabled:bg-muted-foreground/10 disabled:border-dashed disabled:border-muted-foreground/30 disabled:text-muted-foreground disabled:cursor-not-allowed",
              "data-[placeholder]:text-muted-foreground",
              "aria-invalid:border-destructive aria-invalid:ring-3 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40",
              "[&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4",
              a && h ? "pr-8" : "",
              r
            ),
            children: [
              /* @__PURE__ */ i("span", { className: "flex-1 text-left truncate", children: /* @__PURE__ */ i(Sn, { placeholder: o }) }),
              !(a && h) && /* @__PURE__ */ i(wn, { asChild: !0, children: /* @__PURE__ */ i(Me, { size: 16, className: "text-muted-foreground" }) })
            ]
          }
        ),
        /* @__PURE__ */ i(yn, { children: /* @__PURE__ */ R(
          Cn,
          {
            className: P(
              "relative z-50 max-h-[--radix-select-content-available-height] min-w-36 overflow-x-hidden overflow-y-auto rounded-2xl bg-glass shadow-lg ring-1 ring-foreground/5 dark:ring-foreground/10 duration-100",
              "data-[state=open]:animate-in data-[state=open]:fade-in-0 data-[state=open]:zoom-in-95",
              "data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=closed]:zoom-out-95",
              "data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2"
            ),
            children: [
              /* @__PURE__ */ i(kn, { className: "flex cursor-default items-center justify-center bg-card py-1", children: /* @__PURE__ */ i(To, { size: 16 }) }),
              /* @__PURE__ */ i(Rn, { className: "p-1.5", children: t }),
              /* @__PURE__ */ i(Tn, { className: "flex cursor-default items-center justify-center bg-card py-1", children: /* @__PURE__ */ i(Me, { size: 16 }) })
            ]
          }
        ) })
      ] }, m),
      a && h && /* @__PURE__ */ i(
        "button",
        {
          type: "button",
          onClick: v,
          disabled: u,
          className: "absolute right-2 top-1/2 -translate-y-1/2 p-1 text-muted-foreground hover:text-foreground cursor-pointer bg-transparent border-none rounded-sm transition-colors",
          children: /* @__PURE__ */ i(je, { size: 14 })
        }
      )
    ] })
  ] });
}
function Ir({ children: e, className: o, ...t }) {
  return /* @__PURE__ */ R(
    En,
    {
      className: P(
        "flex w-full items-center justify-between gap-2 rounded-xl py-2 px-3 text-sm font-medium cursor-pointer select-none outline-none transition-colors",
        "focus:bg-accent focus:text-accent-foreground",
        "data-[disabled]:opacity-50 data-[disabled]:pointer-events-none",
        o
      ),
      ...t,
      children: [
        /* @__PURE__ */ i(In, { children: /* @__PURE__ */ i("span", { className: "flex items-center gap-2", children: e }) }),
        /* @__PURE__ */ i(_n, { className: "flex items-center", children: /* @__PURE__ */ i(Qe, { className: "shrink-0 size-4" }) })
      ]
    }
  );
}
function _r({ children: e, label: o, ...t }) {
  return /* @__PURE__ */ R(Nn, { ...t, children: [
    o && /* @__PURE__ */ i(Pn, { className: "px-3 py-2.5 text-xs text-muted-foreground", children: o }),
    e
  ] });
}
function kr({ label: e, description: o, checked: t, onChange: r, disabled: n, labelPosition: a = "right", className: s }) {
  const c = /* @__PURE__ */ i(
    Fo,
    {
      checked: t,
      onCheckedChange: r,
      disabled: n,
      className: P(
        "group relative inline-flex h-5 w-9 shrink-0 cursor-pointer items-center rounded-full border-[3px] border-transparent transition-colors duration-200 ease-out",
        "focus-visible:outline-none focus-visible:border-ring focus-visible:ring-3 focus-visible:ring-ring/50",
        "data-[state=checked]:bg-primary data-[state=unchecked]:bg-secondary",
        "disabled:cursor-not-allowed disabled:opacity-50"
      ),
      children: /* @__PURE__ */ i(
        $o,
        {
          className: P(
            "pointer-events-none block h-3.5 w-3.5 rounded-full shadow-sm transition-transform duration-200 ease-out",
            "data-[state=checked]:translate-x-4 data-[state=checked]:bg-primary-foreground",
            "data-[state=unchecked]:translate-x-0 data-[state=unchecked]:bg-muted-foreground"
          )
        }
      )
    }
  ), d = e || o ? /* @__PURE__ */ R("div", { className: "flex flex-col gap-0.5", children: [
    e && /* @__PURE__ */ i("span", { className: P("text-sm text-foreground", n && "opacity-50"), children: e }),
    o && /* @__PURE__ */ i("span", { className: P("text-xs text-muted-foreground", n && "opacity-50"), children: o })
  ] }) : null;
  return /* @__PURE__ */ R("label", { className: P("inline-flex items-center gap-3", n ? "cursor-not-allowed" : "cursor-pointer", s), children: [
    a === "left" && d,
    c,
    a === "right" && d
  ] });
}
function Tr({ label: e, description: o, checked: t, onChange: r, disabled: n, labelPosition: a = "right", className: s }) {
  const c = /* @__PURE__ */ i(
    Bt,
    {
      checked: t,
      onCheckedChange: r,
      disabled: n,
      className: P(
        "mt-0.5 h-4 w-4 shrink-0 rounded-[5px] border border-border bg-input/90 transition-[color,box-shadow,border-color]",
        "focus-visible:outline-none focus-visible:border-ring focus-visible:ring-3 focus-visible:ring-ring/50",
        "data-[state=checked]:bg-primary data-[state=checked]:border-primary",
        "disabled:cursor-not-allowed disabled:opacity-50"
      ),
      children: /* @__PURE__ */ i(Vt, { className: "flex items-center justify-center text-primary-foreground", children: /* @__PURE__ */ i(Qe, { size: 12, strokeWidth: 3 }) })
    }
  ), d = e || o ? /* @__PURE__ */ R("div", { className: "flex flex-col gap-0.5", children: [
    e && /* @__PURE__ */ i("span", { className: P("text-sm text-foreground", n && "opacity-50"), children: e }),
    o && /* @__PURE__ */ i("span", { className: P("text-xs text-muted-foreground", n && "opacity-50"), children: o })
  ] }) : null;
  return /* @__PURE__ */ R("label", { className: P("inline-flex items-start gap-3", n ? "cursor-not-allowed" : "cursor-pointer", s), children: [
    a === "left" && d,
    c,
    a === "right" && d
  ] });
}
function Mr({ label: e, value: o, onValueChange: t, disabled: r, className: n, children: a }) {
  return /* @__PURE__ */ R("div", { className: "flex flex-col gap-1", children: [
    e && /* @__PURE__ */ i(ae, { children: e }),
    /* @__PURE__ */ i(
      Gn,
      {
        value: o,
        onValueChange: t,
        disabled: r,
        className: P("flex flex-col gap-2", n),
        children: a
      }
    )
  ] });
}
function Dr({ value: e, label: o, description: t, disabled: r, className: n }) {
  return /* @__PURE__ */ R("label", { className: P("inline-flex items-start gap-3", r ? "cursor-not-allowed" : "cursor-pointer", n), children: [
    /* @__PURE__ */ i(
      Wn,
      {
        value: e,
        disabled: r,
        className: P(
          "mt-0.5 h-4 w-4 shrink-0 rounded-full border border-border bg-input/90 transition-[color,box-shadow,border-color]",
          "focus-visible:outline-none focus-visible:border-ring focus-visible:ring-3 focus-visible:ring-ring/50",
          "data-[state=checked]:border-primary",
          "disabled:cursor-not-allowed disabled:opacity-50"
        ),
        children: /* @__PURE__ */ i(jn, { className: "flex items-center justify-center", children: /* @__PURE__ */ i("span", { className: "block h-2 w-2 rounded-full bg-primary" }) })
      }
    ),
    (o || t) && /* @__PURE__ */ R("div", { className: "flex flex-col gap-0.5", children: [
      o && /* @__PURE__ */ i("span", { className: P("text-sm text-foreground", r && "opacity-50"), children: o }),
      t && /* @__PURE__ */ i("span", { className: P("text-xs text-muted-foreground", r && "opacity-50"), children: t })
    ] })
  ] });
}
function Ar({ label: e, value: o, onValueChange: t, min: r = 0, max: n = 100, step: a = 1, disabled: s, renderValue: c, className: d, ...u }) {
  const m = c || String, f = o !== void 0 ? o.length > 1 ? `${m(o[0])} – ${m(o[1])}` : m(o[0]) : null;
  return /* @__PURE__ */ R("div", { className: P("flex flex-col gap-2", d), children: [
    (e || f) && /* @__PURE__ */ R("div", { className: "flex items-center justify-between", children: [
      e && /* @__PURE__ */ i(ae, { children: e }),
      f && /* @__PURE__ */ i("span", { className: "text-sm text-muted-foreground font-mono", children: f })
    ] }),
    /* @__PURE__ */ R(
      lr,
      {
        value: o,
        onValueChange: t,
        min: r,
        max: n,
        step: a,
        disabled: s,
        className: P(
          "relative flex w-full touch-none select-none items-center",
          s && "opacity-50 cursor-not-allowed"
        ),
        ...u,
        children: [
          /* @__PURE__ */ i(dr, { className: "relative h-1.5 w-full grow overflow-hidden rounded-full bg-secondary", children: /* @__PURE__ */ i(ur, { className: "absolute h-full bg-primary" }) }),
          (o || [0]).map((h, v) => /* @__PURE__ */ i(
            fr,
            {
              className: P(
                "block h-4 w-4 rounded-full border-2 border-primary bg-background shadow-md transition-[color,box-shadow,border-color]",
                "focus-visible:outline-none focus-visible:border-ring focus-visible:ring-3 focus-visible:ring-ring/50",
                !s && "cursor-grab active:cursor-grabbing"
              )
            },
            v
          ))
        ]
      }
    )
  ] });
}
function Or({ label: e, error: o, icon: t, iconPosition: r = "left", prefix: n, suffix: a, className: s, ...c }) {
  const d = t && r === "left", u = t && r === "right";
  return /* @__PURE__ */ R("div", { className: "flex flex-col gap-1", children: [
    e && /* @__PURE__ */ i(ae, { children: e }),
    t || n || a ? /* @__PURE__ */ R(
      "div",
      {
        className: P(
          "flex items-start gap-2 bg-input/50 border rounded-xl pl-3 text-sm transition-[color,box-shadow,border-color,background-color]",
          "focus-within:border-ring focus-within:ring-3 focus-within:ring-ring/50",
          "has-[textarea:disabled]:bg-muted-foreground/10 has-[textarea:disabled]:cursor-not-allowed has-[textarea:disabled]:border-dashed has-[textarea:disabled]:border-muted-foreground/30 has-[textarea:disabled]:[&_textarea]:text-muted-foreground has-[textarea:disabled]:[&_*]:text-muted-foreground",
          o ? "border-destructive" : "border-transparent"
        ),
        children: [
          d && /* @__PURE__ */ i(t, { size: 16, className: "text-muted-foreground shrink-0 mt-2.5" }),
          n && /* @__PURE__ */ i("span", { className: "text-muted-foreground shrink-0 mt-2", children: n }),
          /* @__PURE__ */ i(
            "textarea",
            {
              className: P(
                "flex-1 min-w-0 bg-transparent py-2 pr-3 text-foreground placeholder:text-muted-foreground outline-none resize-y min-h-[100px]",
                s
              ),
              ...c
            }
          ),
          a && /* @__PURE__ */ i("span", { className: "text-muted-foreground shrink-0 mt-2", children: a }),
          u && /* @__PURE__ */ i(t, { size: 16, className: "text-muted-foreground shrink-0 mt-2.5" })
        ]
      }
    ) : /* @__PURE__ */ i(
      "textarea",
      {
        className: P(
          "w-full bg-input/50 border rounded-xl px-3 py-2 text-sm text-foreground placeholder:text-muted-foreground outline-none transition-[color,box-shadow,border-color,background-color] resize-y min-h-[100px]",
          "focus:border-ring focus:ring-3 focus:ring-ring/50",
          "disabled:bg-muted-foreground/10 disabled:cursor-not-allowed disabled:border-dashed disabled:border-muted-foreground/30 disabled:text-muted-foreground",
          o ? "border-destructive" : "border-transparent",
          s
        ),
        ...c
      }
    ),
    o && /* @__PURE__ */ i("span", { className: "text-xs text-destructive", children: o })
  ] });
}
export {
  Tr as Checkbox,
  Pr as Input,
  ae as Label,
  Mr as RadioGroup,
  Dr as RadioGroupItem,
  Er as Select,
  _r as SelectGroup,
  Ir as SelectItem,
  Ar as Slider,
  Or as Textarea,
  kr as Toggle
};
