// @btcv/ui/dist/LogViewer.js
"use client";
import { jsxs as p, jsx as r } from "react/jsx-runtime";
import * as u from "react";
import { useState as v } from "react";
import { c as f } from "./_shared/utils-eGXXUFl7.js";
import { Check as w, Copy as y } from "lucide-react";
const N = {
  info: "bg-info/15 text-info",
  warn: "bg-warning/15 text-warning",
  error: "bg-destructive/15 text-destructive",
  debug: "bg-muted text-muted-foreground"
};
function C({ level: t = "info", time: n, children: s, className: l }) {
  const [a, d] = u.useState(!1), b = () => {
    const o = [t.toUpperCase(), n, typeof s == "string" ? s : ""].filter(Boolean).join(" ");
    navigator.clipboard.writeText(o), d(!0), setTimeout(() => d(!1), 1500);
  };
  return /* @__PURE__ */ p(
    "div",
    {
      className: f(
        "group relative flex items-baseline gap-3 px-3 py-1.5 font-mono text-xs transition-colors hover:bg-primary/8",
        l
      ),
      children: [
        /* @__PURE__ */ r(
          "span",
          {
            className: f(
              "inline-flex items-center px-1.5 py-0.5 rounded text-[10px] font-bold uppercase shrink-0",
              N[t]
            ),
            children: t.toUpperCase()
          }
        ),
        n && /* @__PURE__ */ r("span", { className: "text-muted-foreground shrink-0", children: n }),
        /* @__PURE__ */ r("span", { className: "text-foreground flex-1", children: s }),
        /* @__PURE__ */ r(
          "button",
          {
            type: "button",
            onClick: b,
            className: "opacity-0 group-hover:opacity-100 shrink-0 p-0.5 rounded text-muted-foreground hover:text-foreground transition-opacity cursor-pointer bg-transparent border-none",
            children: a ? /* @__PURE__ */ r(w, { size: 12, className: "text-success" }) : /* @__PURE__ */ r(y, { size: 12 })
          }
        )
      ]
    }
  );
}
function B({
  logs: t,
  maxHeight: n = "400px",
  filterable: s = !0,
  autoScroll: l = !0,
  title: a,
  className: d,
  ...b
}) {
  const [o, g] = v("all"), c = u.useRef(null), x = u.useMemo(() => {
    const e = { all: t.length, info: 0, warn: 0, error: 0, debug: 0 };
    for (const i of t)
      e[i.level || "info"]++;
    return e;
  }, [t]), m = o === "all" ? t : t.filter((e) => (e.level || "info") === o);
  u.useEffect(() => {
    l && c.current && (c.current.scrollTop = c.current.scrollHeight);
  }, [t, l, o]);
  const h = [
    { key: "all", label: "ALL" },
    { key: "info", label: "INFO" },
    { key: "warn", label: "WARN" },
    { key: "error", label: "ERROR" },
    { key: "debug", label: "DEBUG" }
  ];
  return /* @__PURE__ */ p(
    "div",
    {
      className: f(
        "rounded-2xl border border-border bg-glass overflow-hidden",
        d
      ),
      ...b,
      children: [
        (a || s) && /* @__PURE__ */ p("div", { className: "flex items-center justify-between gap-3 px-4 py-3 border-b border-border", children: [
          a && /* @__PURE__ */ r("span", { className: "text-sm font-medium text-foreground", children: a }),
          s && /* @__PURE__ */ r("div", { className: "flex gap-1.5 flex-wrap", children: h.map(({ key: e, label: i }) => /* @__PURE__ */ p(
            "button",
            {
              type: "button",
              onClick: () => g(e),
              className: f(
                "inline-flex items-center gap-1 px-2 py-0.5 rounded-full text-[10px] font-semibold uppercase tracking-wide transition-all cursor-pointer border",
                e === "all" ? "text-primary" + (o === e ? " bg-primary/15 border-primary/30" : " border-transparent hover:bg-primary/10") : e === "info" ? "text-info" + (o === e ? " bg-info/15 border-info/30" : " border-transparent hover:bg-info/10") : e === "warn" ? "text-warning" + (o === e ? " bg-warning/15 border-warning/30" : " border-transparent hover:bg-warning/10") : e === "error" ? "text-destructive" + (o === e ? " bg-destructive/15 border-destructive/30" : " border-transparent hover:bg-destructive/10") : "text-muted-foreground" + (o === e ? " bg-muted border-border" : " border-transparent hover:bg-muted")
              ),
              children: [
                i,
                /* @__PURE__ */ r("span", { className: "font-mono", children: x[e] })
              ]
            },
            e
          )) })
        ] }),
        /* @__PURE__ */ r(
          "div",
          {
            ref: c,
            className: "overflow-y-auto bg-muted/30",
            style: { maxHeight: typeof n == "number" ? `${n}px` : n },
            children: m.length === 0 ? /* @__PURE__ */ r("div", { className: "px-4 py-8 text-center text-sm text-muted-foreground", children: "Aucun log" }) : m.map((e, i) => /* @__PURE__ */ r(C, { level: e.level, time: e.time, children: e.message }, i))
          }
        )
      ]
    }
  );
}
export {
  C as LogLine,
  B as LogViewer
};
