// @btcv/ui/dist/Menubar.js
"use client";
import { jsx as t } from "react/jsx-runtime";
import * as s from "react";
import { u as W, R as X, I as Y, c as O, a as q } from "./_shared/index-C0gNQvxa.js";
import { u as J, c as v, a as Q } from "./_shared/index-CXeb1OMI.js";
import { u as Z } from "./_shared/index-DLcqcWxM.js";
import { u as y } from "./_shared/index-BC7vfx-u.js";
import { A as ee, P as re, C as ae, I as ne, S as oe, L as te, c as se, G as ue, R as ie, a as ce, b as de, d as le, e as pe, f as be, g as fe, h as me } from "./_shared/index-CukUn3R0.js";
import { P as G } from "./_shared/index-V-Ajw7Ac.js";
import { c as R } from "./_shared/utils-eGXXUFl7.js";
var _ = "Menubar", [I, ge, Me] = q(_), [D] = Q(_, [
  Me,
  O
]), d = se(), k = O(), [ve, S] = D(_), K = s.forwardRef(
  (e, a) => {
    const {
      __scopeMenubar: r,
      value: n,
      onValueChange: o,
      defaultValue: h,
      loop: p = !0,
      dir: c,
      ...i
    } = e, b = W(c), u = k(r), [m, f] = J({
      prop: n,
      onChange: o,
      defaultProp: h ?? "",
      caller: _
    }), [g, l] = s.useState(null);
    return /* @__PURE__ */ t(
      ve,
      {
        scope: r,
        value: m,
        onMenuOpen: s.useCallback(
          (M) => {
            f(M), l(M);
          },
          [f]
        ),
        onMenuClose: s.useCallback(() => f(""), [f]),
        onMenuToggle: s.useCallback(
          (M) => {
            f((P) => P ? "" : M), l(M);
          },
          [f]
        ),
        dir: b,
        loop: p,
        children: /* @__PURE__ */ t(I.Provider, { scope: r, children: /* @__PURE__ */ t(I.Slot, { scope: r, children: /* @__PURE__ */ t(
          X,
          {
            asChild: !0,
            ...u,
            orientation: "horizontal",
            loop: p,
            dir: b,
            currentTabStopId: g,
            onCurrentTabStopIdChange: l,
            children: /* @__PURE__ */ t(G.div, { role: "menubar", ...i, ref: a })
          }
        ) }) })
      }
    );
  }
);
K.displayName = _;
var A = "MenubarMenu", [he, F] = D(A), L = (e) => {
  const { __scopeMenubar: a, value: r, ...n } = e, o = y(), h = r || o || "LEGACY_REACT_AUTO_VALUE", p = S(A, a), c = d(a), i = s.useRef(null), b = s.useRef(!1), u = p.value === h;
  return s.useEffect(() => {
    u || (b.current = !1);
  }, [u]), /* @__PURE__ */ t(
    he,
    {
      scope: a,
      value: h,
      triggerId: y(),
      triggerRef: i,
      contentId: y(),
      wasKeyboardTriggerOpenRef: b,
      children: /* @__PURE__ */ t(
        ie,
        {
          ...c,
          open: u,
          onOpenChange: (m) => {
            m || p.onMenuClose();
          },
          modal: !1,
          dir: p.dir,
          ...n
        }
      )
    }
  );
};
L.displayName = A;
var N = "MenubarTrigger", $ = s.forwardRef(
  (e, a) => {
    const { __scopeMenubar: r, disabled: n = !1, ...o } = e, h = k(r), p = d(r), c = S(N, r), i = F(N, r), b = s.useRef(null), u = Z(a, b, i.triggerRef), [m, f] = s.useState(!1), g = c.value === i.value;
    return /* @__PURE__ */ t(I.ItemSlot, { scope: r, value: i.value, disabled: n, children: /* @__PURE__ */ t(
      Y,
      {
        asChild: !0,
        ...h,
        focusable: !n,
        tabStopId: i.value,
        children: /* @__PURE__ */ t(ee, { asChild: !0, ...p, children: /* @__PURE__ */ t(
          G.button,
          {
            type: "button",
            role: "menuitem",
            id: i.triggerId,
            "aria-haspopup": "menu",
            "aria-expanded": g,
            "aria-controls": g ? i.contentId : void 0,
            "data-highlighted": m ? "" : void 0,
            "data-state": g ? "open" : "closed",
            "data-disabled": n ? "" : void 0,
            disabled: n,
            ...o,
            ref: u,
            onPointerDown: v(e.onPointerDown, (l) => {
              !n && l.button === 0 && l.ctrlKey === !1 && (c.onMenuOpen(i.value), g || l.preventDefault());
            }),
            onPointerEnter: v(e.onPointerEnter, () => {
              c.value && !g && (c.onMenuOpen(i.value), b.current?.focus());
            }),
            onKeyDown: v(e.onKeyDown, (l) => {
              n || (["Enter", " "].includes(l.key) && c.onMenuToggle(i.value), l.key === "ArrowDown" && c.onMenuOpen(i.value), ["Enter", " ", "ArrowDown"].includes(l.key) && (i.wasKeyboardTriggerOpenRef.current = !0, l.preventDefault()));
            }),
            onFocus: v(e.onFocus, () => f(!0)),
            onBlur: v(e.onBlur, () => f(!1))
          }
        ) })
      }
    ) });
  }
);
$.displayName = N;
var xe = "MenubarPortal", B = (e) => {
  const { __scopeMenubar: a, ...r } = e, n = d(a);
  return /* @__PURE__ */ t(re, { ...n, ...r });
};
B.displayName = xe;
var C = "MenubarContent", U = s.forwardRef(
  (e, a) => {
    const { __scopeMenubar: r, align: n = "start", ...o } = e, h = d(r), p = S(C, r), c = F(C, r), i = ge(r), b = s.useRef(!1);
    return /* @__PURE__ */ t(
      ae,
      {
        id: c.contentId,
        "aria-labelledby": c.triggerId,
        "data-radix-menubar-content": "",
        ...h,
        ...o,
        ref: a,
        align: n,
        onCloseAutoFocus: v(e.onCloseAutoFocus, (u) => {
          !p.value && !b.current && c.triggerRef.current?.focus(), b.current = !1, u.preventDefault();
        }),
        onFocusOutside: v(e.onFocusOutside, (u) => {
          const m = u.target;
          i().some((g) => g.ref.current?.contains(m)) && u.preventDefault();
        }),
        onInteractOutside: v(e.onInteractOutside, () => {
          b.current = !0;
        }),
        onEntryFocus: (u) => {
          c.wasKeyboardTriggerOpenRef.current || u.preventDefault();
        },
        onKeyDown: v(
          e.onKeyDown,
          (u) => {
            if (["ArrowRight", "ArrowLeft"].includes(u.key)) {
              const m = u.target, f = m.hasAttribute("data-radix-menubar-subtrigger"), g = m.closest("[data-radix-menubar-content]") !== u.currentTarget, M = (p.dir === "rtl" ? "ArrowRight" : "ArrowLeft") === u.key;
              if (!M && f || g && M) return;
              let x = i().filter((w) => !w.disabled).map((w) => w.value);
              M && x.reverse();
              const E = x.indexOf(c.value);
              x = p.loop ? Le(x, E + 1) : x.slice(E + 1);
              const [T] = x;
              T && p.onMenuOpen(T);
            }
          },
          { checkForDefaultPrevented: !1 }
        ),
        style: {
          ...e.style,
          "--radix-menubar-content-transform-origin": "var(--radix-popper-transform-origin)",
          "--radix-menubar-content-available-width": "var(--radix-popper-available-width)",
          "--radix-menubar-content-available-height": "var(--radix-popper-available-height)",
          "--radix-menubar-trigger-width": "var(--radix-popper-anchor-width)",
          "--radix-menubar-trigger-height": "var(--radix-popper-anchor-height)"
        }
      }
    );
  }
);
U.displayName = C;
var Re = "MenubarGroup", V = s.forwardRef(
  (e, a) => {
    const { __scopeMenubar: r, ...n } = e, o = d(r);
    return /* @__PURE__ */ t(ue, { ...o, ...n, ref: a });
  }
);
V.displayName = Re;
var _e = "MenubarLabel", z = s.forwardRef(
  (e, a) => {
    const { __scopeMenubar: r, ...n } = e, o = d(r);
    return /* @__PURE__ */ t(te, { ...o, ...n, ref: a });
  }
);
z.displayName = _e;
var we = "MenubarItem", H = s.forwardRef(
  (e, a) => {
    const { __scopeMenubar: r, ...n } = e, o = d(r);
    return /* @__PURE__ */ t(ne, { ...o, ...n, ref: a });
  }
);
H.displayName = we;
var ye = "MenubarCheckboxItem", Ie = s.forwardRef(
  (e, a) => {
    const { __scopeMenubar: r, ...n } = e, o = d(r);
    return /* @__PURE__ */ t(ce, { ...o, ...n, ref: a });
  }
);
Ie.displayName = ye;
var Ne = "MenubarRadioGroup", Ce = s.forwardRef(
  (e, a) => {
    const { __scopeMenubar: r, ...n } = e, o = d(r);
    return /* @__PURE__ */ t(de, { ...o, ...n, ref: a });
  }
);
Ce.displayName = Ne;
var Se = "MenubarRadioItem", Ae = s.forwardRef(
  (e, a) => {
    const { __scopeMenubar: r, ...n } = e, o = d(r);
    return /* @__PURE__ */ t(le, { ...o, ...n, ref: a });
  }
);
Ae.displayName = Se;
var Pe = "MenubarItemIndicator", Ee = s.forwardRef((e, a) => {
  const { __scopeMenubar: r, ...n } = e, o = d(r);
  return /* @__PURE__ */ t(pe, { ...o, ...n, ref: a });
});
Ee.displayName = Pe;
var Te = "MenubarSeparator", j = s.forwardRef(
  (e, a) => {
    const { __scopeMenubar: r, ...n } = e, o = d(r);
    return /* @__PURE__ */ t(oe, { ...o, ...n, ref: a });
  }
);
j.displayName = Te;
var Oe = "MenubarArrow", Ge = s.forwardRef(
  (e, a) => {
    const { __scopeMenubar: r, ...n } = e, o = d(r);
    return /* @__PURE__ */ t(be, { ...o, ...n, ref: a });
  }
);
Ge.displayName = Oe;
var De = "MenubarSubTrigger", ke = s.forwardRef(
  (e, a) => {
    const { __scopeMenubar: r, ...n } = e, o = d(r);
    return /* @__PURE__ */ t(
      fe,
      {
        "data-radix-menubar-subtrigger": "",
        ...o,
        ...n,
        ref: a
      }
    );
  }
);
ke.displayName = De;
var Ke = "MenubarSubContent", Fe = s.forwardRef(
  (e, a) => {
    const { __scopeMenubar: r, ...n } = e, o = d(r);
    return /* @__PURE__ */ t(
      me,
      {
        ...o,
        "data-radix-menubar-content": "",
        ...n,
        ref: a,
        style: {
          ...e.style,
          "--radix-menubar-content-transform-origin": "var(--radix-popper-transform-origin)",
          "--radix-menubar-content-available-width": "var(--radix-popper-available-width)",
          "--radix-menubar-content-available-height": "var(--radix-popper-available-height)",
          "--radix-menubar-trigger-width": "var(--radix-popper-anchor-width)",
          "--radix-menubar-trigger-height": "var(--radix-popper-anchor-height)"
        }
      }
    );
  }
);
Fe.displayName = Ke;
function Le(e, a) {
  return e.map((r, n) => e[(a + n) % e.length]);
}
var $e = K, Be = L, Ue = $, Ve = B, ze = U, He = V, je = z, We = H, Xe = j;
const Ye = s.forwardRef(({ className: e, ...a }, r) => /* @__PURE__ */ t(
  $e,
  {
    ref: r,
    className: R(
      "flex items-center gap-1 border border-border rounded-3xl bg-card p-1",
      e
    ),
    ...a
  }
));
Ye.displayName = "Menubar";
const dr = Be, qe = s.forwardRef(({ className: e, ...a }, r) => /* @__PURE__ */ t(
  Ue,
  {
    ref: r,
    className: R(
      "flex items-center gap-1.5 px-3 py-1.5 text-sm font-medium rounded-2xl text-foreground cursor-pointer select-none outline-none transition-colors",
      "hover:bg-muted hover:text-foreground",
      "data-[state=open]:bg-muted data-[state=open]:text-foreground",
      "focus-visible:border-ring focus-visible:ring-3 focus-visible:ring-ring/50",
      e
    ),
    ...a
  }
));
qe.displayName = "MenubarTrigger";
const Je = s.forwardRef(({ className: e, sideOffset: a = 8, alignOffset: r = -4, ...n }, o) => /* @__PURE__ */ t(Ve, { children: /* @__PURE__ */ t(
  ze,
  {
    ref: o,
    sideOffset: a,
    alignOffset: r,
    className: R(
      "z-50 min-w-[180px] overflow-hidden rounded-3xl border border-border bg-glass p-1 shadow-lg",
      "animate-in fade-in-0 zoom-in-95",
      "data-[side=bottom]:slide-in-from-top-2",
      "data-[side=top]:slide-in-from-bottom-2",
      "data-[side=left]:slide-in-from-right-2",
      "data-[side=right]:slide-in-from-left-2",
      e
    ),
    ...n
  }
) }));
Je.displayName = "MenubarContent";
const Qe = s.forwardRef(({ className: e, inset: a, ...r }, n) => /* @__PURE__ */ t(
  We,
  {
    ref: n,
    className: R(
      "relative flex items-center gap-2 rounded-2xl px-2 py-1.5 text-sm text-foreground cursor-pointer select-none outline-none transition-colors",
      "data-[highlighted]:bg-primary/10 data-[highlighted]:text-foreground",
      "data-[disabled]:opacity-50 data-[disabled]:pointer-events-none",
      "[&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 [&_svg]:shrink-0",
      a && "pl-8",
      e
    ),
    ...r
  }
));
Qe.displayName = "MenubarItem";
const Ze = s.forwardRef(({ className: e, ...a }, r) => /* @__PURE__ */ t(
  Xe,
  {
    ref: r,
    className: R("-mx-1 my-1 h-px bg-border", e),
    ...a
  }
));
Ze.displayName = "MenubarSeparator";
const er = s.forwardRef(({ className: e, inset: a, ...r }, n) => /* @__PURE__ */ t(
  je,
  {
    ref: n,
    className: R(
      "px-2 py-1.5 text-xs font-semibold uppercase tracking-wider text-muted-foreground",
      a && "pl-8",
      e
    ),
    ...r
  }
));
er.displayName = "MenubarLabel";
const lr = He;
export {
  Ye as Menubar,
  Je as MenubarContent,
  lr as MenubarGroup,
  Qe as MenubarItem,
  er as MenubarLabel,
  dr as MenubarMenu,
  Ze as MenubarSeparator,
  qe as MenubarTrigger
};
