// @btcv/ui/dist/Pagination.js
import { jsx as a, jsxs as s } from "react/jsx-runtime";
import * as t from "react";
import { ChevronLeft as m, ChevronRight as c, MoreHorizontal as p } from "lucide-react";
import { c as r } from "./_shared/utils-eGXXUFl7.js";
function P({ className: n, ...e }) {
  return /* @__PURE__ */ a(
    "nav",
    {
      role: "navigation",
      "aria-label": "pagination",
      className: r("mx-auto flex w-full justify-center", n),
      ...e
    }
  );
}
const f = t.forwardRef(({ className: n, ...e }, i) => /* @__PURE__ */ a(
  "ul",
  {
    ref: i,
    className: r("flex flex-row items-center gap-1 list-none", n),
    ...e
  }
));
f.displayName = "PaginationContent";
const g = t.forwardRef(
  ({ className: n, ...e }, i) => /* @__PURE__ */ a("li", { ref: i, className: r("", n), ...e })
);
g.displayName = "PaginationItem";
const u = {
  sm: "h-7 min-w-7 text-xs rounded-xl",
  md: "h-9 min-w-9 text-sm rounded-2xl",
  lg: "h-11 min-w-11 text-base rounded-2xl"
}, l = t.forwardRef(
  ({ className: n, isActive: e = !1, size: i = "md", ...o }, d) => /* @__PURE__ */ a(
    "button",
    {
      ref: d,
      type: "button",
      "aria-current": e ? "page" : void 0,
      className: r(
        "inline-flex items-center justify-center font-medium transition-colors cursor-pointer select-none",
        "outline-none focus-visible:border-ring focus-visible:ring-3 focus-visible:ring-ring/50",
        "disabled:opacity-40 disabled:cursor-not-allowed disabled:pointer-events-none",
        u[i],
        e ? "bg-primary/15 text-primary border border-primary/20" : "bg-transparent text-muted-foreground border border-transparent hover:bg-secondary/50 hover:text-foreground",
        n
      ),
      ...o
    }
  )
);
l.displayName = "PaginationLink";
const x = t.forwardRef(({ className: n, label: e, ...i }, o) => /* @__PURE__ */ s(
  l,
  {
    ref: o,
    "aria-label": "Go to previous page",
    className: r("gap-1 px-2.5", n),
    ...i,
    children: [
      /* @__PURE__ */ a(m, { className: "size-4" }),
      e && /* @__PURE__ */ a("span", { children: e })
    ]
  }
));
x.displayName = "PaginationPrevious";
const b = t.forwardRef(({ className: n, label: e, ...i }, o) => /* @__PURE__ */ s(
  l,
  {
    ref: o,
    "aria-label": "Go to next page",
    className: r("gap-1 px-2.5", n),
    ...i,
    children: [
      e && /* @__PURE__ */ a("span", { children: e }),
      /* @__PURE__ */ a(c, { className: "size-4" })
    ]
  }
));
b.displayName = "PaginationNext";
function v({ className: n, ...e }) {
  return /* @__PURE__ */ s(
    "span",
    {
      "aria-hidden": !0,
      className: r(
        "flex h-9 w-9 items-center justify-center text-muted-foreground",
        n
      ),
      ...e,
      children: [
        /* @__PURE__ */ a(p, { className: "size-4" }),
        /* @__PURE__ */ a("span", { className: "sr-only", children: "More pages" })
      ]
    }
  );
}
export {
  P as Pagination,
  f as PaginationContent,
  v as PaginationEllipsis,
  g as PaginationItem,
  l as PaginationLink,
  b as PaginationNext,
  x as PaginationPrevious
};
