// @btcv/ui/dist/ProgressBar.js
import { jsx as l, jsxs as f } from "react/jsx-runtime";
import * as c from "react";
import "react-dom";
import { c as R } from "./_shared/index-BrLJJgkl.js";
import { c as h } from "./_shared/utils-eGXXUFl7.js";
function V(e, r = []) {
  let i = [];
  function o(u, s) {
    const n = c.createContext(s);
    n.displayName = u + "Context";
    const a = i.length;
    i = [...i, s];
    const d = (m) => {
      const { scope: g, children: x, ...p } = m, I = g?.[e]?.[a] || n, E = c.useMemo(() => p, Object.values(p));
      return /* @__PURE__ */ l(I.Provider, { value: E, children: x });
    };
    d.displayName = u + "Provider";
    function N(m, g) {
      const x = g?.[e]?.[a] || n, p = c.useContext(x);
      if (p) return p;
      if (s !== void 0) return s;
      throw new Error(`\`${m}\` must be used within \`${u}\``);
    }
    return [d, N];
  }
  const t = () => {
    const u = i.map((s) => c.createContext(s));
    return function(n) {
      const a = n?.[e] || u;
      return c.useMemo(
        () => ({ [`__scope${e}`]: { ...n, [e]: a } }),
        [n, a]
      );
    };
  };
  return t.scopeName = e, [o, A(t, ...r)];
}
function A(...e) {
  const r = e[0];
  if (e.length === 1) return r;
  const i = () => {
    const o = e.map((t) => ({
      useScope: t(),
      scopeName: t.scopeName
    }));
    return function(u) {
      const s = o.reduce((n, { useScope: a, scopeName: d }) => {
        const m = a(u)[`__scope${d}`];
        return { ...n, ...m };
      }, {});
      return c.useMemo(() => ({ [`__scope${r.scopeName}`]: s }), [s]);
    };
  };
  return i.scopeName = r.scopeName, i;
}
var D = [
  "a",
  "button",
  "div",
  "form",
  "h2",
  "h3",
  "img",
  "input",
  "label",
  "li",
  "nav",
  "ol",
  "p",
  "select",
  "span",
  "svg",
  "ul"
], $ = D.reduce((e, r) => {
  const i = R(`Primitive.${r}`), o = c.forwardRef((t, u) => {
    const { asChild: s, ...n } = t, a = s ? i : r;
    return typeof window < "u" && (window[/* @__PURE__ */ Symbol.for("radix-ui")] = !0), /* @__PURE__ */ l(a, { ...n, ref: u });
  });
  return o.displayName = `Primitive.${r}`, { ...e, [r]: o };
}, {}), P = "Progress", b = 100, [O] = V(P), [j, L] = O(P), y = c.forwardRef(
  (e, r) => {
    const {
      __scopeProgress: i,
      value: o = null,
      max: t,
      getValueLabel: u = k,
      ...s
    } = e;
    (t || t === 0) && !w(t) && console.error(T(`${t}`, "Progress"));
    const n = w(t) ? t : b;
    o !== null && !S(o, n) && console.error(B(`${o}`, "Progress"));
    const a = S(o, n) ? o : null, d = v(a) ? u(a, n) : void 0;
    return /* @__PURE__ */ l(j, { scope: i, value: a, max: n, children: /* @__PURE__ */ l(
      $.div,
      {
        "aria-valuemax": n,
        "aria-valuemin": 0,
        "aria-valuenow": v(a) ? a : void 0,
        "aria-valuetext": d,
        role: "progressbar",
        "data-state": M(a, n),
        "data-value": a ?? void 0,
        "data-max": n,
        ...s,
        ref: r
      }
    ) });
  }
);
y.displayName = P;
var C = "ProgressIndicator", _ = c.forwardRef(
  (e, r) => {
    const { __scopeProgress: i, ...o } = e, t = L(C, i);
    return /* @__PURE__ */ l(
      $.div,
      {
        "data-state": M(t.value, t.max),
        "data-value": t.value ?? void 0,
        "data-max": t.max,
        ...o,
        ref: r
      }
    );
  }
);
_.displayName = C;
function k(e, r) {
  return `${Math.round(e / r * 100)}%`;
}
function M(e, r) {
  return e == null ? "indeterminate" : e === r ? "complete" : "loading";
}
function v(e) {
  return typeof e == "number";
}
function w(e) {
  return v(e) && !isNaN(e) && e > 0;
}
function S(e, r) {
  return v(e) && !isNaN(e) && e <= r && e >= 0;
}
function T(e, r) {
  return `Invalid prop \`max\` of value \`${e}\` supplied to \`${r}\`. Only numbers greater than 0 are valid max values. Defaulting to \`${b}\`.`;
}
function B(e, r) {
  return `Invalid prop \`value\` of value \`${e}\` supplied to \`${r}\`. The \`value\` prop must be:
  - a positive number
  - less than the value passed to \`max\` (or ${b} if no \`max\` prop is set)
  - \`null\` or \`undefined\` if the progress is indeterminate.

Defaulting to \`null\`.`;
}
var G = y, z = _;
const F = {
  gold: "bg-primary",
  success: "bg-success",
  destructive: "bg-destructive",
  info: "bg-info"
}, H = {
  sm: "h-1",
  md: "h-1.5",
  lg: "h-2"
};
function K({ value: e = 0, variant: r = "gold", size: i = "md", indeterminate: o = !1, label: t, className: u }) {
  const s = Math.min(100, Math.max(0, e));
  return /* @__PURE__ */ f("div", { className: h("w-full", u), children: [
    t && /* @__PURE__ */ f("div", { className: "flex items-center justify-between mb-2", children: [
      /* @__PURE__ */ l("span", { className: "text-sm text-muted-foreground", children: t }),
      !o && /* @__PURE__ */ f("span", { className: "text-xs font-mono text-muted-foreground", children: [
        s,
        "%"
      ] })
    ] }),
    /* @__PURE__ */ f(
      G,
      {
        value: o ? 100 : s,
        className: h(
          "relative w-full overflow-hidden rounded-full bg-secondary",
          H[i]
        ),
        children: [
          /* @__PURE__ */ l(
            z,
            {
              className: h(
                "h-full rounded-full transition-[width] duration-500 ease-out",
                F[r]
              ),
              style: { width: o ? "100%" : `${s}%` }
            }
          ),
          o && /* @__PURE__ */ l(
            "div",
            {
              className: "absolute inset-0 rounded-full",
              style: {
                background: "linear-gradient(90deg, transparent 0%, rgba(255,255,255,0.6) 50%, transparent 100%)",
                animation: "shimmer 1.5s ease-in-out infinite"
              }
            }
          )
        ]
      }
    )
  ] });
}
export {
  K as ProgressBar
};
