// @btcv/ui/dist/Resizable.js
"use client";
import { jsx as we } from "react/jsx-runtime";
import { GripVertical as Rn } from "lucide-react";
import * as Ln from "react";
import { createContext as An, forwardRef as en, createElement as de, useContext as nn, useRef as F, useImperativeHandle as tn, useState as ue, useEffect as te, useCallback as T, useMemo as Dn, useLayoutEffect as Mn } from "react";
import { c as rn } from "./_shared/utils-eGXXUFl7.js";
const Re = An(null);
Re.displayName = "PanelGroupContext";
const A = {
  group: "data-panel-group",
  groupDirection: "data-panel-group-direction",
  groupId: "data-panel-group-id",
  panel: "data-panel",
  panelCollapsible: "data-panel-collapsible",
  panelId: "data-panel-id",
  panelSize: "data-panel-size",
  resizeHandle: "data-resize-handle",
  resizeHandleActive: "data-resize-handle-active",
  resizeHandleEnabled: "data-panel-resize-handle-enabled",
  resizeHandleId: "data-panel-resize-handle-id",
  resizeHandleState: "data-resize-handle-state"
}, Te = 10, re = Mn, Oe = Ln.useId, Hn = typeof Oe == "function" ? Oe : () => null;
let Nn = 0;
function Fe(e = null) {
  const n = Hn(), t = F(e || n || null);
  return t.current === null && (t.current = "" + Nn++), e ?? t.current;
}
function on({
  children: e,
  className: n = "",
  collapsedSize: t,
  collapsible: r,
  defaultSize: i,
  forwardedRef: o,
  id: l,
  maxSize: a,
  minSize: s,
  onCollapse: m,
  onExpand: P,
  onResize: f,
  order: u,
  style: v,
  tagName: h = "div",
  ...I
}) {
  const w = nn(Re);
  if (w === null)
    throw Error("Panel components must be rendered within a PanelGroup container");
  const {
    collapsePanel: S,
    expandPanel: R,
    getPanelSize: M,
    getPanelStyle: N,
    groupId: q,
    isPanelCollapsed: C,
    reevaluatePanelConstraints: b,
    registerPanel: K,
    resizePanel: J,
    unregisterPanel: V
  } = w, _ = Fe(l), D = F({
    callbacks: {
      onCollapse: m,
      onExpand: P,
      onResize: f
    },
    constraints: {
      collapsedSize: t,
      collapsible: r,
      defaultSize: i,
      maxSize: a,
      minSize: s
    },
    id: _,
    idIsFromProps: l !== void 0,
    order: u
  });
  F({
    didLogMissingDefaultSizeWarning: !1
  }), re(() => {
    const {
      callbacks: k,
      constraints: $
    } = D.current, G = {
      ...$
    };
    D.current.id = _, D.current.idIsFromProps = l !== void 0, D.current.order = u, k.onCollapse = m, k.onExpand = P, k.onResize = f, $.collapsedSize = t, $.collapsible = r, $.defaultSize = i, $.maxSize = a, $.minSize = s, (G.collapsedSize !== $.collapsedSize || G.collapsible !== $.collapsible || G.maxSize !== $.maxSize || G.minSize !== $.minSize) && b(D.current, G);
  }), re(() => {
    const k = D.current;
    return K(k), () => {
      V(k);
    };
  }, [u, _, K, V]), tn(o, () => ({
    collapse: () => {
      S(D.current);
    },
    expand: (k) => {
      R(D.current, k);
    },
    getId() {
      return _;
    },
    getSize() {
      return M(D.current);
    },
    isCollapsed() {
      return C(D.current);
    },
    isExpanded() {
      return !C(D.current);
    },
    resize: (k) => {
      J(D.current, k);
    }
  }), [S, R, M, C, _, J]);
  const Q = N(D.current, i);
  return de(h, {
    ...I,
    children: e,
    className: n,
    id: _,
    style: {
      ...Q,
      ...v
    },
    // CSS selectors
    [A.groupId]: q,
    [A.panel]: "",
    [A.panelCollapsible]: r || void 0,
    [A.panelId]: _,
    [A.panelSize]: parseFloat("" + Q.flexGrow).toFixed(1)
  });
}
const ln = en((e, n) => de(on, {
  ...e,
  forwardedRef: n
}));
on.displayName = "Panel";
ln.displayName = "forwardRef(Panel)";
let Be = null, Ie = -1, Z = null;
function $n(e, n) {
  if (n) {
    const t = (n & dn) !== 0, r = (n & fn) !== 0, i = (n & pn) !== 0, o = (n & gn) !== 0;
    if (t)
      return i ? "se-resize" : o ? "ne-resize" : "e-resize";
    if (r)
      return i ? "sw-resize" : o ? "nw-resize" : "w-resize";
    if (i)
      return "s-resize";
    if (o)
      return "n-resize";
  }
  switch (e) {
    case "horizontal":
      return "ew-resize";
    case "intersection":
      return "move";
    case "vertical":
      return "ns-resize";
  }
}
function kn() {
  Z !== null && (document.head.removeChild(Z), Be = null, Z = null, Ie = -1);
}
function Ne(e, n) {
  var t, r;
  const i = $n(e, n);
  if (Be !== i) {
    if (Be = i, Z === null && (Z = document.createElement("style"), document.head.appendChild(Z)), Ie >= 0) {
      var o;
      (o = Z.sheet) === null || o === void 0 || o.removeRule(Ie);
    }
    Ie = (t = (r = Z.sheet) === null || r === void 0 ? void 0 : r.insertRule(`*{cursor: ${i} !important;}`)) !== null && t !== void 0 ? t : -1;
  }
}
function an(e) {
  return e.type === "keydown";
}
function sn(e) {
  return e.type.startsWith("pointer");
}
function un(e) {
  return e.type.startsWith("mouse");
}
function Le(e) {
  if (sn(e)) {
    if (e.isPrimary)
      return {
        x: e.clientX,
        y: e.clientY
      };
  } else if (un(e))
    return {
      x: e.clientX,
      y: e.clientY
    };
  return {
    x: 1 / 0,
    y: 1 / 0
  };
}
function _n() {
  if (typeof matchMedia == "function")
    return matchMedia("(pointer:coarse)").matches ? "coarse" : "fine";
}
function Gn(e, n, t) {
  return e.x < n.x + n.width && e.x + e.width > n.x && e.y < n.y + n.height && e.y + e.height > n.y;
}
function Bn(e, n) {
  if (e === n) throw new Error("Cannot compare node with itself");
  const t = {
    a: je(e),
    b: je(n)
  };
  let r;
  for (; t.a.at(-1) === t.b.at(-1); )
    e = t.a.pop(), n = t.b.pop(), r = e;
  y(r, "Stacking order can only be calculated for elements with a common ancestor");
  const i = {
    a: Xe(qe(t.a)),
    b: Xe(qe(t.b))
  };
  if (i.a === i.b) {
    const o = r.childNodes, l = {
      a: t.a.at(-1),
      b: t.b.at(-1)
    };
    let a = o.length;
    for (; a--; ) {
      const s = o[a];
      if (s === l.a) return 1;
      if (s === l.b) return -1;
    }
  }
  return Math.sign(i.a - i.b);
}
const Tn = /\b(?:position|zIndex|opacity|transform|webkitTransform|mixBlendMode|filter|webkitFilter|isolation)\b/;
function Fn(e) {
  var n;
  const t = getComputedStyle((n = cn(e)) !== null && n !== void 0 ? n : e).display;
  return t === "flex" || t === "inline-flex";
}
function Vn(e) {
  const n = getComputedStyle(e);
  return !!(n.position === "fixed" || n.zIndex !== "auto" && (n.position !== "static" || Fn(e)) || +n.opacity < 1 || "transform" in n && n.transform !== "none" || "webkitTransform" in n && n.webkitTransform !== "none" || "mixBlendMode" in n && n.mixBlendMode !== "normal" || "filter" in n && n.filter !== "none" || "webkitFilter" in n && n.webkitFilter !== "none" || "isolation" in n && n.isolation === "isolate" || Tn.test(n.willChange) || n.webkitOverflowScrolling === "touch");
}
function qe(e) {
  let n = e.length;
  for (; n--; ) {
    const t = e[n];
    if (y(t, "Missing node"), Vn(t)) return t;
  }
  return null;
}
function Xe(e) {
  return e && Number(getComputedStyle(e).zIndex) || 0;
}
function je(e) {
  const n = [];
  for (; e; )
    n.push(e), e = cn(e);
  return n;
}
function cn(e) {
  const {
    parentNode: n
  } = e;
  return n && n instanceof ShadowRoot ? n.host : n;
}
const dn = 1, fn = 2, pn = 4, gn = 8, Wn = _n() === "coarse";
let O = [], ce = !1, ne = /* @__PURE__ */ new Map(), Ae = /* @__PURE__ */ new Map();
const xe = /* @__PURE__ */ new Set();
function Kn(e, n, t, r, i) {
  var o;
  const {
    ownerDocument: l
  } = n, a = {
    direction: t,
    element: n,
    hitAreaMargins: r,
    setResizeHandlerState: i
  }, s = (o = ne.get(l)) !== null && o !== void 0 ? o : 0;
  return ne.set(l, s + 1), xe.add(a), Ee(), function() {
    var P;
    Ae.delete(e), xe.delete(a);
    const f = (P = ne.get(l)) !== null && P !== void 0 ? P : 1;
    if (ne.set(l, f - 1), Ee(), f === 1 && ne.delete(l), O.includes(a)) {
      const u = O.indexOf(a);
      u >= 0 && O.splice(u, 1), We(), i("up", !0, null);
    }
  };
}
function Un(e) {
  const {
    target: n
  } = e, {
    x: t,
    y: r
  } = Le(e);
  ce = !0, Ve({
    target: n,
    x: t,
    y: r
  }), Ee(), O.length > 0 && (Ce("down", e), e.preventDefault(), mn(n) || e.stopImmediatePropagation());
}
function $e(e) {
  const {
    x: n,
    y: t
  } = Le(e);
  if (ce && e.buttons === 0 && (ce = !1, Ce("up", e)), !ce) {
    const {
      target: r
    } = e;
    Ve({
      target: r,
      x: n,
      y: t
    });
  }
  Ce("move", e), We(), O.length > 0 && e.preventDefault();
}
function ke(e) {
  const {
    target: n
  } = e, {
    x: t,
    y: r
  } = Le(e);
  Ae.clear(), ce = !1, O.length > 0 && (e.preventDefault(), mn(n) || e.stopImmediatePropagation()), Ce("up", e), Ve({
    target: n,
    x: t,
    y: r
  }), We(), Ee();
}
function mn(e) {
  let n = e;
  for (; n; ) {
    if (n.hasAttribute(A.resizeHandle))
      return !0;
    n = n.parentElement;
  }
  return !1;
}
function Ve({
  target: e,
  x: n,
  y: t
}) {
  O.splice(0);
  let r = null;
  (e instanceof HTMLElement || e instanceof SVGElement) && (r = e), xe.forEach((i) => {
    const {
      element: o,
      hitAreaMargins: l
    } = i, a = o.getBoundingClientRect(), {
      bottom: s,
      left: m,
      right: P,
      top: f
    } = a, u = Wn ? l.coarse : l.fine;
    if (n >= m - u && n <= P + u && t >= f - u && t <= s + u) {
      if (r !== null && document.contains(r) && o !== r && !o.contains(r) && !r.contains(o) && // Calculating stacking order has a cost, so we should avoid it if possible
      // That is why we only check potentially intersecting handles,
      // and why we skip if the event target is within the handle's DOM
      Bn(r, o) > 0) {
        let h = r, I = !1;
        for (; h && !h.contains(o); ) {
          if (Gn(h.getBoundingClientRect(), a)) {
            I = !0;
            break;
          }
          h = h.parentElement;
        }
        if (I)
          return;
      }
      O.push(i);
    }
  });
}
function _e(e, n) {
  Ae.set(e, n);
}
function We() {
  let e = !1, n = !1;
  O.forEach((r) => {
    const {
      direction: i
    } = r;
    i === "horizontal" ? e = !0 : n = !0;
  });
  let t = 0;
  Ae.forEach((r) => {
    t |= r;
  }), e && n ? Ne("intersection", t) : e ? Ne("horizontal", t) : n ? Ne("vertical", t) : kn();
}
let Ge = new AbortController();
function Ee() {
  Ge.abort(), Ge = new AbortController();
  const e = {
    capture: !0,
    signal: Ge.signal
  };
  xe.size && (ce ? (O.length > 0 && ne.forEach((n, t) => {
    const {
      body: r
    } = t;
    n > 0 && (r.addEventListener("contextmenu", ke, e), r.addEventListener("pointerleave", $e, e), r.addEventListener("pointermove", $e, e));
  }), window.addEventListener("pointerup", ke, e), window.addEventListener("pointercancel", ke, e)) : ne.forEach((n, t) => {
    const {
      body: r
    } = t;
    n > 0 && (r.addEventListener("pointerdown", Un, e), r.addEventListener("pointermove", $e, e));
  }));
}
function Ce(e, n) {
  xe.forEach((t) => {
    const {
      setResizeHandlerState: r
    } = t, i = O.includes(t);
    r(e, i, n);
  });
}
function On() {
  const [e, n] = ue(0);
  return T(() => n((t) => t + 1), []);
}
function y(e, n) {
  if (!e)
    throw console.error(n), Error(n);
}
function oe(e, n, t = Te) {
  return e.toFixed(t) === n.toFixed(t) ? 0 : e > n ? 1 : -1;
}
function Y(e, n, t = Te) {
  return oe(e, n, t) === 0;
}
function W(e, n, t) {
  return oe(e, n, t) === 0;
}
function qn(e, n, t) {
  if (e.length !== n.length)
    return !1;
  for (let r = 0; r < e.length; r++) {
    const i = e[r], o = n[r];
    if (!W(i, o, t))
      return !1;
  }
  return !0;
}
function se({
  panelConstraints: e,
  panelIndex: n,
  size: t
}) {
  const r = e[n];
  y(r != null, `Panel constraints not found for index ${n}`);
  let {
    collapsedSize: i = 0,
    collapsible: o,
    maxSize: l = 100,
    minSize: a = 0
  } = r;
  if (oe(t, a) < 0)
    if (o) {
      const s = (i + a) / 2;
      oe(t, s) < 0 ? t = i : t = a;
    } else
      t = a;
  return t = Math.min(l, t), t = parseFloat(t.toFixed(Te)), t;
}
function ye({
  delta: e,
  initialLayout: n,
  panelConstraints: t,
  pivotIndices: r,
  prevLayout: i,
  trigger: o
}) {
  if (W(e, 0))
    return n;
  const l = [...n], [a, s] = r;
  y(a != null, "Invalid first pivot index"), y(s != null, "Invalid second pivot index");
  let m = 0;
  if (o === "keyboard") {
    {
      const f = e < 0 ? s : a, u = t[f];
      y(u, `Panel constraints not found for index ${f}`);
      const {
        collapsedSize: v = 0,
        collapsible: h,
        minSize: I = 0
      } = u;
      if (h) {
        const w = n[f];
        if (y(w != null, `Previous layout not found for panel index ${f}`), W(w, v)) {
          const S = I - w;
          oe(S, Math.abs(e)) > 0 && (e = e < 0 ? 0 - S : S);
        }
      }
    }
    {
      const f = e < 0 ? a : s, u = t[f];
      y(u, `No panel constraints found for index ${f}`);
      const {
        collapsedSize: v = 0,
        collapsible: h,
        minSize: I = 0
      } = u;
      if (h) {
        const w = n[f];
        if (y(w != null, `Previous layout not found for panel index ${f}`), W(w, I)) {
          const S = w - v;
          oe(S, Math.abs(e)) > 0 && (e = e < 0 ? 0 - S : S);
        }
      }
    }
  }
  {
    const f = e < 0 ? 1 : -1;
    let u = e < 0 ? s : a, v = 0;
    for (; ; ) {
      const I = n[u];
      y(I != null, `Previous layout not found for panel index ${u}`);
      const S = se({
        panelConstraints: t,
        panelIndex: u,
        size: 100
      }) - I;
      if (v += S, u += f, u < 0 || u >= t.length)
        break;
    }
    const h = Math.min(Math.abs(e), Math.abs(v));
    e = e < 0 ? 0 - h : h;
  }
  {
    let u = e < 0 ? a : s;
    for (; u >= 0 && u < t.length; ) {
      const v = Math.abs(e) - Math.abs(m), h = n[u];
      y(h != null, `Previous layout not found for panel index ${u}`);
      const I = h - v, w = se({
        panelConstraints: t,
        panelIndex: u,
        size: I
      });
      if (!W(h, w) && (m += h - w, l[u] = w, m.toPrecision(3).localeCompare(Math.abs(e).toPrecision(3), void 0, {
        numeric: !0
      }) >= 0))
        break;
      e < 0 ? u-- : u++;
    }
  }
  if (qn(i, l))
    return i;
  {
    const f = e < 0 ? s : a, u = n[f];
    y(u != null, `Previous layout not found for panel index ${f}`);
    const v = u + m, h = se({
      panelConstraints: t,
      panelIndex: f,
      size: v
    });
    if (l[f] = h, !W(h, v)) {
      let I = v - h, S = e < 0 ? s : a;
      for (; S >= 0 && S < t.length; ) {
        const R = l[S];
        y(R != null, `Previous layout not found for panel index ${S}`);
        const M = R + I, N = se({
          panelConstraints: t,
          panelIndex: S,
          size: M
        });
        if (W(R, N) || (I -= N - R, l[S] = N), W(I, 0))
          break;
        e > 0 ? S-- : S++;
      }
    }
  }
  const P = l.reduce((f, u) => u + f, 0);
  return W(P, 100) ? l : i;
}
function Xn({
  layout: e,
  panelsArray: n,
  pivotIndices: t
}) {
  let r = 0, i = 100, o = 0, l = 0;
  const a = t[0];
  y(a != null, "No pivot index found"), n.forEach((f, u) => {
    const {
      constraints: v
    } = f, {
      maxSize: h = 100,
      minSize: I = 0
    } = v;
    u === a ? (r = I, i = h) : (o += I, l += h);
  });
  const s = Math.min(i, 100 - o), m = Math.max(r, 100 - l), P = e[a];
  return {
    valueMax: s,
    valueMin: m,
    valueNow: P
  };
}
function Se(e, n = document) {
  return Array.from(n.querySelectorAll(`[${A.resizeHandleId}][data-panel-group-id="${e}"]`));
}
function hn(e, n, t = document) {
  const i = Se(e, t).findIndex((o) => o.getAttribute(A.resizeHandleId) === n);
  return i ?? null;
}
function zn(e, n, t) {
  const r = hn(e, n, t);
  return r != null ? [r, r + 1] : [-1, -1];
}
function yn(e, n = document) {
  var t;
  if (n instanceof HTMLElement && (n == null || (t = n.dataset) === null || t === void 0 ? void 0 : t.panelGroupId) == e)
    return n;
  const r = n.querySelector(`[data-panel-group][data-panel-group-id="${e}"]`);
  return r || null;
}
function De(e, n = document) {
  const t = n.querySelector(`[${A.resizeHandleId}="${e}"]`);
  return t || null;
}
function jn(e, n, t, r = document) {
  var i, o, l, a;
  const s = De(n, r), m = Se(e, r), P = s ? m.indexOf(s) : -1, f = (i = (o = t[P]) === null || o === void 0 ? void 0 : o.id) !== null && i !== void 0 ? i : null, u = (l = (a = t[P + 1]) === null || a === void 0 ? void 0 : a.id) !== null && l !== void 0 ? l : null;
  return [f, u];
}
function Jn({
  committedValuesRef: e,
  eagerValuesRef: n,
  groupId: t,
  layout: r,
  panelDataArray: i,
  panelGroupElement: o,
  setLayout: l
}) {
  F({
    didWarnAboutMissingResizeHandle: !1
  }), re(() => {
    if (!o)
      return;
    const a = Se(t, o);
    for (let s = 0; s < i.length - 1; s++) {
      const {
        valueMax: m,
        valueMin: P,
        valueNow: f
      } = Xn({
        layout: r,
        panelsArray: i,
        pivotIndices: [s, s + 1]
      }), u = a[s];
      if (u != null) {
        const v = i[s];
        y(v, `No panel data found for index "${s}"`), u.setAttribute("aria-controls", v.id), u.setAttribute("aria-valuemax", "" + Math.round(m)), u.setAttribute("aria-valuemin", "" + Math.round(P)), u.setAttribute("aria-valuenow", f != null ? "" + Math.round(f) : "");
      }
    }
    return () => {
      a.forEach((s, m) => {
        s.removeAttribute("aria-controls"), s.removeAttribute("aria-valuemax"), s.removeAttribute("aria-valuemin"), s.removeAttribute("aria-valuenow");
      });
    };
  }, [t, r, i, o]), te(() => {
    if (!o)
      return;
    const a = n.current;
    y(a, "Eager values not found");
    const {
      panelDataArray: s
    } = a, m = yn(t, o);
    y(m != null, `No group found for id "${t}"`);
    const P = Se(t, o);
    y(P, `No resize handles found for group id "${t}"`);
    const f = P.map((u) => {
      const v = u.getAttribute(A.resizeHandleId);
      y(v, "Resize handle element has no handle id attribute");
      const [h, I] = jn(t, v, s, o);
      if (h == null || I == null)
        return () => {
        };
      const w = (S) => {
        if (!S.defaultPrevented)
          switch (S.key) {
            case "Enter": {
              S.preventDefault();
              const R = s.findIndex((M) => M.id === h);
              if (R >= 0) {
                const M = s[R];
                y(M, `No panel data found for index ${R}`);
                const N = r[R], {
                  collapsedSize: q = 0,
                  collapsible: C,
                  minSize: b = 0
                } = M.constraints;
                if (N != null && C) {
                  const K = ye({
                    delta: W(N, q) ? b - q : q - N,
                    initialLayout: r,
                    panelConstraints: s.map((J) => J.constraints),
                    pivotIndices: zn(t, v, o),
                    prevLayout: r,
                    trigger: "keyboard"
                  });
                  r !== K && l(K);
                }
              }
              break;
            }
          }
      };
      return u.addEventListener("keydown", w), () => {
        u.removeEventListener("keydown", w);
      };
    });
    return () => {
      f.forEach((u) => u());
    };
  }, [o, e, n, t, r, i, l]);
}
function Je(e, n) {
  if (e.length !== n.length)
    return !1;
  for (let t = 0; t < e.length; t++)
    if (e[t] !== n[t])
      return !1;
  return !0;
}
function vn(e, n) {
  const t = e === "horizontal", {
    x: r,
    y: i
  } = Le(n);
  return t ? r : i;
}
function Yn(e, n, t, r, i) {
  const o = t === "horizontal", l = De(n, i);
  y(l, `No resize handle element found for id "${n}"`);
  const a = l.getAttribute(A.groupId);
  y(a, "Resize handle element has no group id attribute");
  let {
    initialCursorPosition: s
  } = r;
  const m = vn(t, e), P = yn(a, i);
  y(P, `No group element found for id "${a}"`);
  const f = P.getBoundingClientRect(), u = o ? f.width : f.height;
  return (m - s) / u * 100;
}
function Zn(e, n, t, r, i, o) {
  if (an(e)) {
    const l = t === "horizontal";
    let a = 0;
    e.shiftKey ? a = 100 : i != null ? a = i : a = 10;
    let s = 0;
    switch (e.key) {
      case "ArrowDown":
        s = l ? 0 : a;
        break;
      case "ArrowLeft":
        s = l ? -a : 0;
        break;
      case "ArrowRight":
        s = l ? a : 0;
        break;
      case "ArrowUp":
        s = l ? 0 : -a;
        break;
      case "End":
        s = 100;
        break;
      case "Home":
        s = -100;
        break;
    }
    return s;
  } else
    return r == null ? 0 : Yn(e, n, t, r, o);
}
function Qn({
  panelDataArray: e
}) {
  const n = Array(e.length), t = e.map((o) => o.constraints);
  let r = 0, i = 100;
  for (let o = 0; o < e.length; o++) {
    const l = t[o];
    y(l, `Panel constraints not found for index ${o}`);
    const {
      defaultSize: a
    } = l;
    a != null && (r++, n[o] = a, i -= a);
  }
  for (let o = 0; o < e.length; o++) {
    const l = t[o];
    y(l, `Panel constraints not found for index ${o}`);
    const {
      defaultSize: a
    } = l;
    if (a != null)
      continue;
    const s = e.length - r, m = i / s;
    r++, n[o] = m, i -= m;
  }
  return n;
}
function le(e, n, t) {
  n.forEach((r, i) => {
    const o = e[i];
    y(o, `Panel data not found for index ${i}`);
    const {
      callbacks: l,
      constraints: a,
      id: s
    } = o, {
      collapsedSize: m = 0,
      collapsible: P
    } = a, f = t[s];
    if (f == null || r !== f) {
      t[s] = r;
      const {
        onCollapse: u,
        onExpand: v,
        onResize: h
      } = l;
      h && h(r, f), P && (u || v) && (v && (f == null || Y(f, m)) && !Y(r, m) && v(), u && (f == null || !Y(f, m)) && Y(r, m) && u());
    }
  });
}
function Pe(e, n) {
  if (e.length !== n.length)
    return !1;
  for (let t = 0; t < e.length; t++)
    if (e[t] != n[t])
      return !1;
  return !0;
}
function et({
  defaultSize: e,
  dragState: n,
  layout: t,
  panelData: r,
  panelIndex: i,
  precision: o = 3
}) {
  const l = t[i];
  let a;
  return l == null ? a = e != null ? e.toPrecision(o) : "1" : r.length === 1 ? a = "1" : a = l.toPrecision(o), {
    flexBasis: 0,
    flexGrow: a,
    flexShrink: 1,
    // Without this, Panel sizes may be unintentionally overridden by their content
    overflow: "hidden",
    // Disable pointer events inside of a panel during resize
    // This avoid edge cases like nested iframes
    pointerEvents: n !== null ? "none" : void 0
  };
}
function nt(e, n = 10) {
  let t = null;
  return (...i) => {
    t !== null && clearTimeout(t), t = setTimeout(() => {
      e(...i);
    }, n);
  };
}
function Ye(e) {
  try {
    if (typeof localStorage < "u")
      e.getItem = (n) => localStorage.getItem(n), e.setItem = (n, t) => {
        localStorage.setItem(n, t);
      };
    else
      throw new Error("localStorage not supported in this environment");
  } catch (n) {
    console.error(n), e.getItem = () => null, e.setItem = () => {
    };
  }
}
function xn(e) {
  return `react-resizable-panels:${e}`;
}
function Sn(e) {
  return e.map((n) => {
    const {
      constraints: t,
      id: r,
      idIsFromProps: i,
      order: o
    } = n;
    return i ? r : o ? `${o}:${JSON.stringify(t)}` : JSON.stringify(t);
  }).sort((n, t) => n.localeCompare(t)).join(",");
}
function bn(e, n) {
  try {
    const t = xn(e), r = n.getItem(t);
    if (r) {
      const i = JSON.parse(r);
      if (typeof i == "object" && i != null)
        return i;
    }
  } catch {
  }
  return null;
}
function tt(e, n, t) {
  var r, i;
  const o = (r = bn(e, t)) !== null && r !== void 0 ? r : {}, l = Sn(n);
  return (i = o[l]) !== null && i !== void 0 ? i : null;
}
function rt(e, n, t, r, i) {
  var o;
  const l = xn(e), a = Sn(n), s = (o = bn(e, i)) !== null && o !== void 0 ? o : {};
  s[a] = {
    expandToSizes: Object.fromEntries(t.entries()),
    layout: r
  };
  try {
    i.setItem(l, JSON.stringify(s));
  } catch (m) {
    console.error(m);
  }
}
function Ze({
  layout: e,
  panelConstraints: n
}) {
  const t = [...e], r = t.reduce((o, l) => o + l, 0);
  if (t.length !== n.length)
    throw Error(`Invalid ${n.length} panel layout: ${t.map((o) => `${o}%`).join(", ")}`);
  if (!W(r, 100) && t.length > 0)
    for (let o = 0; o < n.length; o++) {
      const l = t[o];
      y(l != null, `No layout data found for index ${o}`);
      const a = 100 / r * l;
      t[o] = a;
    }
  let i = 0;
  for (let o = 0; o < n.length; o++) {
    const l = t[o];
    y(l != null, `No layout data found for index ${o}`);
    const a = se({
      panelConstraints: n,
      panelIndex: o,
      size: l
    });
    l != a && (i += l - a, t[o] = a);
  }
  if (!W(i, 0))
    for (let o = 0; o < n.length; o++) {
      const l = t[o];
      y(l != null, `No layout data found for index ${o}`);
      const a = l + i, s = se({
        panelConstraints: n,
        panelIndex: o,
        size: a
      });
      if (l !== s && (i -= s - l, t[o] = s, W(i, 0)))
        break;
    }
  return t;
}
const ot = 100, ve = {
  getItem: (e) => (Ye(ve), ve.getItem(e)),
  setItem: (e, n) => {
    Ye(ve), ve.setItem(e, n);
  }
}, Qe = {};
function Pn({
  autoSaveId: e = null,
  children: n,
  className: t = "",
  direction: r,
  forwardedRef: i,
  id: o = null,
  onLayout: l = null,
  keyboardResizeBy: a = null,
  storage: s = ve,
  style: m,
  tagName: P = "div",
  ...f
}) {
  const u = Fe(o), v = F(null), [h, I] = ue(null), [w, S] = ue([]), R = On(), M = F({}), N = F(/* @__PURE__ */ new Map()), q = F(0), C = F({
    autoSaveId: e,
    direction: r,
    dragState: h,
    id: u,
    keyboardResizeBy: a,
    onLayout: l,
    storage: s
  }), b = F({
    layout: w,
    panelDataArray: [],
    panelDataArrayChanged: !1
  });
  F({
    didLogIdAndOrderWarning: !1,
    didLogPanelConstraintsWarning: !1,
    prevPanelIds: []
  }), tn(i, () => ({
    getId: () => C.current.id,
    getLayout: () => {
      const {
        layout: c
      } = b.current;
      return c;
    },
    setLayout: (c) => {
      const {
        onLayout: g
      } = C.current, {
        layout: x,
        panelDataArray: p
      } = b.current, d = Ze({
        layout: c,
        panelConstraints: p.map((z) => z.constraints)
      });
      Je(x, d) || (S(d), b.current.layout = d, g && g(d), le(p, d, M.current));
    }
  }), []), re(() => {
    C.current.autoSaveId = e, C.current.direction = r, C.current.dragState = h, C.current.id = u, C.current.onLayout = l, C.current.storage = s;
  }), Jn({
    committedValuesRef: C,
    eagerValuesRef: b,
    groupId: u,
    layout: w,
    panelDataArray: b.current.panelDataArray,
    setLayout: S,
    panelGroupElement: v.current
  }), te(() => {
    const {
      panelDataArray: c
    } = b.current;
    if (e) {
      if (w.length === 0 || w.length !== c.length)
        return;
      let g = Qe[e];
      g == null && (g = nt(rt, ot), Qe[e] = g);
      const x = [...c], p = new Map(N.current);
      g(e, x, p, w, s);
    }
  }, [e, w, s]), te(() => {
  });
  const K = T((c) => {
    const {
      onLayout: g
    } = C.current, {
      layout: x,
      panelDataArray: p
    } = b.current;
    if (c.constraints.collapsible) {
      const d = p.map((X) => X.constraints), {
        collapsedSize: z = 0,
        panelSize: E,
        pivotIndices: H
      } = ee(p, c, x);
      if (y(E != null, `Panel size not found for panel "${c.id}"`), !Y(E, z)) {
        N.current.set(c.id, E);
        const j = ae(p, c) === p.length - 1 ? E - z : z - E, L = ye({
          delta: j,
          initialLayout: x,
          panelConstraints: d,
          pivotIndices: H,
          prevLayout: x,
          trigger: "imperative-api"
        });
        Pe(x, L) || (S(L), b.current.layout = L, g && g(L), le(p, L, M.current));
      }
    }
  }, []), J = T((c, g) => {
    const {
      onLayout: x
    } = C.current, {
      layout: p,
      panelDataArray: d
    } = b.current;
    if (c.constraints.collapsible) {
      const z = d.map((U) => U.constraints), {
        collapsedSize: E = 0,
        panelSize: H = 0,
        minSize: X = 0,
        pivotIndices: j
      } = ee(d, c, p), L = g ?? X;
      if (Y(H, E)) {
        const U = N.current.get(c.id), he = U != null && U >= L ? U : L, He = ae(d, c) === d.length - 1 ? H - he : he - H, B = ye({
          delta: He,
          initialLayout: p,
          panelConstraints: z,
          pivotIndices: j,
          prevLayout: p,
          trigger: "imperative-api"
        });
        Pe(p, B) || (S(B), b.current.layout = B, x && x(B), le(d, B, M.current));
      }
    }
  }, []), V = T((c) => {
    const {
      layout: g,
      panelDataArray: x
    } = b.current, {
      panelSize: p
    } = ee(x, c, g);
    return y(p != null, `Panel size not found for panel "${c.id}"`), p;
  }, []), _ = T((c, g) => {
    const {
      panelDataArray: x
    } = b.current, p = ae(x, c);
    return et({
      defaultSize: g,
      dragState: h,
      layout: w,
      panelData: x,
      panelIndex: p
    });
  }, [h, w]), D = T((c) => {
    const {
      layout: g,
      panelDataArray: x
    } = b.current, {
      collapsedSize: p = 0,
      collapsible: d,
      panelSize: z
    } = ee(x, c, g);
    return y(z != null, `Panel size not found for panel "${c.id}"`), d === !0 && Y(z, p);
  }, []), Q = T((c) => {
    const {
      layout: g,
      panelDataArray: x
    } = b.current, {
      collapsedSize: p = 0,
      collapsible: d,
      panelSize: z
    } = ee(x, c, g);
    return y(z != null, `Panel size not found for panel "${c.id}"`), !d || oe(z, p) > 0;
  }, []), k = T((c) => {
    const {
      panelDataArray: g
    } = b.current;
    g.push(c), g.sort((x, p) => {
      const d = x.order, z = p.order;
      return d == null && z == null ? 0 : d == null ? -1 : z == null ? 1 : d - z;
    }), b.current.panelDataArrayChanged = !0, R();
  }, [R]);
  re(() => {
    if (b.current.panelDataArrayChanged) {
      b.current.panelDataArrayChanged = !1;
      const {
        autoSaveId: c,
        onLayout: g,
        storage: x
      } = C.current, {
        layout: p,
        panelDataArray: d
      } = b.current;
      let z = null;
      if (c) {
        const H = tt(c, d, x);
        H && (N.current = new Map(Object.entries(H.expandToSizes)), z = H.layout);
      }
      z == null && (z = Qn({
        panelDataArray: d
      }));
      const E = Ze({
        layout: z,
        panelConstraints: d.map((H) => H.constraints)
      });
      Je(p, E) || (S(E), b.current.layout = E, g && g(E), le(d, E, M.current));
    }
  }), re(() => {
    const c = b.current;
    return () => {
      c.layout = [];
    };
  }, []);
  const $ = T((c) => {
    let g = !1;
    const x = v.current;
    return x && window.getComputedStyle(x, null).getPropertyValue("direction") === "rtl" && (g = !0), function(d) {
      d.preventDefault();
      const z = v.current;
      if (!z)
        return () => null;
      const {
        direction: E,
        dragState: H,
        id: X,
        keyboardResizeBy: j,
        onLayout: L
      } = C.current, {
        layout: U,
        panelDataArray: he
      } = b.current, {
        initialLayout: be
      } = H ?? {}, He = zn(X, c, z);
      let B = Zn(d, c, E, H, j, z);
      const Ke = E === "horizontal";
      Ke && g && (B = -B);
      const En = he.map((Cn) => Cn.constraints), ze = ye({
        delta: B,
        initialLayout: be ?? U,
        panelConstraints: En,
        pivotIndices: He,
        prevLayout: U,
        trigger: an(d) ? "keyboard" : "mouse-or-touch"
      }), Ue = !Pe(U, ze);
      (sn(d) || un(d)) && q.current != B && (q.current = B, !Ue && B !== 0 ? Ke ? _e(c, B < 0 ? dn : fn) : _e(c, B < 0 ? pn : gn) : _e(c, 0)), Ue && (S(ze), b.current.layout = ze, L && L(ze), le(he, ze, M.current));
    };
  }, []), G = T((c, g) => {
    const {
      onLayout: x
    } = C.current, {
      layout: p,
      panelDataArray: d
    } = b.current, z = d.map((U) => U.constraints), {
      panelSize: E,
      pivotIndices: H
    } = ee(d, c, p);
    y(E != null, `Panel size not found for panel "${c.id}"`);
    const j = ae(d, c) === d.length - 1 ? E - g : g - E, L = ye({
      delta: j,
      initialLayout: p,
      panelConstraints: z,
      pivotIndices: H,
      prevLayout: p,
      trigger: "imperative-api"
    });
    Pe(p, L) || (S(L), b.current.layout = L, x && x(L), le(d, L, M.current));
  }, []), fe = T((c, g) => {
    const {
      layout: x,
      panelDataArray: p
    } = b.current, {
      collapsedSize: d = 0,
      collapsible: z
    } = g, {
      collapsedSize: E = 0,
      collapsible: H,
      maxSize: X = 100,
      minSize: j = 0
    } = c.constraints, {
      panelSize: L
    } = ee(p, c, x);
    L != null && (z && H && Y(L, d) ? Y(d, E) || G(c, E) : L < j ? G(c, j) : L > X && G(c, X));
  }, [G]), pe = T((c, g) => {
    const {
      direction: x
    } = C.current, {
      layout: p
    } = b.current;
    if (!v.current)
      return;
    const d = De(c, v.current);
    y(d, `Drag handle element not found for id "${c}"`);
    const z = vn(x, g);
    I({
      dragHandleId: c,
      dragHandleRect: d.getBoundingClientRect(),
      initialCursorPosition: z,
      initialLayout: p
    });
  }, []), ge = T(() => {
    I(null);
  }, []), Me = T((c) => {
    const {
      panelDataArray: g
    } = b.current, x = ae(g, c);
    x >= 0 && (g.splice(x, 1), delete M.current[c.id], b.current.panelDataArrayChanged = !0, R());
  }, [R]), ie = Dn(() => ({
    collapsePanel: K,
    direction: r,
    dragState: h,
    expandPanel: J,
    getPanelSize: V,
    getPanelStyle: _,
    groupId: u,
    isPanelCollapsed: D,
    isPanelExpanded: Q,
    reevaluatePanelConstraints: fe,
    registerPanel: k,
    registerResizeHandle: $,
    resizePanel: G,
    startDragging: pe,
    stopDragging: ge,
    unregisterPanel: Me,
    panelGroupElement: v.current
  }), [K, h, r, J, V, _, u, D, Q, fe, k, $, G, pe, ge, Me]), me = {
    display: "flex",
    flexDirection: r === "horizontal" ? "row" : "column",
    height: "100%",
    overflow: "hidden",
    width: "100%"
  };
  return de(Re.Provider, {
    value: ie
  }, de(P, {
    ...f,
    children: n,
    className: t,
    id: o,
    ref: v,
    style: {
      ...me,
      ...m
    },
    // CSS selectors
    [A.group]: "",
    [A.groupDirection]: r,
    [A.groupId]: u
  }));
}
const wn = en((e, n) => de(Pn, {
  ...e,
  forwardedRef: n
}));
Pn.displayName = "PanelGroup";
wn.displayName = "forwardRef(PanelGroup)";
function ae(e, n) {
  return e.findIndex((t) => t === n || t.id === n.id);
}
function ee(e, n, t) {
  const r = ae(e, n), o = r === e.length - 1 ? [r - 1, r] : [r, r + 1], l = t[r];
  return {
    ...n.constraints,
    panelSize: l,
    pivotIndices: o
  };
}
function it({
  disabled: e,
  handleId: n,
  resizeHandler: t,
  panelGroupElement: r
}) {
  te(() => {
    if (e || t == null || r == null)
      return;
    const i = De(n, r);
    if (i == null)
      return;
    const o = (l) => {
      if (!l.defaultPrevented)
        switch (l.key) {
          case "ArrowDown":
          case "ArrowLeft":
          case "ArrowRight":
          case "ArrowUp":
          case "End":
          case "Home": {
            l.preventDefault(), t(l);
            break;
          }
          case "F6": {
            l.preventDefault();
            const a = i.getAttribute(A.groupId);
            y(a, `No group element found for id "${a}"`);
            const s = Se(a, r), m = hn(a, n, r);
            y(m !== null, `No resize element found for id "${n}"`);
            const P = l.shiftKey ? m > 0 ? m - 1 : s.length - 1 : m + 1 < s.length ? m + 1 : 0;
            s[P].focus();
            break;
          }
        }
    };
    return i.addEventListener("keydown", o), () => {
      i.removeEventListener("keydown", o);
    };
  }, [r, e, n, t]);
}
function In({
  children: e = null,
  className: n = "",
  disabled: t = !1,
  hitAreaMargins: r,
  id: i,
  onBlur: o,
  onClick: l,
  onDragging: a,
  onFocus: s,
  onPointerDown: m,
  onPointerUp: P,
  style: f = {},
  tabIndex: u = 0,
  tagName: v = "div",
  ...h
}) {
  var I, w;
  const S = F(null), R = F({
    onClick: l,
    onDragging: a,
    onPointerDown: m,
    onPointerUp: P
  });
  te(() => {
    R.current.onClick = l, R.current.onDragging = a, R.current.onPointerDown = m, R.current.onPointerUp = P;
  });
  const M = nn(Re);
  if (M === null)
    throw Error("PanelResizeHandle components must be rendered within a PanelGroup container");
  const {
    direction: N,
    groupId: q,
    registerResizeHandle: C,
    startDragging: b,
    stopDragging: K,
    panelGroupElement: J
  } = M, V = Fe(i), [_, D] = ue("inactive"), [Q, k] = ue(!1), [$, G] = ue(null), fe = F({
    state: _
  });
  re(() => {
    fe.current.state = _;
  }), te(() => {
    if (t)
      G(null);
    else {
      const ie = C(V);
      G(() => ie);
    }
  }, [t, V, C]);
  const pe = (I = r?.coarse) !== null && I !== void 0 ? I : 15, ge = (w = r?.fine) !== null && w !== void 0 ? w : 5;
  return te(() => {
    if (t || $ == null)
      return;
    const ie = S.current;
    y(ie, "Element ref not attached");
    let me = !1;
    return Kn(V, ie, N, {
      coarse: pe,
      fine: ge
    }, (g, x, p) => {
      if (!x) {
        D("inactive");
        return;
      }
      switch (g) {
        case "down": {
          D("drag"), me = !1, y(p, 'Expected event to be defined for "down" action'), b(V, p);
          const {
            onDragging: d,
            onPointerDown: z
          } = R.current;
          d?.(!0), z?.();
          break;
        }
        case "move": {
          const {
            state: d
          } = fe.current;
          me = !0, d !== "drag" && D("hover"), y(p, 'Expected event to be defined for "move" action'), $(p);
          break;
        }
        case "up": {
          D("hover"), K();
          const {
            onClick: d,
            onDragging: z,
            onPointerUp: E
          } = R.current;
          z?.(!1), E?.(), me || d?.();
          break;
        }
      }
    });
  }, [pe, N, t, ge, C, V, $, b, K]), it({
    disabled: t,
    handleId: V,
    resizeHandler: $,
    panelGroupElement: J
  }), de(v, {
    ...h,
    children: e,
    className: n,
    id: i,
    onBlur: () => {
      k(!1), o?.();
    },
    onFocus: () => {
      k(!0), s?.();
    },
    ref: S,
    role: "separator",
    style: {
      ...{
        touchAction: "none",
        userSelect: "none"
      },
      ...f
    },
    tabIndex: u,
    // CSS selectors
    [A.groupDirection]: N,
    [A.groupId]: q,
    [A.resizeHandle]: "",
    [A.resizeHandleActive]: _ === "drag" ? "pointer" : Q ? "keyboard" : void 0,
    [A.resizeHandleEnabled]: !t,
    [A.resizeHandleId]: V,
    [A.resizeHandleState]: _
  });
}
In.displayName = "PanelResizeHandle";
const ct = ({
  className: e,
  ...n
}) => /* @__PURE__ */ we(
  wn,
  {
    className: rn(
      "flex h-full w-full data-[panel-group-direction=vertical]:flex-col",
      e
    ),
    ...n
  }
), dt = ln, ft = ({
  withHandle: e,
  className: n,
  ...t
}) => /* @__PURE__ */ we(
  In,
  {
    className: rn(
      "relative flex w-px items-center justify-center bg-border after:absolute after:inset-y-0 after:left-1/2 after:w-1 after:-translate-x-1/2 focus-visible:outline-none focus-visible:border-ring focus-visible:ring-3 focus-visible:ring-ring/50 data-[panel-group-direction=vertical]:h-px data-[panel-group-direction=vertical]:w-full data-[panel-group-direction=vertical]:after:left-0 data-[panel-group-direction=vertical]:after:h-1 data-[panel-group-direction=vertical]:after:w-full data-[panel-group-direction=vertical]:after:-translate-y-1/2 data-[panel-group-direction=vertical]:after:translate-x-0 [&[data-panel-group-direction=vertical]>div]:rotate-90",
      n
    ),
    ...t,
    children: e && /* @__PURE__ */ we("div", { className: "z-10 flex h-4 w-3 items-center justify-center rounded-sm border border-border bg-card", children: /* @__PURE__ */ we(Rn, { className: "h-2.5 w-2.5 text-muted-foreground" }) })
  }
);
export {
  ft as ResizableHandle,
  dt as ResizablePanel,
  ct as ResizablePanelGroup
};
