// @btcv/ui/dist/Separator.js
"use client";
import { jsx as n } from "react/jsx-runtime";
import { c as p } from "./_shared/utils-eGXXUFl7.js";
import * as m from "react";
import { P as d } from "./_shared/index-V-Ajw7Ac.js";
var f = "Separator", e = "horizontal", v = ["horizontal", "vertical"], s = m.forwardRef((r, o) => {
  const { decorative: a, orientation: t = e, ...c } = r, i = u(t) ? t : e, l = a ? { role: "none" } : { "aria-orientation": i === "vertical" ? i : void 0, role: "separator" };
  return /* @__PURE__ */ n(
    d.div,
    {
      "data-orientation": i,
      ...l,
      ...c,
      ref: o
    }
  );
});
s.displayName = f;
function u(r) {
  return v.includes(r);
}
var h = s;
function T({
  className: r,
  orientation: o = "horizontal",
  decorative: a = !0,
  ...t
}) {
  return /* @__PURE__ */ n(
    h,
    {
      "data-slot": "separator",
      decorative: a,
      orientation: o,
      className: p(
        "shrink-0 bg-border",
        o === "horizontal" ? "h-px w-full" : "w-px self-stretch",
        r
      ),
      ...t
    }
  );
}
export {
  T as Separator
};
