// @btcv/ui/dist/Sheet.js
"use client";
import { jsx as o, jsxs as i } from "react/jsx-runtime";
import * as r from "react";
import { O as f, P as m, C as p, a as n, T as u, D as h, R as b, b as g } from "./_shared/index-DlSuDb9N.js";
import { c as x } from "./_shared/index-uPOYJZpG.js";
import { X as S } from "lucide-react";
import { c as s } from "./_shared/utils-eGXXUFl7.js";
const z = b, P = g, k = n, y = m, d = r.forwardRef(({ className: e, ...t }, a) => /* @__PURE__ */ o(
  f,
  {
    ref: a,
    className: s(
      "fixed inset-0 z-50 bg-black/30 backdrop-blur-sm",
      "data-[state=open]:animate-in data-[state=open]:fade-in-0",
      "data-[state=closed]:animate-out data-[state=closed]:fade-out-0",
      e
    ),
    ...t
  }
));
d.displayName = "SheetOverlay";
const v = x(
  "fixed z-50 flex flex-col gap-4 bg-glass border border-border shadow-lg transition-transform focus:outline-none data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:duration-300 data-[state=open]:duration-300",
  {
    variants: {
      side: {
        top: "inset-x-0 top-0 h-[320px] border-b data-[state=open]:slide-in-from-top data-[state=closed]:slide-out-to-top",
        bottom: "inset-x-0 bottom-0 h-[320px] border-t data-[state=open]:slide-in-from-bottom data-[state=closed]:slide-out-to-bottom",
        left: "inset-y-0 left-0 w-[320px] border-r data-[state=open]:slide-in-from-left data-[state=closed]:slide-out-to-left",
        right: "inset-y-0 right-0 w-[320px] border-l data-[state=open]:slide-in-from-right data-[state=closed]:slide-out-to-right"
      }
    },
    defaultVariants: {
      side: "right"
    }
  }
), N = r.forwardRef(({ side: e = "right", className: t, children: a, ...l }, c) => /* @__PURE__ */ i(y, { children: [
  /* @__PURE__ */ o(d, {}),
  /* @__PURE__ */ i(
    p,
    {
      ref: c,
      className: s(v({ side: e }), t),
      ...l,
      children: [
        a,
        /* @__PURE__ */ o(n, { className: "absolute right-4 top-4 w-7 h-7 rounded-xl flex items-center justify-center text-muted-foreground hover:text-foreground hover:bg-secondary transition-colors cursor-pointer outline-none focus-visible:border-ring focus-visible:ring-3 focus-visible:ring-ring/50", children: /* @__PURE__ */ o(S, { size: 18 }) })
      ]
    }
  )
] }));
N.displayName = "SheetContent";
function V({ className: e, ...t }) {
  return /* @__PURE__ */ o(
    "div",
    {
      className: s("flex flex-col gap-0.5 px-5 pt-5", e),
      ...t
    }
  );
}
function F({ className: e, ...t }) {
  return /* @__PURE__ */ o(
    "div",
    {
      className: s(
        "flex items-center justify-end gap-2 py-4 px-5 mt-auto border-t border-border",
        e
      ),
      ...t
    }
  );
}
const w = r.forwardRef(({ className: e, ...t }, a) => /* @__PURE__ */ o(
  u,
  {
    ref: a,
    className: s("text-base font-semibold text-foreground", e),
    ...t
  }
));
w.displayName = "SheetTitle";
const R = r.forwardRef(({ className: e, ...t }, a) => /* @__PURE__ */ o(
  h,
  {
    ref: a,
    className: s("text-sm text-muted-foreground", e),
    ...t
  }
));
R.displayName = "SheetDescription";
export {
  z as Sheet,
  k as SheetClose,
  N as SheetContent,
  R as SheetDescription,
  F as SheetFooter,
  V as SheetHeader,
  d as SheetOverlay,
  y as SheetPortal,
  w as SheetTitle,
  P as SheetTrigger,
  v as sheetVariants
};
