// @btcv/ui/dist/Sidebar.js
"use client";
import { jsxs as b, jsx as r, Fragment as z } from "react/jsx-runtime";
import * as p from "react";
import { S as N, a as S } from "./_shared/index-BrLJJgkl.js";
import { ChevronRight as O, X as j, Menu as E } from "lucide-react";
import { c as s } from "./_shared/utils-eGXXUFl7.js";
const C = p.createContext({
  open: !1,
  setOpen: () => {
  }
});
function g() {
  return p.useContext(C);
}
function J({
  navbarHeight: n = 56,
  className: t,
  children: a,
  ...o
}) {
  const [l, i] = p.useState(!1);
  return /* @__PURE__ */ r(C.Provider, { value: { open: l, setOpen: i }, children: /* @__PURE__ */ r(
    "div",
    {
      className: s("relative flex min-h-screen", t),
      ...o,
      children: a
    }
  ) });
}
function K({ className: n, ...t }) {
  const { open: a, setOpen: o } = g();
  return /* @__PURE__ */ r(
    "button",
    {
      type: "button",
      className: s(
        "inline-flex items-center justify-center w-8 h-8 rounded-2xl",
        "text-muted-foreground hover:text-foreground hover:bg-muted",
        "transition-colors cursor-pointer md:hidden",
        n
      ),
      onClick: () => o((l) => !l),
      ...t,
      children: a ? /* @__PURE__ */ r(j, { size: 18 }) : /* @__PURE__ */ r(E, { size: 18 })
    }
  );
}
function $({ navbarHeight: n = 56, variant: t = "default", className: a, children: o, ...l }) {
  const { open: i, setOpen: c } = g(), d = t === "floating";
  return /* @__PURE__ */ b(z, { children: [
    i && /* @__PURE__ */ r(
      "div",
      {
        className: "fixed inset-0 z-30 bg-black/30 md:hidden",
        onClick: () => c(!1)
      }
    ),
    /* @__PURE__ */ r(
      "aside",
      {
        className: s(
          "fixed left-0 z-30 w-[220px] flex flex-col bg-card transition-transform duration-200",
          "md:sticky md:translate-x-0 md:shrink-0",
          d ? "rounded-[10px] border border-border shadow-lg overflow-hidden" : "border-r border-border",
          // Margin gives the floating panel its gap, except while it's hidden
          // off-screen on mobile (then no margin so it tucks fully away).
          d && (i ? "m-2" : "m-0 md:m-2"),
          i ? "translate-x-0" : "-translate-x-full",
          a
        ),
        style: {
          // Floating shifts the sticky offset down by its top margin (m-2 = 0.5rem)
          // so the panel doesn't jump upward by that amount once it sticks on scroll.
          top: d ? `calc(${n}px + 0.5rem)` : n,
          height: d ? `calc(100vh - ${n}px - 1rem)` : `calc(100vh - ${n}px)`
        },
        ...l,
        children: o
      }
    )
  ] });
}
function Q({ navbarHeight: n = 56, className: t, children: a, ...o }) {
  return /* @__PURE__ */ r(
    "main",
    {
      className: s("flex-1 min-w-0", t),
      style: { paddingTop: n },
      ...o,
      children: a
    }
  );
}
function I({ className: n, ...t }) {
  return /* @__PURE__ */ r(
    "div",
    {
      className: s("px-3 py-3 shrink-0", n),
      ...t
    }
  );
}
function F({ className: n, ...t }) {
  return /* @__PURE__ */ r(
    "div",
    {
      className: s("flex-1 overflow-y-auto py-1 px-2 flex flex-col gap-2", n),
      ...t
    }
  );
}
function M({ className: n, ...t }) {
  return /* @__PURE__ */ r(
    "div",
    {
      className: s("px-2 py-2 shrink-0 border-t border-border", n),
      ...t
    }
  );
}
function A({ label: n, className: t, children: a, ...o }) {
  return /* @__PURE__ */ b("div", { className: s("flex flex-col gap-0.5", t), ...o, children: [
    n && /* @__PURE__ */ r("span", { className: "px-3 py-1.5 text-[0.6875rem] font-semibold uppercase tracking-wider text-muted-foreground", children: n }),
    a
  ] });
}
function G({ className: n, ...t }) {
  return /* @__PURE__ */ r(
    "div",
    {
      className: s("mx-2 my-2 h-px bg-border", n),
      ...t
    }
  );
}
function v({
  icon: n,
  active: t,
  badge: a,
  asChild: o,
  className: l,
  children: i,
  ...c
}) {
  const d = o ? N : "button", { setOpen: h } = g(), f = (m) => {
    h(!1), c.onClick && c.onClick(m);
  };
  return /* @__PURE__ */ b(
    d,
    {
      "data-active": t || void 0,
      className: s(
        "flex items-center gap-2.5 w-full px-3 py-2 text-[0.8125rem] font-medium rounded-2xl transition-all duration-150 cursor-pointer",
        "border-none bg-transparent text-left",
        t ? "bg-primary/15 text-primary" : "text-muted-foreground hover:bg-muted hover:text-foreground",
        l
      ),
      ...c,
      onClick: f,
      children: [
        n && /* @__PURE__ */ r(n, { size: 16, className: "shrink-0" }),
        o ? /* @__PURE__ */ r(S, { children: i }) : /* @__PURE__ */ r("span", { className: "flex-1 truncate", children: i }),
        a && /* @__PURE__ */ r("span", { className: "shrink-0", children: a })
      ]
    }
  );
}
function w({ className: n, ...t }) {
  return /* @__PURE__ */ r(
    "div",
    {
      className: s(
        "flex flex-col gap-0.5 ml-[1.125rem] pl-3 border-l border-border",
        n
      ),
      ...t
    }
  );
}
function k({
  active: n,
  badge: t,
  asChild: a,
  className: o,
  children: l,
  ...i
}) {
  const c = a ? N : "button", { setOpen: d } = g(), h = (f) => {
    d(!1), i.onClick && i.onClick(f);
  };
  return /* @__PURE__ */ b(
    c,
    {
      "data-active": n || void 0,
      className: s(
        "flex items-center gap-2 w-full px-3 py-1 text-[0.8125rem] font-medium rounded-2xl transition-all duration-150 cursor-pointer",
        "border-none bg-transparent text-left",
        n ? "bg-primary/15 text-primary" : "text-muted-foreground hover:bg-muted hover:text-foreground",
        o
      ),
      ...i,
      onClick: h,
      children: [
        a ? /* @__PURE__ */ r(S, { children: l }) : /* @__PURE__ */ r("span", { className: "flex-1 truncate", children: l }),
        t && /* @__PURE__ */ r("span", { className: "shrink-0", children: t })
      ]
    }
  );
}
function R({
  icon: n,
  label: t,
  badge: a,
  active: o,
  defaultOpen: l,
  className: i,
  children: c,
  ...d
}) {
  const h = p.useMemo(
    () => p.Children.toArray(c).some(
      (x) => p.isValidElement(x) && x.props.active === !0
    ),
    [c]
  ), [f, m] = p.useState(
    l !== void 0 ? l : h
  );
  return /* @__PURE__ */ b("div", { className: s("flex flex-col gap-0.5", i), ...d, children: [
    /* @__PURE__ */ b(
      "button",
      {
        type: "button",
        "data-active": o || void 0,
        "data-state": f ? "open" : "closed",
        onClick: () => m((x) => !x),
        className: s(
          "flex items-center gap-2.5 w-full px-3 py-2 text-[0.8125rem] font-medium rounded-2xl transition-all duration-150 cursor-pointer",
          "border-none bg-transparent text-left",
          o ? "bg-primary/15 text-primary" : "text-muted-foreground hover:bg-muted hover:text-foreground"
        ),
        children: [
          n && /* @__PURE__ */ r(n, { size: 16, className: "shrink-0" }),
          /* @__PURE__ */ r("span", { className: "flex-1 truncate", children: t }),
          a && /* @__PURE__ */ r("span", { className: "shrink-0", children: a }),
          /* @__PURE__ */ r(
            O,
            {
              size: 14,
              className: s(
                "shrink-0 transition-transform duration-200",
                f && "rotate-90"
              )
            }
          )
        ]
      }
    ),
    f && /* @__PURE__ */ r(w, { children: c })
  ] });
}
function T(n) {
  return typeof n == "object" && "items" in n;
}
function P(n) {
  return n === "separator";
}
function U({
  items: n,
  header: t,
  headerClassName: a,
  footer: o,
  renderLink: l,
  onNavigate: i,
  navbarHeight: c = 56,
  variant: d = "default",
  className: h,
  ...f
}) {
  const m = (e) => l ? /* @__PURE__ */ r(
    k,
    {
      active: e.active,
      badge: e.badge,
      asChild: !0,
      children: l(e)
    },
    e.key
  ) : e.href ? /* @__PURE__ */ r(
    k,
    {
      active: e.active,
      badge: e.badge,
      asChild: !0,
      children: /* @__PURE__ */ r("a", { href: e.href, children: e.label })
    },
    e.key
  ) : /* @__PURE__ */ r(
    k,
    {
      active: e.active,
      badge: e.badge,
      onClick: () => i?.(e.key),
      children: e.label
    },
    e.key
  ), x = (e) => {
    if (e.children?.length && e.collapsible)
      return /* @__PURE__ */ r(
        R,
        {
          icon: e.icon,
          label: e.label,
          badge: e.badge,
          active: e.active,
          defaultOpen: e.defaultOpen,
          children: e.children.map(m)
        },
        e.key
      );
    let u;
    return l ? u = /* @__PURE__ */ r(
      v,
      {
        icon: e.icon,
        active: e.active,
        badge: e.badge,
        asChild: !0,
        children: l(e)
      }
    ) : e.href ? u = /* @__PURE__ */ r(
      v,
      {
        icon: e.icon,
        active: e.active,
        badge: e.badge,
        asChild: !0,
        children: /* @__PURE__ */ r("a", { href: e.href, children: e.label })
      }
    ) : u = /* @__PURE__ */ r(
      v,
      {
        icon: e.icon,
        active: e.active,
        badge: e.badge,
        onClick: () => i?.(e.key),
        children: e.label
      }
    ), e.children?.length ? /* @__PURE__ */ b(p.Fragment, { children: [
      u,
      /* @__PURE__ */ r(w, { children: e.children.map(m) })
    ] }, e.key) : p.cloneElement(u, { key: e.key });
  };
  return /* @__PURE__ */ b($, { navbarHeight: c, variant: d, className: h, ...f, children: [
    t && /* @__PURE__ */ r(I, { className: a, children: t }),
    /* @__PURE__ */ r(F, { children: ((e) => e.map((u, y) => P(u) ? /* @__PURE__ */ r(G, {}, `sep-${y}`) : T(u) ? /* @__PURE__ */ r(A, { label: u.label, children: u.items.map(x) }, u.label || `group-${y}`) : x(u)))(n) }),
    o && /* @__PURE__ */ r(M, { children: o })
  ] });
}
function W({
  items: n,
  onNavigate: t,
  header: a,
  footer: o,
  className: l,
  ...i
}) {
  return /* @__PURE__ */ b(
    "aside",
    {
      className: s(
        "w-12 shrink-0 bg-card border-r border-border flex flex-col items-center py-3 gap-1",
        l
      ),
      ...i,
      children: [
        a && /* @__PURE__ */ r("div", { className: "mb-2", children: a }),
        /* @__PURE__ */ r("nav", { className: "flex flex-col items-center gap-1 flex-1", children: n.map((c) => /* @__PURE__ */ r(
          "button",
          {
            type: "button",
            title: c.label,
            onClick: () => t?.(c.key),
            className: s(
              "h-8 w-8 flex items-center justify-center rounded-xl border-none cursor-pointer transition-colors",
              c.active ? "bg-primary/15 text-primary" : "bg-transparent text-muted-foreground hover:bg-muted hover:text-foreground"
            ),
            children: /* @__PURE__ */ r(c.icon, { size: 16 })
          },
          c.key
        )) }),
        o && /* @__PURE__ */ r("div", { className: "mt-2", children: o })
      ]
    }
  );
}
export {
  U as EasySidebar,
  W as IconSidebar,
  $ as Sidebar,
  R as SidebarCollapsible,
  F as SidebarContent,
  M as SidebarFooter,
  A as SidebarGroup,
  I as SidebarHeader,
  Q as SidebarInset,
  v as SidebarItem,
  J as SidebarLayout,
  G as SidebarSeparator,
  k as SidebarSubItem,
  w as SidebarSubMenu,
  K as SidebarTrigger,
  g as useSidebar
};
