// @btcv/ui/dist/Table.js
import { jsxs as i, jsx as t } from "react/jsx-runtime";
import { c as d } from "./_shared/utils-eGXXUFl7.js";
function u({ className: e, ...r }) {
  return /* @__PURE__ */ t("div", { className: "overflow-x-auto rounded-2xl border border-border ring-1 ring-foreground/5 dark:ring-foreground/10 bg-glass", children: /* @__PURE__ */ t(
    "table",
    {
      "data-slot": "table",
      className: d("w-full border-collapse text-sm", e),
      ...r
    }
  ) });
}
function c({ className: e, ...r }) {
  return /* @__PURE__ */ t(
    "thead",
    {
      "data-slot": "table-header",
      className: d("border-b border-border", e),
      ...r
    }
  );
}
function m({ className: e, ...r }) {
  return /* @__PURE__ */ t(
    "tbody",
    {
      "data-slot": "table-body",
      className: d("[&_tr:last-child]:border-0", e),
      ...r
    }
  );
}
function l({ className: e, ...r }) {
  return /* @__PURE__ */ t(
    "tr",
    {
      "data-slot": "table-row",
      className: d(
        "border-b border-border transition-colors hover:bg-primary/8",
        e
      ),
      ...r
    }
  );
}
function f({ className: e, ...r }) {
  return /* @__PURE__ */ t(
    "th",
    {
      "data-slot": "table-head",
      className: d(
        "text-left text-xs font-semibold uppercase tracking-wider text-muted-foreground px-4 py-3 bg-muted",
        e
      ),
      ...r
    }
  );
}
function h({ className: e, ...r }) {
  return /* @__PURE__ */ t(
    "td",
    {
      "data-slot": "table-cell",
      className: d("px-4 py-3 text-foreground", e),
      ...r
    }
  );
}
function g({ className: e, ...r }) {
  return /* @__PURE__ */ t(
    "caption",
    {
      "data-slot": "table-caption",
      className: d("mt-4 text-sm text-muted-foreground", e),
      ...r
    }
  );
}
function y({ columns: e, data: r, className: n, ...s }) {
  return /* @__PURE__ */ i(u, { className: n, ...s, children: [
    /* @__PURE__ */ t(c, { children: /* @__PURE__ */ t(l, { children: e.map((a) => /* @__PURE__ */ t(f, { style: a.width ? { width: a.width } : void 0, children: a.label }, a.key)) }) }),
    /* @__PURE__ */ t(m, { children: r.map((a, b) => /* @__PURE__ */ t(l, { children: e.map((o) => /* @__PURE__ */ t(h, { children: o.render ? o.render(a[o.key], a) : a[o.key] }, o.key)) }, b)) })
  ] });
}
export {
  y as EasyTable,
  u as Table,
  m as TableBody,
  g as TableCaption,
  h as TableCell,
  f as TableHead,
  c as TableHeader,
  l as TableRow
};
