// @btcv/ui/dist/Tabs.js
"use client";
import { jsx as s, jsxs as y } from "react/jsx-runtime";
import * as f from "react";
import { u as S, c as x, a as E, P as F } from "./_shared/index-CXeb1OMI.js";
import { u as w, R as V, I as D, c as T } from "./_shared/index-C0gNQvxa.js";
import { P as v } from "./_shared/index-V-Ajw7Ac.js";
import { u as G } from "./_shared/index-BC7vfx-u.js";
import { c as b } from "./_shared/utils-eGXXUFl7.js";
var m = "Tabs", [$] = E(m, [
  T
]), C = T(), [L, h] = $(m), I = f.forwardRef(
  (o, n) => {
    const {
      __scopeTabs: i,
      value: e,
      onValueChange: r,
      defaultValue: a,
      orientation: t = "horizontal",
      dir: d,
      activationMode: p = "automatic",
      ...g
    } = o, l = w(d), [c, u] = S({
      prop: e,
      onChange: r,
      defaultProp: a ?? "",
      caller: m
    });
    return /* @__PURE__ */ s(
      L,
      {
        scope: i,
        baseId: G(),
        value: c,
        onValueChange: u,
        orientation: t,
        dir: l,
        activationMode: p,
        children: /* @__PURE__ */ s(
          v.div,
          {
            dir: l,
            "data-orientation": t,
            ...g,
            ref: n
          }
        )
      }
    );
  }
);
I.displayName = m;
var N = "TabsList", R = f.forwardRef(
  (o, n) => {
    const { __scopeTabs: i, loop: e = !0, ...r } = o, a = h(N, i), t = C(i);
    return /* @__PURE__ */ s(
      V,
      {
        asChild: !0,
        ...t,
        orientation: a.orientation,
        dir: a.dir,
        loop: e,
        children: /* @__PURE__ */ s(
          v.div,
          {
            role: "tablist",
            "aria-orientation": a.orientation,
            ...r,
            ref: n
          }
        )
      }
    );
  }
);
R.displayName = N;
var A = "TabsTrigger", P = f.forwardRef(
  (o, n) => {
    const { __scopeTabs: i, value: e, disabled: r = !1, ...a } = o, t = h(A, i), d = C(i), p = k(t.baseId, e), g = M(t.baseId, e), l = e === t.value;
    return /* @__PURE__ */ s(
      D,
      {
        asChild: !0,
        ...d,
        focusable: !r,
        active: l,
        children: /* @__PURE__ */ s(
          v.button,
          {
            type: "button",
            role: "tab",
            "aria-selected": l,
            "aria-controls": g,
            "data-state": l ? "active" : "inactive",
            "data-disabled": r ? "" : void 0,
            disabled: r,
            id: p,
            ...a,
            ref: n,
            onMouseDown: x(o.onMouseDown, (c) => {
              !r && c.button === 0 && c.ctrlKey === !1 ? t.onValueChange(e) : c.preventDefault();
            }),
            onKeyDown: x(o.onKeyDown, (c) => {
              [" ", "Enter"].includes(c.key) && t.onValueChange(e);
            }),
            onFocus: x(o.onFocus, () => {
              const c = t.activationMode !== "manual";
              !l && !r && c && t.onValueChange(e);
            })
          }
        )
      }
    );
  }
);
P.displayName = A;
var _ = "TabsContent", j = f.forwardRef(
  (o, n) => {
    const { __scopeTabs: i, value: e, forceMount: r, children: a, ...t } = o, d = h(_, i), p = k(d.baseId, e), g = M(d.baseId, e), l = e === d.value, c = f.useRef(l);
    return f.useEffect(() => {
      const u = requestAnimationFrame(() => c.current = !1);
      return () => cancelAnimationFrame(u);
    }, []), /* @__PURE__ */ s(F, { present: r || l, children: ({ present: u }) => /* @__PURE__ */ s(
      v.div,
      {
        "data-state": l ? "active" : "inactive",
        "data-orientation": d.orientation,
        role: "tabpanel",
        "aria-labelledby": p,
        hidden: !u,
        id: g,
        tabIndex: 0,
        ...t,
        ref: n,
        style: {
          ...o.style,
          animationDuration: c.current ? "0s" : void 0
        },
        children: u && a
      }
    ) });
  }
);
j.displayName = _;
function k(o, n) {
  return `${o}-trigger-${n}`;
}
function M(o, n) {
  return `${o}-content-${n}`;
}
var z = I, K = R, B = P;
function W({ tabs: o, active: n, onChange: i, variant: e = "line", className: r }) {
  return /* @__PURE__ */ s(z, { value: n, onValueChange: i, children: /* @__PURE__ */ s(
    K,
    {
      className: b(
        "inline-flex items-center",
        e === "line" && "gap-1 border-b border-border",
        e === "pill" && "h-9 gap-0 rounded-2xl bg-secondary p-1",
        r
      ),
      children: o.map((a) => {
        const t = a.key === n;
        return /* @__PURE__ */ y(
          B,
          {
            value: a.key,
            className: b(
              "inline-flex items-center justify-center gap-2 text-sm font-medium transition-all outline-none cursor-pointer focus-visible:border-ring focus-visible:ring-3 focus-visible:ring-ring/50",
              e === "line" && b(
                "px-3 py-2 border-b-2 -mb-px hover:text-foreground focus-visible:rounded-sm",
                t ? "text-primary border-b-primary" : "text-muted-foreground border-transparent"
              ),
              e === "pill" && b(
                "h-full flex-1 px-3 py-1 rounded-xl border hover:text-foreground",
                t ? "bg-background text-foreground border-transparent dark:bg-foreground/15" : "text-muted-foreground border-transparent"
              )
            ),
            children: [
              a.icon && /* @__PURE__ */ s(a.icon, { size: 15 }),
              a.label
            ]
          },
          a.key
        );
      })
    }
  ) });
}
function X({ items: o, active: n, onChange: i, className: e }) {
  return /* @__PURE__ */ s("nav", { className: b("flex items-center gap-2", e), children: o.map((r) => /* @__PURE__ */ y(
    "button",
    {
      onClick: () => i(r.key),
      className: b(
        "flex items-center gap-2 px-4 py-2 text-sm font-medium rounded-3xl border border-transparent transition-all outline-none cursor-pointer",
        n === r.key ? "bg-accent text-accent-foreground border-primary/40" : "text-muted-foreground hover:bg-muted hover:text-foreground",
        "focus-visible:border-ring focus-visible:ring-3 focus-visible:ring-ring/50"
      ),
      children: [
        r.icon && /* @__PURE__ */ s(r.icon, { size: 15 }),
        r.label
      ]
    },
    r.key
  )) });
}
export {
  X as NavPills,
  W as Tabs
};
