// @btcv/ui/dist/Toast.js
"use client";
import { jsx as H, jsxs as Ct } from "react/jsx-runtime";
import * as Gt from "react";
import a from "react";
import Qt from "react-dom";
import { X as Jt, Info as Zt, AlertTriangle as te, XCircle as ee, CheckCircle as ae } from "lucide-react";
import { c as Ht } from "./_shared/utils-eGXXUFl7.js";
function oe(n) {
  if (typeof document > "u") return;
  let o = document.head || document.getElementsByTagName("head")[0], e = document.createElement("style");
  e.type = "text/css", o.appendChild(e), e.styleSheet ? e.styleSheet.cssText = n : e.appendChild(document.createTextNode(n));
}
const se = (n) => {
  switch (n) {
    case "success":
      return ie;
    case "info":
      return de;
    case "warning":
      return le;
    case "error":
      return ce;
    default:
      return null;
  }
}, ne = Array(12).fill(0), re = ({ visible: n, className: o }) => /* @__PURE__ */ a.createElement("div", {
  className: [
    "sonner-loading-wrapper",
    o
  ].filter(Boolean).join(" "),
  "data-visible": n
}, /* @__PURE__ */ a.createElement("div", {
  className: "sonner-spinner"
}, ne.map((e, s) => /* @__PURE__ */ a.createElement("div", {
  className: "sonner-loading-bar",
  key: `spinner-bar-${s}`
})))), ie = /* @__PURE__ */ a.createElement("svg", {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 20 20",
  fill: "currentColor",
  height: "20",
  width: "20"
}, /* @__PURE__ */ a.createElement("path", {
  fillRule: "evenodd",
  d: "M10 18a8 8 0 100-16 8 8 0 000 16zm3.857-9.809a.75.75 0 00-1.214-.882l-3.483 4.79-1.88-1.88a.75.75 0 10-1.06 1.061l2.5 2.5a.75.75 0 001.137-.089l4-5.5z",
  clipRule: "evenodd"
})), le = /* @__PURE__ */ a.createElement("svg", {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 24 24",
  fill: "currentColor",
  height: "20",
  width: "20"
}, /* @__PURE__ */ a.createElement("path", {
  fillRule: "evenodd",
  d: "M9.401 3.003c1.155-2 4.043-2 5.197 0l7.355 12.748c1.154 2-.29 4.5-2.599 4.5H4.645c-2.309 0-3.752-2.5-2.598-4.5L9.4 3.003zM12 8.25a.75.75 0 01.75.75v3.75a.75.75 0 01-1.5 0V9a.75.75 0 01.75-.75zm0 8.25a.75.75 0 100-1.5.75.75 0 000 1.5z",
  clipRule: "evenodd"
})), de = /* @__PURE__ */ a.createElement("svg", {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 20 20",
  fill: "currentColor",
  height: "20",
  width: "20"
}, /* @__PURE__ */ a.createElement("path", {
  fillRule: "evenodd",
  d: "M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7-4a1 1 0 11-2 0 1 1 0 012 0zM9 9a.75.75 0 000 1.5h.253a.25.25 0 01.244.304l-.459 2.066A1.75 1.75 0 0010.747 15H11a.75.75 0 000-1.5h-.253a.25.25 0 01-.244-.304l.459-2.066A1.75 1.75 0 009.253 9H9z",
  clipRule: "evenodd"
})), ce = /* @__PURE__ */ a.createElement("svg", {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 20 20",
  fill: "currentColor",
  height: "20",
  width: "20"
}, /* @__PURE__ */ a.createElement("path", {
  fillRule: "evenodd",
  d: "M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-8-5a.75.75 0 01.75.75v4.5a.75.75 0 01-1.5 0v-4.5A.75.75 0 0110 5zm0 10a1 1 0 100-2 1 1 0 000 2z",
  clipRule: "evenodd"
})), ue = /* @__PURE__ */ a.createElement("svg", {
  xmlns: "http://www.w3.org/2000/svg",
  width: "12",
  height: "12",
  viewBox: "0 0 24 24",
  fill: "none",
  stroke: "currentColor",
  strokeWidth: "1.5",
  strokeLinecap: "round",
  strokeLinejoin: "round"
}, /* @__PURE__ */ a.createElement("line", {
  x1: "18",
  y1: "6",
  x2: "6",
  y2: "18"
}), /* @__PURE__ */ a.createElement("line", {
  x1: "6",
  y1: "6",
  x2: "18",
  y2: "18"
})), fe = () => {
  const [n, o] = a.useState(document.hidden);
  return a.useEffect(() => {
    const e = () => {
      o(document.hidden);
    };
    return document.addEventListener("visibilitychange", e), () => window.removeEventListener("visibilitychange", e);
  }, []), n;
};
let Tt = 1;
class me {
  constructor() {
    this.subscribe = (o) => (this.subscribers.push(o), () => {
      const e = this.subscribers.indexOf(o);
      this.subscribers.splice(e, 1);
    }), this.publish = (o) => {
      this.subscribers.forEach((e) => e(o));
    }, this.addToast = (o) => {
      this.publish(o), this.toasts = [
        ...this.toasts,
        o
      ];
    }, this.create = (o) => {
      var e;
      const { message: s, ...T } = o, d = typeof o?.id == "number" || ((e = o.id) == null ? void 0 : e.length) > 0 ? o.id : Tt++, g = this.toasts.find((h) => h.id === d), k = o.dismissible === void 0 ? !0 : o.dismissible;
      return this.dismissedToasts.has(d) && this.dismissedToasts.delete(d), g ? this.toasts = this.toasts.map((h) => h.id === d ? (this.publish({
        ...h,
        ...o,
        id: d,
        title: s
      }), {
        ...h,
        ...o,
        id: d,
        dismissible: k,
        title: s
      }) : h) : this.addToast({
        title: s,
        ...T,
        dismissible: k,
        id: d
      }), d;
    }, this.dismiss = (o) => (o ? (this.dismissedToasts.add(o), requestAnimationFrame(() => this.subscribers.forEach((e) => e({
      id: o,
      dismiss: !0
    })))) : this.toasts.forEach((e) => {
      this.subscribers.forEach((s) => s({
        id: e.id,
        dismiss: !0
      }));
    }), o), this.message = (o, e) => this.create({
      ...e,
      message: o
    }), this.error = (o, e) => this.create({
      ...e,
      message: o,
      type: "error"
    }), this.success = (o, e) => this.create({
      ...e,
      type: "success",
      message: o
    }), this.info = (o, e) => this.create({
      ...e,
      type: "info",
      message: o
    }), this.warning = (o, e) => this.create({
      ...e,
      type: "warning",
      message: o
    }), this.loading = (o, e) => this.create({
      ...e,
      type: "loading",
      message: o
    }), this.promise = (o, e) => {
      if (!e)
        return;
      let s;
      e.loading !== void 0 && (s = this.create({
        ...e,
        promise: o,
        type: "loading",
        message: e.loading,
        description: typeof e.description != "function" ? e.description : void 0
      }));
      const T = Promise.resolve(o instanceof Function ? o() : o);
      let d = s !== void 0, g;
      const k = T.then(async (i) => {
        if (g = [
          "resolve",
          i
        ], a.isValidElement(i))
          d = !1, this.create({
            id: s,
            type: "default",
            message: i
          });
        else if (ge(i) && !i.ok) {
          d = !1;
          const t = typeof e.error == "function" ? await e.error(`HTTP error! status: ${i.status}`) : e.error, w = typeof e.description == "function" ? await e.description(`HTTP error! status: ${i.status}`) : e.description, E = typeof t == "object" && !a.isValidElement(t) ? t : {
            message: t
          };
          this.create({
            id: s,
            type: "error",
            description: w,
            ...E
          });
        } else if (i instanceof Error) {
          d = !1;
          const t = typeof e.error == "function" ? await e.error(i) : e.error, w = typeof e.description == "function" ? await e.description(i) : e.description, E = typeof t == "object" && !a.isValidElement(t) ? t : {
            message: t
          };
          this.create({
            id: s,
            type: "error",
            description: w,
            ...E
          });
        } else if (e.success !== void 0) {
          d = !1;
          const t = typeof e.success == "function" ? await e.success(i) : e.success, w = typeof e.description == "function" ? await e.description(i) : e.description, E = typeof t == "object" && !a.isValidElement(t) ? t : {
            message: t
          };
          this.create({
            id: s,
            type: "success",
            description: w,
            ...E
          });
        }
      }).catch(async (i) => {
        if (g = [
          "reject",
          i
        ], e.error !== void 0) {
          d = !1;
          const N = typeof e.error == "function" ? await e.error(i) : e.error, t = typeof e.description == "function" ? await e.description(i) : e.description, j = typeof N == "object" && !a.isValidElement(N) ? N : {
            message: N
          };
          this.create({
            id: s,
            type: "error",
            description: t,
            ...j
          });
        }
      }).finally(() => {
        d && (this.dismiss(s), s = void 0), e.finally == null || e.finally.call(e);
      }), h = () => new Promise((i, N) => k.then(() => g[0] === "reject" ? N(g[1]) : i(g[1])).catch(N));
      return typeof s != "string" && typeof s != "number" ? {
        unwrap: h
      } : Object.assign(s, {
        unwrap: h
      });
    }, this.custom = (o, e) => {
      const s = e?.id || Tt++;
      return this.create({
        jsx: o(s),
        id: s,
        ...e
      }), s;
    }, this.getActiveToasts = () => this.toasts.filter((o) => !this.dismissedToasts.has(o.id)), this.subscribers = [], this.toasts = [], this.dismissedToasts = /* @__PURE__ */ new Set();
  }
}
const S = new me(), pe = (n, o) => {
  const e = o?.id || Tt++;
  return S.addToast({
    title: n,
    ...o,
    id: e
  }), e;
}, ge = (n) => n && typeof n == "object" && "ok" in n && typeof n.ok == "boolean" && "status" in n && typeof n.status == "number", he = pe, be = () => S.toasts, ve = () => S.getActiveToasts(), jt = Object.assign(he, {
  success: S.success,
  info: S.info,
  warning: S.warning,
  error: S.error,
  custom: S.custom,
  message: S.message,
  promise: S.promise,
  dismiss: S.dismiss,
  loading: S.loading
}, {
  getHistory: be,
  getToasts: ve
});
oe("[data-sonner-toaster][dir=ltr],html[dir=ltr]{--toast-icon-margin-start:-3px;--toast-icon-margin-end:4px;--toast-svg-margin-start:-1px;--toast-svg-margin-end:0px;--toast-button-margin-start:auto;--toast-button-margin-end:0;--toast-close-button-start:0;--toast-close-button-end:unset;--toast-close-button-transform:translate(-35%, -35%)}[data-sonner-toaster][dir=rtl],html[dir=rtl]{--toast-icon-margin-start:4px;--toast-icon-margin-end:-3px;--toast-svg-margin-start:0px;--toast-svg-margin-end:-1px;--toast-button-margin-start:0;--toast-button-margin-end:auto;--toast-close-button-start:unset;--toast-close-button-end:0;--toast-close-button-transform:translate(35%, -35%)}[data-sonner-toaster]{position:fixed;width:var(--width);font-family:ui-sans-serif,system-ui,-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica Neue,Arial,Noto Sans,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol,Noto Color Emoji;--gray1:hsl(0, 0%, 99%);--gray2:hsl(0, 0%, 97.3%);--gray3:hsl(0, 0%, 95.1%);--gray4:hsl(0, 0%, 93%);--gray5:hsl(0, 0%, 90.9%);--gray6:hsl(0, 0%, 88.7%);--gray7:hsl(0, 0%, 85.8%);--gray8:hsl(0, 0%, 78%);--gray9:hsl(0, 0%, 56.1%);--gray10:hsl(0, 0%, 52.3%);--gray11:hsl(0, 0%, 43.5%);--gray12:hsl(0, 0%, 9%);--border-radius:8px;box-sizing:border-box;padding:0;margin:0;list-style:none;outline:0;z-index:999999999;transition:transform .4s ease}@media (hover:none) and (pointer:coarse){[data-sonner-toaster][data-lifted=true]{transform:none}}[data-sonner-toaster][data-x-position=right]{right:var(--offset-right)}[data-sonner-toaster][data-x-position=left]{left:var(--offset-left)}[data-sonner-toaster][data-x-position=center]{left:50%;transform:translateX(-50%)}[data-sonner-toaster][data-y-position=top]{top:var(--offset-top)}[data-sonner-toaster][data-y-position=bottom]{bottom:var(--offset-bottom)}[data-sonner-toast]{--y:translateY(100%);--lift-amount:calc(var(--lift) * var(--gap));z-index:var(--z-index);position:absolute;opacity:0;transform:var(--y);touch-action:none;transition:transform .4s,opacity .4s,height .4s,box-shadow .2s;box-sizing:border-box;outline:0;overflow-wrap:anywhere}[data-sonner-toast][data-styled=true]{padding:16px;background:var(--normal-bg);border:1px solid var(--normal-border);color:var(--normal-text);border-radius:var(--border-radius);box-shadow:0 4px 12px rgba(0,0,0,.1);width:var(--width);font-size:13px;display:flex;align-items:center;gap:6px}[data-sonner-toast]:focus-visible{box-shadow:0 4px 12px rgba(0,0,0,.1),0 0 0 2px rgba(0,0,0,.2)}[data-sonner-toast][data-y-position=top]{top:0;--y:translateY(-100%);--lift:1;--lift-amount:calc(1 * var(--gap))}[data-sonner-toast][data-y-position=bottom]{bottom:0;--y:translateY(100%);--lift:-1;--lift-amount:calc(var(--lift) * var(--gap))}[data-sonner-toast][data-styled=true] [data-description]{font-weight:400;line-height:1.4;color:#3f3f3f}[data-rich-colors=true][data-sonner-toast][data-styled=true] [data-description]{color:inherit}[data-sonner-toaster][data-sonner-theme=dark] [data-description]{color:#e8e8e8}[data-sonner-toast][data-styled=true] [data-title]{font-weight:500;line-height:1.5;color:inherit}[data-sonner-toast][data-styled=true] [data-icon]{display:flex;height:16px;width:16px;position:relative;justify-content:flex-start;align-items:center;flex-shrink:0;margin-left:var(--toast-icon-margin-start);margin-right:var(--toast-icon-margin-end)}[data-sonner-toast][data-promise=true] [data-icon]>svg{opacity:0;transform:scale(.8);transform-origin:center;animation:sonner-fade-in .3s ease forwards}[data-sonner-toast][data-styled=true] [data-icon]>*{flex-shrink:0}[data-sonner-toast][data-styled=true] [data-icon] svg{margin-left:var(--toast-svg-margin-start);margin-right:var(--toast-svg-margin-end)}[data-sonner-toast][data-styled=true] [data-content]{display:flex;flex-direction:column;gap:2px}[data-sonner-toast][data-styled=true] [data-button]{border-radius:4px;padding-left:8px;padding-right:8px;height:24px;font-size:12px;color:var(--normal-bg);background:var(--normal-text);margin-left:var(--toast-button-margin-start);margin-right:var(--toast-button-margin-end);border:none;font-weight:500;cursor:pointer;outline:0;display:flex;align-items:center;flex-shrink:0;transition:opacity .4s,box-shadow .2s}[data-sonner-toast][data-styled=true] [data-button]:focus-visible{box-shadow:0 0 0 2px rgba(0,0,0,.4)}[data-sonner-toast][data-styled=true] [data-button]:first-of-type{margin-left:var(--toast-button-margin-start);margin-right:var(--toast-button-margin-end)}[data-sonner-toast][data-styled=true] [data-cancel]{color:var(--normal-text);background:rgba(0,0,0,.08)}[data-sonner-toaster][data-sonner-theme=dark] [data-sonner-toast][data-styled=true] [data-cancel]{background:rgba(255,255,255,.3)}[data-sonner-toast][data-styled=true] [data-close-button]{position:absolute;left:var(--toast-close-button-start);right:var(--toast-close-button-end);top:0;height:20px;width:20px;display:flex;justify-content:center;align-items:center;padding:0;color:var(--gray12);background:var(--normal-bg);border:1px solid var(--gray4);transform:var(--toast-close-button-transform);border-radius:50%;cursor:pointer;z-index:1;transition:opacity .1s,background .2s,border-color .2s}[data-sonner-toast][data-styled=true] [data-close-button]:focus-visible{box-shadow:0 4px 12px rgba(0,0,0,.1),0 0 0 2px rgba(0,0,0,.2)}[data-sonner-toast][data-styled=true] [data-disabled=true]{cursor:not-allowed}[data-sonner-toast][data-styled=true]:hover [data-close-button]:hover{background:var(--gray2);border-color:var(--gray5)}[data-sonner-toast][data-swiping=true]::before{content:'';position:absolute;left:-100%;right:-100%;height:100%;z-index:-1}[data-sonner-toast][data-y-position=top][data-swiping=true]::before{bottom:50%;transform:scaleY(3) translateY(50%)}[data-sonner-toast][data-y-position=bottom][data-swiping=true]::before{top:50%;transform:scaleY(3) translateY(-50%)}[data-sonner-toast][data-swiping=false][data-removed=true]::before{content:'';position:absolute;inset:0;transform:scaleY(2)}[data-sonner-toast][data-expanded=true]::after{content:'';position:absolute;left:0;height:calc(var(--gap) + 1px);bottom:100%;width:100%}[data-sonner-toast][data-mounted=true]{--y:translateY(0);opacity:1}[data-sonner-toast][data-expanded=false][data-front=false]{--scale:var(--toasts-before) * 0.05 + 1;--y:translateY(calc(var(--lift-amount) * var(--toasts-before))) scale(calc(-1 * var(--scale)));height:var(--front-toast-height)}[data-sonner-toast]>*{transition:opacity .4s}[data-sonner-toast][data-x-position=right]{right:0}[data-sonner-toast][data-x-position=left]{left:0}[data-sonner-toast][data-expanded=false][data-front=false][data-styled=true]>*{opacity:0}[data-sonner-toast][data-visible=false]{opacity:0;pointer-events:none}[data-sonner-toast][data-mounted=true][data-expanded=true]{--y:translateY(calc(var(--lift) * var(--offset)));height:var(--initial-height)}[data-sonner-toast][data-removed=true][data-front=true][data-swipe-out=false]{--y:translateY(calc(var(--lift) * -100%));opacity:0}[data-sonner-toast][data-removed=true][data-front=false][data-swipe-out=false][data-expanded=true]{--y:translateY(calc(var(--lift) * var(--offset) + var(--lift) * -100%));opacity:0}[data-sonner-toast][data-removed=true][data-front=false][data-swipe-out=false][data-expanded=false]{--y:translateY(40%);opacity:0;transition:transform .5s,opacity .2s}[data-sonner-toast][data-removed=true][data-front=false]::before{height:calc(var(--initial-height) + 20%)}[data-sonner-toast][data-swiping=true]{transform:var(--y) translateY(var(--swipe-amount-y,0)) translateX(var(--swipe-amount-x,0));transition:none}[data-sonner-toast][data-swiped=true]{user-select:none}[data-sonner-toast][data-swipe-out=true][data-y-position=bottom],[data-sonner-toast][data-swipe-out=true][data-y-position=top]{animation-duration:.2s;animation-timing-function:ease-out;animation-fill-mode:forwards}[data-sonner-toast][data-swipe-out=true][data-swipe-direction=left]{animation-name:swipe-out-left}[data-sonner-toast][data-swipe-out=true][data-swipe-direction=right]{animation-name:swipe-out-right}[data-sonner-toast][data-swipe-out=true][data-swipe-direction=up]{animation-name:swipe-out-up}[data-sonner-toast][data-swipe-out=true][data-swipe-direction=down]{animation-name:swipe-out-down}@keyframes swipe-out-left{from{transform:var(--y) translateX(var(--swipe-amount-x));opacity:1}to{transform:var(--y) translateX(calc(var(--swipe-amount-x) - 100%));opacity:0}}@keyframes swipe-out-right{from{transform:var(--y) translateX(var(--swipe-amount-x));opacity:1}to{transform:var(--y) translateX(calc(var(--swipe-amount-x) + 100%));opacity:0}}@keyframes swipe-out-up{from{transform:var(--y) translateY(var(--swipe-amount-y));opacity:1}to{transform:var(--y) translateY(calc(var(--swipe-amount-y) - 100%));opacity:0}}@keyframes swipe-out-down{from{transform:var(--y) translateY(var(--swipe-amount-y));opacity:1}to{transform:var(--y) translateY(calc(var(--swipe-amount-y) + 100%));opacity:0}}@media (max-width:600px){[data-sonner-toaster]{position:fixed;right:var(--mobile-offset-right);left:var(--mobile-offset-left);width:100%}[data-sonner-toaster][dir=rtl]{left:calc(var(--mobile-offset-left) * -1)}[data-sonner-toaster] [data-sonner-toast]{left:0;right:0;width:calc(100% - var(--mobile-offset-left) * 2)}[data-sonner-toaster][data-x-position=left]{left:var(--mobile-offset-left)}[data-sonner-toaster][data-y-position=bottom]{bottom:var(--mobile-offset-bottom)}[data-sonner-toaster][data-y-position=top]{top:var(--mobile-offset-top)}[data-sonner-toaster][data-x-position=center]{left:var(--mobile-offset-left);right:var(--mobile-offset-right);transform:none}}[data-sonner-toaster][data-sonner-theme=light]{--normal-bg:#fff;--normal-border:var(--gray4);--normal-text:var(--gray12);--success-bg:hsl(143, 85%, 96%);--success-border:hsl(145, 92%, 87%);--success-text:hsl(140, 100%, 27%);--info-bg:hsl(208, 100%, 97%);--info-border:hsl(221, 91%, 93%);--info-text:hsl(210, 92%, 45%);--warning-bg:hsl(49, 100%, 97%);--warning-border:hsl(49, 91%, 84%);--warning-text:hsl(31, 92%, 45%);--error-bg:hsl(359, 100%, 97%);--error-border:hsl(359, 100%, 94%);--error-text:hsl(360, 100%, 45%)}[data-sonner-toaster][data-sonner-theme=light] [data-sonner-toast][data-invert=true]{--normal-bg:#000;--normal-border:hsl(0, 0%, 20%);--normal-text:var(--gray1)}[data-sonner-toaster][data-sonner-theme=dark] [data-sonner-toast][data-invert=true]{--normal-bg:#fff;--normal-border:var(--gray3);--normal-text:var(--gray12)}[data-sonner-toaster][data-sonner-theme=dark]{--normal-bg:#000;--normal-bg-hover:hsl(0, 0%, 12%);--normal-border:hsl(0, 0%, 20%);--normal-border-hover:hsl(0, 0%, 25%);--normal-text:var(--gray1);--success-bg:hsl(150, 100%, 6%);--success-border:hsl(147, 100%, 12%);--success-text:hsl(150, 86%, 65%);--info-bg:hsl(215, 100%, 6%);--info-border:hsl(223, 43%, 17%);--info-text:hsl(216, 87%, 65%);--warning-bg:hsl(64, 100%, 6%);--warning-border:hsl(60, 100%, 9%);--warning-text:hsl(46, 87%, 65%);--error-bg:hsl(358, 76%, 10%);--error-border:hsl(357, 89%, 16%);--error-text:hsl(358, 100%, 81%)}[data-sonner-toaster][data-sonner-theme=dark] [data-sonner-toast] [data-close-button]{background:var(--normal-bg);border-color:var(--normal-border);color:var(--normal-text)}[data-sonner-toaster][data-sonner-theme=dark] [data-sonner-toast] [data-close-button]:hover{background:var(--normal-bg-hover);border-color:var(--normal-border-hover)}[data-rich-colors=true][data-sonner-toast][data-type=success]{background:var(--success-bg);border-color:var(--success-border);color:var(--success-text)}[data-rich-colors=true][data-sonner-toast][data-type=success] [data-close-button]{background:var(--success-bg);border-color:var(--success-border);color:var(--success-text)}[data-rich-colors=true][data-sonner-toast][data-type=info]{background:var(--info-bg);border-color:var(--info-border);color:var(--info-text)}[data-rich-colors=true][data-sonner-toast][data-type=info] [data-close-button]{background:var(--info-bg);border-color:var(--info-border);color:var(--info-text)}[data-rich-colors=true][data-sonner-toast][data-type=warning]{background:var(--warning-bg);border-color:var(--warning-border);color:var(--warning-text)}[data-rich-colors=true][data-sonner-toast][data-type=warning] [data-close-button]{background:var(--warning-bg);border-color:var(--warning-border);color:var(--warning-text)}[data-rich-colors=true][data-sonner-toast][data-type=error]{background:var(--error-bg);border-color:var(--error-border);color:var(--error-text)}[data-rich-colors=true][data-sonner-toast][data-type=error] [data-close-button]{background:var(--error-bg);border-color:var(--error-border);color:var(--error-text)}.sonner-loading-wrapper{--size:16px;height:var(--size);width:var(--size);position:absolute;inset:0;z-index:10}.sonner-loading-wrapper[data-visible=false]{transform-origin:center;animation:sonner-fade-out .2s ease forwards}.sonner-spinner{position:relative;top:50%;left:50%;height:var(--size);width:var(--size)}.sonner-loading-bar{animation:sonner-spin 1.2s linear infinite;background:var(--gray11);border-radius:6px;height:8%;left:-10%;position:absolute;top:-3.9%;width:24%}.sonner-loading-bar:first-child{animation-delay:-1.2s;transform:rotate(.0001deg) translate(146%)}.sonner-loading-bar:nth-child(2){animation-delay:-1.1s;transform:rotate(30deg) translate(146%)}.sonner-loading-bar:nth-child(3){animation-delay:-1s;transform:rotate(60deg) translate(146%)}.sonner-loading-bar:nth-child(4){animation-delay:-.9s;transform:rotate(90deg) translate(146%)}.sonner-loading-bar:nth-child(5){animation-delay:-.8s;transform:rotate(120deg) translate(146%)}.sonner-loading-bar:nth-child(6){animation-delay:-.7s;transform:rotate(150deg) translate(146%)}.sonner-loading-bar:nth-child(7){animation-delay:-.6s;transform:rotate(180deg) translate(146%)}.sonner-loading-bar:nth-child(8){animation-delay:-.5s;transform:rotate(210deg) translate(146%)}.sonner-loading-bar:nth-child(9){animation-delay:-.4s;transform:rotate(240deg) translate(146%)}.sonner-loading-bar:nth-child(10){animation-delay:-.3s;transform:rotate(270deg) translate(146%)}.sonner-loading-bar:nth-child(11){animation-delay:-.2s;transform:rotate(300deg) translate(146%)}.sonner-loading-bar:nth-child(12){animation-delay:-.1s;transform:rotate(330deg) translate(146%)}@keyframes sonner-fade-in{0%{opacity:0;transform:scale(.8)}100%{opacity:1;transform:scale(1)}}@keyframes sonner-fade-out{0%{opacity:1;transform:scale(1)}100%{opacity:0;transform:scale(.8)}}@keyframes sonner-spin{0%{opacity:1}100%{opacity:.15}}@media (prefers-reduced-motion){.sonner-loading-bar,[data-sonner-toast],[data-sonner-toast]>*{transition:none!important;animation:none!important}}.sonner-loader{position:absolute;top:50%;left:50%;transform:translate(-50%,-50%);transform-origin:center;transition:opacity .2s,transform .2s}.sonner-loader[data-visible=false]{opacity:0;transform:scale(.8) translate(-50%,-50%)}");
function pt(n) {
  return n.label !== void 0;
}
const ye = 3, xe = "24px", we = "16px", zt = 4e3, Ee = 356, Te = 14, ke = 45, Ne = 200;
function A(...n) {
  return n.filter(Boolean).join(" ");
}
function Se(n) {
  const [o, e] = n.split("-"), s = [];
  return o && s.push(o), e && s.push(e), s;
}
const _e = (n) => {
  var o, e, s, T, d, g, k, h, i;
  const { invert: N, toast: t, unstyled: w, interacting: j, setHeights: E, visibleToasts: ht, heights: U, index: c, toasts: st, expanded: W, removeToast: nt, defaultRichColors: bt, closeButton: P, style: tt, cancelButtonStyle: $, actionButtonStyle: vt, className: rt = "", descriptionClassName: yt = "", duration: et, position: z, gap: it, expandByDefault: at, classNames: u, icons: v, closeButtonAriaLabel: O = "Close toast" } = n, [F, M] = a.useState(null), [V, lt] = a.useState(null), [l, p] = a.useState(!1), [m, R] = a.useState(!1), [K, f] = a.useState(!1), [q, dt] = a.useState(!1), [ct, G] = a.useState(!1), [Pt, xt] = a.useState(0), [Lt, kt] = a.useState(0), ot = a.useRef(t.duration || et || zt), Nt = a.useRef(null), L = a.useRef(null), Ot = c === 0, Yt = c + 1 <= ht, _ = t.type, Q = t.dismissible !== !1, $t = t.className || "", Ft = t.descriptionClassName || "", ut = a.useMemo(() => U.findIndex((r) => r.toastId === t.id) || 0, [
    U,
    t.id
  ]), Vt = a.useMemo(() => {
    var r;
    return (r = t.closeButton) != null ? r : P;
  }, [
    t.closeButton,
    P
  ]), St = a.useMemo(() => t.duration || et || zt, [
    t.duration,
    et
  ]), wt = a.useRef(0), J = a.useRef(0), _t = a.useRef(0), Z = a.useRef(null), [Xt, Ut] = z.split("-"), Dt = a.useMemo(() => U.reduce((r, b, x) => x >= ut ? r : r + b.height, 0), [
    U,
    ut
  ]), Rt = fe(), Wt = t.invert || N, Et = _ === "loading";
  J.current = a.useMemo(() => ut * it + Dt, [
    ut,
    Dt
  ]), a.useEffect(() => {
    ot.current = St;
  }, [
    St
  ]), a.useEffect(() => {
    p(!0);
  }, []), a.useEffect(() => {
    const r = L.current;
    if (r) {
      const b = r.getBoundingClientRect().height;
      return kt(b), E((x) => [
        {
          toastId: t.id,
          height: b,
          position: t.position
        },
        ...x
      ]), () => E((x) => x.filter((D) => D.toastId !== t.id));
    }
  }, [
    E,
    t.id
  ]), a.useLayoutEffect(() => {
    if (!l) return;
    const r = L.current, b = r.style.height;
    r.style.height = "auto";
    const x = r.getBoundingClientRect().height;
    r.style.height = b, kt(x), E((D) => D.find((y) => y.toastId === t.id) ? D.map((y) => y.toastId === t.id ? {
      ...y,
      height: x
    } : y) : [
      {
        toastId: t.id,
        height: x,
        position: t.position
      },
      ...D
    ]);
  }, [
    l,
    t.title,
    t.description,
    E,
    t.id,
    t.jsx,
    t.action,
    t.cancel
  ]);
  const Y = a.useCallback(() => {
    R(!0), xt(J.current), E((r) => r.filter((b) => b.toastId !== t.id)), setTimeout(() => {
      nt(t);
    }, Ne);
  }, [
    t,
    nt,
    E,
    J
  ]);
  a.useEffect(() => {
    if (t.promise && _ === "loading" || t.duration === 1 / 0 || t.type === "loading") return;
    let r;
    return W || j || Rt ? (() => {
      if (_t.current < wt.current) {
        const D = (/* @__PURE__ */ new Date()).getTime() - wt.current;
        ot.current = ot.current - D;
      }
      _t.current = (/* @__PURE__ */ new Date()).getTime();
    })() : ot.current !== 1 / 0 && (wt.current = (/* @__PURE__ */ new Date()).getTime(), r = setTimeout(() => {
      t.onAutoClose == null || t.onAutoClose.call(t, t), Y();
    }, ot.current)), () => clearTimeout(r);
  }, [
    W,
    j,
    t,
    _,
    Rt,
    Y
  ]), a.useEffect(() => {
    t.delete && (Y(), t.onDismiss == null || t.onDismiss.call(t, t));
  }, [
    Y,
    t.delete
  ]);
  function Kt() {
    var r;
    if (v?.loading) {
      var b;
      return /* @__PURE__ */ a.createElement("div", {
        className: A(u?.loader, t == null || (b = t.classNames) == null ? void 0 : b.loader, "sonner-loader"),
        "data-visible": _ === "loading"
      }, v.loading);
    }
    return /* @__PURE__ */ a.createElement(re, {
      className: A(u?.loader, t == null || (r = t.classNames) == null ? void 0 : r.loader),
      visible: _ === "loading"
    });
  }
  const qt = t.icon || v?.[_] || se(_);
  var Bt, Mt;
  return /* @__PURE__ */ a.createElement("li", {
    tabIndex: 0,
    ref: L,
    className: A(rt, $t, u?.toast, t == null || (o = t.classNames) == null ? void 0 : o.toast, u?.default, u?.[_], t == null || (e = t.classNames) == null ? void 0 : e[_]),
    "data-sonner-toast": "",
    "data-rich-colors": (Bt = t.richColors) != null ? Bt : bt,
    "data-styled": !(t.jsx || t.unstyled || w),
    "data-mounted": l,
    "data-promise": !!t.promise,
    "data-swiped": ct,
    "data-removed": m,
    "data-visible": Yt,
    "data-y-position": Xt,
    "data-x-position": Ut,
    "data-index": c,
    "data-front": Ot,
    "data-swiping": K,
    "data-dismissible": Q,
    "data-type": _,
    "data-invert": Wt,
    "data-swipe-out": q,
    "data-swipe-direction": V,
    "data-expanded": !!(W || at && l),
    "data-testid": t.testId,
    style: {
      "--index": c,
      "--toasts-before": c,
      "--z-index": st.length - c,
      "--offset": `${m ? Pt : J.current}px`,
      "--initial-height": at ? "auto" : `${Lt}px`,
      ...tt,
      ...t.style
    },
    onDragEnd: () => {
      f(!1), M(null), Z.current = null;
    },
    onPointerDown: (r) => {
      r.button !== 2 && (Et || !Q || (Nt.current = /* @__PURE__ */ new Date(), xt(J.current), r.target.setPointerCapture(r.pointerId), r.target.tagName !== "BUTTON" && (f(!0), Z.current = {
        x: r.clientX,
        y: r.clientY
      })));
    },
    onPointerUp: () => {
      var r, b, x;
      if (q || !Q) return;
      Z.current = null;
      const D = Number(((r = L.current) == null ? void 0 : r.style.getPropertyValue("--swipe-amount-x").replace("px", "")) || 0), ft = Number(((b = L.current) == null ? void 0 : b.style.getPropertyValue("--swipe-amount-y").replace("px", "")) || 0), y = (/* @__PURE__ */ new Date()).getTime() - ((x = Nt.current) == null ? void 0 : x.getTime()), B = F === "x" ? D : ft, mt = Math.abs(B) / y;
      if (Math.abs(B) >= ke || mt > 0.11) {
        xt(J.current), t.onDismiss == null || t.onDismiss.call(t, t), lt(F === "x" ? D > 0 ? "right" : "left" : ft > 0 ? "down" : "up"), Y(), dt(!0);
        return;
      } else {
        var I, C;
        (I = L.current) == null || I.style.setProperty("--swipe-amount-x", "0px"), (C = L.current) == null || C.style.setProperty("--swipe-amount-y", "0px");
      }
      G(!1), f(!1), M(null);
    },
    onPointerMove: (r) => {
      var b, x, D;
      if (!Z.current || !Q || ((b = window.getSelection()) == null ? void 0 : b.toString().length) > 0) return;
      const y = r.clientY - Z.current.y, B = r.clientX - Z.current.x;
      var mt;
      const I = (mt = n.swipeDirections) != null ? mt : Se(z);
      !F && (Math.abs(B) > 1 || Math.abs(y) > 1) && M(Math.abs(B) > Math.abs(y) ? "x" : "y");
      let C = {
        x: 0,
        y: 0
      };
      const It = (X) => 1 / (1.5 + Math.abs(X) / 20);
      if (F === "y") {
        if (I.includes("top") || I.includes("bottom"))
          if (I.includes("top") && y < 0 || I.includes("bottom") && y > 0)
            C.y = y;
          else {
            const X = y * It(y);
            C.y = Math.abs(X) < Math.abs(y) ? X : y;
          }
      } else if (F === "x" && (I.includes("left") || I.includes("right")))
        if (I.includes("left") && B < 0 || I.includes("right") && B > 0)
          C.x = B;
        else {
          const X = B * It(B);
          C.x = Math.abs(X) < Math.abs(B) ? X : B;
        }
      (Math.abs(C.x) > 0 || Math.abs(C.y) > 0) && G(!0), (x = L.current) == null || x.style.setProperty("--swipe-amount-x", `${C.x}px`), (D = L.current) == null || D.style.setProperty("--swipe-amount-y", `${C.y}px`);
    }
  }, Vt && !t.jsx && _ !== "loading" ? /* @__PURE__ */ a.createElement("button", {
    "aria-label": O,
    "data-disabled": Et,
    "data-close-button": !0,
    onClick: Et || !Q ? () => {
    } : () => {
      Y(), t.onDismiss == null || t.onDismiss.call(t, t);
    },
    className: A(u?.closeButton, t == null || (s = t.classNames) == null ? void 0 : s.closeButton)
  }, (Mt = v?.close) != null ? Mt : ue) : null, (_ || t.icon || t.promise) && t.icon !== null && (v?.[_] !== null || t.icon) ? /* @__PURE__ */ a.createElement("div", {
    "data-icon": "",
    className: A(u?.icon, t == null || (T = t.classNames) == null ? void 0 : T.icon)
  }, t.promise || t.type === "loading" && !t.icon ? t.icon || Kt() : null, t.type !== "loading" ? qt : null) : null, /* @__PURE__ */ a.createElement("div", {
    "data-content": "",
    className: A(u?.content, t == null || (d = t.classNames) == null ? void 0 : d.content)
  }, /* @__PURE__ */ a.createElement("div", {
    "data-title": "",
    className: A(u?.title, t == null || (g = t.classNames) == null ? void 0 : g.title)
  }, t.jsx ? t.jsx : typeof t.title == "function" ? t.title() : t.title), t.description ? /* @__PURE__ */ a.createElement("div", {
    "data-description": "",
    className: A(yt, Ft, u?.description, t == null || (k = t.classNames) == null ? void 0 : k.description)
  }, typeof t.description == "function" ? t.description() : t.description) : null), /* @__PURE__ */ a.isValidElement(t.cancel) ? t.cancel : t.cancel && pt(t.cancel) ? /* @__PURE__ */ a.createElement("button", {
    "data-button": !0,
    "data-cancel": !0,
    style: t.cancelButtonStyle || $,
    onClick: (r) => {
      pt(t.cancel) && Q && (t.cancel.onClick == null || t.cancel.onClick.call(t.cancel, r), Y());
    },
    className: A(u?.cancelButton, t == null || (h = t.classNames) == null ? void 0 : h.cancelButton)
  }, t.cancel.label) : null, /* @__PURE__ */ a.isValidElement(t.action) ? t.action : t.action && pt(t.action) ? /* @__PURE__ */ a.createElement("button", {
    "data-button": !0,
    "data-action": !0,
    style: t.actionButtonStyle || vt,
    onClick: (r) => {
      pt(t.action) && (t.action.onClick == null || t.action.onClick.call(t.action, r), !r.defaultPrevented && Y());
    },
    className: A(u?.actionButton, t == null || (i = t.classNames) == null ? void 0 : i.actionButton)
  }, t.action.label) : null);
};
function At() {
  if (typeof window > "u" || typeof document > "u") return "ltr";
  const n = document.documentElement.getAttribute("dir");
  return n === "auto" || !n ? window.getComputedStyle(document.documentElement).direction : n;
}
function De(n, o) {
  const e = {};
  return [
    n,
    o
  ].forEach((s, T) => {
    const d = T === 1, g = d ? "--mobile-offset" : "--offset", k = d ? we : xe;
    function h(i) {
      [
        "top",
        "right",
        "bottom",
        "left"
      ].forEach((N) => {
        e[`${g}-${N}`] = typeof i == "number" ? `${i}px` : i;
      });
    }
    typeof s == "number" || typeof s == "string" ? h(s) : typeof s == "object" ? [
      "top",
      "right",
      "bottom",
      "left"
    ].forEach((i) => {
      s[i] === void 0 ? e[`${g}-${i}`] = k : e[`${g}-${i}`] = typeof s[i] == "number" ? `${s[i]}px` : s[i];
    }) : h(k);
  }), e;
}
const Re = /* @__PURE__ */ a.forwardRef(function(o, e) {
  const { id: s, invert: T, position: d = "bottom-right", hotkey: g = [
    "altKey",
    "KeyT"
  ], expand: k, closeButton: h, className: i, offset: N, mobileOffset: t, theme: w = "light", richColors: j, duration: E, style: ht, visibleToasts: U = ye, toastOptions: c, dir: st = At(), gap: W = Te, icons: nt, containerAriaLabel: bt = "Notifications" } = o, [P, tt] = a.useState([]), $ = a.useMemo(() => s ? P.filter((l) => l.toasterId === s) : P.filter((l) => !l.toasterId), [
    P,
    s
  ]), vt = a.useMemo(() => Array.from(new Set([
    d
  ].concat($.filter((l) => l.position).map((l) => l.position)))), [
    $,
    d
  ]), [rt, yt] = a.useState([]), [et, z] = a.useState(!1), [it, at] = a.useState(!1), [u, v] = a.useState(w !== "system" ? w : typeof window < "u" && window.matchMedia && window.matchMedia("(prefers-color-scheme: dark)").matches ? "dark" : "light"), O = a.useRef(null), F = g.join("+").replace(/Key/g, "").replace(/Digit/g, ""), M = a.useRef(null), V = a.useRef(!1), lt = a.useCallback((l) => {
    tt((p) => {
      var m;
      return (m = p.find((R) => R.id === l.id)) != null && m.delete || S.dismiss(l.id), p.filter(({ id: R }) => R !== l.id);
    });
  }, []);
  return a.useEffect(() => S.subscribe((l) => {
    if (l.dismiss) {
      requestAnimationFrame(() => {
        tt((p) => p.map((m) => m.id === l.id ? {
          ...m,
          delete: !0
        } : m));
      });
      return;
    }
    setTimeout(() => {
      Qt.flushSync(() => {
        tt((p) => {
          const m = p.findIndex((R) => R.id === l.id);
          return m !== -1 ? [
            ...p.slice(0, m),
            {
              ...p[m],
              ...l
            },
            ...p.slice(m + 1)
          ] : [
            l,
            ...p
          ];
        });
      });
    });
  }), [
    P
  ]), a.useEffect(() => {
    if (w !== "system") {
      v(w);
      return;
    }
    if (w === "system" && (window.matchMedia && window.matchMedia("(prefers-color-scheme: dark)").matches ? v("dark") : v("light")), typeof window > "u") return;
    const l = window.matchMedia("(prefers-color-scheme: dark)");
    try {
      l.addEventListener("change", ({ matches: p }) => {
        v(p ? "dark" : "light");
      });
    } catch {
      l.addListener(({ matches: m }) => {
        try {
          v(m ? "dark" : "light");
        } catch (R) {
          console.error(R);
        }
      });
    }
  }, [
    w
  ]), a.useEffect(() => {
    P.length <= 1 && z(!1);
  }, [
    P
  ]), a.useEffect(() => {
    const l = (p) => {
      var m;
      if (g.every((f) => p[f] || p.code === f)) {
        var K;
        z(!0), (K = O.current) == null || K.focus();
      }
      p.code === "Escape" && (document.activeElement === O.current || (m = O.current) != null && m.contains(document.activeElement)) && z(!1);
    };
    return document.addEventListener("keydown", l), () => document.removeEventListener("keydown", l);
  }, [
    g
  ]), a.useEffect(() => {
    if (O.current)
      return () => {
        M.current && (M.current.focus({
          preventScroll: !0
        }), M.current = null, V.current = !1);
      };
  }, [
    O.current
  ]), // Remove item from normal navigation flow, only available via hotkey
  /* @__PURE__ */ a.createElement("section", {
    ref: e,
    "aria-label": `${bt} ${F}`,
    tabIndex: -1,
    "aria-live": "polite",
    "aria-relevant": "additions text",
    "aria-atomic": "false",
    suppressHydrationWarning: !0
  }, vt.map((l, p) => {
    var m;
    const [R, K] = l.split("-");
    return $.length ? /* @__PURE__ */ a.createElement("ol", {
      key: l,
      dir: st === "auto" ? At() : st,
      tabIndex: -1,
      ref: O,
      className: i,
      "data-sonner-toaster": !0,
      "data-sonner-theme": u,
      "data-y-position": R,
      "data-x-position": K,
      style: {
        "--front-toast-height": `${((m = rt[0]) == null ? void 0 : m.height) || 0}px`,
        "--width": `${Ee}px`,
        "--gap": `${W}px`,
        ...ht,
        ...De(N, t)
      },
      onBlur: (f) => {
        V.current && !f.currentTarget.contains(f.relatedTarget) && (V.current = !1, M.current && (M.current.focus({
          preventScroll: !0
        }), M.current = null));
      },
      onFocus: (f) => {
        f.target instanceof HTMLElement && f.target.dataset.dismissible === "false" || V.current || (V.current = !0, M.current = f.relatedTarget);
      },
      onMouseEnter: () => z(!0),
      onMouseMove: () => z(!0),
      onMouseLeave: () => {
        it || z(!1);
      },
      onDragEnd: () => z(!1),
      onPointerDown: (f) => {
        f.target instanceof HTMLElement && f.target.dataset.dismissible === "false" || at(!0);
      },
      onPointerUp: () => at(!1)
    }, $.filter((f) => !f.position && p === 0 || f.position === l).map((f, q) => {
      var dt, ct;
      return /* @__PURE__ */ a.createElement(_e, {
        key: f.id,
        icons: nt,
        index: q,
        toast: f,
        defaultRichColors: j,
        duration: (dt = c?.duration) != null ? dt : E,
        className: c?.className,
        descriptionClassName: c?.descriptionClassName,
        invert: T,
        visibleToasts: U,
        closeButton: (ct = c?.closeButton) != null ? ct : h,
        interacting: it,
        position: l,
        style: c?.style,
        unstyled: c?.unstyled,
        classNames: c?.classNames,
        cancelButtonStyle: c?.cancelButtonStyle,
        actionButtonStyle: c?.actionButtonStyle,
        closeButtonAriaLabel: c?.closeButtonAriaLabel,
        removeToast: lt,
        toasts: $.filter((G) => G.position == f.position),
        heights: rt.filter((G) => G.position == f.position),
        setHeights: yt,
        expandByDefault: k,
        gap: W,
        expanded: et,
        swipeDirections: o.swipeDirections
      });
    })) : null;
  }));
});
function Pe({ position: n = "bottom-right", className: o, ...e }) {
  return /* @__PURE__ */ H(
    Re,
    {
      position: n,
      className: Ht("toaster group", o),
      toastOptions: { style: { opacity: 1, background: "transparent", border: "none", padding: 0, boxShadow: "none" } },
      ...e
    }
  );
}
const Be = {
  success: {
    icon: /* @__PURE__ */ H(ae, { size: 18, className: "text-success shrink-0" }),
    tint: "before:bg-success/15 dark:before:bg-success/10",
    border: "border-success/30 dark:border-success/20"
  },
  error: {
    icon: /* @__PURE__ */ H(ee, { size: 18, className: "text-destructive shrink-0" }),
    tint: "before:bg-destructive/15 dark:before:bg-destructive/10",
    border: "border-destructive/30 dark:border-destructive/20"
  },
  warning: {
    icon: /* @__PURE__ */ H(te, { size: 18, className: "text-warning shrink-0" }),
    tint: "before:bg-warning/15 dark:before:bg-warning/10",
    border: "border-warning/30 dark:border-warning/20"
  },
  info: {
    icon: /* @__PURE__ */ H(Zt, { size: 18, className: "text-info shrink-0" }),
    tint: "before:bg-info/15 dark:before:bg-info/10",
    border: "border-info/30 dark:border-info/20"
  }
};
function Me({ id: n, variant: o, title: e, message: s, icon: T }) {
  const d = Be[o];
  return /* @__PURE__ */ Ct(
    "div",
    {
      className: Ht(
        "isolate relative flex items-center gap-3 border rounded-3xl shadow-lg text-sm px-4 py-3 w-[356px] bg-card backdrop-blur-xl",
        "before:absolute before:inset-0 before:-z-10 before:rounded-[inherit] before:pointer-events-none",
        d.tint,
        d.border
      ),
      children: [
        T || d.icon,
        /* @__PURE__ */ Ct("div", { className: "flex flex-col gap-0.5 flex-1 min-w-0", children: [
          e && /* @__PURE__ */ H("p", { className: "text-foreground font-medium", children: e }),
          /* @__PURE__ */ H("p", { className: "text-muted-foreground", children: s })
        ] }),
        /* @__PURE__ */ H(
          "button",
          {
            onClick: () => jt.dismiss(n),
            className: "text-muted-foreground hover:text-foreground shrink-0 cursor-pointer flex items-center",
            children: /* @__PURE__ */ H(Jt, { size: 14 })
          }
        )
      ]
    }
  );
}
function gt(n, o) {
  const s = typeof o == "object" && o !== null && !Gt.isValidElement(o) ? o : void 0, T = s?.title, d = s ? s.message : o, g = s?.icon, k = s?.duration;
  return jt.custom(
    (h) => /* @__PURE__ */ H(
      Me,
      {
        id: h,
        variant: n,
        title: T,
        message: d,
        icon: g
      }
    ),
    { duration: k }
  );
}
const Le = {
  success: (n) => gt("success", n),
  error: (n) => gt("error", n),
  warning: (n) => gt("warning", n),
  info: (n) => gt("info", n)
};
export {
  Pe as ToastProvider,
  Le as btcvToast,
  Le as toast
};
