// @btcv/ui/dist/Tooltip.js
"use client";
import { jsx as u, Fragment as q, jsxs as X } from "react/jsx-runtime";
import * as s from "react";
import { P as S, a as K, u as W, c as x } from "./_shared/index-CXeb1OMI.js";
import { u as H } from "./_shared/index-DLcqcWxM.js";
import { P as J, D as Q } from "./_shared/index-B03TCNO5.js";
import { u as Z } from "./_shared/index-BC7vfx-u.js";
import { C as ee, c as F, R as te, A as oe, a as re } from "./_shared/index-uu9PT5Nu.js";
import { P as ne } from "./_shared/index-V-Ajw7Ac.js";
import { R as se } from "./_shared/index-B1f-hyuh.js";
import { c as ae } from "./_shared/utils-eGXXUFl7.js";
var ie = /* @__PURE__ */ Symbol("radix.slottable");
// @__NO_SIDE_EFFECTS__
function le(t) {
  const o = ({ children: e }) => /* @__PURE__ */ u(q, { children: e });
  return o.displayName = `${t}.Slottable`, o.__radixId = ie, o;
}
var [O] = K("Tooltip", [
  F
]), D = F(), G = "TooltipProvider", ce = 700, L = "tooltip.open", [ue, k] = O(G), j = (t) => {
  const {
    __scopeTooltip: o,
    delayDuration: e = ce,
    skipDelayDuration: r = 300,
    disableHoverableContent: n = !1,
    children: l
  } = t, i = s.useRef(!0), v = s.useRef(!1), a = s.useRef(0);
  return s.useEffect(() => {
    const p = a.current;
    return () => window.clearTimeout(p);
  }, []), /* @__PURE__ */ u(
    ue,
    {
      scope: o,
      isOpenDelayedRef: i,
      delayDuration: e,
      onOpen: s.useCallback(() => {
        window.clearTimeout(a.current), i.current = !1;
      }, []),
      onClose: s.useCallback(() => {
        window.clearTimeout(a.current), a.current = window.setTimeout(
          () => i.current = !0,
          r
        );
      }, [r]),
      isPointerInTransitRef: v,
      onPointerInTransitChange: s.useCallback((p) => {
        v.current = p;
      }, []),
      disableHoverableContent: n,
      children: l
    }
  );
};
j.displayName = G;
var R = "Tooltip", [de, _] = O(R), $ = (t) => {
  const {
    __scopeTooltip: o,
    children: e,
    open: r,
    defaultOpen: n,
    onOpenChange: l,
    disableHoverableContent: i,
    delayDuration: v
  } = t, a = k(R, t.__scopeTooltip), p = D(o), [c, f] = s.useState(null), h = Z(), d = s.useRef(0), m = i ?? a.disableHoverableContent, g = v ?? a.delayDuration, T = s.useRef(!1), [C, y] = W({
    prop: r,
    defaultProp: n ?? !1,
    onChange: (M) => {
      M ? (a.onOpen(), document.dispatchEvent(new CustomEvent(L))) : a.onClose(), l?.(M);
    },
    caller: R
  }), w = s.useMemo(() => C ? T.current ? "delayed-open" : "instant-open" : "closed", [C]), P = s.useCallback(() => {
    window.clearTimeout(d.current), d.current = 0, T.current = !1, y(!0);
  }, [y]), E = s.useCallback(() => {
    window.clearTimeout(d.current), d.current = 0, y(!1);
  }, [y]), N = s.useCallback(() => {
    window.clearTimeout(d.current), d.current = window.setTimeout(() => {
      T.current = !0, y(!0), d.current = 0;
    }, g);
  }, [g, y]);
  return s.useEffect(() => () => {
    d.current && (window.clearTimeout(d.current), d.current = 0);
  }, []), /* @__PURE__ */ u(te, { ...p, children: /* @__PURE__ */ u(
    de,
    {
      scope: o,
      contentId: h,
      open: C,
      stateAttribute: w,
      trigger: c,
      onTriggerChange: f,
      onTriggerEnter: s.useCallback(() => {
        a.isOpenDelayedRef.current ? N() : P();
      }, [a.isOpenDelayedRef, N, P]),
      onTriggerLeave: s.useCallback(() => {
        m ? E() : (window.clearTimeout(d.current), d.current = 0);
      }, [E, m]),
      onOpen: P,
      onClose: E,
      disableHoverableContent: m,
      children: e
    }
  ) });
};
$.displayName = R;
var A = "TooltipTrigger", B = s.forwardRef(
  (t, o) => {
    const { __scopeTooltip: e, ...r } = t, n = _(A, e), l = k(A, e), i = D(e), v = s.useRef(null), a = H(o, v, n.onTriggerChange), p = s.useRef(!1), c = s.useRef(!1), f = s.useCallback(() => p.current = !1, []);
    return s.useEffect(() => () => document.removeEventListener("pointerup", f), [f]), /* @__PURE__ */ u(oe, { asChild: !0, ...i, children: /* @__PURE__ */ u(
      ne.button,
      {
        "aria-describedby": n.open ? n.contentId : void 0,
        "data-state": n.stateAttribute,
        ...r,
        ref: a,
        onPointerMove: x(t.onPointerMove, (h) => {
          h.pointerType !== "touch" && !c.current && !l.isPointerInTransitRef.current && (n.onTriggerEnter(), c.current = !0);
        }),
        onPointerLeave: x(t.onPointerLeave, () => {
          n.onTriggerLeave(), c.current = !1;
        }),
        onPointerDown: x(t.onPointerDown, () => {
          n.open && n.onClose(), p.current = !0, document.addEventListener("pointerup", f, { once: !0 });
        }),
        onFocus: x(t.onFocus, () => {
          p.current || n.onOpen();
        }),
        onBlur: x(t.onBlur, n.onClose),
        onClick: x(t.onClick, n.onClose)
      }
    ) });
  }
);
B.displayName = A;
var I = "TooltipPortal", [pe, fe] = O(I, {
  forceMount: void 0
}), z = (t) => {
  const { __scopeTooltip: o, forceMount: e, children: r, container: n } = t, l = _(I, o);
  return /* @__PURE__ */ u(pe, { scope: o, forceMount: e, children: /* @__PURE__ */ u(S, { present: e || l.open, children: /* @__PURE__ */ u(J, { asChild: !0, container: n, children: r }) }) });
};
z.displayName = I;
var b = "TooltipContent", U = s.forwardRef(
  (t, o) => {
    const e = fe(b, t.__scopeTooltip), { forceMount: r = e.forceMount, side: n = "top", ...l } = t, i = _(b, t.__scopeTooltip);
    return /* @__PURE__ */ u(S, { present: r || i.open, children: i.disableHoverableContent ? /* @__PURE__ */ u(V, { side: n, ...l, ref: o }) : /* @__PURE__ */ u(ve, { side: n, ...l, ref: o }) });
  }
), ve = s.forwardRef((t, o) => {
  const e = _(b, t.__scopeTooltip), r = k(b, t.__scopeTooltip), n = s.useRef(null), l = H(o, n), [i, v] = s.useState(null), { trigger: a, onClose: p } = e, c = n.current, { onPointerInTransitChange: f } = r, h = s.useCallback(() => {
    v(null), f(!1);
  }, [f]), d = s.useCallback(
    (m, g) => {
      const T = m.currentTarget, C = { x: m.clientX, y: m.clientY }, y = ye(C, T.getBoundingClientRect()), w = Ce(C, y), P = xe(g.getBoundingClientRect()), E = we([...w, ...P]);
      v(E), f(!0);
    },
    [f]
  );
  return s.useEffect(() => () => h(), [h]), s.useEffect(() => {
    if (a && c) {
      const m = (T) => d(T, c), g = (T) => d(T, a);
      return a.addEventListener("pointerleave", m), c.addEventListener("pointerleave", g), () => {
        a.removeEventListener("pointerleave", m), c.removeEventListener("pointerleave", g);
      };
    }
  }, [a, c, d, h]), s.useEffect(() => {
    if (i) {
      const m = (g) => {
        const T = g.target, C = { x: g.clientX, y: g.clientY }, y = a?.contains(T) || c?.contains(T), w = !be(C, i);
        y ? h() : w && (h(), p());
      };
      return document.addEventListener("pointermove", m), () => document.removeEventListener("pointermove", m);
    }
  }, [a, c, i, p, h]), /* @__PURE__ */ u(V, { ...t, ref: l });
}), [he, me] = O(R, { isInside: !1 }), Te = /* @__PURE__ */ le("TooltipContent"), V = s.forwardRef(
  (t, o) => {
    const {
      __scopeTooltip: e,
      children: r,
      "aria-label": n,
      onEscapeKeyDown: l,
      onPointerDownOutside: i,
      ...v
    } = t, a = _(b, e), p = D(e), { onClose: c } = a;
    return s.useEffect(() => (document.addEventListener(L, c), () => document.removeEventListener(L, c)), [c]), s.useEffect(() => {
      if (a.trigger) {
        const f = (h) => {
          h.target?.contains(a.trigger) && c();
        };
        return window.addEventListener("scroll", f, { capture: !0 }), () => window.removeEventListener("scroll", f, { capture: !0 });
      }
    }, [a.trigger, c]), /* @__PURE__ */ u(
      Q,
      {
        asChild: !0,
        disableOutsidePointerEvents: !1,
        onEscapeKeyDown: l,
        onPointerDownOutside: i,
        onFocusOutside: (f) => f.preventDefault(),
        onDismiss: c,
        children: /* @__PURE__ */ X(
          ee,
          {
            "data-state": a.stateAttribute,
            ...p,
            ...v,
            ref: o,
            style: {
              ...v.style,
              "--radix-tooltip-content-transform-origin": "var(--radix-popper-transform-origin)",
              "--radix-tooltip-content-available-width": "var(--radix-popper-available-width)",
              "--radix-tooltip-content-available-height": "var(--radix-popper-available-height)",
              "--radix-tooltip-trigger-width": "var(--radix-popper-anchor-width)",
              "--radix-tooltip-trigger-height": "var(--radix-popper-anchor-height)"
            },
            children: [
              /* @__PURE__ */ u(Te, { children: r }),
              /* @__PURE__ */ u(he, { scope: e, isInside: !0, children: /* @__PURE__ */ u(se, { id: a.contentId, role: "tooltip", children: n || r }) })
            ]
          }
        )
      }
    );
  }
);
U.displayName = b;
var Y = "TooltipArrow", ge = s.forwardRef(
  (t, o) => {
    const { __scopeTooltip: e, ...r } = t, n = D(e);
    return me(
      Y,
      e
    ).isInside ? null : /* @__PURE__ */ u(re, { ...n, ...r, ref: o });
  }
);
ge.displayName = Y;
function ye(t, o) {
  const e = Math.abs(o.top - t.y), r = Math.abs(o.bottom - t.y), n = Math.abs(o.right - t.x), l = Math.abs(o.left - t.x);
  switch (Math.min(e, r, n, l)) {
    case l:
      return "left";
    case n:
      return "right";
    case e:
      return "top";
    case r:
      return "bottom";
    default:
      throw new Error("unreachable");
  }
}
function Ce(t, o, e = 5) {
  const r = [];
  switch (o) {
    case "top":
      r.push(
        { x: t.x - e, y: t.y + e },
        { x: t.x + e, y: t.y + e }
      );
      break;
    case "bottom":
      r.push(
        { x: t.x - e, y: t.y - e },
        { x: t.x + e, y: t.y - e }
      );
      break;
    case "left":
      r.push(
        { x: t.x + e, y: t.y - e },
        { x: t.x + e, y: t.y + e }
      );
      break;
    case "right":
      r.push(
        { x: t.x - e, y: t.y - e },
        { x: t.x - e, y: t.y + e }
      );
      break;
  }
  return r;
}
function xe(t) {
  const { top: o, right: e, bottom: r, left: n } = t;
  return [
    { x: n, y: o },
    { x: e, y: o },
    { x: e, y: r },
    { x: n, y: r }
  ];
}
function be(t, o) {
  const { x: e, y: r } = t;
  let n = !1;
  for (let l = 0, i = o.length - 1; l < o.length; i = l++) {
    const v = o[l], a = o[i], p = v.x, c = v.y, f = a.x, h = a.y;
    c > r != h > r && e < (f - p) * (r - c) / (h - c) + p && (n = !n);
  }
  return n;
}
function we(t) {
  const o = t.slice();
  return o.sort((e, r) => e.x < r.x ? -1 : e.x > r.x ? 1 : e.y < r.y ? -1 : e.y > r.y ? 1 : 0), Pe(o);
}
function Pe(t) {
  if (t.length <= 1) return t.slice();
  const o = [];
  for (let r = 0; r < t.length; r++) {
    const n = t[r];
    for (; o.length >= 2; ) {
      const l = o[o.length - 1], i = o[o.length - 2];
      if ((l.x - i.x) * (n.y - i.y) >= (l.y - i.y) * (n.x - i.x)) o.pop();
      else break;
    }
    o.push(n);
  }
  o.pop();
  const e = [];
  for (let r = t.length - 1; r >= 0; r--) {
    const n = t[r];
    for (; e.length >= 2; ) {
      const l = e[e.length - 1], i = e[e.length - 2];
      if ((l.x - i.x) * (n.y - i.y) >= (l.y - i.y) * (n.x - i.x)) e.pop();
      else break;
    }
    e.push(n);
  }
  return e.pop(), o.length === 1 && e.length === 1 && o[0].x === e[0].x && o[0].y === e[0].y ? o : o.concat(e);
}
var Ee = j, Re = $, _e = B, Oe = z, De = U;
const je = Ee, $e = Re, Be = _e, Le = s.forwardRef(({ className: t, sideOffset: o = 6, ...e }, r) => /* @__PURE__ */ u(Oe, { children: /* @__PURE__ */ u(
  De,
  {
    ref: r,
    sideOffset: o,
    className: ae(
      "z-50 overflow-hidden rounded-xl border border-border bg-glass px-3 py-1.5 text-sm shadow-md",
      "animate-in fade-in-0 zoom-in-95",
      "data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=closed]:zoom-out-95",
      "data-[side=bottom]:slide-in-from-top-2",
      "data-[side=left]:slide-in-from-right-2",
      "data-[side=right]:slide-in-from-left-2",
      "data-[side=top]:slide-in-from-bottom-2",
      t
    ),
    ...e
  }
) }));
Le.displayName = "TooltipContent";
export {
  $e as Tooltip,
  Le as TooltipContent,
  je as TooltipProvider,
  Be as TooltipTrigger
};
