// @btcv/ui/dist/Typography.js
import { jsxs as i, jsx as n } from "react/jsx-runtime";
import { c as o } from "./_shared/utils-eGXXUFl7.js";
function f({ children: t, icon: e, className: r, ...s }) {
  return /* @__PURE__ */ i(
    "h2",
    {
      className: o(
        "text-2xl font-bold text-foreground flex items-center gap-3",
        !e && "before:content-[''] before:w-1 before:h-6 before:bg-primary before:rounded-sm before:shrink-0",
        r
      ),
      ...s,
      children: [
        e && /* @__PURE__ */ n("span", { className: "w-9 h-9 rounded-xl bg-primary/15 text-primary flex items-center justify-center shrink-0", children: /* @__PURE__ */ n(e, { size: 18 }) }),
        t
      ]
    }
  );
}
function u({ children: t, className: e, ...r }) {
  return /* @__PURE__ */ n(
    "p",
    {
      className: o(
        "text-[0.8125rem] text-muted-foreground",
        e
      ),
      ...r,
      children: t
    }
  );
}
function m({ children: t, className: e, ...r }) {
  return /* @__PURE__ */ n(
    "h3",
    {
      className: o(
        "text-xs font-semibold text-muted-foreground uppercase tracking-wide",
        e
      ),
      ...r,
      children: t
    }
  );
}
function x({ children: t, className: e, ...r }) {
  return /* @__PURE__ */ n(
    "h1",
    {
      className: o("text-2xl font-semibold text-foreground", e),
      ...r,
      children: t
    }
  );
}
function d({ children: t, className: e, ...r }) {
  return /* @__PURE__ */ n(
    "h2",
    {
      className: o("text-lg font-semibold text-foreground", e),
      ...r,
      children: t
    }
  );
}
function l({ children: t, className: e, ...r }) {
  return /* @__PURE__ */ n(
    "h3",
    {
      className: o("text-sm font-semibold text-foreground", e),
      ...r,
      children: t
    }
  );
}
function p({ children: t, className: e, ...r }) {
  return /* @__PURE__ */ n(
    "span",
    {
      className: o("text-xs text-muted-foreground", e),
      ...r,
      children: t
    }
  );
}
function g({ children: t, className: e, ...r }) {
  return /* @__PURE__ */ n(
    "code",
    {
      className: o("font-mono text-[13px] text-foreground", e),
      ...r,
      children: t
    }
  );
}
function b({ name: t, suffix: e, className: r, ...s }) {
  return /* @__PURE__ */ i(
    "span",
    {
      className: o(
        "text-sm font-bold uppercase tracking-wider text-foreground",
        r
      ),
      ...s,
      children: [
        t,
        " ",
        /* @__PURE__ */ n("span", { className: "text-primary", children: e })
      ]
    }
  );
}
export {
  b as AppName,
  p as Caption,
  x as H1,
  d as H2,
  l as H3,
  g as InlineCode,
  u as PageDescription,
  f as PageTitle,
  m as SectionTitle
};
