// @btcv/ui/dist/Widget.js
import { jsxs as l, jsx as s } from "react/jsx-runtime";
import { Card as m } from "./Card.js";
import { c as o } from "./_shared/utils-eGXXUFl7.js";
const x = "p-5 transition-all hover:border-secondary", f = {
  primary: "bg-primary/15 text-primary",
  info: "bg-info/15 text-info",
  success: "bg-success/15 text-success",
  destructive: "bg-destructive/15 text-destructive",
  warning: "bg-warning/15 text-warning"
}, h = {
  primary: "text-primary",
  info: "text-info",
  success: "text-success",
  destructive: "text-destructive",
  warning: "text-warning"
};
function b({
  icon: t,
  title: r,
  value: e,
  variant: c,
  accent: n = !!c,
  className: i,
  ...d
}) {
  return /* @__PURE__ */ s(
    m,
    {
      accent: n ? c : void 0,
      className: o(x, i),
      ...d,
      children: /* @__PURE__ */ l("div", { className: "flex items-center gap-3", children: [
        t && /* @__PURE__ */ s(
          "div",
          {
            className: o(
              "w-9 h-9 rounded-xl flex items-center justify-center shrink-0",
              c ? f[c] : "text-foreground"
            ),
            children: /* @__PURE__ */ s(t, { size: c ? 18 : 22 })
          }
        ),
        /* @__PURE__ */ l("div", { className: "flex flex-col gap-0.5 min-w-0", children: [
          /* @__PURE__ */ s("span", { className: "text-sm text-muted-foreground truncate", children: r }),
          /* @__PURE__ */ s("span", { className: "text-xl font-bold text-foreground", children: e })
        ] })
      ] })
    }
  );
}
function y({
  title: t,
  icon: r,
  variant: e,
  accent: c = !!e,
  children: n,
  className: i,
  ...d
}) {
  return /* @__PURE__ */ l(
    m,
    {
      accent: c ? e : void 0,
      className: o(x, "flex flex-col", i),
      ...d,
      children: [
        (t || r) && /* @__PURE__ */ l("div", { className: "flex items-center gap-2", children: [
          r && /* @__PURE__ */ s(
            "div",
            {
              className: o(
                "w-7 h-7 rounded-xl flex items-center justify-center shrink-0",
                e ? f[e] : "text-foreground"
              ),
              children: /* @__PURE__ */ s(r, { size: e ? 14 : 18 })
            }
          ),
          t && /* @__PURE__ */ s("span", { className: "text-sm font-semibold text-foreground", children: t })
        ] }),
        /* @__PURE__ */ s("div", { className: "flex flex-col -my-2.5", children: n })
      ]
    }
  );
}
function w({
  icon: t,
  iconClassName: r,
  imageUrl: e,
  variant: c,
  children: n,
  subtitle: i,
  suffix: d,
  className: a,
  ...u
}) {
  return /* @__PURE__ */ l(
    "div",
    {
      className: o("flex items-center gap-3 py-2.5 border-b border-border last:border-b-0", a),
      ...u,
      children: [
        (t || e) && /* @__PURE__ */ s(
          "div",
          {
            className: o(
              "w-8 h-8 rounded-xl bg-secondary flex items-center justify-center shrink-0 overflow-hidden",
              c ? h[c] : "text-muted-foreground",
              r
            ),
            children: e ? /* @__PURE__ */ s(
              "img",
              {
                src: e,
                alt: "",
                className: "w-full h-full object-cover"
              }
            ) : t ? /* @__PURE__ */ s(t, { size: 14 }) : null
          }
        ),
        /* @__PURE__ */ l("div", { className: "flex-1 min-w-0", children: [
          /* @__PURE__ */ s("div", { className: "text-[0.8125rem] font-medium text-foreground", children: n }),
          i && /* @__PURE__ */ s("div", { className: "text-xs text-muted-foreground", children: i })
        ] }),
        d && /* @__PURE__ */ s("div", { className: "shrink-0 flex items-center", children: d })
      ]
    }
  );
}
function v({
  title: t,
  icon: r,
  variant: e,
  accent: c = !!e,
  children: n,
  className: i,
  ...d
}) {
  return /* @__PURE__ */ l(
    m,
    {
      accent: c ? e : void 0,
      className: o(x, "flex flex-col", i),
      ...d,
      children: [
        (t || r) && /* @__PURE__ */ l("div", { className: "flex items-center gap-2", children: [
          r && /* @__PURE__ */ s(
            "div",
            {
              className: o(
                "w-7 h-7 rounded-xl flex items-center justify-center shrink-0",
                e ? f[e] : "text-foreground"
              ),
              children: /* @__PURE__ */ s(r, { size: e ? 14 : 18 })
            }
          ),
          t && /* @__PURE__ */ s("span", { className: "text-sm font-semibold text-foreground", children: t })
        ] }),
        n
      ]
    }
  );
}
function j({
  title: t,
  icon: r,
  variant: e,
  accent: c = !!e,
  children: n,
  className: i,
  ...d
}) {
  return /* @__PURE__ */ l(
    m,
    {
      accent: c ? e : void 0,
      className: o(x, "flex flex-col gap-3", i),
      ...d,
      children: [
        (t || r) && /* @__PURE__ */ l("div", { className: "flex items-center gap-2", children: [
          r && /* @__PURE__ */ s(
            "div",
            {
              className: o(
                "w-7 h-7 rounded-xl flex items-center justify-center shrink-0",
                e ? f[e] : "text-foreground"
              ),
              children: /* @__PURE__ */ s(r, { size: e ? 14 : 18 })
            }
          ),
          t && /* @__PURE__ */ s("span", { className: "text-sm font-semibold text-foreground", children: t })
        ] }),
        n && /* @__PURE__ */ s("div", { className: "text-sm text-muted-foreground", children: n })
      ]
    }
  );
}
export {
  v as CustomWidget,
  y as ListWidget,
  w as ListWidgetLine,
  b as StatWidget,
  j as TextWidget
};
